<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Afterqc on Biowulf';</script>
<div class="title">Afterqc on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
Automatic Filtering, Trimming, Error Removing and Quality Control for fastq data. AfterQC can simply go through all fastq files in a folder and then output three folders: good, bad and QC folders, which contains good reads, bad reads and the QC results of each fastq file/pair.
</p>

<h3>References:</h3>
<ul>
<li>Shifu Chen, Tanxiao Huang, Yanqing Zhou, Yue Han, Mingyan Xu and Jia Gu. AfterQC: automatic filtering, trimming, error removing and quality control for fastq data. BMC Bioinformatics 2017 18(Suppl 3):80 https://doi.org/10.1186/s12859-017-1469-3</li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/OpenGene/AfterQC">Afterqc Main Site</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>afterqc</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
<li>Single threaded app
<li>environment variables set <!--for ones users should be aware of -->
  <ul>
    <li><tt>afterqc_HOME=/usr/local/apps/afterqc/$version</tt></li>
  </ul>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session (user input in <b>bold</b>:</p>
<pre class="term">
[user@biowulf ~]$ <b>sinteractive -c4 --mem=8g --gres=lscratch:10</b>
salloc.exe: Pending job allocation 11086981
salloc.exe: job 11086981 queued and waiting for resources
salloc.exe: job 11086981 has been allocated resources
salloc.exe: Granted job allocation 11086981
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn0882 are ready for job
srun: error: x11: no local DISPLAY defined, skipping
error: unable to open file /tmp/slurm-spank-x11.11086981.0
slurmstepd: error: x11: unable to read DISPLAY value

[user@cn0882 ~]$ <b>cd /lscratch/$SLURM_JOB_ID</b>

[user@cn0882 11086981]$ <b>module load afterqc</b>
[+] Loading afterqc  0.9.7  on cn0882
[+] Loading python 3.7  ...

[user@cn0882 11086981]$ <b>python2 $afterqc_HOME/after.py  \
     -1 /fdb/app_testdata/fastq/H_sapiens/hg100_1m_pe1.fq.gz \
     -2 /fdb/app_testdata/fastq/H_sapiens/hg100_1m_pe2.fq.gz</b>

/fdb/app_testdata/fastq/H_sapiens/hg100_1m_pe1.fq.gz options:
{'qc_only': False, 'version': '0.9.6', 'seq_len_req': 35, 'index1_file': None, 'trim_tail': 0, 'report_output_folder': None, 'trim_pair_same': True, 
'no_correction': False, 'debubble_dir': 'debubble', 'barcode_flag': 'barcode', 'read2_file': '/fdb/app_testdata/fastq/H_sapiens/hg100_1m_pe2.fq.gz', 
'barcode_length': 12, 'trim_tail2': 0, 'unqualified_base_limit': 60, 'allow_mismatch_in_poly': 2, 'read2_flag': 'R2', 'store_overlap': False, 
'debubble': False, 'read1_flag': 'R1', 'index2_flag': 'I2', 'draw': True, 'index1_flag': 'I1', 'mask_mismatch': False, 'barcode': False, 
'gzip': False, 'overlap_output_folder': None, 'barcode_verify': 'CAGTA', 'compression': 2, 'index2_file': None, 'qualified_quality_phred': 15, 
'trim_front': 1, 'good_output_folder': 'good', 'poly_size_limit': 35, 'n_base_limit': 5, 'qc_sample': 200000, 'trim_front2': 1, 'no_overlap': False, 
'input_dir': None, 'read1_file': '/fdb/app_testdata/fastq/H_sapiens/hg100_1m_pe1.fq.gz', 'qc_kmer': 8, 'bad_output_folder': None}

Time used: 544.626863003

[user@cn0882 11086981]$ <b>ls -lh</b>
total 12K
drwxr-xr-x 2 user user 4.0K Mar 22 17:47 bad
drwxr-xr-x 2 user user 4.0K Mar 22 17:47 good
drwxr-xr-x 2 user user 4.0K Mar 22 17:54 QC

[user@cn0882 11086981]$ <b>exit</b>
exit
salloc.exe: Relinquishing job allocation 11086981
salloc.exe: Job allocation 11086981 has been revoked.

[user@biowulf ~]$ 
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. afterqc.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load afterqc
python2 $afterqc_HOME/after.py -1 /fdb/app_testdata/fastq/H_sapiens/hg100_1m_pe1.fq.gz -2 /fdb/app_testdata/fastq/H_sapiens/hg100_1m_pe2.fq.gz 
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch afterqc.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. afterqc.swarm). For example:</p>

<pre class="term">
python2 $afterqc_HOME/after.py -1 sample1_1.fq.gz -2 sample1_2.fq.gz 
python2 $afterqc_HOME/after.py -1 sample2_1.fq.gz -2 sample2_2.fq.gz 
python2 $afterqc_HOME/after.py -1 sample3_1.fq.gz -2 sample3_2.fq.gz 
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f afterqc.swarm --module afterqc</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module afterqc</tt> <td>Loads the afterqc module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
