<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Bamliquidator on Biowulf';</script>
<div class="title">Bamliquidator on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
Bamliquidator is a set of tools for analyzing the density of short DNA sequence read alignments in the BAM file format

<ul>
<li>the read counts across multiple genomes are grouped, normalized, summarized, and graphed in interactive html files
</li>
<li>for an interactive graph example, see this summary and this breakdown for a single chromosome
</li>
<li>a BAM file is a binary sequence alignment map -- see SAMtools for more info
</li>
<li>the read counts and summaries are stored in HDF5 format where they can be efficiently read via Python PyTables or the HDF5 C apis
</li>
<ul>
<li>the HDF5 files can be viewed directly with the cross platform tool HDFView
</li>
<li>there is an option to output the data in tab delimited text files as well
</li>
</ul>
<li>there is also a command line utility for counting the number of reads in specified portion of a chromosome, and the count is output to the console</li>
</ul>
</p>

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/BradnerLab/pipeline/wiki/bamliquidator">bamliquidator Main Site</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>bamliquidator</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
<li>Multithreaded app
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>cd /lscratch/$SLURM_JOB_ID</b>

[user@cn3144 ~]$ <b>cp -r /usr/local/apps/bamliquidator/test .</b>

[user@cn3144 ~]$ <b>module load bamliqudiator</b>

[user@cn3144 ~]$ <b>bamliquidator_batch test/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam</b>
Liquidating test/wgEncodeUwRepliSeqBg02esG1bAlnRep1.bam (file 1 of 1)
Liquidation completed: 4.188048 seconds, 1784867 reads, 0.238775 millions of reads per second
Cell Types: test
Normalizing and calculating percentiles for cell type test
Indexing normalized counts
Plotting
-- skipping plotting chrM because not enough bins (only 1)
-- skipping plotting chrM because not enough bins (only 1)
Summarizing
Post liquidation processing took 2.364760 seconds

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. bamliquidator.sh). For example:</p>

<pre class="term">
#!/bin/bash
module load bamliquidator
bamliquidator_batch.py -n 16 -o test.output directory_of_bam_files/
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=16 --mem=10g bamliquidator.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. bamliquidator.swarm). For example:</p>

<pre class="term">
bamliquidator sorted_and_indexed.bam chr1 0 1000000 + 2 200
bamliquidator sorted_and_indexed.bam chr1 1000000 2000000 + 2 200
bamliquidator sorted_and_indexed.bam chr1 2000000 3000000 + 2 200
bamliquidator sorted_and_indexed.bam chr1 3000000 4000000 + 2 200
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f bamliquidator.swarm [-g 4] [-t 2] --module bamliquidator</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module bamliquidator</tt> <td>Loads the bamliquidator module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
