<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Bedops on Biowulf';</script>
    <div class="title">Bedops on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</td></tr></table>

<p>
<p>Bedops is a software package for manipulating and analyzing genomic interval
data.  It contains tools to</p>
<ul>
<li>perform boolean and set operations on genome intervals</li>
<li>carry out statistical calculations on genome intervals</li>
<li>sort bed files consistently (important for a lot of operations)</li>
<li>compress genomic data using it's own <em>starch</em> format</li>
<li>conversions of various formats (bam, gff/gtf, wig, vcf) 
  to <em>bed</em> and <em>starch</em></li>
</ul>
<p>Operations can be parallelized efficiently by chromosome.</p>
<p>Each tool is designed to use unix input and output streams for building
efficient pipelines.</p>

</p>

<h3>References:</h3>
Shane Neph, et al. <em>BEDOPS: high-performance genomic feature operations</em>
  Bioinformatics 2012, 28:1919-1920.
  <a href="http://www.ncbi.nlm.nih.gov/pubmed/22576172">Pubmed</a>&nbsp;|&nbsp;
  <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3389768/">PMC</a>&nbsp;|&nbsp;
  <a href="http://bioinformatics.oxfordjournals.org/content/28/14/1919.long">Journal</a>

<p></p>
<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/bedops/bedops">GitHub</a></li>
<li><a href="http://bedops.readthedocs.org/en/latest/">Manual</a></li>
<li><a href="http://bedops.readthedocs.org/en/latest/content/overview.html#overview">Overview</a></li>
<li><a href="http://bedops.readthedocs.org/en/latest/content/usage-examples.html">Examples</a></li>
</ul>

<a Name="notes"></a>
    <div class="heading"><a name="notes" id="notes"></a>Important Notes</div>
    <ul>
      <li>Module Name: <tt>bedops</tt> (see <a href="/apps/modules.html">the modules 
        page</a> for more information) </li>
    </ul>
<P>

<a Name="int"></a></p><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
    <pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <strong>convert2bed --input=bam --output=starch --starch-note "wt/H3ac/a13" &lt; bam_inputfile &gt; starch/a13.starch</strong>
[user@cn3144 ~]$ <strong>unstarch --note starch/a13.starch wt/H3ac/a13</strong>
[user@cn3144 ~]$ <strong>unstarch --elements starch/a13.starch</strong>
[user@cn3144 ~]$ <strong>unstarch --bases-uniq starch/a13.starch</strong>

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a></p><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
    <p>Create a batch input file (e.g. bedops.sh). For example:</p>

    <pre class="term">
#!/bin/bash
set -e
module load bedops
sort-bed in.bed &gt; out.bed</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

    <pre class="term">sbatch [--mem=#] bedops.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

    <p>Create a swarmfile (e.g. bedops.swarm). For example:</p>

<pre class="term">
cd dir1; sort-bed in.bed &gt; out.bed
cd dir2; sort-bed in.bed &gt; out.bed
cd dir3; sort-bed in.bed &gt; out.bed
cd dir4; sort-bed in.bed &gt; out.bed

</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

    <pre class="term">swarm -f bedops.swarm [-g #] --module bedops</pre>
where
    <table width="55%" border=0>
      <tr>
        <td width=12%><tt>-g <i>#</i> </tt></td>
        <td width="88%">Number of Gigabytes of memory required for each process 
          (1 line in the swarm command file) </td>
      </tr>
      <tr> 
        <td><tt>--module </tt></td>
        <td>Loads the module for each subjob in the swarm </td>
      </tr>
    </table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
