<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'bioawk on Biowulf &amp; Helix';</script>
<div class="title">bioawk on Biowulf &amp; Helix</div>
<p></p>
<table width=25% align=right> <tr><td>
    <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><a href="#int">Interactive job on Biowulf</a></div>
        <div class="tocItem"><a href="#serial">Batch job on Biowulf</a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs</a></div>
    </div>
</table>

<!-- ======================================================================= -->
<div class="heading">Description</div>

<p>Bioawk extends awk with support for several common biological data formats,
including optionally gzip'ed BED, GFF, SAM, VCF, FASTA/Q and TAB-delimited
formats with column names. It also adds a few built-in functions and an command
line option to use TAB as the input/output delimiter. When the new
functionality is not used, bioawk is intended to behave exactly the same as the
original BWK awk.  </p>

<p>There may be multiple versions of bioawk available. An easy way of selecting the
version is to use <a href="/apps/modules.html">modules</a>.  To see the modules
available, type</p>
<pre class="term">
module avail bioawk 
</pre>

<p>To select a module use</p>
<pre class="term">
module load bioawk/[version]
</pre>
<p>where <code>[version]</code> is the version of choice.</p>


<h3>Environment variables set</h3>
<ul>
<li> <code>$PATH</code></li>
<li> <code>$BIOAWK_TEST_DATA</code></li>
</ul>

<h3>Documentation</h3>
<ul>
    <li><a href="https://github.com/lh3/bioawk">GitHub page</a>
    </li>
</ul>

<!-- ======================================================================= -->
<a class="navbar-safe-anchor"  Name="int"></a>
<div class="heading">Interactive job on Biowulf</div>


<p>Allocate an interactive session with <a href="/docs/userguide.html#int">sinteractive</a>
and use as described below</p>
<pre class="term">
biowulf$ <b>sinteractive </b>
node$ <b>module load bioawk samtools</b>
node$ # what formats are supported and what automatic variables are created for each of them?
node$ <b>bioawk -c help</b>
bed:
        1:chrom 2:start 3:end 4:name 5:score 6:strand 7:thickstart 8:thickend 9:rgb 10:blockcount 11:blocksizes 12:blockstarts 
sam:
        1:qname 2:flag 3:rname 4:pos 5:mapq 6:cigar 7:rnext 8:pnext 9:tlen 10:seq 11:qual 
vcf:
        1:chrom 2:pos 3:id 4:ref 5:alt 6:qual 7:filter 8:info 
gff:
        1:seqname 2:source 3:feature 4:start 5:end 6:score 7:filter 8:strand 9:group 10:attribute 
fastx:
        1:name 2:seq 3:qual 4:comment 
node$ # create fasta from bam file. Reverse complement the read sequence if the
node$ # read aligned to the minus strand
node$ <b>samtools view $BIOAWK_TEST_DATA/aln.bam \
             | bioawk -c sam '{s=$seq; if(and($flag, 16)) {s=revcomp($seq)} print ">"$qname"\n"s}' \
             | head</b>
&gt;D00777:83:C8R9PACXX:3:2111:6151:30663
CCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCAAACCCTAACACTGACGCGAACGGAAAGAGTAA
&gt;D00777:83:C8R9PACXX:3:2111:6151:30663
CCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCAAACCCTAACACTGACGCGAACGGAAAGAGTAA
&gt;D00777:83:C8R9PACXX:3:2111:6151:30663
CCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCAAACCCTAACACTGACGCGAACGGAAAGAGTAA
&gt;D00777:83:C8R9PACXX:3:2111:6151:30663
CCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCAAACCCTAACACTGACGCGAACGGAAAGAGTAA
&gt;D00777:83:C8R9PACXX:3:2111:6151:30663
CCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCAAACCCTAACACTGACGCGAACGGAAAGAGTAA
node$ <b>samtools view $BIOAWK_TEST_DATA/aln.bam | bioawk -c sam '$seq ~ /^ATG/ {printf("%s:%i\t%s\n", $rname, $pos, $seq)}'</b>
chr1:14686      ATGGAGCCCCCTACGATTCCCAGTCGTCCTCGTCCTCCTCTGCCTGTGGCTGCTGCGGTGGCGGCAGAGGAGGGATGGAGTCTGACACGCGGGCAAAGGCT
chr1:17535      ATGCCCTGGGTCCCCACTAAGCCAGGCCGGGCCTCCCGCCCACACCCCTCGGCCCTGCCCTCTGGCCATACAGGTTCTCGGTGGTGTTGAAGAGCAGCAAG
chr1:17535      ATGCCCTGGGTCCCCACTAAGCCAGGCCGGGCCTCCCGCCCACACCCCTCGGCCCTGCCCTCTGGCCATACAGGTTCTCGGTGGTGTTGAAGAGCAGCAAG
node$ <b>exit</b>
biowulf$
</pre>

<!-- ======================================================================= -->
<a class="navbar-safe-anchor"  Name="serial"></a>
<div class="heading">Batch job on Biowulf</div>

<p>Create a batch script similar to the following example:</p>
<pre class="term">
#! /bin/bash
# this file is bioawk.batch

module load bioawk || exit 1
bioawk -c gff '$feature == "exon" and (end - start) &lt; 100' refseq.gff
</pre>

<p>Submit to the queue with <a href="/docs/userguide.html">sbatch</a>:</p>
<pre class="term">
biowulf$ <b>sbatch --time=10 bioawk.batch</b>
</pre>

<!-- ======================================================================= -->
<a class="navbar-safe-anchor"  Name="swarm"></a>
<div class="heading">Swarm of jobs on Biowulf</div>

<p>Create a swarm command file similar to the following example:</p>
<pre class="term">
# this file is bioawk.swarm
bioawk -c fastx '{print "&gt;" $name; print revcomp($seq)}' seq1.fa.gz | gzip -c &gt; seq1.rc.fa.gz
bioawk -c fastx '{print "&gt;" $name; print revcomp($seq)}' seq2.fa.gz | gzip -c &gt; seq2.rc.fa.gz
bioawk -c fastx '{print "&gt;" $name; print revcomp($seq)}' seq3.fa.gz | gzip -c &gt; seq3.rc.fa.gz
</pre>

<p>And submit to the queue with <a href="/apps/swarm.html">swarm</a></p>
<pre class="term">
biowulf$ <b>swarm -f bioawk.swarm --module bioawk --time 10</b>
</pre>


<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
