<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Bsmap on HPC';</script>
    <div class="title">Bsmap on HPC</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</td></tr></table>    <p> BSMAP is a short reads mapping software for bisulfite sequencing reads. Bisulfite 
      treatment converts unmethylated Cytosines into Uracils (sequenced as Thymine) 
      and leave methylated Cytosines unchanged, hence provides a way to study 
      DNA cytosine methylation at single nucleotide resolution. BSMAP aligns the 
      Ts in the reads to both Cs and Ts in the reference. </p>
    <p>RRBSMAP is a specifically designed version of BSMAP for reduced representation 
      bisulfite sequencing (RRBS), it indexes the genome only on the enzyme digestion 
      sites and therefore guarantees all reads were mapped to digestion sites, 
      and greatly reduces the CPU/memory usage. Since BSMAP-2.0, RRBSMAP has been 
      merged into BSMAP.</p>
    <p></p>
    <h3>References:</h3>    <ul>
      <li><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-10-232">https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-10-232</a></li>
    </ul>


<a Name="doc"></a><div class="heading">Documentation</div>    <ul>
      <li><a href="https://github.com/genome-vendor/bsmap">https://github.com/genome-vendor/bsmap</a></li>
    </ul>

<a Name="notes"></a>
    <div class="heading"><a name="notes" id="notes"></a>Important Notes</div>    <ul>
      <li>Module Name: <tt>bsmap</tt> (see <a href="/apps/modules.html">the modules 
        page</a> for more information) </li>
      <li>Multithreaded</li>
    </ul>
<P>

<a Name="int"></a></p><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
    <pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=4</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <strong>module load bsmap</strong>
[user@cn3144 ~]$ <strong>bsmap -a infile -d ref.fa -o out.bam -p $SLURM_CPUS_PER_TASK</strong>

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a></p><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
    <p>Create a batch input file (e.g. batch.sh). For example:</p>    
    <pre class="term">
#!/bin/bash
set -e
module load bsmap
bsmap -a infile -d ref.fa -o out.bam -p $SLURM_CPUS_PER_TASK</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

    <pre class="term">sbatch --cpus-per-task=4 batch.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

    <p>Create a swarmfile (e.g. job.swarm). For example:</p>

<pre class="term">
cd dir1; bsmap -a infile -d ref.fa -o out.bam -p $SLURM_CPUS_PER_TASK
cd dir2; bsmap -a infile -d ref.fa -o out.bam -p $SLURM_CPUS_PER_TASK
cd dir3; bsmap -a infile -d ref.fa -o out.bam -p $SLURM_CPUS_PER_TASK
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

    <pre class="term">swarm -f job.swarm -t 4 --module bsmap</pre>
where
    <table width="63%" border=0>
      <tr>
        <td width="18%"><tt>-t <i>#</i></tt> </td>
        <td width="82%">Number of threads/CPUs required for each process (1 line 
          in the swarm command file). </td>
      </tr>
      <tr> 
        <td><tt>--module </tt></td>
        <td>Loads the module for each subjob in the swarm </td>
      </tr>
    </table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
