<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Burrows-Wheeler Alignment (BWA) Tool on Biowulf';</script>
<div class="title">Burrows-Wheeler Alignment (BWA) Tool on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>BWA is a fast light-weighted tool that aligns short sequences to a sequence database, such as the human reference genome. By default, BWA finds an alignment within edit distance 2 to the query sequence, except for disallowing gaps close to the end of the query. It can also be tuned to find a fraction of longer gaps at the cost of speed and of more false alignments.
</p>

<p>BWA excels in its speed. Mapping 2 million high-quality 35bp short reads against the human genome can be done in 20 minutes. Usually the speed is gained at the cost of huge memory, disallowing gaps and/or the hard limits on the maximum read length and the maximum mismatches. BWA does not. It is still relatively light-weighted (2.3GB memory for human alignment), performs gapped alignment, and does not set a hard limit on read length or maximum mismatches.
</p>

<p>Given a database file in FASTA format, BWA first builds BWT index with the 'index' command. The alignments in suffix array (SA) coordinates are then generated with the 'aln' command. The resulting file contains ALL the alignments found by BWA. The 'samse/sampe' command converts SA coordinates to chromosomal coordinates. For single-end reads, most of computing time is spent on finding the SA coordinates (the aln command). For paired-end reads, half of computing time may be spent on pairing (the sampe command) given 32bp reads. Using longer reads would reduce the fraction of time spent on pairing because each end in a pair would be mapped to fewer places.
</p>

<h3>References:</h3>
<ul>
<li><a href="http://www.ncbi.nlm.nih.gov/pubmed/19451168">Li H. and Durbin R. (2009) Fast and accurate short read alignment with Burrows-Wheeler Transform. Bioinformatics, 25:1754-60.</a>
<li>If you use BWA-SW, please cite:<br/>
<a href="http://www.ncbi.nlm.nih.gov/pubmed/20080505">Li H. and Durbin R. (2010) Fast and accurate long-read alignment with Burrows-Wheeler Transform. Bioinformatics, Epub.</a>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
  <li><a href="http://bio-bwa.sourceforge.net/">bwa Main Site</a></li>
  <li>Type <tt><b>bwa</b></tt> at the prompt</li>
  <li>Type <tt><b>bwa <em>command</em></b></tt> at the prompt, e.g. <tt><b>bwa index</b></tt></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
  <li>Module Name: <tt>bwa</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
  <li>Multithreaded</li>
  <li>Example files in <tt><b>/fdb/app_testdata/fastq/</b></tt></li>
  <li>Reference data in:
    <ul>
      <li><tt><b>/fdb/bwa/</b></tt></li>
      <li><tt><b>/fdb/igenomes/<em>organism</em>/<em>source</em>/<em>build</em>/Sequence/BWAIndex/</b></tt>, where<br />
        <ul>
          <li><tt><b><em>organism</em></b></tt> is the specific organism of interest (Gallus_gallus, Rattus_norvegicus, etc.)</li>
          <li><tt><b><em>source</em></b></tt> is the source for the sequence (NCBI, Ensembl, UCSC)</li>
          <li><tt><b><em>build</em></b></tt> is the specific genome draft of interest (hg19, build37.2, GRCh37)</li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=8 --mem=24g</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ ml bwa
[user@cn3144 ~]$ bwa aln -t 8 /fdb/bwa/indexes/hg19.fa file.fastq &gt; file.sai

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. bwa.sh). For example:</p>

<pre class="term">
module load bwa
cd /data/$USER/dir
bwa index -a bwtsw file.csfasta
bwa aln -t $SLURM_CPUS_PER_TASK file.csfasta file.fastq &gt; file.sai
bwa samse file.csfasta file.sai file.fastq &gt; file.sam
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] bwa.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. bwa.swarm). For example:</p>

<pre class="term">
cd /data/$USER/dir1; bwa aln -t $SLURM_CPUS_PER_TASK file.csfasta file.fastq &gt; file.sai
cd /data/$USER/dir2; bwa aln -t $SLURM_CPUS_PER_TASK file.csfasta file.fastq &gt; file.sai
cd /data/$USER/dir3; bwa aln -t $SLURM_CPUS_PER_TASK file.csfasta file.fastq &gt; file.sai
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f bwa.swarm [-g #] [-t #] --module bwa</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module bwa</tt> <td>Loads the bwa module for each subjob in the swarm 
</table>


<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
