<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'cellranger-arc on Biowulf';</script>
<div class="title">cellranger-arc on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>From the Cell Ranger Arc manual:</p>

<blockquote> 
    Cell Ranger ARC is a set of analysis pipelines that process
    Chromium Single Cell Multiome ATAC + Gene Expression sequencing data to
    generate a variety of analyses pertaining to gene expression, chromatin
    accessibility and their linkage. Furthermore, since the ATAC and gene
    expression measurements are on the very same cell, we are able to perform
    analyses that link chromatin accessibility and gene expression.

<ul> 
    <li><em>cellranger-arc mkfastq</em> demultiplexes raw base call (BCL)
    files generated by Illumina sequencers into FASTQ files. It is a wrapper
    around Illumina's bcl2fastq, with additional useful features that are
    specific to 10x libraries and a simplified sample sheet format. The same
    command can be used to demultiplex both ATAC and GEX flow cells.</li>

    <li><em>cellranger-arc count</em> takes FASTQ files from cellranger-arc
    mkfastq and performs alignment, filtering, barcode counting, peak calling
    and counting of both ATAC and GEX molecules. Furthermore, it uses the
    Chromium cellular barcodes to generate feature-barcode matrices, perform
    dimensionality reduction, determine clusters, perform differential analysis
    on clusters and identify linkages between peaks and genes. The count
    pipeline can take input from multiple sequencing runs on the same GEM
    well.</li>
</ul>


These pipelines combine Chromium-specific algorithms with the widely used
RNA-seq aligner STAR. Output is delivered in standard BAM, MEX, CSV, HDF5, and HTML
formats that are augmented with cellular information and a .cloupe file for use
with the Loupe browser.
</blockquote>

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="https://support.10xgenomics.com/single-cell-multiome-atac-gex/software/pipelines/latest/what-is-cell-ranger-arc">Home page + Manual</a>
    </li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: cellranger-arc (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>cellranger can operate in <span style="background-color: #ffff99">local mode</span> or 
    <span style="background-color: #ffff99">cluster mode</span>. In both cases, the local part of the job will use
    multiple CPUs. Users have to specify the number of allocated CPUs and amount of memory
    with <code style="background-color: #ffff99">--localcores=# --localmem=#</code> to cellranger.</li>
    <li style="background-color: #ffff99">cellranger may attempt to start more processes or open more files than the default limits
    on our compute nodes allow. If you encounter errors or strange results, you may have to raise these limits.
    See below for more deails.</li>
    <li>Reference data can be found in  <code>$CELLRANGER_ARC_REF</code></li>
    <li>Test data can be found in  <code>$CELLRANGER_ARC_TEST_DATA</code></li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>

<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>

<p>Copy the bcl format test data and run the demux pipeline</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=6 --mem=35g</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load cellranger-arc</b>
[user@cn3144 ~]$ <b>cp ${CELLRANGER_ARC_TEST_DATA:-none}/* .</b>
[user@cn3144 ~]$ <b>tar -xzf cellranger-arc-tiny-bcl-atac-1.0.0.tar.gz</b>
[user@cn3144 ~]$ <b>tar -xzf cellranger-arc-tiny-bcl-gex-1.0.0.tar.gz</b>
[user@cn3144 ~]$ # demultiplex the ATAC flowcell
[user@cn3144 ~]$ <b>cellranger-arc mkfastq --id=tiny-bcl-atac \
                     --csv=cellranger-arc-tiny-bcl-atac-simple-1.0.0.csv \
                     --run=cellranger-arc-tiny-bcl-atac-1.0.0 \
                     --localcores=$SLURM_CPUS_PER_TASK \
                     --localmem=34</b>
cellranger-arc mkfastq (cellranger-arc-1.0.0)
Copyright (c) 2020 10x Genomics, Inc.  All rights reserved.
-------------------------------------------------------------------------------

Martian Runtime - v4.0.1
Serving UI at http://cn1038:38947?auth=OKNOP2vgDnOXFmUoe1dK2y4rCKFuNCkYs16KtaTMqfw

Running preflight checks (please wait)...
Checking run folder...
Checking RunInfo.xml...
...
Pipestance completed successfully!

2020-10-06 20:30:54 Shutting down.
Saving pipestance info to "tiny-bcl-atac/tiny-bcl-atac.mri.tgz"

[user@cn3144 ~]$ # demultiplex the GEX flowcell
[user@cn3144 ~]$ <b>cellranger-arc mkfastq --id=tiny-bcl-gex \
                     --csv=cellranger-arc-tiny-bcl-gex-simple-1.0.0.csv \
                     --run=cellranger-arc-tiny-bcl-gex-1.0.0 \
                     --localcores=$SLURM_CPUS_PER_TASK \
                     --localmem=34</b>
...

</pre>

<p style="background-color: #ffff99">Note that it is necessary to specify
<code>--localcores</code> and <code>--localmem</code>.</p>

<p>Cellranger Arc may start an unreasonable number of processes or open too many
files. If you encounter errors that include</p>

<pre class="term">
...
 self.pid = os.fork()
OSError: [Errno 11] Resource temporarily unavailable 
</pre>

<p>or see unexpected results despite specifying <code>--localcores</code> and
<code>--localmem</code>, you may have to raise the limit on the number of
processes and/or open files allowed in your batch script:</p>

<pre class="term">
[user@cn3144 ~]$ <b>ulimit -u 10240 -n 16384</b>
</pre>

<p>If running in slurm mode, it may be necessary to add <code>--jobinterval=3000</code> if encountering
errors mentioning empty batch scripts.</p>

<p>Generate counts per gene per cell</p>
<pre class="term">
[user@cn3144 ~]$ <b>cat > gex_atac.csv &lt;&lt;__EOF__</b>
fastqs,sample,library_type
tiny-bcl-gex/outs/fastq_path/,test_sample_gex,Gene Expression
tiny-bcl-atac/outs/fastq_path/,test_sample_atac,Chromatin Accessibility
<b>__EOF__</b>
[user@cn3144 ~]$ <b>cellranger-arc count \
                      --id=sample123 \
                      --reference=$CELLRANGER_ARC_REF/refdata-cellranger-arc-GRCh38-2020-A \
                      --libraries=gex_atac.csv \
                      --localcores=$SLURM_CPUS_PER_TASK \
                      --localmem=32</b>
### note: example data currently fails at this step. waiting for reply from 10x support
</pre>

<p>The same job could also be run in cluster mode where pipeline tasks
are submitted as batch jobs. This can be done by setting jobmode to slurm
and limiting the max. number of concurrent jobs:</p>

<pre class="term">
[user@cn3144 ~]$ <b>cellranger-arc count \
                      --jobmode=slurm --maxjobs=10 \
                      --id=sample123 \
                      --reference=$CELLRANGER_ARC_REF/refdata-cellranger-arc-GRCh38-2020-A \
                      --libraries=gex_atac.csv \
                      --localcores=$SLURM_CPUS_PER_TASK \
                      --localmem=32</b>
</pre>

<p>Don't forget to close the interactive session when done</p>
<pre class="term">
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>





<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. cellranger.sh), which uses the input file 'cellranger.in'. For example:</p>

<pre class="term">
#! /bin/bash
module load cellranger || exit 1
## uncomment the following line if encountering 'resource unavailable' errors
## despite using --localcores and --localmem
# ulimit -u 4096
cellranger-arc mkfastq --id=tiny-bcl-atac \
     --csv=cellranger-arc-tiny-bcl-atac-simple-1.0.0.csv \
     --run=cellranger-arc-tiny-bcl-atac-1.0.0 \
     --localcores=$SLURM_CPUS_PER_TASK \
     --localmem=34
cellranger-arc mkfastq --id=tiny-bcl-gex \
     --csv=cellranger-arc-tiny-bcl-gex-simple-1.0.0.csv \
     --run=cellranger-arc-tiny-bcl-gex-1.0.0 \
     --localcores=$SLURM_CPUS_PER_TASK \
     --localmem=34

cat > gex_atac.csv &lt;&lt;__EOF__
fastqs,sample,library_type
tiny-bcl-gex/outs/fastq_path/,test_sample_gex,Gene Expression
tiny-bcl-atac/outs/fastq_path/,test_sample_atac,Chromatin Accessibility
__EOF__
cellranger-arc count \
      --id=sample123 \
      --reference=$CELLRANGER_ARC_REF/refdata-cellranger-arc-GRCh38-2020-A \
      --libraries=gex_atac.csv \
      --localcores=6 \
      --localmem=32
</pre>
<p>Again, please remember to include <code>--localcoes</code> and <code>--localmem</code>.</p>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=12 --mem=35g cellranger.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. cellranger.swarm). For example:</p>

<pre class="term">
cellranger-arc mkfastq --run=./run1 --localcores=$SLURM_CPUS_PER_TASK --localmem=34
cellranger-arc mkfastq --run=./run2 --localcores=$SLURM_CPUS_PER_TASK --localmem=34
cellranger-arc mkfastq --run=./run3 --localcores=$SLURM_CPUS_PER_TASK --localmem=34
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f cellranger.swarm -g 35 -t 12 --module cellranger</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>-t # <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td>--module cellranger <td>Loads the cellranger module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
