<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type="text/javascript" language="JavaScript">document.querySelector('title').textContent = "checkm2";</script>

<div class="title">checkm2 on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
      </div>
</table>

<p>
Rapid assessment of genome bin quality using machine learning. From the documentation:
</p>
<blockquote>
    Unlike CheckM1, CheckM2 has universally trained machine learning models it
    applies regardless of taxonomic lineage to predict the completeness and
    contamination of genomic bins. This allows it to incorporate many lineages in
    its training set that have few - or even just one - high-quality genomic
    representatives, by putting it in the context of all other organisms in the
    training set. As a result of this machine learning framework, CheckM2 is also
    highly accurate on organisms with reduced genomes or unusual biology, such as
    the Nanoarchaeota or Patescibacteria.
</blockquote>

<!-- <h3>References:</h3>
<ul>

  <li>
    Blow J.
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/00000000"><b><u>A really amazing research paper.</u></b></a>
    <em>J Mol Biol. 2012 Jan 13;415(2):406-18.</em>
  </li>
  <li>
    Blow J., Doe J.
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/00000000"><b><u>A retread of another amazing research paper.</u></b></a>
    <em>J Struct Biol. 2012 Dec;180(3):519-30.</em>
  </li>

</ul>
-->

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li>CheckM2 on <a href="https://github.com/chklovski/CheckM2">GitHub</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>


<ul>
    <li>Module Name: <tt>checkm2</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>This application is multithreaded. Please match the number of allocated CPUs to the number of threads.</li>
    <li>Example files in <tt>$CHECKM2_TEST_DATA</tt></li>
    <li>CheckM2 will automatically use lscratch as a temp dir if lscratch has been allocated.</li>
    <li>Benchmarking suggests that CheckM2 may not scale efficiently to more than 16 CPUs</li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=12g --cpus-per-task=16 --gres=lscratch:20</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load checkm2</b>
[user@cn3144 ~]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144 ~]$ <b>mkdir tmp</b>
[user@cn3144 ~]$ <b>cp -Lr $CHECKM2_TEST_DATA/fasta .</b>
[user@cn3144 ~]$ <b>checkm2 predict --threads=$SLURM_CPUS_PER_TASK \
    --input ./fasta \
    --output-directory ./checkm2</b>
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<p>Timing and efficiency with different numbers of CPUs for the 36 genomic bins in the test
data set:</p>

<div style="width: 50%; margin: 0 auto;">
    <table class="horztable" style="text-align: right;">
        <tr>
            <th class="softBottom">CPUs</th>
            <th class="softBottom">Memory [GiB]</th>
            <th class="softBottom">Runtime [minutes]</th>
            <th class="softBottom">Est. efficiency</th>
        </tr>
        <tr>
            <td>2</td>
            <td>9</td>
            <td>29.9</td>
            <td>100%</td>
        </tr>
        <tr>
            <td>4</td>
            <td>9</td>
            <td>15.3</td>
            <td>98%</td>
        </tr>
        <tr>
            <td>8</td>
            <td>9</td>
            <td>10.5</td>
            <td>72%</td>
        </tr>
        <tr>
            <td>16</td>
            <td>9</td>
            <td>6.2</td>
            <td>60%</td>
        </tr>
    </table>
</div>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. checkm2.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load checkm2/1.0.2
# uses lscratch automatically as tmp dir
checkm2 predict \
    --threads=$SLURM_CPUS_PER_TASK \
    --input $CHECKM2_TEST_DATA/fasta \
    --output-directory test-checkm2.out
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=8 --mem=12g --gres=lscratch:10 checkm2.sh</pre>


<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
