<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Cnvkit on HPC';</script>
    <div class="title">Cnvkit on HPC</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</td></tr></table>    <p> CNVkit is a Python library and command-line software toolkit to infer 
      and visualize copy number from targeted DNA sequencing data. It is designed 
      for use with hybrid capture, including both whole-exome and custom target 
      panels, and short-read sequencing platforms such as Illumina and Ion Torrent. 
      <a Name="doc"></a></p>

<div class="heading">Documentation</div>    <ul>
      <li><a href="http://cnvkit.readthedocs.io/en/stable/quickstart.html">http://cnvkit.readthedocs.io/en/stable/quickstart.html</a></li>
    </ul>

<a Name="notes"></a>
    <div class="heading"><a name="notes" id="notes"></a>Important Notes</div>    
    <ul>
      <li>Module Name: <tt>cnvkit</tt> (see <a href="/apps/modules.html">the modules 
        page</a> for more information) </li>
      <li> Example files in /usr/local/apps/cnvkit/cnvkit-examples. </li>
	  To test cnvkit with the example files:
  <pre class="term">
  $ cp -r /usr/local/apps/cnvkit/cnvkit-examples /data/$USER
  $ cd /data/$USER/cnvkit-examples
  $ sinteractive --mem=5g
  $ module load cnvkit
  $ make
  </pre>
      <li>Reference data in 
	    <pre class="term">/fdb/igenomes/Homo_sapiens/UCSC/hg19/Sequence/WholeGenomeFasta/genome.fa
/fdb/igenomes/Mus_musculus/UCSC/mm10/Sequence/WholeGenomeFasta/genome.fa</pre>
      </li>
	  <li>Sequencing-accessible regions files are under</li>
	  <pre class="term">/usr/local/apps/cnvkit/data
	  </pre>
    </ul>
<P>

<a Name="int"></a></p><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>    
    <pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <strong>module load cnvkit
[user@cn3144 ~]$ cnvkit.py autobin input.bam</strong>
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a></p><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
    <p>Create a batch input file (e.g. cnvkit.sh). For example:</p>

    <pre class="term">
#!/bin/bash
set -e
module load cnvkit
cnvkit.py autobin input.bam</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

    <pre class="term">sbatch cnvkit.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

    <p>Create a swarmfile (e.g. cnvkit.swarm). For example:</p>

<pre class="term">
cd dir1; cnvkit.py autobin input.bam
cd dir2; cnvkit.py autobin input.bam
cd dir3; cnvkit.py autobin input.bam
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

    <pre class="term">swarm -f cnvkit.swarm [-g #] --module TEMPLATE</pre>
where
    <table width="55%" border=0>
      <tr>
        <td width=12%><tt>-g <i>#</i> </tt></td>
        <td width="88%">Number of Gigabytes of memory required for each process 
          (1 line in the swarm command file) </td>
      </tr>
      <tr> 
        <td><tt>--module </tt></td>
        <td>Loads the module for each subjob in the swarm </td>
      </tr>
    </table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
