<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'colabfold on Biowulf';</script>
<style>
    .hl { background-color: #ffff99; }
    code {background-color: #eeeeee; border: 1px solid #bbbbbb; padding: 0;}
    dt {font-weight: bold; margin-top: 5px;}
    dd {padding-left: 2px; border-left: 1px solid #bbbbbb;}
    .btt {border: 1px solid silver;
          background-color: white;
          padding: 5px;
          position: relative;
          margin: 5px 0px 10px 10px;
          float: right;
          top: -25px;
          left: 10px;
          }
</style>
<div class="title">colabfold on Biowulf</div>

<table width=25% align=right>
    <tr>
        <td>
            <div class="toc">
                <div class="tocHeading">Quick Links</div>
                <div class="tocItem"><A href="#doc">Documentation</a></div>
                <div class="tocItem"><a href="#notes">Notes</a></div>
                <div class="tocItem"><a href="#int">Interactive job </a></div>
                <div class="tocItem"><a href="#sbatch">Batch job </a></div>
            </div>
        </td>
    </tr>
</table>

<p>
This module provides the batch scripts of the ColabFold implementation of alphafold.
</p>

<h3>References:</h3>
<ul>
    <li>Mirdita M, Sch√ºtze K, Moriwaki Y, Heo L, Ovchinnikov S and Steinegger M. 
        <em>ColabFold: Making protein folding accessible to all.</em>
        Nature Methods (2022) doi: 10.1038/s41592-022-01488-1. 
        <a href="https://pubmed.ncbi.nlm.nih.gov/35637307/">PubMed</a>&nbsp;|&nbsp;
        <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9184281/">PMC</a>&nbsp;|&nbsp;
        <a href="https://www.nature.com/articles/s41592-022-01488-1">Journal</a>
    </li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li>colabfold on <a href="https://github.com/sokrypton/ColabFold">GitHub</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: colabfold (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>The MSA generation is a multithreaded process. Model generation uses GPU</li>
    <li>Example files in <code>$COLABFOLD_TEST_DATA</code></li>
    <li>Reference data in <code>/fdb/colabfold/</code></li>
    <li>Currently including templates is not yet supported</li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>

<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> for
the generation of the multiple sequence applications (MSAs). This session
requires <span class="hl">at least 128GB of memory</span>. Note that
<em>colabfold_search is optimized for running many query sequences</em> in a
single job. <span class="hl">Never run single sequences in the search part of
the analysis. It is inefficient and if you run many will strain the file
system</span>. For example: Below we will create alignments for 17 polymerase
related proteins of the mimivirus genome which takes about 100 minutes.
Creating MSAs for all 979 proteins of the mimivirus genome take takes about 285
minutes or only 2.8x longer for 50x more proteins.

Note that colabfold_search
treats each protein in a fasta file as a separate monomer.</p>

<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=128G --cpus-per-task=16 --gres=lscratch:100</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144]$ <b>module load colabfold</b>
[user@cn3144]$ <b>cp $COLABFOLD_TEST_DATA/mimi_poly.fa .</b>
[user@cn3144]$ <b>colabfold_search --threads $SLURM_CPUS_PER_TASK \
    mimi_poly.fa $COLABFOLD_DB mimi_poly</b>
[...much output...]
[user@cn3144]$ <b>ls -lh mimi_poly</b>
total 66M
-rw-r--r-- 1 user group  17M Sep 22 16:14 0.a3m
-rw-r--r-- 1 user group 2.7M Sep 22 16:14 10.a3m
-rw-r--r-- 1 user group  92K Sep 22 16:14 11.a3m
-rw-r--r-- 1 user group 296K Sep 22 16:14 12.a3m
-rw-r--r-- 1 user group  33K Sep 22 16:14 13.a3m
-rw-r--r-- 1 user group 6.4M Sep 22 16:14 14.a3m
-rw-r--r-- 1 user group  42K Sep 22 16:14 15.a3m
-rw-r--r-- 1 user group 7.1K Sep 22 16:14 16.a3m
-rw-r--r-- 1 user group  23M Sep 22 16:14 1.a3m
-rw-r--r-- 1 user group  12M Sep 22 16:14 2.a3m
-rw-r--r-- 1 user group 797K Sep 22 16:14 3.a3m
-rw-r--r-- 1 user group 200K Sep 22 16:14 4.a3m
-rw-r--r-- 1 user group 580K Sep 22 16:14 5.a3m
-rw-r--r-- 1 user group 273K Sep 22 16:14 6.a3m
-rw-r--r-- 1 user group 160K Sep 22 16:14 7.a3m
-rw-r--r-- 1 user group 464K Sep 22 16:14 8.a3m
-rw-r--r-- 1 user group 2.6M Sep 22 16:14 9.a3m
[user@cn3144]$ <b>exit</b>
</pre>

<p>Now we can build the structure predictions for the 17 proteins above. This step requires
GPU. Each of the 17 proteins in this example is modeled as a monomer. See below for
multimer predictions. With the default settings on a A100 model generation for the 17
proteins takes approximately 4h</p>


<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=48G --cpus-per-task=8 --gres=lscratch:100,gpu:a100:1</b>
salloc.exe: Pending job allocation 46116227
salloc.exe: job 46116227 queued and waiting for resources
salloc.exe: job 46116227 has been allocated resources
salloc.exe: Granted job allocation 46116227
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job
[user@cn2144]$ <b>module load colabfold</b>
[user@cn2144]$ <b>colabfold_batch --amber --use-gpu-relax \
     mimi_poly mimi_poly_models</b>
2022-09-26 09:54:12,026 Running colabfold 1.3.0 (22671664ac2c9dcb30086c3e654414d950ccb297)
2022-09-26 09:54:19,458 Found 6 citations for tools or databases
2022-09-26 09:54:24,671 Query 1/17: 12 (length 73)
2022-09-26 09:54:24,720 Running model_3
2022-09-26 09:54:50,065 model_3 took 23.3s (3 recycles) with pLDDT 86.1 and ptmscore 0.622
2022-09-26 09:55:15,397 Relaxation took 15.9s
2022-09-26 09:55:15,400 Running model_4
2022-09-26 09:55:24,884 model_4 took 8.6s (3 recycles) with pLDDT 89 and ptmscore 0.668
2022-09-26 09:55:31,969 Relaxation took 5.7s
[...snip...]
[user@cn2144]$ <b>ls -lh mimi_poly_models | head</b>
total 570M
-rw-r--r-- 1 user group  17M Sep 26 15:36 0.a3m
-rw-r--r-- 1 user group 142K Sep 26 16:19 0_coverage.png
-rw-r--r-- 1 user group    0 Sep 26 16:19 0.done.txt
-rw-r--r-- 1 user group 743K Sep 26 16:19 0_PAE.png
-rw-r--r-- 1 user group 196K Sep 26 16:19 0_plddt.png
-rw-r--r-- 1 user group  18M Sep 26 16:19 0_predicted_aligned_error_v1.json
-rw-r--r-- 1 user group 1.5M Sep 26 16:19 0_relaxed_rank_1_model_3.pdb
-rw-r--r-- 1 user group 1.5M Sep 26 16:19 0_relaxed_rank_2_model_4.pdb
-rw-r--r-- 1 user group 1.5M Sep 26 16:19 0_relaxed_rank_3_model_2.pdb
[user@cn2144]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf]$
</pre>

<p>For each protein the output includes some diagnostic plots, the predicted alignment error
of the ranked models in json, as well as the ranked relaxed and unrelaxed models in PDB format.
As an example below see the predicted best ranked model (rank 1) for YP_003986740 (DNA dependent
RNA polymerase subunit B) along with some of the diagnostic plots.</p>


<div style="width:60% margin:0 auto;">
<figure>
    <img src="/images/colabfold_fig1.png" style="display: block; margin: 0 auto;" width="600"
        alt="colabfold predicted structure for YP_003986740">

    <figcaption><strong>Figure 1</strong>. <strong>(A)</strong> The highest confidence
        structure prediction (<code>0_relaxed_rank_1_model_3.pdb</code>) of protein
        YP_003986740 colored by pLDDT. <strong>(B)</strong> pLDDT for all 5 models.
        <strong>(C)</strong> Coverage of YP_003986740 in the MSA generated by colabfold_search
        with mmseqs2. <strong>(D)</strong> Predicted aligned error (PAE) for all 5 models.
</figure>
</div>

<p>Commandline for generating two predictions for each of the 5 models per protein but only
relax the top 5 results</p>

<pre class="term">
[user@cn2144]$ <b>colabfold_batch --amber --use-gpu-relax --num-relax 5 --num-seeds 2 ...</b>
</pre>



<h4>Predicting multimers with colabfold batch tools</h4>

<p>Multimers are predicted by concatenating all sequences of a multimer separated by ':' into a
single sequence</p>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. colabfold-search.sh) to create the MSAs:</p>

<pre class="term">
#!/bin/bash
module load colabfold
cp $COLABFOLD_TEST_DATA/mimi_poly.fa .
colabfold_search --threads $SLURM_CPUS_PER_TASK \
    mimi_poly.fa $COLABFOLD_DB mimi_poly
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=16 --mem=128g --gres=lscratch:100 colabfold-search.sh</pre>

<p>Then build the corresponding models</p>

<pre class="term">
#!/bin/bash
module load colabfold
colabfold_batch --amber --use-gpu-relax mimi_poly mimi_poly_models
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=8 --mem=48g --gres=lscratch:100,gpu:a100:1 --partition=gpu colabfold-batch.sh</pre>





<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
