<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'EDD on Biowulf';</script>
<div class="title">EDD on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
EDD is a ChIP-seq peak caller for detection of megabase domains of enrichment.
</p>

<h3>References:</h3>
<ul>
<li>
  E Lund, AR Oldenburg, and P Collas
  <a href="https://academic.oup.com/nar/article/42/11/e92/1432049"
  <b><u>Enriched domain detector: a program for detection of wide genomic enrichment domains robust against local variations</b></u>
  </a>
  <em>Nucl. Acids Res. 42 (2014)</em>
</li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/CollasLab/edd">EDD on GitHub</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
  <li>Module Name: <tt>edd</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
  <li>Multithreaded
  <li>Environment variables set:
    <ul>
      <li><tt><b>EDD_EXAMPLES</b></tt></li>
    </ul>
  </li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=8</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load edd</b>
[user@cn3144 ~]$ <b>ln -s $EDD_EXAMPLES/hg19.sizes .</b>
[user@cn3144 ~]$ <b>ln -s $EDD_EXAMPLES/*.bam .</b>
[user@cn3144 ~]$ <b>touch unalign.bed</b>
[user@cn3144 ~]$ <b>edd -p $SLURM_CPUS_PER_TASK -n 10000 hg19.sizes unalign.bed SRX447385.bam SRX447386.bam output_dir</b>

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. edd.sh). For example:</p>

<pre class="term">
#!/bin/bash

# Generate hg19 chrom sizes
module load ucsc
fetchChromSizes hg19 &gt; hg19.sizes
module unload ucsc

# Create empty file for unalignable regions
touch unalign_empty

# Run the command, collecing output in output directory
ml edd
edd -p $SLURM_CPUS_PER_TASK -n 10000 hg19.sizes unalign.bed chip_seq.bam input.bam output_dir
</pre>

<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] edd.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. edd.swarm). For example:</p>

<pre class="term">
edd -p $SLURM_CPUS_PER_TASK hg19.sizes unalign.bed chip_seq_1.bam input_1.bam output_dir_1
edd -p $SLURM_CPUS_PER_TASK hg19.sizes unalign.bed chip_seq_2.bam input_2.bam output_dir_2
edd -p $SLURM_CPUS_PER_TASK hg19.sizes unalign.bed chip_seq_3.bam input_3.bam output_dir_3
edd -p $SLURM_CPUS_PER_TASK hg19.sizes unalign.bed chip_seq_4.bam input_4.bam output_dir_4
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f edd.swarm [-g #] [-t #] --module edd</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module edd</tt> <td>Loads the edd module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
