<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = ' Epic2 on Biowulf';</script>
<div class="title"> Epic2 on Biowulf</div>
<table width=25% align=right> <tr><td>
<div class="toc">
<div class="tocHeading">Quick Links</div>
<div class="tocItem"><a href="#int">Interactive Jobs</a></div>
<div class="tocItem"><a href="#batch">Batch job on Biowulf</a></div>
<div class="tocItem"><a href="#swarm">Using Swarm</a></div>
<div class="tocItem"><A href="#doc">Documentation</a></div>
</div></table>

<p>epic2 is an ultraperformant reimplementation of SICER. It focuses on speed, low memory overhead and ease of use.

It also contains a reimplementation of the SICER-df scripts for differential enrichment and a script to create many kinds of bigwigs from your data.
<P><b>Features:</b>
<ul>
    <li>easy to install and use</li>
    <li>reads sam, single-end bam, bed and bedpe (.gz)</li>
    <li>extremely fast</li>
    <li>very low memory requirements</li>
    <li>works both with and without input</li>
    <li>metadata for ~80 UCSC genomes built in</li>
    <li>easily use custom genomes and assemblies with --chromsizes and --effective-genome-fraction args</li>
    <li>differential enrichment for WT vs. KO works with and without input (epic2-df)</li>
    <li>fixes two bugs in the original SICER and one bug in epic</li>
    <li>create many types of useful bigwigs for visualization in genome browsers</li>
</ul> 


<a Name="doc"></a><div class="heading">Documentation</div>
<p><a href="https://github.com/biocore-ntnu/epic2">https://github.com/biocore-ntnu/epic2</a></p>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>epic2</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
<li>Example files are under <tt>/usr/local/apps/epic2/examples</tt></li>
</ul>

<a Name="batch"></a><div class="heading">Submitting an interactive job</div>
<P>
Allocate an interactive session  and run the interactive job there.</p>
<pre class="term">[biowulf]$ <b>sinteractive  --mem=5g</b>
salloc.exe: Granted job allocation 789523
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn0135 are ready for job

[cn0135]$ <b>cd /data/$USER/</b>

[cn0135]$ <b>module load epic2</b>

[cn0135]$ <b>epic2 -ex</b>
# An example of command can be copied from the output.
[cn0135]$ <b>epic2 -t /opt/conda/envs/app/lib/python3.7/site-packages/epic2/examples/test.bed.gz \
-c /opt/conda/envs/app/lib/python3.7/site-packages/epic2/examples/control.bed.gz > deleteme.txt</b>

[cn0135]$ <b>exit</b>
salloc.exe: Job allocation 789523 has been revoked.

[biowulf]$</pre>
<p>

<a Name="batch"></a><div class="heading">Submitting a single batch job</div>
    
<p>1. Create a script file (myscript) similar to the one below
<pre>#! /bin/bash
# myscript
set -e

module load epic2 || exit 1
cd /data/$USER/
epic2 -t /opt/conda/envs/app/lib/python3.7/site-packages/epic2/examples/test.bed.gz \
-c /opt/conda/envs/app/lib/python3.7/site-packages/epic2/examples/control.bed.gz > deleteme.txt</pre>
  <p>2. Submit the script on biowulf:   
      
  <pre class="term">[biowulf]$ sbatch --mem=5g myscript</pre>

<p>

<a Name="swarm"></a><div class="heading">Using Swarm</div>
<P>
Using the 'swarm' utility, one can submit many jobs to the cluster to run concurrently. </p>

  <p>Set up a swarm command file (eg /data/$USER/cmdfile). </p>

  <pre class="term">
cd /data/$USER/dir1; epic2 ...
cd /data/$USER/dir2; epic2 ...
cd /data/$USER/dir3; epic2 ...
...
cd /data/$USER/dir20; epic2 ...
</pre>
<P>

<p> submit the swarm job:
  <pre class="term">$ swarm -f cmdfile --module epic2 -g 5</pre>

<p>For more information regarding running swarm, see <a href="swarm.html">swarm.html</a></p>


</p>


<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
