<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'FastQC on Biowulf';</script>
    <div class="title">FastQC on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>    <p> FastQC aims to provide a simple way to do some quality control checks on raw 
      sequence data coming from high throughput sequencing pipelines. It provides 
      a modular set of analyses which you can use to give a quick impression of 
      whether your data has any problems of which you should be aware before doing 
      any further analysis.</p>
    <p>The main functions of FastQC are</p>
    <p>- Import of data from BAM, SAM or FastQ files (any variant)<br />
      - Providing a quick overview to tell you in which areas there may be problems<br />
      - Summary graphs and tables to quickly assess your data<br />
      - Export of results to an HTML based permanent report<br />
      - Offline operation to allow automated generation of reports without running 
      the interactive application<br />
    </p>
    <h3>References:</h3>    <ul>
      <li>FastQC is developed by <a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">Simon 
        Andrews, Babraham Bioinformatics</a>.</li>
    </ul>


<a Name="doc"></a><div class="heading">Documentation</div>    <ul>
      <li><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">http://www.bioinformatics.babraham.ac.uk/projects/fastqc/</a></li>
    </ul>

<div class="heading">Important Notes</div>    <ul>
      <li>Module Name: <tt>fastqc</tt> (see <a href="/apps/modules.html">the modules 
        page</a> for more information) 
    </ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
    <pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <strong>module load fastqc</strong>
[user@cn3144 ~]$ <strong>fastqc -o output_dir [-f fastq|bam|sam] -c contaminant_file seqfile1 .. seqfileN</strong>

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>    <p>Create a batch input file (e.g. fastqc.sh). For example:</p>    
    <pre class="term">
#!/bin/bash
set -e
module load fastqc
fastqc -o output_dir [-f fastq|bam|sam] -c contaminant_file seqfile1 .. seqfileN</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

    <pre class="term">sbatch --mem=10g fastqc.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

    <p>Create a swarmfile (e.g. fastqc.swarm). For example:</p>

    <pre class="term">cd dir1;fastqc -o output_dir [-f fastq|bam|sam] -c contaminant_file seqfile1 .. seqfileN
cd dir2;fastqc -o output_dir [-f fastq|bam|sam] -c contaminant_file seqfile1 .. seqfileN
cd dir3;fastqc -o output_dir [-f fastq|bam|sam] -c contaminant_file seqfile1 .. seqfileN
cd dir4;fastqc -o output_dir [-f fastq|bam|sam] -c contaminant_file seqfile1 .. seqfileN
cd dir5;fastqc -o output_dir [-f fastq|bam|sam] -c contaminant_file seqfile1 .. seqfileN
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

    <pre class="term">swarm -f fastqc.swarm -g 10 --module fastqc</pre>
where
    <table border=0>
      <tr>
        <td width=20%><tt>-g <i>#</i> </tt>
        <td>Number of Gigabytes of memory required for each process (1 line in 
          the swarm command file) 
      <tr>
        <td><tt>--module fastqc</tt> 
        <td>Loads the fastqc module for each subjob in the swarm 
    </table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
