<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'freebayes on Biowulf';</script>
<div class="title">freebayes on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<blockquote><p> FreeBayes is a Bayesian genetic variant detector designed to find small
polymorphisms, specifically SNPs (single-nucleotide polymorphisms), indels
(insertions and deletions), MNPs (multi-nucleotide polymorphisms), and complex
events (composite insertion and substitution events) smaller than the length of
a short-read sequencing alignment.</p>

<p>FreeBayes is haplotype-based, in the sense that it calls variants based on
the literal sequences of reads aligned to a particular target, not their
precise alignment. This model is a straightforward generalization of previous
ones (e.g. PolyBayes, samtools, GATK) which detect or report variants based on
alignments. This method avoids one of the core problems with alignment-based
variant detection--- that identical sequences may have multiple possible
alignments </p>
</blockquote>

<h3>References:</h3>
<ul>
    <li>Erik Garrison, Gabor Marth. <em>Haplotype-based variant detection from short-read sequencing</em>.
    2012, <a href="https://arxiv.org/abs/1207.3907">arXiv:1207.3907</a></li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li>freebayes <a href="https://github.com/ekg/freebayes">GitHub repo</a>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: freebayes (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>freebayes can be parallelized with <code>freebayes-parallel</code></li>
    <li>Example data in <code>$FREEBAYES_TEST_DATA</code></li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=10g</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load freebayes samtools</b>
[user@cn3144 ~]$ <b>freebayes \
    --fasta-reference /fdb/igenomes/Homo_sapiens/UCSC/hg19/Sequence/WholeGenomeFasta/genome.fa \
    --min-alternate-count 2 \
    --min-alternate-qsum 40 \
    --pvar 0.0001 \
    --use-mapping-quality \
    --site-selection-max-iterations 3 \
    --genotyping-max-iterations 25 \
    /fdb/app_testdata/bam/hg19/gcat_set_053.bam \
  | bgzip -c \
  &gt; test.vcf.gz</b>

# This takes approximately 72 minutes.

[user@cn3144 ~]$ <b>bgzip -r test.vcf.gz</b>
[user@cn3144 ~]$ <b>ls -lh</b>
-rw-r--r-- 1 user group 30M Feb 12 14:14 test.vcf.gz
-rw-r--r-- 1 user group 48K Feb 12 14:17 test.vcf.gz.gzi
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. freebayes.sh), which uses the input file 'freebayes.in'. For example:</p>

<pre class="term">
#! /bin/bash

module load freebayes/1.1.0 samtools || exit 1
freebayes --fasta-reference /fdb/igenomes/Homo_sapiens/UCSC/hg19/Sequence/WholeGenomeFasta/genome.fa \
    --min-alternate-count 2 \
    --min-alternate-qsum 40 \
    --pvar 0.0001 \
    --use-mapping-quality \
    --site-selection-max-iterations 3 \
    --genotyping-max-iterations 25 \
    secret_sample_1.bam \
  | bgzip -c \
  &gt; secret_sample_1.vcf.gz
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=2 --mem=10g freebayes.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. freebayes.swarm). For example:</p>

<pre class="term">
cd /data/$USER/dir1; freebayes --fasta-reference ../fa/h.sapiens.fasta sample1.bam | bgzip -c &gt; sample1.vcf.gz
cd /data/$USER/dir1; freebayes --fasta-reference ../fa/h.sapiens.fasta sample2.bam | bgzip -c &gt; sample2.vcf.gz
cd /data/$USER/dir1; freebayes --fasta-reference ../fa/h.sapiens.fasta sample3.bam | bgzip -c &gt; sample3.vcf.gz
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f freebayes.swarm -g 10 --module freebayes/1.1.0,samtools</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>-t # <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td>--module freebayes <td>Loads the freebayes module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
