<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'GATK on Biowulf';</script>
<!-- ======================================================================== -->
    
<div class="title">GATK on Biowulf</div>

<!-- ======================================================================= -->
<div class="heading">Description</div>

<p>The Genome Analysis Toolkit (GATK) is a software package developed at the
Broad Institute to analyze high-throughput sequencing data. The toolkit
includes a wide variety of tools, with a focus on variant discovery and
genotyping as well as emphasis on data quality assurance.</p>

<p><Hr><P>
<a href="https://hpc.nih.gov/training/gatk_tutorial"><h3>Online Tutorial: A practical introduction to GATK 4 on Biowulf</h3></a>
<p><font size=+1><b>New in May 2021:</b></font> A <a href="https://hpc.nih.gov/training/gatk_tutorial">self-paced, online tutorial</a> to work through 
a GATK example on Biowulf. 
Developed by the Biowulf staff, this tutorial includes a case study of germline variant discovery with WGS data from a trio,  and benchmarks for each step. By working through the tutorial, you will learn NGS data preprocessing and how to optimize your Biowulf batch jobs. 
</p><Hr><P>

<h3>Notes</h3>
<ul>
    <li>With the release of GATK 3.5-0 MuTect2 and ContEst are
    included as part of GATK.</li>
    <li>GATK 4 is significantly different from GATK 3. Both are documented
    separately below.</li>
    <li>GATK 4 is the default module as of April 2021.</li>
    <li>In most cases, GATK jobs can run successfully on norm partition, there is no need/benefit to run GATK on largemem partition.</li>
</ul>

<p>There are multiple versions of GATK available. An easy way of selecting the
version is to use <a href="/apps/modules.html">modules</a>.  To see the modules
available, type</p>
<pre class="term">
    module avail GATK 
</pre>

<p>To select a module use</p>
<pre class="term">
    module load GATK/[version]
</pre>
<p>where <code>[version]</code> is the version of choice.</p>

<h3>Documentation</h3>

<ul>
    <li><a href="https://www.broadinstitute.org/gatk/">Home page</a></li>
    <li><a href="https://www.broadinstitute.org/gatk/guide/">Guide</a></li>
    <li><a href="https://www.broadinstitute.org/gatk/guide/tooldocs/">Tool Documentation</a></li>
    <li><a href="http://gatkforums.broadinstitute.org/">Forum</a></li>
    <li><a href="https://www.broadinstitute.org/gatk/blog">Blog</a></li>
</ul>

<h3>Release notes</h3>
<ul>
    <li><a href="http://gatkforums.broadinstitute.org/discussion/4739/release-notes-for-gatk-version-3-3">3.3 Release notes</a></li>
    <li><a href="http://gatkforums.broadinstitute.org/discussion/5562/release-notes-for-gatk-version-3-4">3.4 Release notes</a> </li>
    <li><a href="http://gatkforums.broadinstitute.org/discussion/6494/release-notes-for-gatk-version-3-5">3.5 Release notes</a> </li>
    <li><a href="http://gatkforums.broadinstitute.org/wdl/discussion/7711/release-notes-for-gatk-version-3-6">3.6 Release notes</a></li>
    <li><a href="http://gatkforums.broadinstitute.org/gatk/discussion/8691/release-notes-for-gatk-version-3-7">3.7 Release notes</a></li>
    <li><a href="https://gatkforums.broadinstitute.org/gatk/discussion/10062/release-notes-for-gatk-version-3-8">3.8 Release notes</a></li>
    <li>Release notes for GATK 4 can be found on the <a href="https://github.com/broadinstitute/gatk/releases">GitHub page</a></li>

</ul>



<h3>References</h3>
<ul>
    <li>Aaron McKenna et al. <em>The Genome Analysis Toolkit: a MapReduce framework for
        analyzing next-generation DNA sequencing data.</em> Genome Research 2010,
    20:1297-1303.
    <a href="http://www.ncbi.nlm.nih.gov/pubmed/20644199">Pubmed</a>&nbsp;|&nbsp;
    <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2928508/">PMC</a>&nbsp;|&nbsp;
    <a href="http://genome.cshlp.org/content/20/9/1297">Journal</a></li>
    <li>Mark A. DePristo et al. <em>A framework for variation discovery and genotyping
        using next-generation DNA sequencing data.</em> Nature Genetics 2011, 43:491-498.
    <a href="http://www.ncbi.nlm.nih.gov/pubmed/21478889">Pubmed</a>&nbsp;|&nbsp;
    <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3083463/">PMC</a>&nbsp;|&nbsp;
    <a href="https://www.nature.com/articles/ng.806">Journal</a></li>
    <li>Geraldine A. Van der Auwera et al. <em>From FastQ Data to High-Confidence
        Variant Calls: The Genome Analysis Toolkit Best Practices Pipeline.</em>
    Current Protocols In Bioinformatics 2013, 11:11.10:11.10.1â€“11.10.33.
    <a href="http://www.ncbi.nlm.nih.gov/pubmed/25431634">Pubmed</a>&nbsp;|&nbsp;
    <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4243306/">PMC</a>&nbsp;|&nbsp;
    <a href="http://onlinelibrary.wiley.com/doi/10.1002/0471250953.bi1110s43/abstract;jsessionid=2E0A8F753564C2126232257E21B17766.f04t01">Journal</a></li>
    <li>Poplin et al. <em>Scaling accurate genetic variant discovery to tens of thousands of samples</em>
    bioRxiv,2017
    <a href="https://doi.org/10.1101/201178">bioRxiv</a>&nbsp;|&nbsp;
    </li>
</ul>
        
<!-- ======================================================================== -->

<div class="heading">Resource bundles</div>

<p>GATK
<a href="http://gatkforums.broadinstitute.org/discussion/1213/whats-in-the-resource-bundle-and-how-can-i-get-it">Resource</a>
<a href="https://www.broadinstitute.org/gatk/guide/article.php?id=1213">bundles</a>
contain reference genome assembles (.fa, .fai, .dict), variation data (dbSNP,
HapMap, 1000 genomes, ...), and some genotype calls for a gold standard
genome.</p>

<p>Resource bundles are available at</p>

<p><b><code>/fdb/GATK_resource_bundle</code></b></p>


<div class="tabs-container">
    <ul class="tabs">
        <li data-tab="tab-3">GATK 3</li>
        <li class="current" data-tab="tab-4">GATK 4</li>
    </ul>
    <!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
    <!--                                 GATK 3                                  -->
    <!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
    <div id="tab-3" class="tab-content">

        <table width=25% align=right> <tr><td>
            <div class="toc">
            <div class="tocHeading">Quick Links</div>
            <div class="tocItem"><a href="#overview3">Overview</a></div>
            <div class="tocItem"><a href="#serial3">Batch job on Biowulf</a></div>
            <div class="tocItem"><a href="#swarm3">Swarm of jobs</a></div>
            <div class="tocItem"><a href="#int3">Interactive job on Biowulf</a></div>
        </div></table>


        <!-- ======================================================================== -->

        <a class="navbar-safe-anchor"  name="overview3"></a>
        <div class="heading">Overview</div>

        <h3>Environment module</h3>
        <p>The environment module for GATK (<code>module load GATK</code> or
        <code>module load GATK/version</code>)</p>

        <ul>
            <li>loads R as a dependency</li>
            <li>sets the environment variables 
                <ul>
                    <li> <code>$GATK_HOME</code></li>
                    <li> <code>$GATK_JAR</code></li>
                    <li> <code>$GATK_JARPATH</code></li>
                    <li> <code>$GATK_KEY</code></li>
                    <li> <code>$GATK_QUEUE_JAR</code></li>
                    <li> <code>$GATK_TEST_DATA</code></li>
                </ul>
            </li>
            <li>adds a GATK wrapper script developed in house and utilities for lifting
            over VFC files between genome builds</li>
        </ul>

        <h3>Wrapper</h3>
        <p>The GATK wrapper script provides a simplified interface to the GATK tools:</p>

<pre class="term">
usage: GATK [options] [-m MEM] tool [gatk_tool_args]

This GATK wrapper script invokes GATK tools with a more friendly interface.
For documentation on each of the tools and their arguments see
https://www.broadinstitute.org/gatk/guide/tooldocs/

positional arguments:
  tool               GATK tool. Use 'LIST' or 'list' to get the current list
                     of available tools

optional arguments:
  -h, --help         show this help message and exit
  -m MEM, --mem MEM  maximal heap memory allowed for the Java process.
                     *Format*: &lt;number&gt;[kmg]. *Examples*: 1024k, 1000m, 2g,
                     32g. Passed to Java with -Xmx (default: 2g)
  -p N, --pgct N     maximal number of parallel GC threads allowed for Java
                     VM. Defaults to 1 less than the number of allocated CPUs.
                     (default: -1)
  -n, --dry-run      show command that would have been executed and set up
                     temp dir (default: False)
</pre>

        <p>It takes care of finding the GATK jar and writing temporary files to either</p>
        <ul>
            <li><code>/lscratch/${SLURM_JOB_ID}</code>: if running on a compute node with
            allocated local disk space</li>
            <li><code>./unique_tmp_dir</code>: if running on a compute 
            node without allocated local disk space</li>
        </ul>

        <p class="alert">Whenever possible, please use lscratch for temporary files. It will be
        more performant and reduce strain on the shared filesystems.</p>
    
        <h3>Calling GATK directly</h3>
        <p>When not using the <code>GATK</code> wrapper script, the basic GATK command
        should be </p>

<pre class="term">
java -Xmx####g -Djava.io.tmpdir=/lscratch/${SLURM_JOB_ID} \
  -XX:ParallelGCThreads=## -jar $GATK_JAR -T &lt;Toolname&gt; [options]
</pre>

<p>which requires requesting lscratch space at job submission.</p>

        <p>Note that <code>/lscratch</code> is the node-local scratch space. See the
        <a href="/docs/b2-userguide.html#local">user guide</a> for more information
        about allocating local scratch space.</p>

        <p>Some tools may perform better with non-default garbage collectors.</p>

        <h4>Memory</h4>
        <p>The <code>-Xmx</code> option to Java specifies the maximal size of the
        memory allocation pool. It accepts suffixes <code>k</code>, <code>m</code>, and
        <code>g</code>. This is the option used by the wrapper script to limit memory
        usage.</p>

        <p>Note that you should request additional memory (1-2GB) for batch/interactive jobs on
        top of the memory given to java to ensure that the whole java process stays
        below the allocated memory limit.</p>

        <h4>Redirect temporary files to <tt>/lscratch/${SLURM_JOB_ID}</tt></h4>

        <p>GATK writes some temporary files during its run. By default, Java writes
        temp files to <code>/tmp</code>. However this is not advisable because
        <code>/tmp</code> is used by the operating system and is very limited in space.
        The tmp files should be redirected to <code>/lscratch/${SLURM_JOB_ID}</code> (if
        allocating local disk space) 

<pre class="term">
-Djava.io.tmpdir=/lscratch/${SLURM_JOB_ID}
</pre>

        <p>to the java command as shown above. The wrapper script takes care of this
        automatically. Note that the temp dir must already exist if
        specifying it manually. The wrapper script will automatically create it
        if necessary.</p>

        <h4>Limit the number of parallel garbage collection threads</h4>

        <p>The Java VM has a tendency to start many threads including parallel threads
        for garbage collection since it automatically detects all CPUs on a compute
        node irrespective of the allocation. This can be aggravated when there is not
        sufficient memory allocated to the JVM.  Use <tt>-XX:ParallelGCThreads=##</tt>
        to limit the number of parallel garbage collection threads to the number of
        allocated CPUs up to 8 CPUs plus 5 for every 8 beyond that.  The GATK wrapper
        by default sets the number of GC threads correctly one less than the number of
        allocated CPUs.</p>

        <p class="alert">GATK 3.8-0 made the intel de/infaltor the default. While
        slightly faster than their jdk equivalents, this did introduce a bug that can
        lead to segfaults when allocating large heaps. Adding <code>--use_jdk_inflater
        --use_jdk_deflater</code> to the GATK tool options avoids this. The GATK
        wrapper takes care of this automatically.</p>

        <h3>Phone home</h3>

        <p>Note that GATK has a
        <a href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_engine_CommandLineGATK.php#--phone_home">phone
        home</a> feature which will send some generic information about each GATK run
        to the developers at the Broad Institute. This feature will not work on the
        Biowulf computational nodes, which are on a private network. This will not
        affect your runs in most cases. If you're getting an error related to
        'phone-home', add <code> -et NO_ET -K $GATK_KEY</code>
        to your GATK command line to disable the phone-home feature. </p>

        <h3>Open files</h3>

        <p>GATK opens many files simultaneously. It can happen that you hit the
        open-file limit on Biowulf, and see errors like:</p>

<pre class="term">
##### ERROR MESSAGE: Couldn't read file .... (Too many open files)
</pre>

        <p>The most important change you can make is to reduce the multithreading. If
        your GATK command has any multithreading flags turned on (e.g. '-nt' and
        'nct'), reduce the number of threads to 1.</p>

        <p>If that happens, the limit on the number of open files has to be
        increased from its default soft limit of 4096. This can be done by adding
        the following command to your batch script: <code>ulimit -n 16384</code>.</p>

        <p>Here are some other possibilities (thanks to Ryan Neff, NHGRI):</p>

        <ul>
            <li> Reduce the number of open files. When doing variant calling, first create gVCF
              files for each sample and then combine them together in batches of 100-200, and
              then do the final variant calls on the remaining 5-20 files (GATK's
              HaplotypeCaller and GenotypeGVCFs). If you are using the UG or a different
              variant caller, try first discovering variant calls on a per-sample basis (-gt
              mode DISCOVERY), combining all of those files together, and then genotyping
              each sample on it's own from the list of all variants (-gt mode
              GENOTYPE_GIVEN_ALLELES).
            </li>
            <li> Reduce the number of open file pointers. This can be done by increasing RAM,
              reducing the number of temporary files open at one time. Sometimes, in the
              GATK, setting --disableAutoIndexCreationAndLockingWhenReadingRods helps with
              performance and other I/O issues.
            </li>
        </ul>

        <p>Relevant discussions from the GATK mailing list:</p>
        <ul>
            <li><a href="http://gatkforums.broadinstitute.org/discussion/1404/gatk-unifiedgenotyper-unable-to-merge-temporary-tribble-output-file">GATK_UnifiedGenotyper_Unable to merge temporary Tribble output file</a></li>
            <li><a href="http://gatkforums.broadinstitute.org/discussion/3627/gatk-unified-genotyper-too-many-files-open-even-with-ulimit">GATK Unified Genotyper Too many files open even with ulimit</a></li>
        </ul>

        <h3>GATK Parallelization</h3>

        <p>Some GATK tools can make use of more than one CPU using option <code>-nt</code> 
        or <code>-nct</code>. See the GATK documentation for more detail.</p>


        <!-- ======================================================================== -->

        <a class="navbar-safe-anchor"  name="serial3"></a>
        <div class="heading">Running a single GATK batch job on Biowulf</div>

        <p>Set up a batch script along the following lines:</p>

<pre class="term">
#! /bin/bash
# gatk.bat
set -e

module load GATK/3.8-0 || exit 1

cd /data/$USER/test_data/GATK

# use 8 GB memory
java -Xmx8g -Djava.io.tmpdir=/lscratch/$SLURM_JOB_ID -jar $GATK_JAR \
  -T CountReads \
  -R exampleFASTA.fasta \
  -I exampleBAM.bam \
  -et NO_ET -K $GATK_KEY
</pre>

        <p>Submit to the queue with <a href="/docs/userguide.html">sbatch</a>:</p>

<pre class="term">
biowulf$ <b>sbatch --mem=9g --gres=lscratch:10 gatk.bat</b>
</pre>

        <p>Since the simple <code>CountReads</code> tool does not support
        multithreading, only a single core (the default) is requested.</p>

        <!-- ======================================================================== -->

        <a class="navbar-safe-anchor"  name="swarm3"></a>
        <div class="heading">Running a swarm of GATK batch jobs on Biowulf</div>

        <p>The following swarm commend file would run the <a
        href="https://www.broadinstitute.org/gatk/guide/tooldocs/org_broadinstitute_gatk_tools_walkers_indels_RealignerTargetCreator.php">RealignerTargetCreator</a>
        ,which determines which genomic intervals require realignment, for a number of
        samples:</p>

<pre class="term">
# this file is gatk-swarm
GATK -m 7g RealignerTargetCreator \
  -R ref.fasta
  -I input1.bam \
  -o output1.intervals \
  --known /fdb/GATK_resource_bundle/hg19/1000G_phase1.indels.hg19.vcf.gz
GATK -m 7g RealignerTargetCreator \
  -R ref.fasta
  -I input2.bam \
  -o output2.intervals \
  --known /fdb/GATK_resource_bundle/hg19/1000G_phase1.indels.hg19.vcf.gz
[...]
</pre>

        <p>These command lines specify that the process will use 7 GB of memory
        (-Xmx7g).  Thus, this swarm job should be submitted with the '-g 7' flag, to
        tell swarm that each process will require 7 GB of memory.</p>

<pre class="term">
biowulf$ <b>swarm -g 7 -f gatk-swarm --module GATK/3.8-1</b>
</pre>

        <!-- ======================================================================= -->
        <a class="navbar-safe-anchor"  Name="int3"></a>
        <div class="heading">Interactive job on Biowulf</div>


        <p>Allocate an interactive session with <a href="/docs/userguide.html#int">sinteractive</a>
        and use as shown below</p>
        <pre class="term">
[user@biowulf]$ <b>sinteractive --gres=lscratch:50 --cpus-per-task=2 --mem=6g</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144]$ <b>module load GATK/3.8-1</b>    
[user@cn3144]$ <b>cp ${GATK_TEST_DATA:-none}/NA12878.ba[im]* .</b>
[user@cn3144]$ <b>ls -lh</b>
total 18M
-rw-r--r-- 1 user group 2.2M Jun 30 13:45 NA12878.bai
-rw-r--r-- 1 user group  15M Jun 30 13:45 NA12878.bam
[user@cn3144]$ <b>GATK CountReads \
                    -R /fdb/GATK_resource_bundle/hg38/Homo_sapiens_assembly38.fasta \
                    -I NA12878.bam</b>
directory for temp files: /lscratch/60475214
Executing ' java -Djava.io.tmpdir=/lscratch/60475214 -Xmx2g -XX:ParallelGCThreads=2 -jar /usr/local/apps/GATK/3.8-1/GenomeAnalysisTK.jar -T CountReads -R /fdb/GATK_resource_bundle/hg38/Homo_sapiens_assembly38.fasta -I NA12878.bam --use_jdk_inflater --use_jdk_deflater '
INFO  13:50:21,043 HelpFormatter - ----------------------------------------------------------------
INFO  13:50:21,047 HelpFormatter - The Genome Analysis Toolkit (GATK) v3.8-1-0-gf15c1c3ef, Compiled...
INFO  13:50:21,048 HelpFormatter - Copyright (c) 2010-2016 The Broad Institute
INFO  13:50:21,048 HelpFormatter - For support and documentation go to https://software.broadinstit...
INFO  13:50:21,049 HelpFormatter - [Tue Jun 30 13:50:21 EDT 2020] Executing on Linux 3.10.0-862....
INFO  13:50:21,049 HelpFormatter - OpenJDK 64-Bit Server VM 1.8.0_181-b13
INFO  13:50:21,056 HelpFormatter - Program Args: -T CountReads -R /fdb/GATK_resource_bundle/hg38/Ho...
INFO  13:50:21,064 HelpFormatter - Executing as user@cn3293 on Linux 3.10.0-862.14.4.el7.x86_64 a...
INFO  13:50:21,065 HelpFormatter - Date/Time: 2020/06/30 13:50:21
INFO  13:50:21,065 HelpFormatter - ----------------------------------------------------------------
INFO  13:50:21,066 HelpFormatter - ----------------------------------------------------------------
INFO  13:50:21,072 GenomeAnalysisEngine - Deflater: JdkDeflater
INFO  13:50:21,073 GenomeAnalysisEngine - Inflater: JdkInflater
INFO  13:50:21,074 GenomeAnalysisEngine - Strictness is SILENT
INFO  13:50:23,245 GenomeAnalysisEngine - Downsampling Settings: No downsampling
INFO  13:50:23,258 SAMDataSource$SAMReaders - Initializing SAMRecords in serial
INFO  13:50:23,540 SAMDataSource$SAMReaders - Done initializing BAM readers: total time 0.28
INFO  13:50:24,955 GenomeAnalysisEngine - Preparing for traversal over 1 BAM files
INFO  13:50:24,961 GenomeAnalysisEngine - Done preparing for traversal
INFO  13:50:24,962 ProgressMeter - [INITIALIZATION COMPLETE; STARTING PROCESSING]
INFO  13:50:24,962 ProgressMeter -                 | processed |    time |    per 1M |           |   total | remaining
INFO  13:50:24,962 ProgressMeter -        Location |     reads | elapsed |     reads | completed | runtime |   runtime
INFO  13:50:24,964 ReadShardBalancer$1 - Loading BAM index data
INFO  13:50:24,966 ReadShardBalancer$1 - Done loading BAM index data
INFO  13:50:26,502 CountReads - CountReads counted 61614 reads in the traversal
INFO  13:50:26,507 ProgressMeter -            done     61614.0     1.0 s      25.0 s       99.9%     1.0 s       0.0 s
INFO  13:50:26,508 ProgressMeter - Total runtime 1.55 secs, 0.03 min, 0.00 hours
INFO  13:50:26,512 MicroScheduler - 0 reads were filtered out during the traversal out of approximately 61614 total reads (0.00%)
INFO  13:50:26,514 MicroScheduler -   -> 0 reads (0.00% of total) failing BadCigarFilter
INFO  13:50:26,515 MicroScheduler -   -> 0 reads (0.00% of total) failing MalformedReadFilter
------------------------------------------------------------------------------------------
Done. There were no warn messages.
------------------------------------------------------------------------------------------
GATK finished (exit code 0)
[user@cn3144]$ <b>exit</b>

        </pre>

        <!-- ======================================================================== -->

        <div class="heading">LiftOverVCF -- converts a VCF file from one reference build to another</div>

        <p class="alert">LiftOverVCF has been deprecated with release 3.5-0. Use picard's LiftoverVCF
        instead.</p>

        <p>The procedure for lifting over VCF file from one genome build to different
        build in GATK is a three step process - (1) LiftoverVCF (2) sort the VCF and
        (3) FilterLiftedVCF. This process is automated by <code>liftOverVCF</code> </p>
<pre class="term">
Usage: liftOverVCF
   -vcf            &lt;input vcf&gt;
   -gatk           &lt;path to gatk trunk&gt;
   -chain          &lt;chain file&gt;
   -newRef         &lt;path to new reference prefix;
                    we will need newRef.dict, .fasta, and .fasta.fai&gt;
   -oldRef         &lt;path to old reference prefix; we will need oldRef.fasta&gt;
   -out            &lt;output vcf&gt;
   -recordOriginalLocation
                   &lt;Should we record what the
                    original location was in the INFO field?; defaults to false&gt;
</pre>

        <p>This procedure is to use liftOverVCF to switch from, say, hg18 to hg19.</p>

        <h3>Required data:</h3>
        <ul>
            <li>Chain files from the
              <a href="ftp://gsapubftp-anonymous@ftp.broadinstitute.org/Liftover_Chain_Files/">Broad ftp site</a>
              are mirrored in <code>fdb/GATK_resource_bundle/liftover_chains</code>. In addtion
              a hg18 to hg19 liftover file from UCSC is included as well.</li>
            <li>Reference sequences and .dict, .fai files for references can also be
              found in the resource bundles.</li>
        </ul>

        <p>Example:</p>

<pre class="term">
liftOverVCF  \
    -vcf test.hg18.vcf.gz \
    -chain /fdb/GATK_resource_bundle/liftover_chains/hg18ToHg19.over.chain \
    -newRef /fdb/GATK_resource_bundle/hg19/ucsc.hg19 \
    -oldRef /fdb/GATK_resource_bundle/hg18/Homo_sapiens_assembly18 \
    -out test.hg19.vzf
</pre>


    </div>
    
    <!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
    <!--                                 GATK 4                                  -->
    <!-- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -->
    <div id="tab-4" class="tab-content current">
            
        <table width=25% align=right> <tr><td>
            <div class="toc">
            <div class="tocHeading">Quick Links</div>
            <div class="tocItem"><a href="#overview">Overview</a></div>
            <div class="tocItem"><a href="#serial">Batch job on Biowulf</a></div>
            <div class="tocItem"><a href="#swarm">Swarm of jobs</a></div>
            <div class="tocItem"><a href="#int">Interactive job on Biowulf</a></div>
        </div></table>
        
        <!-- ======================================================================== -->

        <a class="navbar-safe-anchor"  name="overview"></a>
        <div class="heading">Overview</div>

            
        <h3>Environment module</h3>
        <p>The environment module for GATK (<code>module load GATK</code> or
        <code>module load GATK/version</code>)</p>

        <ul>
            <li>loads R as a dependency</li>
            <li>sets the environment variables
            <ul>
                <li> <code>$GATK_TEST_DATA</code></li>
            </ul>
            </li>
        </ul>


        <p class="alert">Whenever possible, please use lscratch for temporary files. It will be
        more performant and reduce strain on the shared filesystems.</p>

        <h3>Calling GATK</h3>
        <p>GATK now includes its own wrapper script (<code>gatk</code>) and therefore
        the Biowulf-supplied wrapper for GATK 3 is not available for GATK 4. The basic 
        GATK 4 command is </p>

<pre class="term">
gatk --java-options "[java options]" &lt;Toolname&gt; [tool options]
</pre>


        <p>For detailed GATK help use</p>
<pre class="term">
<b>gatk --help</b>

 Usage template for all tools (uses --spark-runner LOCAL when used with a Spark tool)
    gatk AnyTool toolArgs

 Usage template for Spark tools (will NOT work on non-Spark tools)
    gatk SparkTool toolArgs  [ -- --spark-runner <LOCAL | SPARK | GCS> sparkArgs ]

 Getting help
    gatk --list       Print the list of available tools

    gatk Tool --help  Print help on a particular tool

 Configuration File Specification
     --gatk-config-file                PATH/TO/GATK/PROPERTIES/FILE

 gatk forwards commands to GATK and adds some sugar for submitting spark jobs

   --spark-runner <target>    controls how spark tools are run
     valid targets are:
     LOCAL:      run using the in-memory spark runner
     SPARK:      run using spark-submit on an existing cluster
                 --spark-master must be specified
                 --spark-submit-command may be specified to control the Spark submit command
                 arguments to spark-submit may optionally be specified after --
     GCS:        run using Google cloud dataproc
                 commands after the -- will be passed to dataproc
                 --cluster <your-cluster> must be specified after the --
                 spark properties and some common spark-submit parameters will be translated
                 to dataproc equivalents

   --dry-run      may be specified to output the generated command line without running it
   --java-options 'OPTION1[ OPTION2=Y ... ]'   optional - pass the given string of options to the
                 java JVM at runtime.
                 Java options MUST be passed inside a single string with space-separated values.
</pre>

        <h4>Memory</h4>

        <p>The <code>-Xmx</code> option to Java specifies the maximal size of the
        heap memory. It accepts suffixes <code>k</code>, <code>m</code>, and
        <code>g</code>. This is passed to gatk in the <code>--java-otions</code> string. For
        example</p>

<pre class="term">
gatk --java-options "-Xmx6g" &lt;Toolname&gt; [tool options]
</pre>

        <p>Note that you should request additional memory (1-2GB) for batch/interactive jobs on
        top of the memory given to java to ensure that the whole java process stays
        below the allocated memory limit.</p>
        
        <p>Note that some tools may perform better with non-default garbage collectors. For
        example here are some possible options for a large heap allocation using ParallelGCThreads:</p>

<pre class="term">
gatk --java-options "-Djava.io.tmpdir=/lscratch/$SLURM_JOB_ID -Xms48G -Xmx48G \
     -XX:ParallelGCThreads=8" ...
</pre>

<h4>Redirect temporary files to <tt>/lscratch/${SLURM_JOB_ID}</tt></h4>

        <p>GATK writes some temporary files during its run. By default, Java writes
        temp files to <code>/tmp</code>. However this is not advisable because
        <code>/tmp</code> is used by the operating system and is very limited in space.
        The tmp files should be redirected to <code>/lscratch/${SLURM_JOB_ID}</code>
        by adding </p>

<pre class="term">
-Djava.io.tmpdir=/lscratch/${SLURM_JOB_ID}
</pre>

        <p>to the java options.</p>
        
        <p>For more details on the node-local storage under <code>/lscratch</code> see the
        <a href="/docs/b2-userguide.html#local">user guide</a></p>

        <h4>Parallel garbage collection threads</h4>

        <p>If you find that your GATK runs have more active threads than you were
        expecting you may have to limit the number of prallel garbage collection
        threads. The JVM options <code>-XX:ParallelGCThreads</code> and  
        <code>-XX:ConcGCThreads</code> can be used to tune the number of threads
        dedicated to garbage collection.</p>


        <h3>Open files</h3>

        <p>GATK may open many files simultaneously. It can happen that you hit the
        open-file limit on Biowulf, and see errors like:</p>

<pre class="term">
##### ERROR MESSAGE: Couldn't read file .... (Too many open files)
</pre>

        <p>If that happens, the limit on the number of open files has to be
        increased from its default soft limit of 4096. This can be done by adding
        the following command to your batch script: <code>ulimit -n 16384</code>.</p>

        <p>Here are some other possibilities (thanks to Ryan Neff, NHGRI):</p>

        <ul>
            <li> Reduce the number of open files. When doing variant calling, first create gVCF
              files for each sample and then combine them together in batches of 100-200, and
              then do the final variant calls on the remaining 5-20 files (GATK's
              HaplotypeCaller and GenotypeGVCFs).</li>
            </li>
        </ul>

        <div class="heading">GATK Parallelization</div>

        <p>The paradigm for parallelization for GATK 4 is completely different
        from GATK 3. Spark is used to parallelize workloads and the tools
        capable of using this form of parallelism include 'Spark' in their
        toolname. Some of these tools are still in beta and outputs may not be
        compatible with non-parallel variants.  See the GATK documentation for
        more detail.</p>
        

        <!-- ======================================================================== -->

        <a class="navbar-safe-anchor"  name="serial"></a>
        <div class="heading">Running a single GATK batch job on Biowulf</div>

        <p>Set up a batch script along the following lines:</p>

<pre class="term">
#! /bin/bash
# gatk.bat
set -e

module load GATK/4.1.8.0 || exit 1
cd /lscratch/$SLURM_JOB_ID || exit 1

ref=/fdb/GATK_resource_bundle/hg38/Homo_sapiens_assembly38.fasta
bam=${GATK_TEST_DATA:-none}/HG00133.alt_bwamem_GRCh38DH.20150826.GBR.exome.bam
# use 8 GB memory
gatk --java-options "-Xmx8g -Djava.io.tmpdir=/lscratch/$SLURM_JOB_ID" \
  CountReads \
  -R $ref \
  -I $bam \
</pre>

        <p>Submit to the queue with <a href="/docs/userguide.html">sbatch</a>:</p>

<pre class="term">
[user@biowulf]$ <b>sbatch --mem=9g --gres=lscratch:10 gatk.bat</b>
</pre>

        <p>Since the simple <code>CountReads</code> tool does not support
        multithreading, only a single core (the default) is requested.</p>

        <!-- ======================================================================== -->

        <a class="navbar-safe-anchor"  name="swarm"></a>
        <div class="heading">Running a swarm of GATK batch jobs on Biowulf</div>

        <p>The following swarm commend file would run the MarkDuplicatesSpark.
        It will also sort and index the bam file. This is a Spark
        implementation of Picard MarkDuplicates that allows the tool to be run
        in parallel on multiple cores: </p>

<pre class="term">
# this file is gatk-swarm
gatk --java-options "-Xmx8g" MarkDuplicatesSpark \
  -I input1.bam \
  -O input1_markdup.bam \
  --spark-runner LOCAL \
  --spark-master local[$SLURM_CPUS_PER_TASK] \
gatk --java-options "-Xmx8g" MarkDuplicatesSpark \
  -I input2.bam \
  -O input2_markdup.bam \
  --spark-runner LOCAL \
  --spark-master local[$SLURM_CPUS_PER_TASK] \
[...]
</pre>

        <p>These command lines specify that the process will use 8 GB of memory
        (-Xmx8g).  Thus, this swarm job should be submitted with the '-g 9' flag, to
        tell swarm that each process will require 9 GB (8GB plus some buffer) of memory. 
	Please do not use more than 12 CPUs due to the parallel efficiency.</p>

<pre class="term">
biowulf$ <b>swarm -g 9 -f gatk-swarm -t 16 --module GATK/4.1.8.0</b>
</pre>

        <!-- ======================================================================== -->

        <a class="navbar-safe-anchor"  name="int"></a>
        <div class="heading">Running an interactive job on Biowulf</div>
        
        <p>It may be useful for debugging purposes to run GATK jobs interactively. Such
        jobs should not be run on the Biowulf login node. Instead allocate an
        interactive node as described below, and run the interactive job there.</p>

<pre class="term">
[user@biowulf]$ <b>sinteractive --gres=lscratch:50 --cpus-per-task=2 --mem=6g</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144]$ <b>module load GATK/4.1.8.0</b>    
[user@cn3144]$ <b>cp ${GATK_TEST_DATA:-none}/NA12878.ba[im]* .</b>
[user@cn3144]$ <b>ls -lh</b>
total 18M
-rw-r--r-- 1 user group 2.2M Jun 30 13:45 NA12878.bai
-rw-r--r-- 1 user group  15M Jun 30 13:45 NA12878.bam
[user@cn3144]$ <b>gatk --java-options "-Xmx5g -Xms5g -Djava.io.tmpdir=/lscratch/$SLURM_JOB_ID" \
                   CountReads \
                   -R /fdb/igenomes/Homo_sapiens/UCSC/hg38/Sequence/WholeGenomeFasta/genome.fa \
                   -I NA12878.bam</b>
Using GATK jar /usr/local/apps/GATK/4.1.8.0/gatk-package-4.1.8.0-local.jar
Running:
    java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true 
    -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xmx5g -Xms5g 
    -Djava.io.tmpdir=/lscratch/60475214 
    -jar /usr/local/apps/GATK/4.1.8.0/gatk-package-4.1.8.0-local.jar CountReads -R 
     /fdb/igenomes/Homo_sapiens/UCSC/hg38/Sequence/WholeGenomeFasta/genome.fa -I NA12878.bam
17:06:09.007 INFO  NativeLibraryLoader - Loading libgkl_compression.so from jar:file:/usr/local/apps/GAT
K/4.1.8.0/gatk-package-4.1.8.0-local.jar!/com/intel/gkl/native/libgkl_compression.so
17:06:09.513 INFO  CountReads - ------------------------------------------------------------
17:06:09.514 INFO  CountReads - The Genome Analysis Toolkit (GATK) v4.1.8.0
17:06:09.514 INFO  CountReads - For support and documentation go to https://software.broadinstitute.org/
gatk/
17:06:09.515 INFO  CountReads - Executing as user@cn3293 on Linux v3.10.0-862.14.4.el7.x86_64 amd64
17:06:09.515 INFO  CountReads - Java runtime: OpenJDK 64-Bit Server VM v1.8.0_181-b13
17:06:09.516 INFO  CountReads - Start Date/Time: June 30, 2020 5:06:08 PM EDT
17:06:09.517 INFO  CountReads - ------------------------------------------------------------
17:06:09.517 INFO  CountReads - ------------------------------------------------------------
17:06:09.519 INFO  CountReads - HTSJDK Version: 2.22.0
17:06:09.519 INFO  CountReads - Picard Version: 2.22.8
17:06:09.519 INFO  CountReads - HTSJDK Defaults.COMPRESSION_LEVEL : 2
17:06:09.520 INFO  CountReads - HTSJDK Defaults.USE_ASYNC_IO_READ_FOR_SAMTOOLS : false
17:06:09.520 INFO  CountReads - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_SAMTOOLS : true
17:06:09.521 INFO  CountReads - HTSJDK Defaults.USE_ASYNC_IO_WRITE_FOR_TRIBBLE : false
17:06:09.521 INFO  CountReads - Deflater: IntelDeflater
17:06:09.521 INFO  CountReads - Inflater: IntelInflater
17:06:09.521 INFO  CountReads - GCS max retries/reopens: 20
17:06:09.521 INFO  CountReads - Requester pays: disabled
17:06:09.521 INFO  CountReads - Initializing engine
WARNING: BAM index file /lscratch/60475214/NA12878.bai is older than BAM /lscratch/60475214/NA12878.bam
17:06:10.212 INFO  CountReads - Done initializing engine
17:06:10.212 INFO  ProgressMeter - Starting traversal
17:06:10.212 INFO  ProgressMeter -        Current Locus  Elapsed Minutes       Reads Processed     Reads
/Minute
17:06:11.192 INFO  CountReads - 0 read(s) filtered by: WellformedReadFilter

17:06:11.194 INFO  ProgressMeter - chrUn_KN707963v1_decoy:19294              0.0                 61614
      3768440.4
17:06:11.195 INFO  ProgressMeter - Traversal complete. Processed 61614 total reads in 0.0 minutes.
17:06:11.195 INFO  CountReads - CountReads counted 61614 total reads
17:06:11.195 INFO  CountReads - Shutting down engine
[June 30, 2020 5:06:11 PM EDT] org.broadinstitute.hellbender.tools.CountReads done. Elapsed time: 0.04 m
inutes.
Runtime.totalMemory()=5145362432
Tool returned:
61614
</pre>



</div><!-- versions div -->



<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
