<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'gmap-gsnap on Biowulf';</script>
<div class="title">gmap-gsnap on Biowulf</div>
<p>
  <table width=25% align=right>
    <tr>
      <td>
	<div class="toc">
	  <div class="tocHeading">Quick Links</div>
	  <div class="tocItem"><a href="#[user@cn3144]">On Helix</a></div>
	  <div class="tocItem"><a href="#serial">Batch job on Biowulf</a></div>
	  <!-- <div class="tocItem"><a href="#parallel">Parallel gmap-gsnap on Biowulf</a></div> -->
	  <div class="tocItem"><a href="#swarm">Swarm of jobs</a></div>
	  <div class="tocItem"><a href="#int">Interactive job</a></div>
	</div>
      </td>
    </tr>
  </table>
</p>
<div class="heading">Description</div>

<p>GMAP is a tools for rapidly and accurately mapping and aligning cDNA
sequences to genomic sequences. </p>

<p>GSNAP is designed to align short reads from NGS data and allow detection of
short and long range splicing de novo or with a database of know juctions.  In
addtion, SNP-tolerant alignements and alignments of reads obtained from
bisulfite treated DNA are implemented.</p>

<p>There may be multiple versions of gmap-gsnap available. An easy way of selecting the
version is to use <a href="/apps/modules.html">modules</a>.  To see the modules
available, type</p>
<pre class="term">
module avail gmap-gsnap 
</pre>

<p>To select a module use</p>
<pre class="term">
module load gmap-gsnap/[version]
</pre>
<p>where <code>[version]</code> is the version of choice.</p>

<!-- comment out where applicable -->
<p class="alert">gmap-gsnap is a multithreaded application. Make sure to match the
number of cpus requested with the number of threads.</p>

<h3>Environment variables set</h3>
<ul>
<li>$PATH</li>
</ul>

<h3>References</h3>
<ul>
<li>Thomas D. Wu and Colin K. Watanabe.
  <em>GMAP: a genomic mapping and alignment program for mRNA and EST sequences.</em>
  Bioinformatics 2005, 21:1859-1875. 
  <a href="http://www.ncbi.nlm.nih.gov/pubmed/15728110">Pubmed</a>&nbsp;|&nbsp;
  PMC]&nbsp;|&nbsp;
  <a href="http://bioinformatics.oupjournals.org/cgi/content/full/21/9/1859">Journal</a></li>
<li>Thomas D. Wu and Serban Nacu.
  <em>Fast and SNP-tolerant detection of complex variants and splicing in short reads.</em>
  Bioinformatics 2010, 26:873-881.
  <a href="http://www.ncbi.nlm.nih.gov/pubmed/20147302">Pubmed</a>&nbsp;|&nbsp;
  <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2844994/">PMC</a>&nbsp;|&nbsp;
  <a href="http://bioinformatics.oupjournals.org/cgi/content/full/26/7/873">Journal</a></li>
</ul>
<h3>Documentation</h3>
<ul>
<li><a href="http://research-pub.gene.com/gmap/">Home page</a></li>
<li><a href="http://research-pub.gene.com/gmap/src/README">Manual</a></li>
</ul>

<div class="heading">Index files</div>

<p>gmap/gsnap indices are stored under</p>

<p><b><code>/fdb/gmap_post20210308/[organism]/[source]/[build]/Sequence/gmap-gsnap</code></b></p>
<ul>
<li><b><code>[organism]</code></b> is the specific organism of interest
  (Gallus_gallus, Rattus_norvegicus, etc.)</li>
<li><b><code>[source]</code></b> is the source for the sequence (NCBI,
  Ensembl, UCSC)</li>
<li><b><code>[build]</code></b> is the specific genome build of interest
  (hg19, build37.2, GRCh37)</li>
</ul>

<p>in a structure analogous to the igenomes package.  In addition, commonly used
genomes are also available at
<code>/fdb/gmap_post20210308/common_genomes</code>. This path is also the
default path gmap and gsnap use to look for genome files, so a particular genome
is available there it is sufficient to provide the genome name with <code>-d
genome_name</code> to any of the tools without also specifying the genome
directory with <code>-D</code>.</p>

<div class="heading">Performance considerations</div>

<p>Gsnap throughput was measured using different numbers of threads with an
ENCODE data set (100nt single ended RNA-Seq, mouse Purkinje cells; accession 
ENCFF220BOE; biosample <a href="https://www.encodeproject.org/experiments/ENCSR758TAD/">
    ENCSR758TAD</a>). Discovery of novel splice sites was disabled but known
splice sites were provided as input. The input file was compressed and stored
on local scratch space. The output was saved unmodified on local scratch space. Each
process was allocated 30GB of memory. The exact command used was</p>

<pre class="term">
gsnap -d mm10 --nthreads=${SLURM_CPUS_PER_TASK} \
    --use-shared-memory=0 -B 5 \
    --novelsplicing=0 --gunzip \
    --use-splicing=/lscratch/${SLURM_JOBID}/mm10.iit \
    --format=sam -o /lscratch/${SLURM_JOBID}/${n}.sam \
    /lscratch/${SLURM_JOBID}/${fq}
</pre>

<div style="width:75%; margin:0 auto;">
  <img src="/images/gsnap_thread_scaling.png" alt="gsnap thread scaling" width="100%">
  <p>Reads aligned per second per thread for different versions of gsnap with different numbers
  of threads.</p>
  <p>The lower throughput of version 2015-05-01
  with default settings is largely the result of a change in the default for one setting 
  ( <code>--end-detail</code>) from medium to high which has the effect of improving
  alignments in the small number of cases where the distal end of a splice or indel
  contains a mismatch. Changing this version back to medium recovers most the loss
  in throughput. It is therefore largely a tradeoff between throughput and sensitivity.
  As of version 2016-06-30, the default reverted to medium.</p>
  <p>Most benchmarks were run on the phase1 norm nodes (x2650; 10g). Benchmarks ending
  in x2695 were run on the newer phase2 norm nodes (x2695; ibfdr).</p>
</div>

<p>Gsnap scales well with the number of threads all the way to 32, though the most
efficient use is probably around 24 threads with this input data set. Note that 
gsnap does not start any extra processes or threads, so 
<code>--nthreads=$SLURM_CPUS_PER_TASK</code> is appropriate unless the output is
piped into another process such as samtools. In that case the number of threads used
by the second process has to be subtracted from the number of threads given to
gsnap.</p>
<p>Performance was comparable when input and output were kept on the shared storage
space (<code>/data</code>).</p>



<!-- ======================================================================= -->
<a class="navbar-safe-anchor"  name="compute node"></a>
<div class="heading">On interactive session</div>

<p>Gmap and gsnap are multi-threaded applications. The thread count is
determined with <code>-t / --nthreads</code> and defaults to 1. </p>

<p>Map and align a single cDNA to the mouse genome. Note that the mouse genome
is in common genomes, so no path has to be provided.</p>


<pre class="term">
[user@biowulf]$ <b>sinteractive</b>

[user@cn3144]$ <b>module load gmap-gsnap</b>
[user@cn3144]$ <b>cd /data/$USER/test_data</b>
[user@cn3144]$ <b>gmap -d mm10 -A /usr/local/apps/gmap-gsnap/TEST_DATA/myc.fa</b>
</pre>

<p>Map and align single end RNA-Seq data from encode to the mouse genome with
database of known splice junctions. Supress search for novel juctions:</p>

<pre class="term">
[user@cn3144]$ <b>module load samtools/1.2</b>
[user@cn3144]$ <b>gtf_splicesites &lt; /fdb/igenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/genes.gtf \
  | iit_store -o mm10.iit</b>

[user@cn3144]$ <b>gsnap -d mm10 --nthreads=2 --novelsplicing=0 --use-splicing=mm10.iit --gunzip \
            --format=sam \
            /usr/local/apps/gmap-gsnap/TEST_DATA/ENCFF220BOE.fastq.gz \
            2&gt; test.log \
          | samtools view -S -b -F4 -q 20 - \
          &gt; bam/gsnap.bam</b>

</pre>

<p>A number of other small tools are available as part of the package.</p>

<!-- ======================================================================= -->
<a class="navbar-safe-anchor"  name="serial"></a>
<div class="heading">Batch job on Biowulf</div>

<p>Batch scripts should use the <code>$SLURM_CPUS_PER_TASK</code> environment
variable to determine the number of threads to use. This variable is set by
slurm according to the number of requested cores. An example script for gsnap
might look like the following:</p>

<pre class="term">
#! /bin/bash
# gsnap_batch.sh
set -o pipefail
set -e

function fail {
    echo "$@" &gt;&amp;2
    exit 1
}

module load samtools/1.2 || fail "could not load samtools module"
module load gmap-gsnap || fail "could not load gmap-gsnap module"
cd /data/$USER/test_data || fail "no such directory"

gtf_splicesites &lt; /fdb/igenomes/Mus_musculus/UCSC/mm10/Annotation/Genes/genes.gtf \
  | iit_store -o mm10.iit

gsnap -d mm10 --nthreads=$(( SLURM_CPUS_PER_TASK - 2 ))\
        --novelsplicing=0 --use-splicing=mm10.iit --gunzip \
        --format=sam \
        /usr/local/apps/gmap-gsnap/TEST_DATA/ENCFF220BOE.fastq.gz \
        2&gt; test.log \
      | samtools view -S -b -F4 -q 20 -@ 2 - \
      &gt; bam/gsnap.bam

</pre>

<p>The script is submitted to the batch system requesting, for example, 8 cores
and 25GB of memory:</p>

<pre class="term">
biowulf$ <b>sbatch -c 20 --mem=30g gsnap_batch.sh</b>
</pre>



<!-- ======================================================================= -->
<a class="navbar-safe-anchor"  name="swarm"></a>
<div class="heading">Swarm of jobs on Biowulf</div>

<p>Set up a command file for two jobs, each aligning single ended reads. Note
that swarm allows line continuations.</p>

<pre class="term">
# gsnap.swarm
cd /data/$USER/test_data \
  &amp;&amp; gsnap -d mm10 --nthreads=$(( SLURM_CPUS_PER_TASK - 1 ))\
        --novelsplicing=0 --use-splicing=mm10.iit --gunzip \
        --format=sam file1.fastq.gz \
        2&gt; test1.log \
  | samtools view -S -b -F4 -q 20 - \
  &gt; bam/gsnap.bam
cd /data/$USER/test_data \
  &amp;&amp; gsnap -d mm10 --nthreads=$(( SLURM_CPUS_PER_TASK - 1 ))\
        --novelsplicing=0 --use-splicing=mm10.iit --gunzip \
        --format=sam file2.fastq.gz \
        2&gt; test2.log \
  | samtools view -S -b -F4 -q 20 - \
  &gt; bam/gsnap2.bam
</pre>

<p>Create the known splice site database as shown above and start the jobs with swarm</p>

<pre class="term">
biowulf$ <b>swarm -f swarm_command_file -t 20 -g 30 \
                --module samtools/1.2 gmap-gsnap</b>
</pre>

<!-- ======================================================================= -->
<p><a class="navbar-safe-anchor"  name="int"></a>
<div class="heading">Interactive job on Biowulf</div></p>

<p>Interactive work requiring significant resources should be carried out on
interactive compute nodes, not on the head node or [user@cn3144]. Interactive nodes are
allocated with <code>sinteractive</code>. For example, to request a 8 core
interactive job with 25GB memory:</p>

<pre class="term">
biowulf$ <b>sinteractive -c 8 --mem=30g</b>
</pre>

<p>On the interactive node, gmap and gsnap are then used essentially as decribed
<a href="#[user@cn3144]">above</a>.</p>



<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
