<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'gtex_rnaseq on Biowulf';</script>
<div class="title">gtex_rnaseq on Biowulf</div>

<table width=25% align=right>
    <tr>
        <td>
            <div class="toc">
                <div class="tocHeading">Quick Links</div>
                <div class="tocItem"><A href="#doc">Documentation</a></div>
                <div class="tocItem"><a href="#notes">Notes</a></div>
                <div class="tocItem"><a href="#int">Interactive job </a></div>
                <div class="tocItem"><a href="#sbatch">Batch job </a></div>
                <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
            </div>
        </td>
    </tr>
</table>

<p> This module makes available the tools used in the <a
    href="https://github.com/broadinstitute/gtex-pipeline/tree/master/rnaseq">GTEX
    RNA-Seq pipeline</a>. Also planned is the implementation of a combined
    pipeline but that is not yet ready.</p>

<!-- <h3>References:</h3>
<ul>
<li>Paper</li>
</ul> -->


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li>gtex_rnaseq on <a href="https://github.com/broadinstitute/gtex-pipeline/tree/master/rnaseq">GitHub</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: gtex_rnaseq (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
<li>Some of the tools are multithreaded. When running individual tools please match the number
    of allocated CPUs to the number of threads</li>
<li>Reference data in <code>GTEX_RNASEQ_REF</code></li>
<li>Example files in <code>GTEX_RNASEQ_TEST_DATA</code></li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the
individual steps of the pipeline. Note that only some of the steps are resource
intensive. For production work please run individual steps as batch jobs with
the appropriate resource allocations. This full size example takes some hours
to run even though it is downsampled to just 10M reads per sample.</p>

<p>Note that in this interactive examples all steps are run as part of the same allocation.
That is ok for testing and debugging, but since the steps require quite different resources
(e.g. star scales well to multiple threads and requires around 45GB of memory, rnaseqc and
markduplicates are mostly single threaded) for production each step should be run as a separate
job.</p>

<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=40g --time=1-12 --cpus-per-task=16 --gres=lscratch:400</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144]$ <b>module load gtex_rnaseq</b>
# temp files for STAR are located in the same directory as output so better be in lscratch
[user@cn3144]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144]$ <b>cp -Lr ${GTEX_RNASEQ_TEST_DATA:-none} ./data</b>
[user@cn3144]$ <b>samples=( HepG2_ENCLB059ZZZ 
                            HepG2_ENCLB060ZZZ 
                            K562_ENCLB063ZZZ 
                            K562_ENCLB064ZZZ )</b>
[user@cn3144]$ <b> for sample in "${samples[@]}" ; do
    run_STAR.py \
        --output_dir data/star_out \
        -t $SLURM_CPUS_PER_TASK \
        ${GTEX_RNASEQ_REF}/GRCh38_release38/star_index_oh100 \
        data/${sample}_R1.fastq.gz \
        data/${sample}_R2.fastq.gz \
        ${sample}
    done</b>

# note that this script changes into the output directory,
# so the path to the input bam has to be absolute or relative to
# that output directory
[user@cn3144]$ <b>for sample in "${samples[@]}" ; do
    run_MarkDuplicates.py \
        -o data \
        star_out/${sample}.Aligned.sortedByCoord.out.bam \
        ${sample}.Aligned.sortedByCoord.out.md
    done</b>

# need to manually index the markduplicate bam output
[user@cn3144]$ <b>module load samtools</b>
[user@cn3144]$ <b>for sample in "${samples[@]}" ; do
    samtools index data/${sample}.Aligned.sortedByCoord.out.md.bam
    done</b>

# - Syntax for the V8 pipeline:
#   - need to use the older java
#   - need more memory
[user@cn3144]$ <b>for sample in "${samples[@]}" ; do
    run_rnaseqc.py \
        --output_dir=$PWD/data \
        --java /usr/lib/jvm/java-1.7.0-openjdk-amd64/bin/java \
        --memory=16 \
        ${sample}.Aligned.sortedByCoord.out.md.bam \
        ${GTEX_RNASEQ_REF}/GRCh38_release38/gencode.v38.primary_assembly.genes.gtf \
        ${GTEX_RNASEQ_REF}/GRCh38_release38/GRCh38.primary_assembly.genome.fa \
        ${sample}
    done</b>

# - Syntax for the V10 pipeline. Need to change to the directory for this
#   step to work
[user@cn3144]$ <b>pushd data</b>
[user@cn3144]$ <b>for sample in "${samples[@]}" ; do
    run_rnaseqc.py \
        ${GTEX_RNASEQ_REF}/GRCh38_release38/gencode.v38.primary_assembly.genes.gtf \
        ${sample}.Aligned.sortedByCoord.out.md.bam \
        ${sample}
    done</b>
[user@cn3144]$ <b>popd</b> 

[user@cn3144]$ <b>for sample in "${samples[@]}" ; do
    run_RSEM.py \
        --threads $SLURM_CPUS_PER_TASK \
        ${GTEX_RNASEQ_REF}/GRCh38_release38/rsem_reference \
        $PWD/data/star_out/${sample}.Aligned.toTranscriptome.out.bam \
        $PWD/data/${sample}
    done</b>

[user@cn3144]$ <b>combine_GCTs.py data/*.exon_reads.gct.gz data/combined_gcts_exon_reads</b>

# copy relevant results back

[user@cn3144]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. gtex_rnaseq.sh), which uses the input file 'gtex_rnaseq.in'. For example:</p>

<pre class="term">
#!/bin/bash
module load gtex_rnaseq/V8 || exit 1
wd=$PWD
sample=HepG2_ENCLB059ZZZ
cd /lscratch/$SLURM_JOB_ID || exit 1
mkdir data
cp -L ${GTEX_RNASEQ_TEST_DATA:-none}/${sample}* ./data</b>
run_STAR.py \
    --output_dir $PWD/data/star_out \
    -t $SLURM_CPUS_PER_TASK \
    ${GTEX_RNASEQ_REF}/GRCh38_release38/star_index_oh100 \
    data/${sample}_R1.fastq.gz \
    data/${sample}_R2.fastq.gz \
    ${sample}
cp -r data $wd
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=16 --mem=40g --gres=lscratch:75 gtex_rnaseq.sh</pre>






<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
