<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type="text/javascript" language="JavaScript">document.querySelector('title').textContent = "gutSMASH";</script>

<div class="title">gutSMASH on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
    gutSMASH is a tool that has been developed to systematically evaluate the metabolic potential of anaerobic bacteria in the gut by predicting both known and novel anaerobic metabolic gene clusters (MGCs) from the gut microbiome.
    The gutSMASH detection rules have been validated using a curated dataset.
</p>

<h3>References:</h3>
<ul>

    <li>
	Andreu, V.P., Augustijn, H.E., Chen, L., Zhernakova, A., Fu, J., Fischbach, M.A., Dodd, D. and Medema, M.H.
	<a href="https://doi.org/10.1101/2021.02.25.432841"><b><u>A systematic analysis of metabolic pathways in the human gut microbiota.</u></b></a>
    <em>bioRxiv, pp.2021-02..</em>
  </li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="https://gutsmash.bioinformatics.nl/help.html">gutSMASH Main Site</a></li>
    <li><a href="https://github.com/victoriapascal/gutsmash">gutSMASH on GitHub</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>

<p class="NXblock">This application requires a <a href="/docs/connect.html#nx">graphical connection using NX</a></p>

<ul>
    <li>Module Name: <tt>gutsmash</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
	<li>Multithreaded: set the <tt>-c</tt>/<tt>--cpus</tt> flag to match your allocation.
	<li>This application produces HTML reports. You can use <a href="/docs/hpcdrive.html">hpcdrive</a> to view these reports on your local workstation.</li>
<li>Environment variables set <!--for ones users should be aware of -->
  <ul>
    <li><tt>GUTSMASH_HOME</tt></li>
  </ul>
<li>Example files in <tt>$GUTSMASH_HOME/test</tt>
<li>Reference data in <tt>/fdb/gutsmash/</tt>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task 4</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load gutsmash</b>

[user@cn3144 ~]$ <b>run_gutsmash.py --cpus $SLURM_CPUS_PER_TASK --minimal $GUTSMASH_HOME/test/Streptomyces_coelicolor.gbk</b>

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. gutsmash.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load gutsmash
run_gutsmash --cpus $SLURM_CPUS_PER_TASK --minimal $GUTSMASH_HOME/test/Streptomyces_coelicolor.gbk
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] gutsmash.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. gutsmash.swarm). For example:</p>

<pre class="term">
run_gutsmash.py --cpus $SLURM_CPUS_PER_TASK --minimal sample1.gbk
run_gutsmash.py --cpus $SLURM_CPUS_PER_TASK --minimal sample2.gbk
run_gutsmash.py --cpus $SLURM_CPUS_PER_TASK --minimal sample3.gbk
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f gutsmash.swarm [-g #] [-t #] --module gutsmash</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module gutsmash</tt> <td>Loads the gutSMASH module for each subjob in the swarm 
</table>

<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
