

document.querySelector('title').textContent = 'hap.py';
hap.py





|  |
| --- |
| 


Quick Links
[Interactive](#helix)
[Batch job on Biowulf](#sbatch)
[Swarm of jobs on Biowulf](#swarm)
[Documentation](#doc)
 |



*a set of programs based on htslib to benchmark variant calls against gold standard truth datasets*

Hap.py is a tool to compare diploid genotypes at haplotype level. Rather than comparing VCF records row by row, hap.py will generate and match alternate sequences in a superlocus. A superlocus is a small region of the genome (sized between 1 and around 1000 bp) that contains one or more variants.


**Web sites**
* [hap.py on Github](https://github.com/Illumina/hap.py)
* [hap.py User's Manual](https://github.com/Illumina/hap.py/blob/master/doc/happy.md)




hap.py Interactive job
[back to top](hap.py.html)  








