<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Hicexplorer on Biowulf';</script>
<div class="title">Hicexplorer on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>Hicexplorer is a set of tools to process, normalize and visualize Hi-C data.</em>  
</p>

<h3>References:</h3>
<ul>
<li>Wolff J, Rabbani L, Gilsbach R, Richard G, Manke T, Backofen R, Gr√ºning BA.
<em><a href="https://pubmed.ncbi.nlm.nih.gov/32301980/">Galaxy HiCExplorer 3: a web server for reproducible Hi-C, capture Hi-C and single-cell Hi-C data analysis, quality control and visualization. </a></em>Nucleic Acids Res. 2020 Jul 2;48(W1):W177-W184.</li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li>Hicexplorer Main Site: <a href="https://github.com/deeptools/HiCExplorer">GitHub</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: hicexplorer (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=8g --cpus-per-task=8 --gres=lscratch:100</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn4224 are ready for job

[user@cn4224 ~]$ <b>module load hicexplorer bowtie samtools</b>
[+] Loading hicexplorer  3.5.1  on cn4224 
[+] Loading singularity  3.7.0  on cn4224
[+] Loading bowtie  2-2.4.2 
[+] Loading samtools 1.11  ... 

[user@cn4224 ~]$ <b>cd /lscratch/${SLURM_JOB_ID}</b>

[user@cn4224 ~]$ <b>wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/005/SRR3950565/SRR3950565_1.fastq.gz</b>
[user@cn4224 ~]$ <b>wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/005/SRR3950565/SRR3950565_2.fastq.gz</b>
[user@cn4224 ~]$ <b>wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/006/SRR3950566/SRR3950566_1.fastq.gz</b>
[user@cn4224 ~]$ <b>wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/006/SRR3950566/SRR3950566_2.fastq.gz</b>
[user@cn4224 ~]$ <b>wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/009/SRR3950559/SRR3950559_1.fastq.gz</b>
[user@cn4224 ~]$ <b>wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/009/SRR3950559/SRR3950559_2.fastq.gz</b>
[user@cn4224 ~]$ <b>wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/000/SRR3950560/SRR3950560_1.fastq.gz</b>
[user@cn4224 ~]$ <b>wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/000/SRR3950560/SRR3950560_2.fastq.gz</b>

[user@cn4224 ~]$ <b>bowtie2 -x mm9_index --threads 8 -U SRR3950565_1.fastq.gz --reorder | samtools view -Shb - > SRR3950565_1.bam</b>
[user@cn4224 ~]$ <b>bowtie2 -x mm9_index --threads 8 -U SRR3950565_2.fastq.gz --reorder | samtools view -Shb - > SRR3950565_2.bam</b>
[user@cn4224 ~]$ <b>bowtie2 -x mm9_index --threads 8 -U SRR3950566_1.fastq.gz --reorder | samtools view -Shb - > SRR3950566_1.bam</b>
[user@cn4224 ~]$ <b>bowtie2 -x mm9_index --threads 8 -U SRR3950566_2.fastq.gz --reorder | samtools view -Shb - > SRR3950566_2.bam</b>
[user@cn4224 ~]$ <b>bowtie2 -x mm9_index --threads 8 -U SRR3950559_1.fastq.gz --reorder | samtools view -Shb - > SRR3950559_1.bam</b>
[user@cn4224 ~]$ <b>bowtie2 -x mm9_index --threads 8 -U SRR3950559_2.fastq.gz --reorder | samtools view -Shb - > SRR3950559_2.bam</b>
[user@cn4224 ~]$ <b>bowtie2 -x mm9_index --threads 8 -U SRR3950560_1.fastq.gz --reorder | samtools view -Shb - > SRR3950560_1.bam</b>
[user@cn4224 ~]$ <b>bowtie2 -x mm9_index --threads 8 -U SRR3950560_2.fastq.gz --reorder | samtools view -Shb - > SRR3950560_2.bam</b>

[user@cn4224 ~]$ <b>hicBuildMatrix --samFiles SRR3950565_1.bam SRR3950565_2.bam  \
                                   --binSize 1000 \
                                   --restrictionSequence GATC \
                                   --outFileName SRR3950565.cool \
                                   --QCfolder SRR3950565_QC \
                                   --threads 6</b>
[user@cn4224 ~]$ <b>hicBuildMatrix --samFiles SRR3950566_1.bam SRR3950566_2.bam  \
                                   --binSize 1000 --restrictionSequence GATC \
                                   --outFileName SRR3950566.cool \
                                   --QCfolder SRR3950566_QC \
                                   --threads 6</b>
[user@cn4224 ~]$ <b>hicBuildMatrix --samFiles SRR3950559_1.bam SRR3950559_2.bam \
                                   --binSize 1000 --restrictionSequence GATC \
                                   --outFileName SRR3950559.cool \
                                   --QCfolder SRR3950559_QC \
                                   --threads 6</b>
[user@cn4224 ~]$ <b>hicBuildMatrix --samFiles SRR3950560_1.bam SRR3950560_2.bam \
                                   --binSize 1000 --restrictionSequence GATC \
                                   --outFileName SRR3950560.cool \
                                   --QCfolder SRR3950560_QC \
                                   --threads 6</b>
[...]

[user@cn4224 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226

[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. hicexplorer.sh) similar to the following.</p>

<pre class="term">
#! /bin/bash

set -e

module load hicexplorer

cd /lscratch/${SLURM_JOB_ID}

wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/005/SRR3950565/SRR3950565_1.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/005/SRR3950565/SRR3950565_2.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/006/SRR3950566/SRR3950566_1.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/006/SRR3950566/SRR3950566_2.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/009/SRR3950559/SRR3950559_1.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/009/SRR3950559/SRR3950559_2.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/000/SRR3950560/SRR3950560_1.fastq.gz
wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR395/000/SRR3950560/SRR3950560_2.fastq.gz

bowtie2 -x mm9_index --threads 8 -U SRR3950565_1.fastq.gz --reorder | samtools view -Shb - > SRR3950565_1.bam
bowtie2 -x mm9_index --threads 8 -U SRR3950565_2.fastq.gz --reorder | samtools view -Shb - > SRR3950565_2.bam
bowtie2 -x mm9_index --threads 8 -U SRR3950566_1.fastq.gz --reorder | samtools view -Shb - > SRR3950566_1.bam
bowtie2 -x mm9_index --threads 8 -U SRR3950566_2.fastq.gz --reorder | samtools view -Shb - > SRR3950566_2.bam
bowtie2 -x mm9_index --threads 8 -U SRR3950559_1.fastq.gz --reorder | samtools view -Shb - > SRR3950559_1.bam
bowtie2 -x mm9_index --threads 8 -U SRR3950559_2.fastq.gz --reorder | samtools view -Shb - > SRR3950559_2.bam
bowtie2 -x mm9_index --threads 8 -U SRR3950560_1.fastq.gz --reorder | samtools view -Shb - > SRR3950560_1.bam
bowtie2 -x mm9_index --threads 8 -U SRR3950560_2.fastq.gz --reorder | samtools view -Shb - > SRR3950560_2.bam

hicBuildMatrix --samFiles SRR3950565_1.bam SRR3950565_2.bam  \
                                   --binSize 1000 \
                                   --restrictionSequence GATC \
                                   --outFileName SRR3950565.cool \
                                   --QCfolder SRR3950565_QC \
                                   --threads 6
hicBuildMatrix --samFiles SRR3950566_1.bam SRR3950566_2.bam  \
                                   --binSize 1000 --restrictionSequence GATC \
                                   --outFileName SRR3950566.cool \
                                   --QCfolder SRR3950566_QC \
                                   --threads 6
hicBuildMatrix --samFiles SRR3950559_1.bam SRR3950559_2.bam \
                                   --binSize 1000 --restrictionSequence GATC \
                                   --outFileName SRR3950559.cool \
                                   --QCfolder SRR3950559_QC \
                                   --threads 6
hicBuildMatrix --samFiles SRR3950560_1.bam SRR3950560_2.bam \
                                   --binSize 1000 --restrictionSequence GATC \
                                   --outFileName SRR3950560.cool \
                                   --QCfolder SRR3950560_QC \
                                   --threads 6
</pre>

<p>Submit these jobs using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile for the first step of the pipeline (e.g. hicexplorer.swarm). For example:</p>

<pre class="term">
hicBuildMatrix --samFiles SRR3950565_1.bam SRR3950565_2.bam  \
                                   --binSize 1000 \
                                   --restrictionSequence GATC \
                                   --outFileName SRR3950565.cool \
                                   --QCfolder SRR3950565_QC \
                                   --threads 6
hicBuildMatrix --samFiles SRR3950566_1.bam SRR3950566_2.bam  \
                                   --binSize 1000 --restrictionSequence GATC \
                                   --outFileName SRR3950566.cool \
                                   --QCfolder SRR3950566_QC \
                                   --threads 6
hicBuildMatrix --samFiles SRR3950559_1.bam SRR3950559_2.bam \
                                   --binSize 1000 --restrictionSequence GATC \
                                   --outFileName SRR3950559.cool \
                                   --QCfolder SRR3950559_QC \
                                   --threads 6
hicBuildMatrix --samFiles SRR3950560_1.bam SRR3950560_2.bam \
                                   --binSize 1000 --restrictionSequence GATC \
                                   --outFileName SRR3950560.cool \
                                   --QCfolder SRR3950560_QC \
                                   --threads 6
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f hicexplorer.swarm [-g #] --module hicexplorer</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>--module hicexplorer <td>Loads the hicexplorer module for each subjob in the swarm
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
