<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'HLA-LA on Biowulf';</script>
<div class="title">HLA-LA on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><a href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#ref">Additional References</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
      </div>
</table>

<p>
HLA*LA carries out HLA typing based on a population reference graph and employs a new linear projection method to align reads to the graph.

Previously called HLA*PRG:LA, the application was developed by Alexander Dilthey at NHGRI.

<h3>Reference:</h3>
<ul>
  <li>
    Alexander T Dilthey, Alexander J Mentzer, Raphael Carapito, Clare Cutland, Nezih Cereb, Shabir A Madhi, Arang Rhie, Sergey Koren, Seiamak Bahram, Gil McVean, Adam M Phillippy
    <a href="https://academic.oup.com/bioinformatics/article/35/21/4394/5426702"><b><u>HLA*LA—HLA typing from linearly projected graph alignments.</u></b></a><em>Bioinformatics, Volume 35, Issue 21, 1 November 2019, Pages 4394–4396</em>
  </li>
</ul>

<P>

</p>

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li> <a href="https://github.com/DiltheyLab/HLA-LA/">Github Site</a></li>
<li> <a href="https://genomeinformatics.github.io/HLA-PRG-LA/">Description of the algorithm</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>HLA-LA</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
<li>Multithreaded.</li>
<li>Environment variables set:</li>
  <ul>
    <li><tt>HLA_LA_GRAPHS</tt> HLA*LA Graph References</li>
    <li><tt>HLA_LA_TESTDATA</tt> HLA*LA Test Data</li>
  </ul>
</ul>

<a Name="ref"></a><div class="heading">Additional References</div>
<p>
Please contact the HPC staff (<tt>staff@hpc.nih.gov</tt>) if you want additional reference files installed for HLA-LA. 
Note that HLA-LA will break for all users if your reference file is incorrectly formatted.
See <a href="https://github.com/DiltheyLab/HLA-LA/#adding-further-references" target="blank">this documentation</a> for the format.
We highly recommend using an already installed reference as a template. For example: </p> 
<pre class="term">
[user@cn3144 ~] module load HLA-LA
[user@cn3144 ~] cd $HLA_LA_GRAPHS/PRG_MHC_GRCh38_withIMGT/knownReferences
[user@cn3144 knownReferences] head PRG_MHC_GRCh38_withIMGT.txt > ~/testgraph.txt
[user@cn3144 knownReferences] tail PRG_MHC_GRCh38_withIMGT.txt >> ~/testgraph.txt
</pre>

<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --gres=lscratch:50 --cpus-per-task=8 --mem=60g </b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load HLA-LA</b>
[+] Loading singularity  3.10.5  
[+] Loading HLA-LA  1.0.3 

[user@cn3144 ~]$ <b>cd /lscratch/$SLURM_JOBID</b>

[user@cn3144 46116226]$ <b>cp $HLA_LA_TESTDATA/NA12878.mini.cram  .</b>

[user@cn3144 46116226]$ <b>samtools index NA12878.mini.cram</b>

[user@cn3144 46116226]$ <b>HLA-LA.pl --BAM NA12878.mini.cram \
      --graph PRG_MHC_GRCh38_withIMGT --sampleID NA12878 \
	    --maxThreads 7 --workingDir .</b>
HLA-LA.pl

Identified paths:
        samtools_bin: /opt/conda/envs/hla-la/bin/samtools
        bwa_bin: /opt/conda/envs/hla-la/bin/bwa
        java_bin: /opt/conda/envs/hla-la/bin/java
        picard_sam2fastq_bin: /opt/conda/envs/hla-la/bin/picard
        General working directory: /lscratch/4506949
        Sample-specific working directory: /lscratch/4506949/NA12878

Using /opt/conda/envs/hla-la/opt/hla-la/src/../graphs/PRG_MHC_GRCh38_withIMGT/knownReferences/1000G_B38.txt as reference file.
Extract reads from 534 regions...
Extract unmapped reads...
Merging...
Indexing...
Extract FASTQ...
        /opt/conda/envs/hla-la/bin/picard SamToFastq VALIDATION_STRINGENCY=LENIENT I=/lscratch/4506949/NA12878/extraction.bam F=/lscratch/4506949/NA12878/R_1.fastq F2=/lscratch/4506949/NA12878/R_2.fastq FU=/lscratch/4506949/NA12878/R_U.fastq 2>&1

Now executing:
../bin/HLA-LA --action HLA --maxThreads 7 --sampleID NA12878 --outputDirectory /lscratch/4506949/NA12878 --PRG_graph_dir /opt/conda/envs/hla-la/opt/hla-la/src/../graphs/PRG_MHC_GRCh38_withIMGT --FASTQU /lscratch/4506949/NA12878/R_U.fastq.splitLongReads --FASTQ1 /lscratch/4506949/NA12878/R_1.fastq --FASTQ2 /lscratch/4506949/NA12878/R_2.fastq --bwa_bin /opt/conda/envs/hla-la/bin/bwa --samtools_bin /opt/conda/envs/hla-la/bin/samtools --mapAgainstCompleteGenome 1 --longReads 0
Set maxThreads to 7

[...]

[user@cn3144 46116226]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. HLA.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
cd /lscratch/$SLURM_JOBID
module load HLA-LA

cp /data/$USER/myfile.cram .
samtools index myfile.cram

cpus=$(( SLURM_CPUS_PER_TASK - 1 ))
echo "Running on $cpus CPUs"
HLA-LA.pl --BAM myfile.cram --graph PRG_MHC_GRCh38_withIMGT --sampleID myfile --maxThreads $cpus --workingDir .

# copy output from /lscratch back to /data area
cp -r myfile/hla  /data/$USER/

</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=32 --mem=100g --gres=lscratch:100 --time=1-00:00:00 HLA.sh</pre>



<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
