<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'HTSeq on Biowulf';</script>
    <div class="title">HTSeq on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>    <p> HTSeq is a Python package that provides infrastructure to process data from 
      high-throughput sequencing assays. It is developed by <a href="http://www.embl.de/research/units/genome_biology/huber/members/index.php?s_personId=6001">Simon 
      Anders</a> at EMBL Heidelberg.</p>
    <p></p>
    <h3>References:</h3>    <ul>
      <li><a href="https://academic.oup.com/bioinformatics/article/31/2/166/2366196">https://academic.oup.com/bioinformatics/article/31/2/166/2366196</a></li>
    </ul>


<a Name="doc"></a><div class="heading">Documentation</div>    <ul>
      <li><a href="http://www-huber.embl.de/users/anders/HTSeq/doc/tour.html#tour">http://www-huber.embl.de/users/anders/HTSeq/doc/tour.html#tour</a></li>
    </ul>

<div class="heading">Important Notes</div>
    <ul>
      <li>Module Name: <tt>htseq</tt> (see <a href="/apps/modules.html">the modules 
        page</a> for more information) 
      <li>Example files in <a href="http://www-huber.embl.de/users/anders/HTSeq/HTSeq_example_data.tgz">http://www-huber.embl.de/users/anders/HTSeq/HTSeq_example_data.tgz</a>
    </ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
    <pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <strong>module load htseq</strong>
[user@cn3144 ~]$ <strong>htseq-count input.sam input.gff</strong>

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>    <p>Create a batch input file (e.g. htseq.sh). For example:</p>    
    <pre class="term">
#!/bin/bash
set -e
module load htseq
htseq-count input.sam input.gff</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

    <pre class="term">sbatch [--mem=#] htseq.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

    <p>Create a swarmfile (e.g. htseq.swarm). For example:</p>

<pre class="term">
cd dir1; htseq-count input.sam input.gff
cd dir2; htseq-count input.sam input.gff
cd dir3; htseq-count input.sam input.gff
cd dir4; htseq-count input.sam input.gff
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

    <pre class="term">swarm -f htseq.swarm [-t #] --module htseq</pre>
where
    <table border=0>
      <tr>
        <td width=20%><tt>-g <i>#</i> </tt>
        <td>Number of Gigabytes of memory required for each process (1 line in 
          the swarm command file) 
      <tr>
        <td><tt>--module htseq</tt> 
        <td>Loads the htseq module for each subjob in the swarm 
    </table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
