<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'humann on Biowulf';</script>
<div class="title">humann on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p><em>From the humann home page:</em></p>

<blockquote> HUMAnN is a pipeline for efficiently and accurately profiling the
presence/absence and abundance of microbial pathways in a community from
metagenomic or metatranscriptomic sequencing data (typically millions of short
DNA/RNA reads). This process, referred to as functional profiling, aims to
describe the metabolic potential of a microbial community and its members. More
generally, functional profiling answers the question "What are the microbes in
my community-of-interest doing (or capable of doing)?" </blockquote>

<!--
<h3>References:</h3>
<ul>
</ul> 
-->


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="https://github.com/biobakery/humann">GitHub</a> </li>
    <li><a href="https://huttenhower.sph.harvard.edu/humann">Home page</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: humann (see <a href="/apps/modules.html">the modules page</a> 
    for more information)</li>
    <li>For historical reasons there are also modules called humann2 for the series 2 versions
    of humann. They point to the same installation as the equivalent humann modules.</li>
    <li>humann is a multithreaded application</li>
    <li>Reference data in /fdb/humann2/</li>
    <li>Example data in <code>$HUMANN_TEST_DATA</code></li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive -c12 --mem=24g --gres=lscratch:100</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load humann/3.6.0</b>
[user@cn3144 ~]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144 ~]$ <b>cp -r ${HUMANN_TEST_DATA:-none} demo</b>
[user@cn3144 ~]$ <b>humann --threads $SLURM_CPUS_PER_TASK --input demo/demo.fastq --output demo.out</b>
[user@cn3144 ~]$ # note that for the humann series 2 the command is `humann2`
WARNING: While bind mounting '/gs10:/gs10': destination is already in the mount point list
Creating output directory: /lscratch/46116226/demo.out
Output files will be written to: /lscratch/46116226/demo.out

Running metaphlan ........

Found g__Bacteroides.s__Bacteroides_dorei : 57.96% of mapped reads
Found g__Bacteroides.s__Bacteroides_vulgatus : 42.04% of mapped reads

Total species selected from prescreen: 2

Selected species explain 100.00% of predicted community composition


Creating custom ChocoPhlAn database ........


Running bowtie2-build ........


Running bowtie2 ........

Total bugs from nucleotide alignment: 2
g__Bacteroides.s__Bacteroides_vulgatus: 1274 hits
g__Bacteroides.s__Bacteroides_dorei: 1318 hits

Total gene families from nucleotide alignment: 548

Unaligned reads after nucleotide alignment: 87.6571428571 %


Running diamond ........


Aligning to reference database: uniref90_201901b_full.dmnd

Total bugs after translated alignment: 3
g__Bacteroides.s__Bacteroides_vulgatus: 1274 hits
g__Bacteroides.s__Bacteroides_dorei: 1318 hits
unclassified: 1599 hits

Total gene families after translated alignment: 815

Unaligned reads after translated alignment: 80.6190476190 %


Computing gene families ...

Computing pathways abundance and coverage ...

Output files created:
/lscratch/46116226/demo.out/demo_genefamilies.tsv
/lscratch/46116226/demo.out/demo_pathabundance.tsv
/lscratch/46116226/demo.out/demo_pathcoverage.tsv

[user@cn3144 ~]$ <b>ls -lh demo.out</b>
total 168K
-rw-r--r-- 1 user group 104K Dec 11 12:25 demo_genefamilies.tsv
drwxr-xr-x 2 user group 4.0K Dec 11 12:25 demo_humann_temp
-rw-r--r-- 1 user group 1.4K Dec 11 12:25 demo_pathabundance.tsv
-rw-r--r-- 1 user group 1.3K Dec 11 12:25 demo_pathcoverage.tsv
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. humann.sh), which uses the input file 'humann2.in'. For example:</p>

<pre class="term">
#! /bin/bash

module load humann/3.0.0 || exit 1
cd /lscratch/$SLURM_JOB_ID || exit 1
cp $HUMANN_TEST_DATA/demo.fastq .
mkdir out

# for humann version 2 modules, this command would be humann2
humann --threads $SLURM_CPUS_PER_TASK \
  --input demo.fastq \
  --output out
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] humann.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. humann.swarm). For example:</p>

<pre class="term">
humann --input bample1.bam --output sample1.out
humann --input bample2.bam --output sample2.out
humann --input bample3.bam --output sample3.out
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f humann.swarm -g 10 -t 4 --module humann/3.0.0</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>-t # <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td>--module humann/XXXX <td>Loads the humann module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
