<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'HYPHY on Biowulf';</script>
<div class="title">HYPHY on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int-threaded">Interactive job (threaded version)</a></div>
        <div class="tocItem"><a href="#int-mpi">Interactive job (MPI version)</a></div>
	<div class="tocItem"><a href="#sbatch-threaded">Batch job -- threaded</a></div>
	<div class="tocItem"><a href="#sbatch-MPI">Batch job -- MPI</a></div>
      </div>
</table>

<p>
HyPhy (Hypothesis Testing using Phylogenies) is an open-source software package for the analysis of genetic sequences (in particular the inference of natural selection) using techniques in phylogenetics, molecular evolution, and machine learning. It features a rich scripting language for limitless customization of analyses. 
</p>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="http://hyphy.org/">HYPHY Main Site</a></li>
<li><A href="http://hyphy.org/tutorials/CLI-tutorial/">Command-line tool documentation</a>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
    <li>Module Name: <tt>hyphy</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>Multithreaded/MPI</li>
    <li>Environment variables set <!--for ones users should be aware of --></li>
    <ul>
        <li><tt>HYPHY_TEMPLATES -- the location for the hyphy template .bf files</tt></li>
        <li><tt>HYPHY_TESTDATA -- the location for the test data used in the examples below</tt></li>
    </ul>
</ul>

<p class="alert">HYPHYMPI v2.5.29 does not work on e7543 nodes. See the <a href="#int-mpi">MPI Interactive session</a> below for an example of how to allocate a different node type</p>
<P>

<a Name="int-threaded"></a><div class="heading">Interactive job </div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=8 --mem=20g --gres=lscratch:5</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load hyphy</b>

[user@cn3144 ~]$ <b>cd /lscratch/$SLURM_JOB_ID</b>

[user@cn3144 ~]$ <b>cp $HYPHY_TESTDATA/CD2.*  .</b>

[user@cn3144 ~]$ <b>hyphy CPU=$SLURM_CPUS_PER_TASK GARD --alignment CD2.nex --tree CD2.newick</b>

Analysis Description
--------------------
GARD : Genetic Algorithms for Recombination Detection. Implements a
heuristic approach to screening alignments of sequences for
recombination, by using the CHC genetic algorithm to search for
phylogenetic incongruence among different partitions of the data. The
number of partitions is determined using a step-up procedure, while the
placement of breakpoints is searched for with the GA. The best fitting
model (based on c-AIC) is returned; and additional post-hoc tests run to
distinguish topological incongruence from rate-variation. v0.2 adds and
spooling results to JSON after each breakpoint search conclusion

- __Requirements__: A sequence alignment.

- __Citation__: **Automated Phylogenetic Detection of Recombination Using a Genetic
Algorithm**, _Mol Biol Evol 23(10), 1891â€“1901

- __Written by__: Sergei L Kosakovsky Pond

- __Contact Information__: spond@temple.edu

- __Analysis Version__: 0.2

type: nucleotide
rv: None
>Loaded a nucleotide multiple sequence alignment with **10** sequences, **561** sites (390 of which are variable) from `/lscratch/46116226/CD2.nex`
>Minimum size of a partition is set to be 17 sites


### Fitting the baseline (single-partition; no breakpoints) model
* Log(L) = -3529.89, AIC-c =  7112.21 (25 estimated parameters)

### Performing an exhaustive single breakpoint analysis
Done with single breakpoint analysis.
   Best sinlge break point location: 25
   c-AIC  = 7106.656403051507

### Performing multi breakpoint analysis using a genetic algorithm
Done with 2 breakpoint analysis.
    Best break point locations: 25, 65
    c-AIC = 7088.977422652543
Done with 3 breakpoint analysis.
    Best break point locations: 25, 65, 173
    c-AIC = 7101.784222144825

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<a Name="int-mpi"></a><div class="heading">Interactive job - MPI version</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> with ntasks and run the program with mpirun.</p>
<p>HYPHYMPI does not run on e7543 nodes. Use the <a href="https://slurm.schedmd.com/sbatch.html#OPT_constraint"><tt>--constraint</tt></a> option to choose another node type like x6140 or x2695. See the feature table in the output of <tt>freen</tt> command to find different node types. </p>
<p>Sample session (user input in <b>bold</b>):</p>

<P>
<pre class="term">
[user@biowulf]$ <b>sinteractive --ntasks=8 --ntasks-per-core=1 --mem=20g --constraint=x6140</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load hyphy</b>

[user@cn3144 ~]$ <b>mkdir /data/$USER/hyphy</b>

[user@cn3144 ~]$ <b>cp $HYPHY_TESTDATA/CD2.*   /data/$USER/hyphy/ </b>

[user@cn3144 ~]$ <b>mpirun -np $SLURM_NTASKS HYPHYMPI gard </b>


Analysis Description
--------------------
GARD : Genetic Algorithms for Recombination Detection. Implements a
heuristic approach to screening alignments of sequences for
recombination, by using the CHC genetic algorithm to search for
phylogenetic incongruence among different partitions of the data. The
number of partitions is determined using a step-up procedure, while the
placement of breakpoints is searched for with the GA. The best fitting
model (based on c-AIC) is returned; and additional post-hoc tests run to
distinguish topological incongruence from rate-variation.

- __Requirements__: A sequence alignment.

- __Citation__: **Automated Phylogenetic Detection of Recombination Using a Genetic
Algorithm**, _Mol Biol Evol 23(10), 1891-1901

- __Written by__: Sergei L Kosakovsky Pond

- __Contact Information__: spond@temple.edu

- __Analysis Version__: 0.1

type: Nucleotide

Select a sequence alignment file <b>/data/$USER/hyphy/CD2.nex</b>
rv: None
>Loaded a Nucleotide multiple sequence alignment with **10** sequences, **561** sites (390 of which are variable) from `/data/$USER/hyphy/CD2.nex`
>Minimum size of a partition is set to be 17 sites


### Fitting the baseline (single-partition; no breakpoints) model
* Log(L) = -3529.89, AIC-c =  7112.21 (25 estimated parameters)

### Performing an exhaustive single breakpoint analysis
Done with single breakpoint analysis.
   Best sinlge break point location: 25
   c-AIC  = 7107.094096738777

### Performing multi breakpoint analysis using a genetic algorithm
[...]

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$

</pre>

<P>
<a Name="sbatch-threaded"></a><div class="heading">Batch job - threaded version</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. hyphy.sh). For example, to run the threaded version of hyphy:</p>

<pre class="term">
#!/bin/bash
set -e
module load hyphy
cp $HYPHY_TESTDATA/CD2* .

hyphy CPU=$SLURM_CPUS_PER_TASK slac --alignment CD2.nex --tree CD2.newick
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=# [--mem=#] hyphy.sh</pre>

<P>
<a Name="sbatch-MPI"></a><div class="heading">Batch job - MPI version</div>
<P>
To run the MPI version of hyphy, here is a sample batch script. 
<pre class="term">
#!/bin/bash

module load hyphy
# copy the test data
mkdir /data/$USER/hyphy
cp $HYPHY_TESTDATA/CD2.*   /data/$USER/hyphy/ 

mpirun -np $SLURM_NTASKS HYPHYMPI gard --alignment CD2.nex
</pre>
Submit this job with:
<pre class="term">
sbatch --ntasks=8 --ntasks-per-core=1 --mem=20g  --constraint=x6140 hyphympi.bat
</pre>



<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
