<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Scientific Applications on NIH HPC Systems';</script><div class="title">Scientific Applications on NIH HPC Systems</div><!-- Start content - do not edit above this line  -->

<table width=100% cellpadding=5>
  <tr>
    <td width=25%>
      <div class="toc">
      <div class="tocHeading">Application Areas</div>
        <div class="tocItem"><a href="#compchem" >Computational Chemistry</a></div>
<div class="tocItem"><a href="#deeplearning" >Deep Learning</a></div>
<div class="tocItem"><a href="/development/" >Development Tools</a></div>
<div class="tocItem"><a href="#hts" >High-Throughput Sequencing</a></div>
<div class="tocItem"><a href="#image" >Image Analysis</a></div>
<div class="tocItem"><a href="#linkage" >Linkage/Phylogenetics</a></div>
<div class="tocItem"><a href="#massspec" >Mass Spectrometry</a></div>
<div class="tocItem"><a href="#math" >Mathematical/Statistics</a></div>
<div class="tocItem"><a href="#model" >Molecular Modeling/Graphics</a></div>
<div class="tocItem"><a href="#seqanal" >Sequence Analysis</a></div>
<div class="tocItem"><a href="#structbio" >Structural Biology</a></div>
<div class="tocItem"><a href="#sysbiol" >Systems Biology</a></div>
<div class="tocItem"><a href="#utils" >Utilities</a></div>
<div class="tocItem"><a href="#workflow" >Workflow Managers</a></div>
      </div>
    </td>
    <td>
      <p>
The NIH HPC staff maintains several hundred  scientific programs, packages and databases for our users.
Below is a list of system-installed software available on Biowulf and Helix. Click on the application name to
get to site-specific instructions on how to run a given package on the cluster, including links to the 
original application documentation.
      </p>

      <p>
In almost all cases, applications are made available through the use of
<b><a href="modules.html">environment modules</a></b>.
      </p>

      <p>
Users are welcome to install additional applications in their own /home or /data areas, and create their own <a href="https://hpc.nih.gov/apps/modules.html#personal">personal modules</a> for those applications. Some applications are difficult to install or require admin privileges; in those cases, you can email staff@hpc.nih.gov with a request to install the application.
      </p>

    </td>
  </tr>
</table>

<div class="heading"><a name="compchem" id="compchem">Computational Chemistry</a></div>
<a href="/apps/" style="font-size:12px">back to top</a><br />
<p></p>
<div class="subheading"><a href="acemd.html">Acemd</a> (3.5.1) </div><p class="appDesc">ACEMD is a high performance molecular dynamics code for biomolecular systems designed specifically for NVIDIA GPUs. Simple and fast, ACEMD uses very similar commands and input files of NAMD and output files as NAMD or Gromacs. </p>
<div class="subheading"><a href="AMBER.html">AMBER</a> (22) </div><p class="appDesc">AMBER (Assisted Model Building with Energy Refinement) is a package of molecular simulation programs.</p>
<div class="subheading"><a href="ampl.html">AMPL</a> (1.3.0) </div><p class="appDesc">The Accelerating Therapeutics for Opportunites in Medicine (ATOM) Consortium Modeling PipeLine for Drug Discovery. AMPL is an open-source, modular, extensible software pipeline for building and sharing models to advance in silico drug discovery.</p>
<div class="subheading"><a href="APBS.html">APBS</a> (3.4.1) </div><p class="appDesc">APBS (Adaptive Poisson-Boltzmann Solver) is a software package for the numerical solution of the Poisson-Boltzmann equation (PBE), one of the most popular continuum models for describing electrostatic interactions between molecular solutes in salty, aqueous media.</p>
<div class="subheading">Autodock (4.2.6) </div><p class="appDesc">Autodock is a suite of automated docking tools. It is designed to predict how small molecules, such as substrates or drug candidates, bind to a receptor of known 3D structure.</p>
<div class="subheading"><a href="AutodockVina.html">AutodockVina</a> (1_1_2) </div><p class="appDesc">AutoDock Vina is a program for drug discovery, molecular docking and virtual screening, offering multi-core capability, high performance and enhanced accuracy and ease of use. It is closely tied to Autodock.</p>
<div class="subheading"><a href="charmm/index.html">CHARMM</a> (c46b1) </div><p class="appDesc">CHARMM is a  general and flexible software application for modeling the structure and behavior of molecular systems.</p>
<div class="subheading"><a href="cryoDRGN.html">cryoDRGN</a> (1.1.0) </div><p class="appDesc">CryoDRGN is an algorithm that leverages the representation power of deep neural networks
to directly reconstruct continuous distributions of 3D density maps
and map per-particle heterogeneity of single-particle cryo-EM datasets.
It contains interactive tools to visualize a dataset’s distribution of per-particle
variability, generate density maps for exploratory analysis, extract particle
subsets for use with other tools and generate trajectories to visualize molecular motions.</p>
<div class="subheading"><a href="GAMESS.html">GAMESS</a> (30Sep22-R2) </div><p class="appDesc">GAMESS is a general ab initio quantum chemistry package.</p>
<div class="subheading"><a href="Gaussian.html">Gaussian</a> (G16-C02) </div><p class="appDesc">Gaussian is a connected system of programs for performing semiempirical and ab initio molecular orbital (MO) calculations.</p>
<div class="subheading"><a href="Gromacs.html">gromacs</a> (2022.4) </div><p class="appDesc">Gromacs  is a versatile package to perform molecular dynamics, i.e. simulate the Newtonian equations of motion for systems with hundreds to millions of particles. It is primarily designed for biochemical molecules like proteins and lipids that have a lot of complicated bonded interactions, but since GROMACS is extremely fast at calculating the nonbonded interactions (that usually dominate simulations) many groups are also using it for research on non-biological systems, e.g. polymers.
</p>
<div class="subheading"><a href="NAMD.html">NAMD</a> (2.14) </div><p class="appDesc">NAMD is a parallel molecular dynamics program for UNIX platforms designed for high-performance simulations in structural biology. VMD, the associated molecular visualization program, is also available.</p>
<div class="subheading"><a href="parmed.html">parmed</a> (4.0.0) </div><p class="appDesc">Cross-program parameter and topology file editor and molecular mechanical simulator engine.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/psi4.html">Psi4</a> (1.6.1) </div><p class="appDesc">Psi4 is an ab-initio electronic structure code that supports various methods for calculating energies and gradients of molecular systems.</p>
<div class="subheading"><a href="qchem.html">Q-Chem</a> (5.0.1) </div><p class="appDesc">Q-Chem is a comprehensive ab initio quantum chemistry package for accurate predictions of molecular structures, reactivities, and vibrational, electronic and NMR spectra.</p>
<div class="subheading"><a href="schrodinger/index.html">Schrodinger</a> (2023.1) </div><p class="appDesc">A limited number of Schrödinger applications are available on the Biowulf cluster through the Molecular Modeling Interest Group. Most are available through the Maestro GUI.</p>
<div class="subheading"><a href="/apps/vmd.html">vmd</a> (1.9.3) </div><p class="appDesc">VMD is a molecular visualization program for displaying, animating, and analyzing large biomolecular systems using 3-D graphics and built-in scripting. To use, type vmd at the prompt.</p>
<div class="heading"><a name="deeplearning" id="deeplearning">Deep Learning</a></div>
<a href="/apps/" style="font-size:12px">back to top</a><br />
<p></p>
<div class="subheading"><a href="b-soid.html">B-SOID</a> (1.3) </div><p class="appDesc">B-SOID (Behavioral Segmentation in Deeplabcut) is an 
unsupervised learning algorithm that serves to discover and classify behaviors that are not pre-defined by users.
It segregates statistically different, sub-second rodent behaviors with a single bottom-up perspective video cameraR by performing a novel expectation maximization
fitting of Gaussian mixture models on t-Distributed Stochastic Neighbor Embedding (t-SNE).</p>
<div class="subheading"><a href="BioBERT.html">BioBERT</a> (v20200409) </div><p class="appDesc">BioBERT is a biomedical language representation model
designed for biomedical text mining tasks
such as biomedical named entity recognition, relation extraction, question answering, etc.</p>
<div class="subheading"><a href="biogans.html">BioGANs</a> (20210916) </div><p class="appDesc">BioGANs is a novel application of Generative Adversarial Networks (GAN) to the synthesis of cells
imaged by fluorescence microscopy. It allows to infer the correlation between the spatial pattern
of different fluorescent proteins that reflects important biological functions.
The synthesized images capture these relationships, which are relevant for biological applications.</p>
<div class="subheading"><a href="cryoDRGN.html">cryoDRGN</a> (1.1.0) </div><p class="appDesc">CryoDRGN is an algorithm that leverages the representation power of deep neural networks
to directly reconstruct continuous distributions of 3D density maps
and map per-particle heterogeneity of single-particle cryo-EM datasets.
It contains interactive tools to visualize a dataset’s distribution of per-particle
variability, generate density maps for exploratory analysis, extract particle
subsets for use with other tools and generate trajectories to visualize molecular motions.</p>
<div class="subheading"><a href="https://hpc.nih.gov/development/cuDNN.html">cuDNN</a> (8.0.3) </div><p class="appDesc">The NVIDIA CUDA Deep Neural Network library (cuDNN) is a GPU-accelerated library of primitives for deep neural networks. cuDNN provides highly tuned implementations for standard routines such as forward and backward convolution, pooling, normalization, and activation layers.</p>
<div class="subheading"><a href="DanQ.html">DanQ</a> (20220825) </div><p class="appDesc">DanQ is a hybrid convolutional and recurrent deep
neural network for quantifying the function of DNA sequences</p>
<div class="subheading"><a href="deepcadrt.html">deepcadrt</a> (0.1.0) </div><p class="appDesc">Real-time denoising of fluorescence time-lapse imaging using deep self-supervised learning</p>
<div class="subheading"><a href="deepcell-tf.html">DeepCell-tf</a> (0.12.6) </div><p class="appDesc">The DeepCell-tf  library allows users to apply pre-existing models to imaging data
as well as to develop new deep learning models for single-cell analysis.
The library specializes in models for cell segmentation (whole-cell and nuclear) in 2D and 3D images
as well as cell tracking in 2D time-lapse datasets.
The models are applicable to data ranging from multiplexed images of tissues to dynamic live-cell imaging movies.</p>
<div class="subheading"><a href="DeepEMhancer.html">DeepEmhancer</a> (0.15) </div><p class="appDesc">A deep learning solution for cryo-EM map.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/DeepLabCut.html">DeepLabCut</a> (2.3.4) </div><p class="appDesc">DeepLabCut is an open source toolbox
that builds on a state-of-the-art human pose estimation algorithm.
It allows training of a deep neural network by using limited training data
to precisely track user-defined features, so that the human labeling accuracy will be matched.</p>
<div class="subheading"><a href="deepmedic.html">deepmedic</a> (0.8.4) </div><p class="appDesc">This project aims to offer easy access to Deep Learning for segmentation of structures of interest in biomedical 3D scans. It is a system that allows the easy creation of a 3D Convolutional Neural Network, which can be trained to detect and segment structures if corresponding ground truth labels are provided for training. The system processes NIFTI images, making its use straightforward for many biomedical tasks.</p>
<div class="subheading"><a href="DeepMM.html">DeepMM</a> (20220830) </div><p class="appDesc">DeepMM implements fully automated de novo structure modeling method, MAINMAST, which builds three-dimensional models of a protein from a near-atomic resolution EM map. The method directly traces the protein’s main-chain and identifies Cα positions as tree-graph structures in the EM map.</p>
<div class="subheading"><a href="GCN_Cancer.html">GCN_Cancer</a> (20221105) </div><p class="appDesc">The GCN_Cancer application employs graph convolutional network (GCN) models
to classify the gene expression data samples from The Cancer Genonme Atlas (TCAG)
as 33 designated tumor types or as normal. It has been trained on  10,340
cancer samples and 731 normal tissue samples from TCGA dataset.</p>
<div class="subheading"><a href="IsoNet.html">IsoNet</a> (0.2.1) </div><p class="appDesc">IsoNet is a deep learning-based software package that iteratively reconstructs the missing-wedge information and increases signal-to-noise ratio, using the knowledge learned from raw tomograms. Without the need for sub-tomogram averaging, IsoNet generates tomograms with significantly reduced resolution anisotropy.</p>
<div class="subheading"><a href="https://keras.io/">keras</a> (2.4.3; 2.9.1) </div><p class="appDesc">Keras is a deep learning API written in Python, running on top of the machine learning platform TensorFlow. It was developed with a focus on enabling fast experimentation. Being able to go from idea to result as fast as possible is key to doing good research.</p>
<div class="subheading"><a href="PyTom.html">PyTom</a> (1.0) </div><p class="appDesc">PyTom is a software package for the analysis of volumetric data
obtained by cryo electron tomography (cryo-ET).
It covers a complete pipeline of processing steps
for tomogram reconstruction, localization of macromolecular complexes in tomograms,
fine alignment of subtomograms extracted at these locations, and their classification.</p>
<div class="subheading"><a href="ReLeaSE.html">ReLeaSE</a> (20220825) </div><p class="appDesc">ReLeaSE (Reinforcement Learning for Structural Evolution) is an application for de-novo Drug Design
based on Reinforcement Learning. It integrates two deep neural networks: generative and
predictive, that are trained separately but are used jointly to generate novel targeted chemical libraries.
ReLeaSE uses simple representation of molecules by their simplified molecular input line entry specification (SMILES)          strings only.</p>
<div class="subheading"><a href="RFdiffusion.html">RFdiffusion</a> (1.1.0) </div><p class="appDesc">Rosetta Fold (RF) dissusion is an open source method for structure generation, with or without conditional information
(a motif, target etc). It can perform motif scaffolding, unconditional protein generation, and other tasks.</p>
<div class="subheading"><a href="SpliceAI.html">SpliceAI</a> (1.3.1) </div><p class="appDesc">SpliceAI is a deep neural network that accurately predicts
splice junctions from an arbitrary pre-mRNA
transcript sequence, enabling precise prediction of
noncoding genetic variants that cause cryptic
splicing.
</p>
<div class="subheading"><a href="talos.html">talos</a> (1.0) </div><p class="appDesc">Talos is a hyperparameter optimization package for deep learning. It works with any Keras, TensorFlow (tf.keras) or PyTorch model, takes minutes to implement,
involves no new syntax to learn and adds zero new overhead to your workflow.</p>
<div class="subheading"><a href="TensorQTL.html">TensorQTL</a> (1.0.7) </div><p class="appDesc">ensoorQTL leverages general-purpose libraries and graphics processing units
(GPUs) to achieve high efficiency of computations at low costR. Using PyTorch or TensorFlow
it allows > 200-fold decreases in runtime and ~ 5–10-fold reductions in cost when
running on GPUs relative to CPUs.</p>
<div class="subheading"><a href="Tybalt.html">Tybalt</a> (202208265) </div><p class="appDesc">Tybalt implements a Variational EutoEncoder (VAE), a deep neural network approach capable of generating meaningful latent spaces for image and text data. Tybalt has been trained on The Cancer Genome Atlas (TCGA) pan-cancer RNA-seq data and used to identify specific patterns in the VAE encoded features.
</p>
<div class="subheading"><a href="UNet.html">UNet</a> (20220825) </div><p class="appDesc">U-Net is an image segmentation tool. It relies on the strong use of data augmentation to use the available
annotated samples more efficiently. The architecture consists of a contracting
path to capture context and a symmetric expanding path that enables precise localization.</p>
<div class="heading"><a name="hts" id="hts">High-Throughput Sequencing</a></div>
<a href="/apps/" style="font-size:12px">back to top</a><br />
<p></p>
<div class="subheading"><a href="abyss.html">abyss</a> (2.3.5) </div><p class="appDesc">Abyss represents Assembly By Short Sequences - a de novo, parallel, paired-end sequence assembler. The parallel version is implemented using MPI and is capable of assembling larger genomes.</p>
<div class="subheading"><a href="AfterQC.html">AfterQC</a> (0.9.7) </div><p class="appDesc">Automatic Filtering, Trimming, Error Removing and Quality Control for fastq data. 
</p>
<div class="subheading"><a href="angsd.html">angsd</a> (0.940) </div><p class="appDesc"> ANGSD is a software for analyzing next generation sequencing data. The software can handle a number of different input types from mapped reads to imputed genotype probabilities. Most methods take genotype uncertainty into account instead of basing the analysis on called genotypes. This is especially useful for low and medium depth data. The software is written in C++ and has been used on large sample sizes. </p>
<div class="subheading"><a href="AnnotSV.html">AnnotSV</a> (3.3.1) </div><p class="appDesc">AnnotSV is a program designed for annotating Structural Variations (SV). This tool compiles functionally, regulatory and clinically relevant information and aims at providing annotations useful to i) interpret SV potential pathogenicity and ii) filter out SV potential false positives.
</p>
<div class="subheading"><a href="ANNOVAR.html">ANNOVAR</a> (2020-06-08) </div><p class="appDesc">ANNOVAR is an efficient software tool to utilize update-to-date information to functionally annotate genetic variants detected from diverse genomes.</p>
<div class="subheading"><a href="ascatNgs.html">ascatNgs</a> (4.5.0) </div><p class="appDesc">AscatNGS contains the Cancer Genome Projects workflow implementation of the ASCAT copy number algorithm for paired end sequencing.</p>
<div class="subheading"><a href="augustus.html">augustus</a> (3.4.0) </div><p class="appDesc">AUGUSTUS is a program that predicts genes in eukaryotic genomic sequences. </p>
<div class="subheading"><a href="autoreject.html">autoreject</a> (0.3.1) </div><p class="appDesc">This is a library to automatically reject bad trials and repair bad sensors in magneto-/electroencephalography (M/EEG) data.</p>
<div class="subheading"><a href="bam2fastq.html">bam2fastq</a> (1.1.0) </div><p class="appDesc">This tool is used to extract raw sequences (with qualities) from bam files. </p>
<div class="subheading"><a href="bamliquidator.html">bamliquidator</a> (1.5.2) </div><p class="appDesc">bamliquidator is a set of tools for analyzing the density of short DNA sequence read alignments in the BAM file format.</p>
<div class="subheading"><a href="bamreadcount.html">bamreadcount</a> (cram-v0.0.1) </div><p class="appDesc">Bam-readcount generates metrics at single nucleotide positions.
There are number of metrics generated which can be useful for filtering out false positive calls.</p>
<div class="subheading"><a href="bamtools.html">bamtools</a> (2.5.2) </div><p class="appDesc">BamTools provides a fast, flexible C++ API & toolkit for reading, writing, and manipulating BAM files. </p>
<div class="subheading"><a href="bamUtil.html">bamUtil</a> (1.0.15) </div><p class="appDesc">bamUtil is a repository that contains several programs that perform operations on SAM/BAM files. All of these programs are built into a single executable, bam.</p>
<div class="subheading"><a href="bartender.html">Bartender</a> (1.1) </div><p class="appDesc">Bartender is an accurate clustering algorithm to detect barcodes
and their abundances from raw next-generation sequencing data.
In contrast with existing methods that cluster based on sequence similarity alone,
Bartender uses a modified two-sample proportion test that also considers cluster size. This modification
results in higher accuracy and lower rates of under- and over-clustering artifacts.</p>
<div class="subheading"><a href="basespace_cli.html">basespace_cli</a> (1.5.2) </div><p class="appDesc">Command line interface for Illumina's BaseSpace</p>
<div class="subheading"><a href="bazam.html">bazam</a> (1.0.1) </div><p class="appDesc">A tool to extract paired reads in FASTQ format from coordinate sorted BAM files</p>
<div class="subheading"><a href="bbtools.html">bbtools</a> (38.96) </div><p class="appDesc">An extensive set of bioinformatics tools including bbmap (short read aligner), bbnorm (kmer based normalization), dedupe (deduplication and clustering of unaligned reads), reformat (formatting and trimming reads) and many more.</p>
<div class="subheading"><a href="bcl-convert.html">bcl-convert</a> (4.1.5) </div><p class="appDesc">The Illumina BCL Convert is a standalone local software app that converts the Binary Base Call (BCL) files produced by Illumina sequencing systems to FASTQ files. BCL Convert also provides adapter handling (through masking and trimming) and UMI trimming and produces metric outputs.</p>
<div class="subheading"><a href="bcl2fastq.html">bcl2fastq</a> (2.20) </div><p class="appDesc">a tool to handle bcl conversion and demultiplexing</p>
<div class="subheading"><a href="bedops.html">bedops</a> (2.4.41) </div><p class="appDesc">Bedops is a suite of tools to address common questions raised in genomic studies - mostly with regard to overlap and proximity relationships between data sets - BEDOPS aims to be scalable and flexible, facilitating the efficient and accurate analysis and management of large-scale genomic data.</p>
<div class="subheading"><a href="bedtools.html">bedtools</a> (2.30.0) </div><p class="appDesc">The BEDTools utilities allow one to address common genomics tasks such finding feature overlaps and computing coverage. In addition, one can develop sophisticated pipelines that answer complicated research questions by "streaming" several BEDTools together.</p>
<div class="subheading"><a href="beta.html">BETA</a> (1.0.7) </div><p class="appDesc">Binding and expression target analysis (BETA) is a software package that integrates ChIP-seq of TFs or chromatin regulators with differential gene expression data to infer direct target genes. The combination of ChIP-seq and transcriptome analysis is a compelling approach to unravel the regulation of gene expression.
</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/bigSCale2.html">bigSCale2</a> (20191119) </div><p class="appDesc">bigSCale is a complete framework for the analysis and visualization of single cell data. It allows to cluster, phenotype, perform pseudotime analysis, infer gene regulatory networks and reduce large datasets in smaller datasets with higher quality.
</p>
<div class="subheading"><a href="bioawk.html">bioawk</a> (1.0) </div><p class="appDesc">Regular awk with support for several common biological data formats, including optionally gzip'ed BED, GFF, SAM, VCF, FASTA/Q and TAB-delimited formats with column names.</p>
<div class="subheading"><a href="biobakery_workflows.html">biobakery_workflows</a> (3.1) </div><p class="appDesc">bioBakery is a meta’omic analysis environment and collection of individual software
tools with the capacity to process raw shotgun sequencing data into actionable microbial community
feature profiles, summary reports, and publication-ready figures.
It includes a collection of preconfigured analysis modules also joined into workflows for reproducibility.
Each individual module has been developed to perform a particular task, e.g. quantitative taxonomic
profiling or statistical analysis.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/biobambam2.html">biobambam2</a> (2.0.185-release-20221211202123) </div><p class="appDesc">Tools for early stage alignment file processing.</p>
<div class="subheading"><a href="biom-format.html">biom-format</a> (2.1.12) </div><p class="appDesc">tool (and library) to manipulate Biological Observation Matrix (BIOM) Format files</p>
<div class="subheading"><a href="bismark.html">bismark</a> (0.23.1) </div><p class="appDesc">Bismark is a program to map bisulfite treated sequencing reads to a genome of interest and perform methylation calls in a single step. The output can be easily imported into a genome viewer, such as SeqMonk, and enables a researcher to analyse the methylation levels of their samples straight away.</p>
<div class="subheading"><a href="bonito.html">bonito</a> (0.7.2) </div><p class="appDesc">A PyTorch Basecaller for Oxford Nanopore Reads</p>
<div class="subheading"><a href="bowtie.html">bowtie</a> (1.3.1) </div><p class="appDesc">bowtie is an ultrafast, memory-efficient short read aligner geared toward quickly aligning large sets of short DNA sequences (reads) to large genomes.</p>
<div class="subheading"><a href="bowtie2.html">bowtie2</a> (2.5.1) </div><p class="appDesc">A version of bowtie that's particularly good at aligning reads of about 50 up to 100s or 1,000s of characters, and particularly good at aligning to relatively long (e.g. mammalian) genomes</p>
<div class="subheading"><a href="BRASS.html">BRASS</a> (6.3.4) </div><p class="appDesc">BRASS analyses one or more related BAM files of paired-end sequencing to determine potential rearrangement breakpoints.
</p>
<div class="subheading"><a href="breakdancer.html">breakdancer</a> (1.4.5) </div><p class="appDesc">provides genome-wide detection of structural variants from next generation paired-end sequencing reads.</p>
<div class="subheading"><a href="breseq.html">breseq</a> (0.37.1) </div><p class="appDesc">breseq is a computational pipeline for finding mutations relative to a reference sequence in short-read DNA re-sequencing data. It is intended for haploid microbial genomes (<20 Mb).</p>
<div class="subheading"><a href="bsmap.html">bsmap</a> (2.90) </div><p class="appDesc">BSMAP is a short reads mapping software for bisulfite sequencing reads. Bisulfite treatment converts unmethylated Cytosines into Uracils (sequenced as Thymine) and leave methylated Cytosines unchanged, hence provides a way to study DNA cytosine methylation at single nucleotide resolution. BSMAP aligns the Ts in the reads to both Cs and Ts in the reference.
</p>
<div class="subheading"><a href="busco.html">busco</a> (5.4.7) </div><p class="appDesc">BUSCO completeness assessments employ sets of Benchmarking Universal Single-Copy Orthologs from OrthoDB (www.orthodb.org) to provide quantitative measures of the completeness of genome assemblies, annotated gene sets, and transcriptomes in terms of expected gene content. </p>
<div class="subheading"><a href="bwa.html">bwa</a> (0.7.17) </div><p class="appDesc">BWA is a  fast light-weighted tool that aligns short sequences to a sequence database, such as the human reference genome.</p>
<div class="subheading"><a href="bwa-mem2.html">bwa-mem2</a> (2.2.1) </div><p class="appDesc">The next version of the bwa-mem algorithm in bwa. </p>
<div class="subheading"><a href="CADD.html">CADD</a> (1.6.post1) </div><p class="appDesc">CADD (Combined Annotation Dependent Depletion) is a tool for scoring the deleteriousness of single nucleotide variants as well as insertion/deletions variants in the human genome. Currently, it supports the builds: GRCh37/hg19 and GRCh38/hg38.</p>
<div class="subheading"><a href="canu.html">Canu</a> (2.1) </div><p class="appDesc">Canu is a fork of the Celera Assembler designed for high-noise single-molecule sequencing (such as the PacBio RSII or Oxford Nanopore MinION). Canu will correct the reads, then trim suspicious regions (such as remaining SMRTbell adapter), then assemble the corrected and cleaned reads into unitigs.

</p>
<div class="subheading"><a href="Canvas.html">Canvas</a> (1.40) </div><p class="appDesc">Canvas is a tool for calling copy number variants (CNVs) from human DNA sequencing data.</p>
<div class="subheading"><a href="https://github.com/CCBR">ccbrpipeliner</a> (4.0.2) </div><p class="appDesc">CCBR Pipeliner provides access to a set of best-practices NGS pipelines developed, tested, and benchmarked by experts at CCBR and NCBR for Biowulf. Contact CCBR_Pipeliner@mail.nih.gov with questions</p>
<div class="subheading"><a href="ceas.html">ceas</a> (1.0.2) </div><p class="appDesc">Cis-regulatory Element Annotation System is a tool designed to characterize genome-wide protein-DNA interaction patterns from ChIP-chip and ChIP-Seq of both sharp and broad binding factors. </p>
<div class="subheading"><a href="cellprofiler.html">cellprofiler</a> (4.2.5) </div><p class="appDesc">An open-source application for biological image analysis</p>
<div class="subheading"><a href="cellranger.html">cellranger</a> (7.2.0) </div><p class="appDesc">Cell Ranger is a set of analysis pipelines that processes Chromium single cell 3’ RNA-seq output to align reads, generate gene-cell matrices and perform clustering and gene expression analysis.</p>
<div class="subheading"><a href="cellranger-arc.html">cellranger-arc</a> (2.0.2) </div><p class="appDesc">Cell Ranger ARC is a set of analysis pipelines that process Chromium Single Cell Multiome ATAC + Gene Expression sequencing data to generate a variety of analyses pertaining to gene expression, chromatin accessibility and their linkage. Furthermore, since the ATAC and gene expression measurements are on the very same cell, we are able to perform analyses that link chromatin accessibility and gene expression.</p>
<div class="subheading"><a href="cellranger-atac.html">cellranger-atac</a> (2.1.0) </div><p class="appDesc">Cell Ranger ATAC is a set of analysis pipelines that process Chromium Single Cell ATAC data. </p>
<div class="subheading"><a href="cellsnp-lite.html">cellsnp-lite</a> (1.2.2) </div><p class="appDesc">Efficient genotyping bi-allelic SNPs on single cells</p>
<div class="subheading"><a href="cgpBattenberg.html">cgpBattenberg</a> (3.5.3) </div><p class="appDesc">Detect subclonality and copy number in matched NGS data</p>
<div class="subheading"><a href="checkm2.html">checkm2</a> (1.0.2) </div><p class="appDesc">Rapid assessment of genome bin quality using machine learning.</p>
<div class="subheading"><a href="chess.html">CHESS</a> (0.3.7) </div><p class="appDesc">The CHESS (Comparison of Hi-C Experiments using Structural Similarity) application
implements an algorithm for the comparison of chromatin contact maps and automatic
differential feature extraction.</p>
<div class="subheading"><a href="ChromHMM.html">ChromHMM</a> (1.23) </div><p class="appDesc">ChromHMM is software for learning and characterizing chromatin states. </p>
<div class="subheading"><a href="circexplorer2.html">circexplorer2</a> (2.3.8) </div><p class="appDesc">A combined strategy to identify circular RNAs (circRNAs and ciRNAs) </p>
<div class="subheading"><a href="circos.html">circos</a> (0.69-9) </div><p class="appDesc">Circos is a program for the generation of publication-quality, circularly composited renditions of genomic data and related annotations. Circos is particularly suited for visualizing alignments, conservation and intra and inter-chromosomal relationships. Also, Circos is useful to visualize any type of information that benefits from a circular layout. Thus, although it has been designed for the field of genomics, it is sufficiently flexible to be used in other data domains.</p>
<div class="subheading"><a href="Clair3.html">Clair3</a> (0.1-r6) </div><p class="appDesc">Clair3 is a small variant caller for Illumina, PacBio and ONT long reads. Compare to PEPPER (r0.4), Clair3 (v0.1) shows a better SNP F1-score with ≤30-fold of ONT data (precisionFDA Truth Challenge V2), and a better Indel F1-score, while runs generally four times faster.   </p>
<div class="subheading"><a href="clark.html">clark</a> (1.2.6.1) </div><p class="appDesc">A method based on a supervised sequence classification using discriminative k-mers</p>
<div class="subheading"><a href="cnvkit.html">cnvkit</a> (0.9.9) </div><p class="appDesc">Copy number variant detection from targeted DNA sequencing</p>
<div class="subheading"><a href="cnvnator.html">cnvnator</a> (0.4.1) </div><p class="appDesc">CNVnator is a tool for CNV discovery and genotyping from depth of read mapping. </p>
<div class="subheading"><a href="cogentap.html">cogentap</a> (2.0.1) </div><p class="appDesc">Cogent NGS Analysis Pipeline (CogentAP) is bioinformatic software for analyzing RNA-seq NGS data generated using various takara kits.</p>
<div class="subheading"><a href="combp.html">combp</a> (0.50.6) </div><p class="appDesc">A library to combine, analyze, group and correct p-values in BED files. Unique tools involve correction for spatial autocorrelation. This is useful for ChIP-Seq probes and Tiling arrays, or any data with spatial correlation.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/conpair.html">conpair</a> (0.2) </div><p class="appDesc">Concordance and contamination estimator for tumor–normal pairs
</p>
<div class="subheading"><a href="crispresso.html">crispresso</a> (2.2.8) </div><p class="appDesc">Software pipeline for the analysis of CRISPR-Cas9 genome editing outcomes from sequencing data
</p>
<div class="subheading"><a href="crossmap.html">crossmap</a> (0.6.5) </div><p class="appDesc">CrossMap is a program for convenient conversion of genome coordinates between different assemblies (e.g. mm9->mm10). It can convert SAM, BAM, bed, GTF, GFF, wig/bigWig, and VCF files</p>
<div class="subheading"><a href="csvkit.html">csvkit</a> (1.3.0) </div><p class="appDesc">csvkit is a suite of command-line tools for converting to and working with CSV, the king of tabular file formats.</p>
<div class="subheading"><a href="cufflinks.html">cufflinks</a> (2.2.1) </div><p class="appDesc">Cufflinks assembles transcripts and estimates their abundances in RNA-Seq samples. It accepts aligned RNA-Seq reads and assembles the alignments into a parsimonious set of transcripts. Cufflinks then estimates the relative abundances of these transcripts based on how many reads support each one.
</p>
<div class="subheading"><a href="cutadapt.html">cutadapt</a> (4.4) </div><p class="appDesc">cutadapt removes adapter sequences from DNA high-throughput
sequencing data. This is usually necessary when the read length of the
machine is longer than the molecule that is sequenced, such as in
microRNA data.</p>
<div class="subheading"><a href="cutruntools2.html">cutruntools2</a> (2.0) </div><p class="appDesc">cutruntools2 is a major update of CutRunTools, including a set of new features specially designed for CUT&RUN and CUT&Tag experiments. Both of the bulk and single-cell data can be processed, analyzed and interpreted.</p>
<div class="subheading"><a href="DANPOS.html">DANPOS</a> (3.1.1) </div><p class="appDesc">A toolkit for Dynamic Analysis of Nucleosome and Protein Occupancy by Sequencing, version 2</p>
<div class="subheading"><a href="deepcell-tf.html">DeepCell-tf</a> (0.12.6) </div><p class="appDesc">The DeepCell-tf  library allows users to apply pre-existing models to imaging data
as well as to develop new deep learning models for single-cell analysis.
The library specializes in models for cell segmentation (whole-cell and nuclear) in 2D and 3D images
as well as cell tracking in 2D time-lapse datasets.
The models are applicable to data ranging from multiplexed images of tissues to dynamic live-cell imaging movies.</p>
<div class="subheading"><a href="deepconsensus.html">deepconsensus</a> (0.3.1) </div><p class="appDesc">DeepConsensus uses gap-aware sequence transformers to correct errors in Pacific Biosciences (PacBio) Circular Consensus Sequencing (CCS) data</p>
<div class="subheading"><a href="deepsignal.html">deepsignal</a> (2-0.1.3) </div><p class="appDesc">A deep-learning method for detecting DNA methylation state from Oxford Nanopore sequencing reads.</p>
<div class="subheading"><a href="deeptools.html">deeptools</a> (3.5.4) </div><p class="appDesc">deepTools is a suite of user-friendly tools for the visualization, quality control and normalization of data from deep-sequencing DNA sequencing experiments.
</p>
<div class="subheading"><a href="deepvariant.html">deepvariant</a> (1.5.0) </div><p class="appDesc">DeepVariant is an analysis pipeline that uses a deep neural network to call genetic variants from next-generation DNA sequencing data. </p>
<div class="subheading"><a href="defuse.html">defuse</a> (0.8.1) </div><p class="appDesc">deFuse is a software package for gene fusion discovery using RNA-Seq data. The software uses clusters of discordant paired end alignments to inform a split read alignment analysis for finding fusion boundaries. The software also employs a number of heuristic filters in an attempt to reduce the number of false positives and produces a fully annotated output for each predicted fusion.</p>
<div class="subheading"><a href="delly.html">delly</a> (1.1.6) </div><p class="appDesc">DELLY is an integrated structural variant prediction method that can detect deletions, tandem duplications, inversions and translocations at single-nucleotide resolution in short-read massively parallel sequencing data. It uses paired-ends and split-reads to sensitively and accurately delineate genomic rearrangements throughout the genome.
</p>
<div class="subheading"><a href="DETONATE.html">DETONATE</a> (1.11) </div><p class="appDesc">DETONATE is a tool for evaluation of de novo transcriptome
assemblies from RNA-Seq data. It consists of two component packages, RSEM-EVAL and REF-EVAL. RSEM-EVAL is a reference-free evaluation method based on a novel probabilistic model that depends only on an assembly and the RNA-Seq reads used for its construction. REF-EVAL is a toolkit of reference-based measures.</p>
<div class="subheading"><a href="distiller-nf.html">distiller-nf</a> (0.3.3) </div><p class="appDesc">A modular Hi-C mapping pipeline for reproducible data analysis, it was used for Micro-C analysis too. </p>
<div class="subheading"><a href="dorado.html">dorado</a> (0.4.2) </div><p class="appDesc">Dorado is a high-performance, easy-to-use, open source basecaller for Oxford Nanopore reads.</p>
<div class="subheading"><a href="drop.html">drop</a> (1.3.3) </div><p class="appDesc">A pipeline to find aberrant events in RNA-Seq data.</p>
<div class="subheading"><a href="https://expressionanalysis.github.io/ea-utils/">ea-utils</a> (1.04.807) </div><p class="appDesc">Command-line tools for processing biological sequencing data. Barcode demultiplexing, adapter trimming, etc. </p>
<div class="subheading"><a href="edd.html">edd</a> (1.1.19) </div><p class="appDesc">EDD is a ChIP-seq peak caller for detection of megabase domains of enrichment.</p>
<div class="subheading"><a href="encode-atac-seq-pipeline.html">encode-atac-seq-pipeline</a> (2.1.0) </div><p class="appDesc">This pipeline is designed for automated end-to-end quality control and processing of ATAC-seq or DNase-seq data.</p>
<div class="subheading"><a href="EPACTS.html">EPACTS</a> (3.4.2) </div><p class="appDesc">EPACTS (Efficient and Parallelizable Association Container Toolbox) is a versatile software pipeline to perform various statistical tests for identifying genome-wide association from sequence data through a user-friendly interface, both to scientific analysts and to method developers.
</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/eukdetect.html">eukdetect</a> (1.2) </div><p class="appDesc">EukDetect: Detect eukaryotes from shotgun metagenomic data</p>
<div class="subheading"><a href="exomiser.html">exomiser</a> (13.0.1) </div><p class="appDesc">The Exomiser is a Java program that functionally annotates variants from whole-exome sequencing data starting from a VCF file.</p>
<div class="subheading"><a href="express.html">express</a> (1.5.1) </div><p class="appDesc">eXpress is a streaming tool for quantifying the abundances of a set of target sequences from sampled subsequences. </p>
<div class="subheading"><a href="fanc.html">fanc</a> (0.9.21) </div><p class="appDesc">FAN-C is a toolkit for the analysis and visualization of Hi-C data. Beyond objects generated within FAN-C, the toolkit is largely compatible with Hi-C files from Cooler and Juicer.</p>
<div class="subheading"><a href="fastqc.html">fastqc</a> (0.11.9) </div><p class="appDesc">It provide quality control functions to next gen sequencing data.</p>
<div class="subheading"><a href="fastqtools.html">fastqtools</a> (0.8.3) </div><p class="appDesc">fastq-tools a collection of small and efficient programs for performing some common and uncommon tasks with FASTQ files. </p>
<div class="subheading"><a href="fastq_screen.html">fastq_screen</a> (0.15.3) </div><p class="appDesc">FastQ Screen allows you to screen a library of sequences in FastQ format against a set of sequence databases so you can see if the composition of the library matches with what you expect.</p>
<div class="subheading"><a href="fastxtoolkit.html">fastxtoolkit</a> (0.0.14) </div><p class="appDesc">The FASTX-Toolkit is a collection of command line tools for Short-Reads FASTA/FASTQ files preprocessing.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/fcs.html">fcs</a> (0.4.0) </div><p class="appDesc">FCS is a toolset to remove contaminant sequences from a genome assembly.</p>
<div class="subheading">fgbio (2.0.2) </div><p class="appDesc">The Fulcrum Genomics tools are a set of utilities for working with BAM files, VCF files, and Unique Molecular IDs.  Theey are accessed as subprograms from a Java jar, like GATK or Picard.</p>
<div class="subheading">fithic (2.0.8) </div><p class="appDesc">Fit-Hi-C is a tool for assigning statistical confidence estimates to intra-chromosomal contact maps produced by genome-wide genome architecture assays such as Hi-C. </p>
<div class="subheading"><a href="flanker.html">flanker</a> (0.1.5) </div><p class="appDesc">Gene-flank analysis tool</p>
<div class="subheading"><a href="flashpca.html">flashpca</a> (2.0) </div><p class="appDesc">FlashPCA performs fast principal component analysis (PCA) of single nucleotide polymorphism (SNP) data, similar to smartpca from EIGENSOFT (http://www.hsph.harvard.edu/alkes-price/software/) and shellfish (https://github.com/dandavison/shellfish). FlashPCA is based on the https://github.com/yixuan/spectra/ library.</p>
<div class="subheading"><a href="flexbar.html">Flexbar</a> (3.5.0) </div><p class="appDesc">Flexbar preprocesses high-throughput sequencing data efficiently. It demultiplexes barcoded runs and removes adapter sequences. Moreover, trimming and filtering features are provided. Flexbar increases read mapping rates and improves genome and transcriptome assemblies. It supports next-generation sequencing data in fasta and fastq format, e.g. from Illumina and the Roche 454 platform</p>
<div class="subheading"><a href="flye.html">flye</a> (2.9.1) </div><p class="appDesc">Fast and accurate de novo assembler for single molecule sequencing reads</p>
<div class="subheading"><a href="fqtools.html">fqtools</a> (2.0) </div><p class="appDesc">Tools for manipulating fastq files</p>
<div class="subheading"><a href="freebayes.html">freebayes</a> (1.3.5) </div><p class="appDesc">Bayesian haplotype-based polymorphism discovery and genotyping</p>
<div class="subheading"><a href="freec.html">freec</a> (11.6) </div><p class="appDesc">Control-FREEC is a tool for detection of copy-number changes and allelic imbalances (including LOH) using deep-sequencing data </p>
<div class="subheading"><a href="fuseq-wes.html">fuseq-wes</a> (1.0.0) </div><p class="appDesc">Fusion Gene Detection Using Whole-Exome Sequencing Data in Cancer Patients </p>
<div class="subheading"><a href="fusioncatcher.html">fusioncatcher</a> (1.33) </div><p class="appDesc">FusionCatcher searches for novel/known somatic fusion genes, translocations, and chimeras in RNA-seq data (paired-end or single-end reads from Illumina NGS platforms like Solexa/HiSeq/NextSeq/MiSeq) from diseased samples.</p>
<div class="subheading"><a href="GATK.html">GATK</a> (4.3.0.0) </div><p class="appDesc">GATK, from the Broad Institute,  is a structured software library that makes writing efficient analysis tools using next-generation sequencing data very easy, and second it's a suite of tools for working with human medical resequencing projects such as 1000 Genomes and The Cancer Genome Atlas. These tools include things like a depth of coverage analyzers, a quality score recalibrator, a SNP/indel caller and a local realigner.</p>
<div class="subheading"><a href="gem.html">gem</a> (3.4) </div><p class="appDesc">High resolution peak calling and motif discovery for ChIP-seq and ChIP-exo data</p>
<div class="subheading"><a href="gemini.html">Gemini</a> (0.30.2) </div><p class="appDesc">GEMINI (GEnome MINIng) is designed to be a flexible framework for exploring genetic variation in the context of the wealth of genome annotations available for the human genome. By placing genetic variants, sample genotypes, and useful genome annotations into an integrated database framework, GEMINI provides a simple, flexible, yet very powerful system for exploring genetic variation for for disease and population genetics.</p>
<div class="subheading"><a href="GEMMA.html">GEMMA</a> (0.98.5) </div><p class="appDesc">GEMMA is the software implementing the Genome-wide Efficient Mixed Model Association algorithm for a standard linear mixed model and some of its close relatives for genome-wide association studies (GWAS). </p>
<div class="subheading"><a href="Genome_Browser.html">Genome Browser</a> (451) </div><p class="appDesc">The <a href="https://hpcnihapps.cit.nih.gov/genome">Genome Browser Mirror Fragments</a> is a mirror of the UCSC Genome Browser.  The URL is https://hpcnihapps.cit.nih.gov/genome.  Users can also access the MySQL databases, supporting files directly, and a huge number of associated executables.</p>
<div class="subheading">genrich (0.6) </div><p class="appDesc">Genrich is a peak-caller for genomic enrichment assays (e.g. ChIP-seq, ATAC-seq). It analyzes alignment files generated following the assay and produces a file detailing peaks of significant enrichment.</p>
<div class="subheading"><a href="geomx_ngs_pipeline.html">GeoMX NGS Pipeline</a> (2.3.3.10) </div><p class="appDesc">The GeoMx NGS Pipeline, processes RNA-sequencing files (FASTQ files) from Illumina sequencers according to parameters defined in the Configuration File (which is generated from the GeoMx DSP run). The Pipeline processes information from these files and outputs .dcc files, which can then be uploaded to the GeoMx DSP system for data analysis.</p>
<div class="subheading"><a href="gistic.html">Gistic</a> (2.0.23) </div><p class="appDesc">Facilitates sensitive and confident localization of the targets of focal somatic copy-number alteration in human cancers.</p>
<div class="subheading"><a href="gmap-gsnap.html">gmap-gsnap</a> (2021-12-17) </div><p class="appDesc">A Genomic Mapping and Alignment Programs</p>
<div class="subheading"><a href="gopeaks.html">gopeaks</a> (1.0.0) </div><p class="appDesc">GoPeaks is a peak caller designed for CUT&TAG/CUT&RUN sequencing data. GoPeaks by default works best with narrow peaks such as H3K4me3 and transcription factors. However, broad epigenetic marks like H3K27Ac/H3K4me1 require different the step, slide, and minwidth parameters.</p>
<div class="subheading"><a href="graphaligner.html">graphaligner</a> (1.0.17b) </div><p class="appDesc">Seed-and-extend program for aligning long error-prone reads to genome graphs.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/gridss.html">gridss</a> (2.13.2) </div><p class="appDesc">GRIDSS is a module software suite containing tools useful for the detection of genomic rearrangements. GRIDSS includes a genome-wide break-end assembler, as well as a structural variation caller for Illumina sequencing data. GRIDSS calls variants based on alignment-guided positional de Bruijn graph genome-wide break-end assembly, split read, and read pair evidence.</p>
<div class="subheading"><a href="gtex_rnaseq.html">gtex_rnaseq</a> (V10) </div><p class="appDesc">This module makes available the tools used in the GTEX RNA-Seq pipeline.</p>
<div class="subheading">gtool (0.7.5) </div><p class="appDesc"> GTOOL is a program for transforming sets of genotype data for use with the programs SNPTEST and IMPUTE.

GTOOL can be used to    (a) generate subsets of genotype data,
                                             (b) to convert genotype data between the PED file format and the FILE FORMAT used by SNPTEST and IMPUTE.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/gunc.html">gunc</a> (1.0.5) </div><p class="appDesc">Genome UNClutterer (GUNC) is a tool for detection of chimerism and contamination in prokaryotic genomes resulting from mis-binning of genomic contigs from unrelated lineages</p>
<div class="subheading"><a href="guppy.html">guppy</a> (6.5.7) </div><p class="appDesc">Local accelerated basecalling for Nanopore data</p>
<div class="subheading"><a href="Hail.html">Hail</a> (0.2.99) </div><p class="appDesc">Hail is an open-source, scalable framework for exploring and analyzing genomic data.</p>
<div class="subheading"><a href="hap.py.html">hap.py</a> (0.3.14) </div><p class="appDesc">A set of programs based on htslib to benchmark variant calls against gold standard truth datasets.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/hicexplorer.html">hicexplorer</a> (3.7.2) </div><p class="appDesc">Tools to process, normalize and visualize Hi-C data</p>
<div class="subheading"><a href="hichipper.html">hichipper</a> (0.7.7) </div><p class="appDesc">hichipper is a preprocessing and QC pipeline for HiChIP data. This package takes output from a HiC-Pro run and a sample manifest file (.yaml) that coordinates optional high-quality peaks (identified through ChIP-Seq) and restriction fragment locations (see folder here) as input and produces output that can be used to 1) determine library quality, 2) identify and characterize DNA loops and 3) interactively visualize loops.</p>
<div class="subheading"><a href="hicpro.html">hicpro</a> (3.1.0) </div><p class="appDesc">HiC-Pro: An optimized and flexible pipeline for Hi-C data processing</p>
<div class="subheading">hic_breakfinder (1.0) </div><p class="appDesc">A framework that integrates optical mapping, high-throughput chromosome conformation capture (Hi-C), and whole genome sequencing to systematically detect SVs in a variety of normal or cancer samples and cell lines.</p>
<div class="subheading"><a href="hifiasm.html">hifiasm</a> (0.19.5) </div><p class="appDesc">    Hifiasm is a fast haplotype-resolved de novo assembler initially designed for PacBio HiFi reads.
    Its latest release supports telomere-to-telomere assembly by utilizing ultralong Oxford Nanopore reads.
    It can produce better haplotype-resolved assemblies when given parental short reads or Hi-C data.</p>
<div class="subheading"><a href="hint.html">hint</a> (2.2.7) </div><p class="appDesc">a computational method to detect CNVs and Translocations from Hi-C data. </p>
<div class="subheading"><a href="hipstr.html">hipstr</a> (0.7) </div><p class="appDesc">Tool for genotyping short tandom repeats from Illumina sequencing data</p>
<div class="subheading"><a href="hisat.html">hisat</a> (2.2.2.1-ngs3.0.1) </div><p class="appDesc">HISAT is a fast and sensitive spliced alignment program which uses Hierarchical Indexing for Spliced Alignment of Transcripts.</p>
<div class="subheading"><a href="HLA-LA.html">HLA-LA</a> (1.0.3) </div><p class="appDesc">Fast HLA type inference from whole-genome data. Previously known as HLA-PRG-LA.</p>
<div class="subheading"><a href="HMMRATAC.html">HMMRATAC</a> (1.2.10) </div><p class="appDesc">HMMRATAC peak caller for ATAC-seq data</p>
<div class="subheading"><a href="homer.html">homer</a> (4.11.1) </div><p class="appDesc">HOMER (Hypergeometric Optimization of Motif EnRichment) is a suite of tools for Motif Discovery and ChIP-Seq analysis.</p>
<div class="subheading"><a href="htgts.html">htgts</a> (2) </div><p class="appDesc">High-Throughput Genome-Wide Translocation Sequencing pipeline</p>
<div class="subheading"><a href="htsbox.html">htsbox</a> (r346) </div><p class="appDesc">HTSbox is a fork of early HTSlib. It is a collection of small experimental tools manipulating HTS-related files. While some of these tools are already part of the official SAMtools package, others are for niche use cases.</p>
<div class="subheading"><a href="htseq.html">htseq</a> (2.0.4) </div><p class="appDesc">HTSeq is a Python package that provides infrastructure to process data from high-throughput sequencing assays.</p>
<div class="subheading"><a href="humann.html">humann</a> (3.6.0) </div><p class="appDesc">HUMAnN is a pipeline for efficiently and accurately profiling the presence/absence and abundance of microbial pathways in a community from metagenomic or metatranscriptomic sequencing data (typically millions of short DNA/RNA reads).</p>
<div class="subheading"><a href="idba.html">IDBA</a> (1.1.3) </div><p class="appDesc">IDBA is a practical iterative De Bruijn Graph De Novo Assembler for sequence assembly in bioinfomatics. Most assemblers based on de Bruijn graph build a de Bruijn graph with a specific k to perform the assembling task. For all of them, it is very crucial to find a specific value of k. If k is too large, there will be a lot of gap problems in the graph. If k is too small, there will a lot of branch problems. IDBA uses not only one specific k but a range of k values to build the iterative de Bruijn graph. It can keep all the information in graphs with different k values.</p>
<div class="subheading"><a href="iDiffIR.html">iDiffIR</a> (20220121) </div><p class="appDesc">iDiffIR is a tool for identifying differential IR from RNA-seq data.
It accepts any sorted, indexed BAM file for single- or paired-end reads.</p>
<div class="subheading"><a href="idr.html">IDR</a> (2.0.3) </div><p class="appDesc">The IDR (Irreproducible Discovery Rate) framework is a uniﬁed approach to measure the reproducibility of ﬁndings identiﬁed from replicate experiments and provide highly stable thresholds based on reproducibility. The IDR method compares a pair of ranked lists of identifications (such as ChIP-seq peaks).</p>
<div class="subheading"><a href="IGV.html">IGV</a> (2.12.3) </div><p class="appDesc">The Integrative Genomics Viewer is a high-performance visualization tool for interactive exploration of large, integrated genomic datasets. </p>
<div class="subheading"><a href="IGV.html">IGVTools</a> (2.12.3) </div><p class="appDesc">IGVTools provides utilities for working with ascii file formats used by the Integrated Genome Viewer.  The files can be sorted, tiled, indexed, and counted.</p>
<div class="subheading"><a href="IMPUTE.html">IMPUTE</a> (2.3.2) </div><p class="appDesc">Impute is a program for estimating ("imputing") unobserved genotypes in SNP association studies.</p>
<div class="subheading"><a href="InterVar.html">InterVar</a> (2.1.2, 2.1.3) </div><p class="appDesc">In 2015, the American College of Medical Genetics and Genomics (ACMG) and the Association for Molecular Pathology (AMP) published
updated standards and guidelines for the clinical interpretation of sequence variants with respect to human diseases on the basis
of 28 criteria. However, variability between individual interpreters can be extensive because of reasons such as the different understandings of these guidelines and the lack of standard algorithms for implementing them, yet computational tools for semi-automated variant interpretation are not available. To address these problems, InterVar implements these criteria to help human reviewers interpret the clinical significance of variants. InterVar can take a pre-annotated or VCF file as input and generate automated interpretation on 18 criteria.</p>
<div class="subheading">intervene (0.6.5) </div><p class="appDesc">a tool for intersection and visualization of multiple genomic region sets</p>
<div class="subheading"><a href="iva.html">iva</a> (1.0.11) </div><p class="appDesc">IVA is a de novo assembler designed to assemble virus genomes that have no repeat sequences, using Illumina read pairs sequenced from mixed populations at extremely high and variable depth.</p>
<div class="subheading"><a href="iVar.html">iVar</a> (1.3.1) </div><p class="appDesc">iVar is a computational package that contains functions broadly useful for viral amplicon-based sequencing. Additional tools for metagenomic sequencing are actively being incorporated into iVar. While each of these functions can be accomplished using existing tools, iVar contains an intersection of functionality from multiple tools that are required to call iSNVs and consensus sequences from viral sequencing data across multiple replicates.</p>
<div class="subheading"><a href="juicer.html">Juicer</a> (1.6) </div><p class="appDesc">A One-Click System for Analyzing Loop-Resolution Hi-C Experiments</p>
<div class="subheading"><a href="jvarkit.html">jvarkit</a> (20211013) </div><p class="appDesc">Java tools for bioinformatics</p>
<div class="subheading"><a href="kat.html">KAT</a> (2.4.2) </div><p class="appDesc"> KAT (K-mer Analysis Toolkit) is a suite of tools that analyse Jellyfish hashes or sequence files (fasta or fastq) using kmer counts.</p>
<div class="subheading"><a href="kb-python.html">kb-python</a> (0.27.3) </div><p class="appDesc">kb-python is a python package for processing single-cell RNA-sequencing. It wraps the kallisto | bustools single-cell RNA-seq command line tools in order to unify multiple processing workflows. </p>
<div class="subheading"><a href="kmc.html">KMC</a> (3.1.0) </div><p class="appDesc">KMC is a disk-based programm for counting k-mers from (possibly gzipped) FASTQ/FASTA files</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/KmerGO2.html">KmerGO2</a> (2.0.1) </div><p class="appDesc">KmerGO is a user-friendly tool to identify the group-specific sequences on two groups of high throughput sequencing datasets. </p>
<div class="subheading"><a href="kneaddata.html">kneaddata</a> (0.11.0) </div><p class="appDesc">KneadData is a tool designed to perform quality control on metagenomic and metatranscriptomic sequencing data, especially data from microbiome experiments.</p>
<div class="subheading"><a href="kraken.html">kraken</a> (2.1.2) </div><p class="appDesc">Kraken is a system for assigning taxonomic labels to short DNA sequences, usually obtained through metagenomic studies</p>
<div class="subheading"><a href="lefse.html">lefse</a> (1.0.8) </div><p class="appDesc">LEfSe (Linear discriminant analysis Effect Size) determines the features (organisms, clades, operational taxonomic units, genes, or functions) most likely to explain differences between classes by coupling standard tests for statistical significance with additional tests encoding biological consistency and effect relevance.</p>
<div class="subheading"><a href="LJA.html">LJA</a> (0.1) </div><p class="appDesc">The La Jolla Assembler (LJA) is a tool for assemblies of long and accurate reads. It reduces the error rate in
these reads by three orders of magnitude (making them nearly error-free) and constructs the de Bruijn
graph for large genomes and large k-mer sizes. Since the de Bruijn graph constructed for a fixed k-mer
size is typically either too tangled or too fragmented, LJA uses a new concept of a multiplex de Bruijn
graph with varying k-mer sizes.</p>
<div class="subheading"><a href="locuszoom.html">locuszoom</a> (1.3) </div><p class="appDesc">LocusZoom is designed to facilitate viewing of local association results together with useful information about a locus, such as the location and orientation of the genes it includes, linkage disequilibrium coefficients and local estimates of recombination rates</p>
<div class="subheading"><a href="lofreq.html">lofreq</a> (2.1.5) </div><p class="appDesc">LoFreq is a fast and sensitive variant-caller for inferring SNVs and indels from next-generation sequencing data.</p>
<div class="subheading"><a href="lordec.html">lordec</a> (141) </div><p class="appDesc">LoRDEC processes data coming from high throughput sequencing machines of the second and third generations. These data are called sequencing reads, or simply reads for short. Technically speaking it processes short reads and long reads to correct errors in the long reads.</p>
<div class="subheading"><a href="lumpy.html">lumpy</a> (0.3.1) </div><p class="appDesc">A probabilistic framework for structural variant discovery.</p>
<div class="subheading"><a href="mach2qtl.html">mach2qtl</a> (1.1.3) </div><p class="appDesc">mach2qtl uses dosages/posterior probabilities inferred with MACH as predictors in a linear regression to test association with a quantitative trait</p>
<div class="subheading"><a href="macs.html">macs</a> (3) </div><p class="appDesc">Model-based Analysis of ChIP-Seq (MACS) on short reads sequencers such as Genome Analyzer (Illumina / Solexa). MACS empirically models the length of the sequenced ChIP fragments, which tends to be shorter than sonication or library construction size estimates, and uses it to improve the spatial resolution of predicted binding sites. MACS also uses a dynamic Poisson distribution to effectively capture local biases in the genome sequence, allowing for more sensitive and robust prediction.</p>
<div class="subheading"><a href="mafft.html">mafft</a> (7.475) </div><p class="appDesc">Multiple alignment program for amino acid or nucleotide sequences
</p>
<div class="subheading"><a href="MAGeCK.html">MAGeCK</a> (0.5.9.2) </div><p class="appDesc">MAGeCK is Model-based Analysis of Genome-wide CRISPR/Cas9 Knockout (MAGeCK) method for prioritizing
single-guide RNAs, genes and pathways in genome-scale CRISPR/Cas9 knockout screens. It demonstrates better performance compared with other methods, identifies both positively and negatively selected genes
simultaneously, and reports robust results across different experimental conditions.
</p>
<div class="subheading"><a href="mageck-vispr.html">mageck-vispr</a> (0.5.6) </div><p class="appDesc">MAGeCK-VISPR is a comprehensive quality control, analysis and visualization workflow for CRISPR/Cas9 screens.</p>
<div class="subheading"><a href="magicblast.html">magicblast</a> (1.7.0) </div><p class="appDesc">Magic-BLAST is a tool for mapping large next-generation RNA or DNA sequencing runs against a whole genome or transcriptome. Each alignment optimizes a composite score, taking into account simultaneously the two reads of a pair, and in case of RNA-seq, locating the candidate introns and adding up the score of all exons. This is very different from other versions of BLAST, where each exon is scored as a separate hit and read-pairing is ignored.</p>
<div class="subheading"><a href="majiq.html">MAJIQ</a> (2.4) </div><p class="appDesc">Modeling Alternative Junction Inclusion Quantification. MAJIQ and Voila are two software packages that together define, quantify, and visualize local splicing variations (LSV) from RNA-Seq data.</p>
<div class="subheading"><a href="manorm.html">manorm</a> (1.1.4) </div><p class="appDesc">MAnorm is for quantitative comparison of ChIP-Seq data sets describing transcription factor binding sites and epigenetic modifications. The quantitative binding differences inferred by MAnorm showed strong correlation with both the changes in expression of target genes and the binding of cell type-specific regulators.</p>
<div class="subheading"><a href="manta.html">manta</a> (1.6.0-fork-jykr) </div><p class="appDesc">Structural variant and indel caller for mapped sequencing data </p>
<div class="subheading">maps (1.1.0) </div><p class="appDesc">a set of multiple scripts used to analyze PLAC-Seq and HiChIP data.</p>
<div class="subheading"><a href="mash.html">mash</a> (2.3) </div><p class="appDesc">mash is a command line tool and library to provide fast genome and metagenome distance estimation using MinHash. Only command line tool is installed</p>
<div class="subheading"><a href="mbg.html">mbg</a> (1.0.15) </div><p class="appDesc">Minimizer based sparse de Bruijn Graph constructor.</p>
<div class="subheading"><a href="medaka.html">medaka</a> (1.10.0) </div><p class="appDesc">medaka is a tool to create a consensus sequence from nanopore sequencing data. This task is performed using neural networks applied from a pileup of individual sequencing reads against a draft assembly.</p>
<div class="subheading"><a href="medusa.html">medusa</a> (1.6) </div><p class="appDesc">A draft genome scaffolder that uses multiple reference genomes in a graph-based approach.</p>
<div class="subheading"><a href="megadepth.html">megadepth</a> (1.2.0) </div><p class="appDesc"></p>
<div class="subheading"><a href="megahit.html">MEGAHIT</a> (1.2.9) </div><p class="appDesc">MEGAHIT is a single node assembler for large and complex metagenomics NGS reads, such as soil. It makes use of succinct de Bruijn graph (SdBG) to achieve low memory assembly. MEGAHIT can optionally utilize a CUDA-enabled GPU to accelerate its SdBG construction.</p>
<div class="subheading"><a href="megalodon.html">megalodon</a> (2.5.0) </div><p class="appDesc">Megalodon is a research command line tool to extract high accuracy modified base and sequence variant calls from raw nanopore reads by anchoring the information rich basecalling neural network output to a reference genome/transcriptome.</p>
<div class="subheading"><a href="MEGAN.html">MEGAN</a> (6.24.11) </div><p class="appDesc">MEtaGenome ANalyzer that takes a file of reads and a Blast output from comparison against a reference genome, and automatically calculate a taxonomic classification of the reads and if desired, a functional classification.</p>
<div class="subheading"><a href="merfin.html">merfin</a> (1.1) </div><p class="appDesc">Improved variant filtering and polishing via k-mer validation</p>
<div class="subheading"><a href="merqury.html">merqury</a> (1.3) </div><p class="appDesc">Evaluate genome assemblies with k-mers and more</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/Meryl.html">Meryl</a> (0.0) </div><p class="appDesc">Meryl: a genomic k-mer counter (and sequence utility) with nice features. It is built into the Celera Assembler
and is also available as a stand-alone application.
Meryl uses a sorting-based approach that sorts the k-mers in lexicographical order.</p>
<div class="subheading"><a href="metabat.html">metabat</a> (2.15) </div><p class="appDesc">MetaBAT: A robust statistical framework for reconstructing genomes from metagenomic data</p>
<div class="subheading"><a href="metal.html">metal</a> (2020-05-05) </div><p class="appDesc">The METAL software is designed to facilitate meta-analysis of large datasets (such as several whole genome scans) in a convenient, rapid and memory efficient manner. </p>
<div class="subheading"><a href="metaphlan.html">metaphlan</a> (4.0.3) </div><p class="appDesc">MetaPhlAn is a computational tool for profiling the composition of microbial communities (Bacteria, Archaea, Eukaryotes and Viruses) from metagenomic shotgun sequencing data (i.e. not 16S) with species-level. With the newly added StrainPhlAn module, it is now possible to perform accurate strain-level microbial profiling.</p>
<div class="subheading">miniasm (0.3.r179) </div><p class="appDesc">Ultrafast de novo assembly for long noisy reads (though having no consensus step)</p>
<div class="subheading"><a href="minimac.html">minimac</a> (4 (1.0.1)) </div><p class="appDesc">minimac is a low memory, computationally efficient implementation of the MaCH algorithm for genotype imputation. It is designed to work on phased genotypes and can handle very large reference panels with hundreds or thousands of haplotypes. 'mini' refers to the low amount of computational resources it needs. </p>
<div class="subheading"><a href="minimap2.html">minimap2</a> (2.26) </div><p class="appDesc">Minimap2 is a fast sequence mapping and alignment program that can find overlaps between long noisy reads, or map long reads or their assemblies to a reference genome optionally with detailed alignment (i.e. CIGAR).</p>
<div class="subheading"><a href="mirdeep.html">mirdeep2</a> (0.1.3) </div><p class="appDesc">miRDeep2 is a completely overhauled tool which discovers microRNA genes by analyzing sequenced RNAs. </p>
<div class="subheading"><a href="misopy.html">misopy</a> (0.5.4) </div><p class="appDesc">MISO (Mixture-of-Isoforms) is a probabilistic framework that quantitates the expression level of alternatively spliced genes from RNA-Seq data</p>
<div class="subheading"><a href="mitosuite.html">mitosuite</a> (1.0.9b) </div><p class="appDesc">mitosuite is a graphical tool for human mitochondrial genome profiling in massively parallel sequencing</p>
<div class="subheading"><a href="mixcr.html">mixcr</a> (4.3.2) </div><p class="appDesc">MiXCR is a universal software for fast and accurate analysis of T- and B- cell receptor repertoire sequencing data.</p>
<div class="subheading"><a href="mixer.html">mixer</a> (1.3) </div><p class="appDesc">MiXeR is Causal Mixture Model for GWAS summary statistics. The version(1.3) contains a Python port of MiXeR, wrapping the C/C++ core. Also data preprocessing code sumstats.py is included too.
</p>
<div class="subheading"><a href="modbamtools.html">modbamtools</a> (0.4.8) </div><p class="appDesc">A set of tools to manipulate and visualize DNA/RNA base modification data that are stored in bam format.</p>
<div class="subheading"><a href="modkit.html">modkit</a> (0.1.11) </div><p class="appDesc">A bioinformatics tool for working with modified bases from Oxford Nanopore. Specifically for converting modBAM to bedMethyl files using best practices, but also manipulating modBAM files and generating summary statistics.</p>
<div class="subheading"><a href="modphred.html">modphred</a> (1.0c) </div><p class="appDesc">modPhred is a pipeline for detection, annotation and visualisation of DNA/RNA modifications.</p>
<div class="subheading"><a href="mosaicforecast.html">mosaicforecast</a> (0.0.1) </div><p class="appDesc">a machine learning method that leverages read-based phasing and read-level features to accurately detect mosaic SNVs (SNPs, small indels) from NGS data.It builds on existing algorithms to achieve a multifold increase in specificity. </p>
<div class="subheading"><a href="mosdepth.html">mosdepth</a> (0.3.3) </div><p class="appDesc">Fast BAM/CRAM depth calculation for WGS, exome, or targeted sequencing. </p>
<div class="subheading"><a href="mtoolbox.html">mtoolbox</a> (1.2.1) </div><p class="appDesc">A bioinformatics pipeline aimed at the analysis of mitochondrial DNA (mtDNA) in high throughput sequencing studies.</p>
<div class="subheading"><a href="multiqc.html">multiqc</a> (1.14) </div><p class="appDesc">aggregates results for various frequently used bioinformatics tools across multiple samples into a nice visual report</p>
<div class="subheading"><a href="http://bioinformatics.mdanderson.org/main/MuSE">MuSE</a> (2.0.1) </div><p class="appDesc">MuSE is an approach to somatic variant calling based on the F81 Markov substitution model for molecular evolution, which models the evolution of the reference allele to the allelic composition of the matched tumor and normal tissue at each genomic locus.</p>
<div class="subheading"><a href="muTect.html">muTect</a> (1.1.7) </div><p class="appDesc">MuTect is a method developed at the Broad Institute for the reliable and accurate identification of somatic point mutations in next generation sequencing data of cancer genomes.</p>
<div class="subheading"><a href="MutSig.html">MutSig</a> (1.41) </div><p class="appDesc">MutSig analyzes lists of mutations discovered in DNA sequencing, to identify genes that were mutated more often than expected by chance given background mutation processes.</p>
<div class="subheading"><a href="nanopolish.html">nanopolish</a> (0.14.0) </div><p class="appDesc">nanopolish is a software package for signal-level analysis of Oxford Nanopore sequencing data. Nanopolish can calculate an improved consensus sequence for a draft genome assembly, detect base modifications, call SNPs and indels with respect to a reference genome and more (see Nanopolish modules, below).

</p>
<div class="subheading"><a href="neusomatic.html">neusomatic</a> (0.2.1) </div><p class="appDesc">NeuSomatic is based on deep convolutional neural networks for accurate somatic mutation detection. With properly trained models, it can robustly perform across sequencing platforms, strategies, and conditions. NeuSomatic summarizes and augments sequence alignments in a novel way and incorporates multi-dimensional features to capture variant signals effectively. It is not only a universal but also accurate somatic mutation detection method.</p>
<div class="subheading"><a href="ngmlr.html">NGMLR</a> (0.2.7) </div><p class="appDesc">NGMLR is a long-read mapper designed to align PacBio or Oxford Nanopore (standard and ultra-long) to a reference genome with a focus on reads that span structural variations.</p>
<div class="subheading"><a href="ngsplot/">ngsplot</a> (2.63) </div><p class="appDesc">ngsplot is an easy-to-use global visualization tool for next-generation sequencing data.</p>
<div class="subheading"><a href="novocraft.html">novocraft</a> (4.03.08) </div><p class="appDesc">Package includes aligner for single-ended and paired-end reads from the Illumina Genome Analyser. Novoalign finds global optimum alignments using full Needleman-Wunsch algorithm with affine gap penalties.</p>
<div class="subheading">Octopus (0.7.4) </div><p class="appDesc">Octopus is a mapping-based variant caller that implements several calling models within a unified haplotype-aware framework. Octopus takes inspiration from particle filtering by constructing a tree of haplotypes and dynamically pruning and extending the tree based on haplotype posterior probabilities in a sequential manner. This allows octopus to implicitly consider all possible haplotypes at a given loci in reasonable time.</p>
<div class="subheading"><a href="ont-fast5-api.html">ont-fast5-api</a> (4.1.0) </div><p class="appDesc">Tools to manipulate HDF5 files of the Oxford Nanopore .fast5 file format</p>
<div class="subheading"><a href="pairtools.html">pairtools</a> (0.3.0; 1.0.2) </div><p class="appDesc">Pairtools is a simple and fast command-line framework to process sequencing data
from a Hi-C experiment. Pairtools perform various operations on Hi-C pairs and occupy the middle position in a typical Hi-C data processing pipeline. Pairtools aim to be an all-in-one tool for processing Hi-C pairs.</p>
<div class="subheading"><a href="panaroo.html">panaroo</a> (1.3.3) </div><p class="appDesc">An updated pipeline for pangenome investigation </p>
<div class="subheading"><a href="parabricks.html">parabricks</a> (4.0.0) </div><p class="appDesc">The Clara Parabricks toolkit is a set of GPU-accelerated genome analysis tools for secondary analysis of next generation sequencing data.</p>
<div class="subheading"><a href="https://partekflow.cit.nih.gov">PartekFlow</a> (10.0.23.0720) </div><p class="appDesc">Web interface designed specifically for the analysis needs of next generation sequencing applications including RNA, small RNA, and DNA sequencing.</p>
<div class="subheading"><a href="pb-cpg-tools.html">pb-cpg-tools</a> (2.3.1) </div><p class="appDesc">Tools for analyzing CpG/5mC data from PacBio HiFi reads aligned to a reference genome</p>
<div class="subheading"><a href="pbipa.html">pbipa</a> (1.8.0) </div><p class="appDesc">Improved Phased Assembler (IPA) is the official PacBio software for HiFi genome assembly. IPA was designed to utilize the accuracy of PacBio HiFi reads to produce high-quality phased genome assemblies. </p>
<div class="subheading"><a href="pbsuite.html">pbsuite</a> (15.8.24) </div><p class="appDesc">The PBSuite contains two projects created for analysis of Pacific Biosciences long-read sequencing data: PBHoney and PBJelly. PBHoney is an implementation of two variant-identification approaches designed to exploit the high mappability of long reads (i.e., greater than 10,000 bp). PBHoney considers both intra-read discordance and soft-clipped tails of long reads to identify structural variants. PBJelly is a highly automated pipeline that aligns long sequencing reads (such as PacBio RS reads or long 454 reads in fasta format) to high-confidence draft assembles. PBJelly fills or reduces as many captured gaps as possible to produce upgraded draft genomes.</p>
<div class="subheading"><a href="pear.html">pear</a> (0.9.11) </div><p class="appDesc">PEAR is an ultrafast, memory-efficient and highly accurate pair-end read merger. It is fully parallelized and can run with as low as just a few kilobytes of memory.</p>
<div class="subheading"><a href="peddy.html">peddy</a> (0.4.8) </div><p class="appDesc">peddy is used to compare sex and familial relationships given in a PED file with those inferred from a VCF file</p>
<div class="subheading"><a href="PennCNV.html">PennCNV</a> ( 1.0.5) </div><p class="appDesc">PennCNV is a free software tool for Copy Number Variation (CNV) detection
from SNP genotyping arrays. Currently it can handle signal intensity data
from Illumina and Affymetrix arrays. With appropriate preparation of file format,
it can also handle other types of SNP arrays and oligonucleotide arrays.</p>
<div class="subheading"><a href="PEPATAC.html">PEPATAC</a> (0.10.3) </div><p class="appDesc">PEPATAC is a robust pipeline for Assay for Transposase-Accessible Chromatin using sequencing (ATAC-seq) built on a loosely coupled modular framework.
It may be easily applied to ATAC-seq projects of any size,
from one-off experiments to large-scale sequencing projects.
It is optimized on unique features of ATAC-seq data to be fast and accurate
and provides several unique analytical approaches.</p>
<div class="subheading"><a href="picard.html">picard</a> (3.1.0) </div><p class="appDesc">Picard comprises Java-based command-line utilities that manipulate SAM files, and a Java API (SAM-JDK) for creating new programs that read and write SAM files. Both SAM text format and SAM binary (BAM) format are supported.</p>
<div class="subheading"><a href="picrust.html">picrust</a> (2.5.2) </div><p class="appDesc">PICRUSt is a bioinformatics software package designed to predict metagenome functional content from marker gene (e.g., 16S rRNA) surveys and full genomes.</p>
<div class="subheading"><a href="pindel.html">pindel</a> (0.2.5) </div><p class="appDesc">Pindel can detect breakpoints of large deletions, medium sized insertions, inversions, tandem duplications and other structural variants at single-based resolution from next-gen sequence data. It uses a pattern growth approach to identify the breakpoints of these variants from paired-end short reads.
</p>
<div class="subheading"><a href="platypus.html">Platypus</a> (0.8.1) </div><p class="appDesc">tool for variant-detection in high-throughput sequencing data. </p>
<div class="subheading"><a href="plink.html">plink</a> (3.6-alpha) </div><p class="appDesc">PLINK is whole genome association analysis toolset, designed to perform a range of basic, large-scale analyses in a computationally efficient manner.

</p>
<div class="subheading"><a href="pod5.html">pod5</a> (0.5.0) </div><p class="appDesc">Tool for manipulating the pod5 format of nanopore reads</p>
<div class="subheading"><a href="pomoxis.html">pomoxis</a> (0.3.10) </div><p class="appDesc">Pomoxis comprises a set of basic bioinformatic tools tailored to nanopore sequencing. Notably tools are included for generating and analysing draft assemblies. Many of these tools are used by the research data analysis group at Oxford Nanopore Technologies.</p>
<div class="subheading"><a href="porechop.html">porechop</a> (0.2.4) </div><p class="appDesc">Trim/demultiplex Oxford Nanopore reads</p>
<div class="subheading"><a href="poretools.html">poretools</a> (0.6.1a1) </div><p class="appDesc">Poretools is a toolkit for manipulating and exploring nanopore sequencing data sets. Poretools operates on individual FAST5 files, directory of FAST5 files, and tar archives of FAST5 files. </p>
<div class="subheading"><a href="preseq.html">preseq</a> (3.1.2) </div><p class="appDesc">predicting library complexity and genome coverage in high-throughput sequencing</p>
<div class="subheading"><a href="prinseq.html">PRINSEQ</a> (0.20.4) </div><p class="appDesc">PRINSEQ is a tool that generates summary statistics of sequence and quality data and that is used to filter, reformat and trim next-generation sequence data. It is particular designed for 454/Roche data, but can also be used for other types of sequence data.</p>
<div class="subheading"><a href="proseq.html">proseq</a> (2.0) </div><p class="appDesc">proseq-2.0 is a pipeline for preprocesses and alignment
of run-on sequencing (PRO/GRO/ChRO-seq) data
from Single-Read or Paired-End Illumina Sequencing
</p>
<div class="subheading"><a href="pvactools.html">pvactools</a> (4.0.1) </div><p class="appDesc">pVACtools is a cancer immunotherapy suite consisting of pVACseq, pVACfuse, pVACvector</p>
<div class="subheading"><a href="pychopper.html">pychopper</a> (2.7.1) </div><p class="appDesc">Pychopper v2 is a tool to identify, orient and trim full-length Nanopore cDNA reads. The tool is also able to rescue fused reads.</p>
<div class="subheading"><a href="pyclone.html">pyclone</a> (0.13.1) </div><p class="appDesc">PyClone is statistical model and software tool designed to infer the prevalence of point mutations in heterogeneous cancer samples. </p>
<div class="subheading"><a href="pycoQC.html">pycoQC</a> (2.5.2) </div><p class="appDesc">pycoQC is a new tool to generate interactive quality control metrics and plots
from basecalled nanopore reads
or summary files generated by the basecallers Albacore, Guppy or MinKNOW.
pycoQC has several novel features, including:
1) python support for creation of dynamic D3.js visualizations and interactive data exploration in Jupyter Notebooks;
2) simple command line interface to generate customizable interactive HTML reports; and
3) multiprocessing FAST5 feature extraction program to generate a summary file directly
from FAST5 files.</p>
<div class="subheading"><a href="pyega3.html">pyega3</a> (5.0.2) </div><p class="appDesc">A download client for the European Genome-phenome Archive (EGA). The EGA is designed to be a repository for all types of sequence and genotype experiments, including case-control, population, and family studies.</p>
<div class="subheading">qctool (2.2) </div><p class="appDesc">QCTOOL is a command-line utility program for basic quality control of gwas datasets. </p>
<div class="subheading"><a href="qualimap.html">qualimap</a> (2.2.1) </div><p class="appDesc">A platform-independednt application written in Java and R that provides both a GUI and a co
mmand-line interface to facilitate the quality control of alignment sequencing data.</p>
<div class="subheading"><a href="quast.html">quast</a> (5.2.0) </div><p class="appDesc">QUAST stands for QUality ASsessment Tool. The tool evaluates genome assemblies by computing various metrics. The package includes the general QUAST tool for genome assemblies, MetaQUAST, the extension for metagenomic datasets, and Icarus, interactive visualizer for these tools.</p>
<div class="subheading">racon (1.4.3) </div><p class="appDesc">Ultrafast consensus module for raw de novo genome assembly of long uncorrected reads.</p>
<div class="subheading"><a href="ragtag.html">ragtag</a> (2.1.0) </div><p class="appDesc">RagTag is a collection of software tools for scaffolding and improving modern genome assemblies.</p>
<div class="subheading"><a href="raremetal.html">raremetal</a> (4.15.1) </div><p class="appDesc">RAREMETAL is a computationally efficient tool for meta-analysis of rare variants using sequencing or genotyping array data.</p>
<div class="subheading"><a href="Rcorrector.html">Rcorrector</a> (1.0.5) </div><p class="appDesc">Rcorrector implements a k-mer based method to correct random sequencing errors
in Illumina RNA-seq reads. Rcorrector uses a De Bruijn graph to compactly represent
all trusted k-mers in the input reads. Unlike WGS read correctors,
which use a global threshold to determine trusted k-mers, Rcorrector computes a local
threshold at every position in a read.
</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/regtools.html">regtools</a> (1.0.0) </div><p class="appDesc">regtools:Tools that integrate DNA-seq and RNA-seq data to help interpret mutations in a regulatory and splicing context</p>
<div class="subheading"><a href="repeatmasker.html">RepeatMasker</a> (4.1.5) </div><p class="appDesc">RepeatMasker is a program that screens DNA sequences for interspersed repeats and low complexity DNA sequences. The output of the program is a detailed annotation of the repeats that are present in the query sequence as well as a modified version of the query sequence in which all the annotated repeats have been masked (default: replaced by Ns). On average, almost 50% of a human genomic DNA sequence currently will be masked by the program. </p>
<div class="subheading"><a href="repeatmodeler.html">repeatmodeler</a> (2.0.1) </div><p class="appDesc">RepeatModeler is a de novo transposable element (TE) family identification and modeling package. RepeatModeler assists in automating the runs of the various algorithms given a genomic database, clustering redundant results, refining and classifying the families and producing a high quality library of TE families suitable for use with RepeatMasker and ultimately for submission to the Dfam database (http://dfam.org).</p>
<div class="subheading"><a href="REViewer.html">REViewer</a> (0.2.7) </div><p class="appDesc">REViewer is a tool for visualizing alignments of reads in regions containing tandem repeats. REViewer requires a BAMlet with graph-realigned reads generated by ExpansionHunter and the corresponding variant catalog.</p>
<div class="subheading"><a href="rgt.html">rgt</a> (0.13.2) </div><p class="appDesc">Regulatory Genomics Toolbox: Python library and set of tools for the integrative analysis of high throughput regulatory genomics data. http://www.regulatory-genomics.org</p>
<div class="subheading"><a href="rilseq.html">rilseq</a> (0.82) </div><p class="appDesc">RILseq computational protocol</p>
<div class="subheading"><a href="rnaseqc.html">rnaseqc</a> (2.4.2) </div><p class="appDesc">RNA-SeQC is a java program which computes a series of quality control metrics for RNA-seq data. </p>
<div class="subheading"><a href="rockhopper.html">rockhopper</a> (2.0.3) </div><p class="appDesc">Rockhopper is a comprehensive and user-friendly system for computational analysis of bacterial RNA-seq data. As input, Rockhopper takes RNA sequencing reads output by high-throughput sequencing technology (FASTQ, QSEQ, FASTA, SAM, or BAM files)</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/ROSE.html">ROSE</a> (1.3.1) </div><p class="appDesc">ROSE (Rank Ordering of Super-Enhancers) is tool for 
(1) creating stitched enhancers, and  
(2) separating super-enhancers from typical enhancers. 
given sequencing data (.bam) and a file of previously identified constituent enhancers (.gff)</p>
<div class="subheading"><a href="RSD.html">RSD</a> (1.1.7) </div><p class="appDesc">Reciprocal Smallest Distance (RSD) is a pairwise orthology algorithm that uses global sequence alignment and maximum likelihood evolutionary distance between sequences to accurately detects orthologs between genomes.</p>
<div class="subheading"><a href="rsem.html">rsem</a> (1.3.3) </div><p class="appDesc">RSEM is a software package for estimating gene and isoform expression levels from RNA-Seq data. </p>
<div class="subheading"><a href="rseqc.html">rseqc</a> (5.0.3) </div><p class="appDesc">Rseqc comprehensively evaluate RNA-seq datasets generated from clinical tissues or other well annotated organisms such as mouse, fly and yeast. </p>
<div class="subheading"><a href="rtg-tools.html">rtg-tools</a> (3.8.4) </div><p class="appDesc">variant detection for singletons, families, large pedigrees and populations, cancer, structural variant and CNV analysis, and microbial and metagenomic analysis</p>
<div class="subheading"><a href="rvtests.html">rvtests</a> (2.1.0) </div><p class="appDesc">Rare Variant tests is a flexible software package for genetic association studies. It is designed to support unrelated individual or related (family-based) individuals</p>
<div class="subheading"><a href="sailfish.html">sailfish</a> (0.10.0) </div><p class="appDesc">Sailfish is a tool for transcript quantification from RNA-seq data. It requires a set of target transcripts (either from a reference or de-novo assembly) to quantify. All that is needed to run sailfish is a fasta file containing your reference transcripts and a (set of) fasta/fastq file(s) containing your RNA-Seq reads.</p>
<div class="subheading"><a href="salmon.html">salmon</a> (1.10.0) </div><p class="appDesc">a tool for quantifying the expression of transcripts using RNA-seq data. </p>
<div class="subheading"><a href="salmonte.html">SalmonTE</a> (0.4) </div><p class="appDesc">SalmonTE is an ultra-Fast and Scalable Quantification Pipeline of Transpose Element (TE) Abundances.</p>
<div class="subheading"><a href="sambamba.html">sambamba</a> (0.8.2) </div><p class="appDesc">Sambamba is a high performance modern robust and fast tool (and library), written in the D programming language, for working with SAM and BAM files. Current parallelised functionality is an important subset of samtools functionality, including view, index, sort, markdup, and depth.</p>
<div class="subheading"><a href="samblaster.html">samblaster</a> (0.1.26) </div><p class="appDesc">samblaster is a program for marking duplicates and finding discordant/split read pairs in read-id grouped paired-end SAM files. When marking duplicates, samblaster will use about 20MB per 1M read pairs. In a read-id grouped SAM file all alignments for a read-id (QNAME) are continuous. Aligners naturally produce such files. They can also be created by sorting a SAM file by read-id. </p>
<div class="subheading"><a href="samtools.html">samtools</a> (1.17) </div><p class="appDesc">The samtools package now provides samtools, bcftools, tabix, and the underlying htslib library. </p>
<div class="subheading"><a href="scallop.html">scallop</a> (0.10.5) </div><p class="appDesc">Scallop is a reference-based transcript assembler.</p>
<div class="subheading"><a href="scalpel.html">scalpel</a> (0.5.4) </div><p class="appDesc">Bioinformatics pipeline for discovery of genetic variants from NGS reads.</p>
<div class="subheading"><a href="scanpy.html">scanpy</a> (1.8.1) </div><p class="appDesc">Scanpy is a scalable toolkit for analyzing single-cell gene expression data. It includes preprocessing, visualization, clustering, pseudotime and trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.</p>
<div class="subheading"><a href="https://schicexplorer.readthedocs.io/">schicexplorer</a> (7) </div><p class="appDesc">scHiCExplorer is a set of programs to process, normalize, analyse and visualize single-cell Hi-C data</p>
<div class="subheading"><a href="Scramble.html">Scramble</a> (1.0.2) </div><p class="appDesc">Scramble is a mobile element insertion (MEI) detection tool. It identifies clusters of soft clipped reads
in a BAM file, builds consensus sequences, aligns to representative L1Ta, AluYa5, and SVA-E sequences, and
outputs MEI calls.</p>
<div class="subheading"><a href="seqan.html">seqan</a> (2.4.0) </div><p class="appDesc"> SeqAn is an open source C++ library of efficient algorithms and data structures for the analysis of sequences with the focus on biological data. It applies a unique generic design that guarantees high performance, generality, extensibility, and integration with other libraries. This package also contains a suite of apps, including Fiona, Gustaf, Mason, RazerS 3, Yara, SeqAn T-Coffee, Stellar, and searchjoin.</p>
<div class="subheading"><a href="seqkit.html">seqkit</a> (2.2.0) </div><p class="appDesc">A cross-platform toolkit for FASTA/Q file manipulation</p>
<div class="subheading"><a href="seqtk.html">seqtk</a> (1.4) </div><p class="appDesc">seqtk is a toolkit for processing sequences in FASTA/Q formats</p>
<div class="subheading"><a href="sequenza-utils.html">sequenza-utils</a> (3.0.0) </div><p class="appDesc">Sequenza-utils is The supporting python library for the sequenza R package.

Sequenza is a project the estimate purity/ploidy and copy number alteration from tumor sequencing experiments. Sequenza-utils provide command lines programs to transform common NGS file type, such as BAM, pileup and VCF, to input files for the R package

</p>
<div class="subheading"><a href="shapeit.html">shapeit</a> (5.1.0) </div><p class="appDesc">SHAPEIT is a fast and accurate haplotype inference software</p>
<div class="subheading"><a href="shasta.html">shasta</a> (0.11.1) </div><p class="appDesc">De novo assembly from Oxford Nanopore reads</p>
<div class="subheading"><a href="shmlast.html">shmlast</a> (1.6) </div><p class="appDesc">shmlast is a reimplementation of the Conditional Reciprocal Best Hits algorithm for finding potential orthologs between a transcriptome and a species-specific protein database. It uses the LAST aligner and the pydata stack to achieve much better performance while staying in the Python ecosystem.
</p>
<div class="subheading"><a href="shrimp.html">shrimp</a> (2_2_3) </div><p class="appDesc">SHRiMP is a software package for aligning genomic reads against a target genome. It was primarily developed with the multitudinous short reads of next generation sequencing machines in mind, as well as Applied Biosystem's colourspace genomic representation. </p>
<div class="subheading"><a href="sicer.html">sicer</a> (2-1.0.3) </div><p class="appDesc">A clustering approach for identification of enriched domains from histone modification ChIP-Seq data</p>
<div class="subheading"><a href="sickle.html">sickle</a> (1.33) </div><p class="appDesc">A windowed adaptive trimming tool for FASTQ files using quality</p>
<div class="subheading"><a href="SIFT.html">SIFT</a> (6.2.1) </div><p class="appDesc">SIFT predicts whether an amino acid substitution affects protein function based on sequence homology and the physical properties of amino acids.</p>
<div class="subheading"><a href="slamdunk.html">slamdunk</a> (0.4.3) </div><p class="appDesc">SlamDunk is a novel, fully automated software tool for automated, robust, scalable and reproducible SLAMseq data analysis. </p>
<div class="subheading"><a href="smrtanalysis.html">smrtanalysis</a> (12.0.0.177059) </div><p class="appDesc">SMRT® Analysis is a bioinformatics software suite available for analysis of DNA sequencing data from Pacific Biosciences’ SMRT technology. Users can choose from a variety of analysis protocols that utilize PacBio® and third-party tools. Analysis protocols include de novo genome assembly, cDNA mapping, DNA base-modification detection, and long-amplicon analysis to determine phased consensus sequences. </p>
<div class="subheading"><a href="sniffles.html">sniffles</a> (2.2) </div><p class="appDesc">Sniffles is a structural variation caller using third generation sequencing (PacBio or Oxford Nanopore). It detects all types of SVs (10bp+) using evidence from split-read alignments, high-mismatch regions, and coverage analysis.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/snippy.html">snippy</a> (4.6.0) </div><p class="appDesc">snippy is a tool for rapid haploid variant calling and core genome alignment</p>
<div class="subheading"><a href="snp2hla.html">snp2hla</a> (1.0.3) </div><p class="appDesc">SNP2HLA is a tool to impute amino acid polymorphisms and single nucleotide polymorphisms in human luekocyte antigenes (HLA) within the major histocompatibility complex (MHC) region in chromosome 6. </p>
<div class="subheading"><a href="snpEff.html">snpEff</a> (5.1d) </div><p class="appDesc">snpEff is a variant annotation and effect prediction tool. It annotates and predicts the effects of variants on genes (such as amino acid changes).</p>
<div class="subheading">snptest (2.5.6) </div><p class="appDesc">SNPTEST is a program for the analysis of single SNP association in genome-wide studies. The tests implemented include

    * Binary (case-control) phenotypes, single and multiple quantitative phenotypes
    * Bayesian and Frequentist tests
    * Ability to condition upon an arbitrary set of covariates
    * Various different methods for the dealing with imputed SNPs.

The program is designed to work seamlessly with the output of both the genotype calling program CHIAMO, the genotype imputation program IMPUTE and the program GTOOL. </p>
<div class="subheading"><a href="somaticsniper.html">somaticsniper</a> (1.0.5.0) </div><p class="appDesc">The purpose of this program is to identify single nucleotide positions that are different between tumor and normal (or, in theory, any two bam files). It takes a tumor bam and a normal bam and compares the two to determine the differences. It outputs a file in a format very similar to Samtools consensus format. </p>
<div class="subheading"><a href="sortmeRNA.html">sortmeRNA</a> (4.3.6) </div><p class="appDesc">SortMeRNA is a local sequence alignment tool for filtering, mapping and clustering.
The core algorithm is based on approximate seeds and allows for sensitive analysis
of NGS reads. The main application of SortMeRNA is filtering rRNA
from metatranscriptomic data. SortMeRNA takes as input a file of reads
(fasta or fastq format) and one or multiple rRNA database file(s),
and sorts apart aligned and rejected reads into two files specified by the user.</p>
<div class="subheading"><a href="spaceranger.html">spaceranger</a> (2.1.1) </div><p class="appDesc">10x pipeline for processing Visium spatial RNA-seq data</p>
<div class="subheading"><a href="spades.html">spades</a> (3.15.5) </div><p class="appDesc">SPAdes – St. Petersburg genome assembler – is intended for both standard isolates and single-cell MDA bacteria assemblies. </p>
<div class="subheading"><a href="speedseq.html">speedseq</a> (0.1.2-20180208-4e60002) </div><p class="appDesc">SpeedSeq is a genome analysis platform designed for rapid whole-genome variant detection and interpretation </p>
<div class="subheading"><a href="sratoolkit.html">sratoolkit</a> (3.0.2) </div><p class="appDesc">The NCBI SRA Toolkit enables reading ("dumping") of sequencing files from the SRA database and writing ("loading") files into the .sra format.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/srst2.html">SRST2</a> (0.2.0) </div><p class="appDesc">SRST2 is a a read mapping-based tool for rapid molecular typing of bacterial pathogens.
It allows fast and accurate detection of genes, alleles and multi-locus
sequence types (MLST) from WGS data. SRST2 is highly accurate and outperforms assembly-based methods in terms of both gene detection and allele assignment.</p>
<div class="subheading"><a href="stampy.html">stampy</a> (1.0.32) </div><p class="appDesc">Short read aligner</p>
<div class="subheading"><a href="STAR.html">STAR</a> (2.7.10b) </div><p class="appDesc">Spliced Transcripts Alignment to a Reference</p>
<div class="subheading"><a href="STAR.html">STAR-Fusion</a> (1.12.0) </div><p class="appDesc">Transcript fusion detection</p>
<div class="subheading"><a href="STREAM.html">STREAM</a> (20180816) </div><p class="appDesc">STREAM stands for Single-cell Trajectories Reconstruction, Exploration And Mapping ofomics data. It is an interactive pipeline capable of disentangling and visualizing
complex branching trajectories from both single-cell transcriptomic and epigenomic data.</p>
<div class="subheading"><a href="strelka.html">strelka</a> (2.9.10) </div><p class="appDesc">Strelka is an analysis package designed to detect somatic SNVs and small indels from the aligned sequencing reads of matched tumor-normal samples.</p>
<div class="subheading"><a href="stringtie.html">stringtie</a> (2.2.1) </div><p class="appDesc">StringTie is a fast and highly efficient assembler of RNA-Seq alignments into potential transcripts. It is primarily a genome-guided transcriptome assembler, although it can borrow algorithmic techniques from de novo genome assembly to help with transcript assembly. </p>
<div class="subheading"><a href="subread.html">subread</a> (2.0.3) </div><p class="appDesc">High-performance read alignment, quantification and mutation discovery</p>
<div class="subheading">suppa (2.3) </div><p class="appDesc">Fast, accurate, and uncertainty-aware differential splicing analysis across multiple conditions </p>
<div class="subheading"><a href="survivor.html">SURVIVOR</a> (1.0.7) </div><p class="appDesc">SURVIVOR is a tool set for simulating/evaluating SVs, merging and comparing SVs within and among samples, and includes various methods to reformat or summarize SVs.</p>
<div class="subheading"><a href="svanna.html">svanna</a> (1.0.3) </div><p class="appDesc">The svanna is an efficient and accurate pathogenicity prediction for coding and regulatory structural variants in long-read genome sequencing. </p>
<div class="subheading"><a href="svtyper.html">svtyper</a> (0.7.1) </div><p class="appDesc">Svtyper is a Bayesian genotyper for structural variants.</p>
<div class="subheading"><a href="svviz.html">svviz</a> (1.6.2) </div><p class="appDesc">svviz visualizes high-throughput sequencing data relevant to a structural variant. Only reads supporting the variant or the reference allele will be shown. svviz can operate in both an interactive web browser view to closely inspect individual variants, or in batch mode, allowing multiple variants (annotated in a VCF file) to be analyzed simultaneously. </p>
<div class="subheading"><a href="talon.html">talon</a> (6.0) </div><p class="appDesc">TALON is a Python package for identifying and quantifying known and novel genes/isoforms in long-read transcriptome data sets.</p>
<div class="subheading"><a href="taxonkit.html">taxonkit</a> (0.12.0) </div><p class="appDesc">A Cross-platform and Efficient NCBI Taxonomy Toolkit
</p>
<div class="subheading"><a href="telescope.html">telescope</a> (6cd5525) </div><p class="appDesc">Single locus resolution of Transposable ELEment expression. Telescope estimates transposable element expression (retrotranscriptome) resolved to specific genomic locations. It directly addresses uncertainty in fragment assignment by reassigning ambiguously mapped fragments to the most probable source transcript as determined within a Bayesian statistical model.</p>
<div class="subheading">telseq (0.0.2) </div><p class="appDesc">TelSeq is a software that estimates telomere length from whole genome sequencing data (BAMs).</p>
<div class="subheading"><a href="tetoolkit.html">tetoolkit</a> (2.2.1) </div><p class="appDesc">A package for including transposable elements in differential enrichment analysis of sequencing datasets. </p>
<div class="subheading"><a href="theta.html">THetA</a> (0.7-20-g94fd772) </div><p class="appDesc">Tumor Heterogeneity Analysis (THetA) is an algorithm used to estimate tumor purity and clonal/subclonal copy number aberrations simultaneously from high-throughput DNA sequencing data.</p>
<div class="subheading"><a href="tombo.html">tombo</a> (1.5.1) </div><p class="appDesc">a suite of tools primarily for the identification of modified nucleotides from nanopore sequencing data. </p>
<div class="subheading"><a href="tophat.html">tophat</a> (2.1.2) </div><p class="appDesc">TopHat is a fast splice junction mapper for RNA-Seq reads. It aligns RNA-Seq reads to mammalian-sized genomes using the ultra high-throughput short read aligner Bowtie, and then analyzes the mapping results to identify splice junctions between exons. </p>
<div class="subheading"><a href="TPMCalculator.html">TPMCalculator</a> (0.0.4) </div><p class="appDesc">TPMCalculator quantifies mRNA abundance directly from the alignments by parsing BAM files</p>
<div class="subheading"><a href="transcript_clean.html">transcript_clean</a> (2.0.3) </div><p class="appDesc">TranscriptClean is a Python program that corrects mismatches, microindels, and noncanonical splice junctions in long reads that have been mapped to the genome. It is designed for use with sam files from the PacBio Iso-seq and Oxford Nanopore transcriptome sequencing technologies</p>
<div class="subheading"><a href="TransDecoder.html">TransDecoder</a> (5.5.0) </div><p class="appDesc">TransDecoder identifies candidate coding regions within transcript sequences, such as those generated by de novo RNA-Seq transcript assembly using Trinity, or constructed based on RNA-Seq alignments to the genome using Tophat and Cufflinks.</p>
<div class="subheading"><a href="transvar.html">transvar</a> (2.5.9) </div><p class="appDesc">TransVar is a versatile annotator for 3-way conversion and annotation among genomic characterization(s) of mutations and transcript-dependent annotation(s). </p>
<div class="subheading"><a href="trimAl.html">trimAl</a> (1.2rev59) </div><p class="appDesc">trimAl is a tool for the automated removal of spurious sequences or poorly aligned regions from a multiple sequence alignment.
It can consider several parameters, alone or in multiple combinations,
in order to select the most-reliable positions in the alignment.
These include the proportion of sequences with a gap, the level of residue similarity and,
if several alignments for the same set of sequences are provided, the consistency level of columns among alignments.
Moreover, trimAl is able to manually select a set of columns to be removed from the alignment.</p>
<div class="subheading"><a href="trimgalore.html">trimgalore</a> (0.6.7) </div><p class="appDesc">Consistent quality and adapter trimming for RRBS or standard FastQ files.</p>
<div class="subheading"><a href="trimmomatic.html">trimmomatic</a> (0.39) </div><p class="appDesc">Trimmomatic performs a variety of useful trimming tasks for illumina paired-end and single ended data.</p>
<div class="subheading"><a href="trinity.html">trinity</a> (2.15.1) </div><p class="appDesc">Trinity, developed at the Broad Institute and the Hebrew University of Jerusalem, represents a novel method for the efficient and robust de novo reconstruction of transcriptomes from RNA-seq data.</p>
<div class="subheading"><a href="tRNAscan-SE.html">tRNAscan-SE</a> (2.0.9) </div><p class="appDesc">tRNAscan-SE 2.0 has advanced the state-of-the-art methodology in tRNA gene detection and functional prediction, captured by rich new content of the companion Genomic tRNA Database</p>
<div class="subheading"><a href="ultraplex.html">ultraplex</a> (1.2.5) </div><p class="appDesc">Ultraplex is primarily designed for the demultiplexing of sequencing data generated using in-house library preparation protocols with custom adaptors</p>
<div class="subheading"><a href="umitools.html">umitools</a> (1.1.2) </div><p class="appDesc">tools for dealing with Unique Molecular Identifiers (UMIs)/Random Molecular Tags (RMTs) and single cell RNA-Seq cell barcodes</p>
<div class="subheading"><a href="vcf2maf.html">vcf2maf</a> (1.6.21) </div><p class="appDesc">A smarter, more reproducible, and more configurable tool for converting a VCF to a MAF.</p>
<div class="subheading"><a href="vcfanno.html">vcfanno</a> (0.3.3) </div><p class="appDesc">annotate a VCF with other VCFs/BEDs/tabixed files</p>
<div class="subheading"><a href="vcflib.html">vcflib</a> (1.0.3) </div><p class="appDesc">a simple C++ library for parsing and manipulating VCF files, + many command-line utilities</p>
<div class="subheading"><a href="vcftools.html">vcftools</a> (0.1.16) </div><p class="appDesc">VCFtools contains a Perl API (Vcf.pm) and a number of Perl scripts that can be used to perform common tasks with VCF files such as file validation, file merging, intersecting, complements, etc. </p>
<div class="subheading"><a href="VEP.html">VEP</a> (110) </div><p class="appDesc">VEP (Variant Effect Predictor) determines the effect of your variants (SNPs, insertions, deletions, CNVs or structural variants) on genes, transcripts, and protein sequence, as well as regulatory regions.</p>
<div class="subheading"><a href="verifybamid.html">verifybamid</a> (2.0.1) </div><p class="appDesc">verifyBamID is a software that verifies whether the reads in particular file match previously known genotypes for an individual (or group of individuals), and checks whether the reads are contaminated as a mixture of two samples. verifyBamID can detect sample contamination and swaps when external genotypes are available. When external genotypes are not available, verifyBamID still robustly detects sample swaps. </p>
<div class="subheading"><a href="verkko.html">verkko</a> (1.4.1) </div><p class="appDesc">Verkko is a hybrid genome assembly pipeline developed for telomere-to-telomere assembly of PacBio HiFi and Oxford Nanopore reads.</p>
<div class="subheading"><a href="VirSorter2.html">VirSorter2</a> (2.2.3) </div><p class="appDesc">VirSorter2 is a DNA and RNA virus identification tool. It leverages genome-informed
database advances across a collection of customized automatic classifiers
to improve the accuracy and range of virus sequence detection.</p>
<div class="subheading"><a href="VIRTUS.html">VIRTUS</a> (2.0.1) </div><p class="appDesc">Bioinformatics pipeline for viral transcriptome detection.</p>
<div class="subheading"><a href="vsearch.html">vsearch</a> (2.22.1) </div><p class="appDesc">VSEARCH supports de novo and reference based chimera detection, clustering, full-length and prefix dereplication, rereplication, reverse complementation, masking, all-vs-all pairwise global alignment, exact and global alignment searching, shuffling, subsampling and sorting. It also supports FASTQ file analysis, filtering, conversion and merging of paired-end reads.</p>
<div class="subheading"><a href="vt.html">vt</a> (0.57721) </div><p class="appDesc">vt is a variant tool set that discovers short variants from Next Generation Sequencing data.
</p>
<div class="subheading"><a href="winnowmap.html">winnowmap</a> (2.03) </div><p class="appDesc">winnowmap is used for mapping ONT and PacBio reads to repetitive reference sequences.</p>
<div class="subheading"><a href="xHLA.html">xHLA</a> (2018-04-04) </div><p class="appDesc">The HLA gene complex on human chromosome 6 is one of the
most polymorphic regions in the human genome and contributes
in large part to the diversity of the immune system. Accurate typing of HLA genes with short-read sequencing data has historically been difficult due to the sequence similarity between the polymorphic alleles. xHLA iteratively refines the mapping results at the amino acid level to
achieve high typing accuracy for both class I and II HLA genes.</p>
<div class="subheading"><a href="XHMM.html">XHMM</a> (2016-01-04) </div><p class="appDesc">XHMM uses principal component analysis (PCA) normalization and a hidden Markov model (HMM) to detect and genotype copy number variation (CNV) from normalized read-depth data from targeted sequencing experiments.</p>
<div class="subheading"><a href="yak.html">yak</a> (r69) </div><p class="appDesc">Yet another K-mer analyzer. Yak is initially developed for two specific use cases: 1) to robustly estimate the base accuracy of CCS reads and assembly contigs, and 2) to investigate the systematic error rate of CCS reads.</p>
<div class="heading"><a name="image" id="image">Image Analysis</a></div>
<a href="/apps/" style="font-size:12px">back to top</a><br />
<p></p>
<div class="subheading"><a href="3DSlicer.html">3DSlicer</a> (5.2.2) </div><p class="appDesc">A software platform for the analysis (including registration and interactive segmentation) and visualization (including volume rendering) of medical images and for research in image guided therapy. </p>
<div class="subheading"><a href="afni.html">AFNI</a> (current-py3) </div><p class="appDesc">AFNI (Analysis of Functional NeuroImages) is a set of C programs for processing, analyzing, and displaying functional MRI (FMRI) data - a technique for mapping human brain activity.

</p>
<div class="subheading"><a href="ANTs.html">ANTs</a> (2.4.2) </div><p class="appDesc">Advanced Normalization Tools (ANTs) extracts information from complex datasets that include imaging. Paired with ANTsR (answer), ANTs is useful for managing, interpreting and visualizing multidimensional data. </p>
<div class="subheading">AreTomo (1.3.4) </div><p class="appDesc">Alignment and Reconstruction for Electron Tomography</p>
<div class="subheading"><a href="b-soid.html">B-SOID</a> (1.3) </div><p class="appDesc">B-SOID (Behavioral Segmentation in Deeplabcut) is an 
unsupervised learning algorithm that serves to discover and classify behaviors that are not pre-defined by users.
It segregates statistically different, sub-second rodent behaviors with a single bottom-up perspective video cameraR by performing a novel expectation maximization
fitting of Gaussian mixture models on t-Distributed Stochastic Neighbor Embedding (t-SNE).</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/baracus.html">baracus</a> (1.1.4) </div><p class="appDesc">Baracus predicts brain age, based on data from Freesurfer. It combines data from cortical thickness, cortical surface area, and subcortical information </p>
<div class="subheading"><a href="BrkRaw.html">BrkRaw</a> (0.3.6) </div><p class="appDesc">The ‘BrkRaw’ is a python module designed to provide a comprehensive tool to access raw data acquired from Bruker Biospin preclinical MRI scanner. This module is also compatible with the zip compressed data to enable use of the archived data directly.
The module is comprised of four components, including graphical user interface (GUI), command-line tools, high-level and low-level python APIs.</p>
<div class="subheading"><a href="Bsoft.html">Bsoft</a> (1.9.0) </div><p class="appDesc">Bsoft is a collection of programs and a platform for development of software for image and molecular processing in structural biology. Problems in structural biology are approached with a highly modular design, allowing fast development of new algorithms without the burden of issues such as file I/O. It provides an easily accessible interface, a resource that can be and has been used in other packages.
</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/c3d.html">c3d</a> (1.1.0) </div><p class="appDesc">C3D is a command-line tool for converting 3D images between common file formats. The tool also includes a growing list of commands for image manipulation, such as thresholding and resampling.</p>
<div class="subheading"><a href="cellpose.html">cellpose</a> (2.1.1) </div><p class="appDesc">A generalist algorithm for cellular segmentation with human-in-the-loop capabilities.</p>
<div class="subheading"><a href="cisTEM.html">cisTEM</a> (1.0.0-beta) </div><p class="appDesc">cisTEM is user-friendly software to process cryo-EM images of macromolecular complexes and obtain high-resolution 3D reconstructions from them.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/civet.html">civet</a> (2.1.1) </div><p class="appDesc">civet is a brain-imaging pipeline for
analysis of large MR data sets. civet
extracts and analyses cortical surfaces 
from MR images, as well as many other 
volumetric and corticometric functions.</p>
<div class="subheading"><a href="cmtk.html">cmtk</a> (3.3.2) </div><p class="appDesc">CMTK is a Software toolkit for computational morphometry of biomedical images. CMTK provides a set of command line tools for processing and I/O.</p>
<div class="subheading">CompuCell3D (4.4.0) </div><p class="appDesc">CompuCell3D is a multiscale multicellular virtual tissue modeling and simulation environment. CompuCell3D is written in C++ and provides Python bindings for model and simulation development in Python. CompuCell3D is supported on Windows, Mac and Linux.</p>
<div class="subheading"><a href="connectome-workbench.html">connectome-workbench</a> (1.5.0) </div><p class="appDesc">Tools to browse, download, explore, and analyze data from the Human Connectome Project (HCP).  Allows users to compare their own data to that of the HCP.  </p>
<div class="subheading"><a href="coolbox.html">coolbox</a> (0.3.8) </div><p class="appDesc">CoolBox is an open-source, user-friendly toolkit for visual analysis of genomics data.
It is highly compatible with the Python ecosystem and customizable with a well-designed user interface.
It can bed used, for example, to produce high-quality genome track
plots or fetch commonly used genomic data files with a Python script or command
line, to explore genomic data interactively within Jupyter environment or web browser.</p>
<div class="subheading"><a href="https://hpc.nih.gov/nih/cryosparc/">cryosparc</a> (4.2.1) </div><p class="appDesc">CryoSPARC (Cryo-EM Single Particle Ab-Initio Reconstruction and Classification) is a state of the art HPC software solution for complete processing of single-particle cryo-electron microscopy (cryo-EM) data. CryoSPARC is useful for solving cryo-EM structures of membrane proteins, viruses, complexes, flexible molecules, small particles, phase plate data and negative stain data.</p>
<div class="subheading"><a href="afni.html">CTF</a> (6.1) </div><p class="appDesc">The CTF MEG software has two main roles:
- Provide a human-machine interface to the CTF MEG elec- tronics to collect MEG and/or EEG data.
- Provide a tool for reviewing and (to a limited extent) ana- lyzing the MEG and/or EEG data acquired by the CTF MEG system.</p>
<div class="subheading"><a href="ctffind.html">ctffind</a> (4.1.14) </div><p class="appDesc">Programs for finding CTFs of electron micrographs</p>
<div class="subheading"><a href="dcm2niix.html">dcm2niix</a> (1.0.20211006) </div><p class="appDesc">DICOM to NIfTI converter</p>
<div class="subheading"><a href="DeepCAD.html">DeepCAD</a> (20210826) </div><p class="appDesc">DeepCAD is a self-supervised deep-learning method
for spatiotemporal enhancement of calcium imaging data
that does not require any high signal-to-noise ratio (SNR) observations.
DeepCAD suppresses detection noise and improves the SNR more than tenfold,
which reinforces the accuracy of neuron extraction and spike inference
and facilitates the functional analysis of neural circuits.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/DeepLabCut.html">DeepLabCut</a> (2.3.4) </div><p class="appDesc">DeepLabCut is an open source toolbox
that builds on a state-of-the-art human pose estimation algorithm.
It allows training of a deep neural network by using limited training data
to precisely track user-defined features, so that the human labeling accuracy will be matched.</p>
<div class="subheading"><a href="deepmedic.html">deepmedic</a> (0.8.4) </div><p class="appDesc">This project aims to offer easy access to Deep Learning for segmentation of structures of interest in biomedical 3D scans. It is a system that allows the easy creation of a 3D Convolutional Neural Network, which can be trained to detect and segment structures if corresponding ground truth labels are provided for training. The system processes NIFTI images, making its use straightforward for many biomedical tasks.</p>
<div class="subheading"><a href="dmriprep.html">dmriprep</a> (0.5.0) </div><p class="appDesc">Pipeline for preprocessing diffusion MRI datasets.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/dynamo.html">dynamo</a> (1.1.532) </div><p class="appDesc">Dynamo is a software environment for subtomogram averaging of cryo-EM data.</p>
<div class="subheading"><a href="elastix.html">elastix</a> (4.9; 5.1.0) </div><p class="appDesc">a toolbox for rigid and nonrigid registration of images.</p>
<div class="subheading"><a href="EMAN2.html">EMAN2</a> (2.99) </div><p class="appDesc">EMAN2 is a broadly based greyscale scientific image processing suite with a primary focus on processing data from transmission electron microscopes. </p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/fastsurfer.html">fastsurfer</a> (1.1.1) </div><p class="appDesc">Fastsurfer is a neuroimaging pipeline based on deep learning.</p>
<div class="subheading"><a href="Fiji.html">Fiji</a> (1.52f) </div><p class="appDesc">Fiji is a distribution of the popular open-source software ImageJ focused on biological-image analysis. Fiji uses modern software engineering practices to combine powerful software libraries with a broad range of scripting languages to enable rapid prototyping of image-processing algorithms. Fiji facilitates the transformation of new algorithms into ImageJ plugins that can be shared with end users through an integrated update system.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/fitlins.html">fitlins</a> (0.9.1) </div><p class="appDesc">Fitlins fits linear models to BIDS neuroimaging datasets.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/fmriprep.html">fmriprep</a> (23.1.4) </div><p class="appDesc">A Robust Preprocessing Pipeline for fMRI Data</p>
<div class="subheading"><a href="Frealign.html">Frealign</a> (9.11_151031) </div><p class="appDesc">Frealign is a program for high-resolution refinement of 3D reconstructions from cryo-EM images of single particles.</p>
<div class="subheading"><a href="freesurfer.html">Freesurfer</a> (7.4.1) </div><p class="appDesc">Freesurfer is a set of automated tools for reconstruction of the brain's cortical surface from structural MRI data, and overlay of functional MRI data onto the reconstructed surface.</p>
<div class="subheading"><a href="fsl.html">FSL</a> (6.0.6) </div><p class="appDesc">FSL is a comprehensive library of image analysis and statistical tools for FMRI, MRI and DTI brain imaging data.</p>
<div class="subheading"><a href="Gctf.html">Gctf</a> (1.06) </div><p class="appDesc">Gctf provides accurate estimation of the contrast transfer function (CTF) for near-atomic resolution cryo electron microscopy (cryoEM) reconstruction using GPUs.  </p>
<div class="subheading"><a href="Huygens.html">Huygens</a> (23.04.0-p6) </div><p class="appDesc">Huygens is an image restoration, deconvolution, resolution and noise reduction. It can process images from all current optical microscopes, including wide-field, confocal, Nipkow (scanning disk confocal), multiple-photon, and 4Pi microscopes.</p>
<div class="subheading"><a href="imaris.html">Imaris</a> (9.7) </div><p class="appDesc">Imaris provides scientists with solutions for processing, visualizing and analyzing multi-dimensional microscopic images. It reads images in many of the most commonly used proprietary formats.</p>
<div class="subheading"><a href="IMOD.html">IMOD</a> (4.12.25) </div><p class="appDesc">IMOD is a set of image processing, modeling and display programs used for tomographic reconstruction and for 3D reconstruction of EM serial sections and optical sections. </p>
<div class="subheading"><a href="IsoNet.html">IsoNet</a> (0.2.1) </div><p class="appDesc">IsoNet is a deep learning-based software package that iteratively reconstructs the missing-wedge information and increases signal-to-noise ratio, using the knowledge learned from raw tomograms. Without the need for sub-tomogram averaging, IsoNet generates tomograms with significantly reduced resolution anisotropy.</p>
<div class="subheading">ITK-SNAP (3.8.0) </div><p class="appDesc">ITK-SNAP is a tool for segmentation of 3D biomedical images.  It requires a graphical connection to run on the cluster.  </p>
<div class="subheading"><a href="ITK-SNAP-BS.html">ITK-SNAP-BS</a> (20131007) </div><p class="appDesc">ITK-SNAP brings active contour
segmentation to the fingertips of clinical researchers.
This application fulfills a specific and pressing need of biomedical
imaging research by providing a combination of manual and
semiautomatic tools for extracting structures in 3D image data of
different modalities and from different anatomical regions.</p>
<div class="subheading">laynii (2.4.0) </div><p class="appDesc">Tools to analyze layer fMRI datasets</p>
<div class="subheading">magetbrain (1.0) </div><p class="appDesc">Given a set of labelled MR images (atlases) and unlabelled images (subjects), MAGeT produces a segmentation for each subject using a multi-atlas voting procedure based on a template library made up of images from the subject set.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/mango.html">mango</a> (4.1) </div><p class="appDesc">Mango (Multi-image Analysis GUI) is a viewer for medical research images. It provides analysis tools and a user interface to navigate image volumes. </p>
<div class="subheading"><a href="https://github.com/BIC-MNI/minc-toolkit-v2">minc-toolkit</a> (1.9.18) </div><p class="appDesc">This metaproject bundles multiple MINC-based packages that historically have been developed somewhat independently</p>
<div class="subheading"><a href="MIPAV.html">MIPAV</a> (11.0.3) </div><p class="appDesc">The MIPAV (Medical Image Processing, Analysis, and Visualization) application enables quantitative analysis and visualization of medical images of numerous modalities such as PET, MRI, CT, or microscopy. </p>
<div class="subheading"><a href="MONAILabel.html">MONAILabel</a> (0.3.2; 0.4.2) </div><p class="appDesc">MONAI Label is a free and open-source platform that facilitates the development of AI-based applications
that aim at reducing the time required to annotate 3D medical image datasets.
It allows researchers to readily deploy their apps as services, which can be made available to clinicians
via their preferred user-interface. Currently, MONAI Label readily supports locally installed (3DSlicer) and web-based (OHIF) frontends, and offers two Active learning strategies to facilitate and speed up the training of segmentation algorithms. MONAI Label allows researchers to make incremental improvements to their labeling apps by making them available to other researchers and clinicians alike.</p>
<div class="subheading"><a href="MotionCor2.html">MotionCor2</a> (1.5.0) </div><p class="appDesc">MotionCor2 is a multi-GPU accelerated program that provides iterative, patch-based motion detection combining spatial and temporal constraints and dose weighting for both single particle and tomographic cryo-electon microscopy images.</p>
<div class="subheading">mricron (1.0.20190902) </div><p class="appDesc">Viewer for several types of brain scan formats</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/mriqc.html">mriqc</a> (23.1.0) </div><p class="appDesc">MRIQC is an MRI quality control tool</p>
<div class="subheading"><a href="mrtrix.html">mrtrix</a> (3.0.4) </div><p class="appDesc">MRtrix provides a large suite of tools for image processing, analysis and visualisation, with a focus on the analysis of white matter using diffusion-weighted MRI.</p>
<div class="subheading"><a href="mrtrix3tissue.html">mrtrix3tissue</a> (5.2.9) </div><p class="appDesc">MRtrix3Tissue is a fork of the MRtrix3 project. It aims to add capabilities for 3-Tissue CSD modelling and analysis to a complete version of the MRtrix3 software.</p>
<div class="subheading">napari (0.4.18) </div><p class="appDesc">napari is an image viewer.Can be used to annotate images.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/nibabies.html">nibabies</a> (22.0.2) </div><p class="appDesc">Preprocessing pipeline for neonate and infant MRI.</p>
<div class="subheading">OpenSlide (3.4.1) </div><p class="appDesc">OpenSlide is a C library for reading and manipulating digital slides of diverse
vendor formats. It provides a simple interface to read whole-slide images
(also known as virtual slides). OpenSlide has been used in the digital pathology projects.</p>
<div class="subheading"><a href="PEET.html">PEET</a> (1.15.0) </div><p class="appDesc">PEET (Particle Estimation for Electron Tomography) is an open-source package for aligning and averaging particles in 3-D subvolumes extracted from tomograms. It seeks the optimal alignment of each particle against a reference volume through several iterations. If PEET and IMOD are both installed, most PEET operations are available from the eTomo graphical user interface in IMOD. </p>
<div class="subheading"><a href="petprep_hmc.html">petprep_hmc</a> (0.06) </div><p class="appDesc">Positron Emission Tomography (PET) is a state-of-the-art neuroimaging tool for quantification of the in vivo spatial distribution of specific molecules in the brain. It is affected by various kinds of patient movement during a scan.
The PETprep_HMC application allows for correction of the PET results in the presence of head motions.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/plastimatch.html">plastimatch</a> (1.9.3) </div><p class="appDesc">Application for registration of medical images such as X-rays, CT, MRI and PET</p>
<div class="subheading"><a href="PyTom.html">PyTom</a> (1.0) </div><p class="appDesc">PyTom is a software package for the analysis of volumetric data
obtained by cryo electron tomography (cryo-ET).
It covers a complete pipeline of processing steps
for tomogram reconstruction, localization of macromolecular complexes in tomograms,
fine alignment of subtomograms extracted at these locations, and their classification.</p>
<div class="subheading"><a href="qsiprep.html">qsiprep</a> (0.16.1) </div><p class="appDesc">qsiprep configures pipelines for processing diffusion-weighted MRI (dMRI) data.</p>
<div class="subheading"><a href="QuPath.html">QuPath</a> (0.4.1) </div><p class="appDesc">QuPath is open source software for bioimage analysis. It is often used for digital pathology applications because it offers a powerful set of tools for working with whole slide images - but it can be applied to lots of other kinds of image as well.</p>
<div class="subheading"><a href="RELION/index.html">RELION</a> (4.0.1) </div><p class="appDesc">RELION (for REgularised LIkelihood OptimisatioN) is a stand-alone computer program for Maximum A Posteriori refinement of (multiple) 3D reconstructions or 2D class averages in cryo-electron microscopy.</p>
<div class="subheading"><a href="ResMap.html">ResMap</a> (1.9) </div><p class="appDesc">ResMap (Resolution Map) is a Python (NumPy/SciPy) application with a Tkinter GUI and a command-line interface. It is a software package for computing the local resolution of 3D density maps studied in structural biology, primarily electron cryo-microscopy (cryo-EM).</p>
<div class="subheading"><a href="SAMsrcV3.html">SAMsrcV3</a> (20180713-c5e1042) </div><p class="appDesc">Synthetic Aperture Magnetometry - The SANsrcV3 suite implements the latest advances in MEG source localization. </p>
<div class="subheading"><a href="simnibs.html">SimNIBS</a> (4.0) </div><p class="appDesc">SimNIBS is a free software package for the Simulation of Non-invasive Brain Stimulation. It allows for realistic calculations of the electric field induced by transcranial magnetic stimulation (TMS) and transcranial direct current stimulation (tDCS).</p>
<div class="subheading"><a href="smriprep.html">smriprep</a> (0.8.3) </div><p class="appDesc">Structural MRI PREProcessing (sMRIPrep) workflows for NIPreps (NeuroImaging PREProcessing tools).</p>
<div class="subheading"><a href="SPHIRE.html">SPHIRE</a> (1.4) </div><p class="appDesc">SPHIRE (SPARX for High-Resolution Electron Microscopy) is an open-source,
user-friendly software suite for the semi-automated
processing of single particle electron cryo-microscopy (cryo-EM) data.
It allows fast and reproducible structure determination from cryo-EM images.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/spm12.html">spm12</a> (7870) </div><p class="appDesc">The (S)tatistical (P)ara(M)etric application analyzes brain imaging data.</p>
<div class="subheading">tedana (23.0.1) </div><p class="appDesc">Tedana is an application to denoise multi-echo fMRI datasets</p>
<div class="subheading"><a href="tomotwin.html">tomotwin</a> (0.5.0) </div><p class="appDesc">TomoTwin - a deep metric learning based particle picking procedure for cryo-ET</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/topaz.html">topaz</a> (0.2.5-ba91e19) </div><p class="appDesc">topaz is a pipeline for particle detection in 
cryo-electron microscopy images using convolutional 
neural networks trained from positive and unlabeled 
examples. Topaz also includes methods for micrograph 
and tomogram denoising using deep denoising models.
</p>
<div class="subheading"><a href="TORTOISE.html">TORTOISE</a> (3.2.0) </div><p class="appDesc">(Tolerably Obsessive Registration and Tensor Optimization
Indolent Software Ensemble)
The TORTOISE software package is for processing diffusion MRI data.</p>
<div class="subheading"><a href="tractseg.html">tractseg</a> (1.7.1) </div><p class="appDesc">Tool for white matter bundle segmentation from Diffusion MRI.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/xcpengine.html">xcpengine</a> (1.2.4) </div><p class="appDesc">xpcEngine performs denoising and estimation of Functional Connectivity on fMRI datasets</p>
<div class="subheading"><a href="xcp_d.html">xcp_d</a> (0.5.0) </div><p class="appDesc">xcp_d is a postprocessing and noise regression pipeline for fMRI datasets (can use output from fmriprep and nibabies).</p>
<div class="heading"><a name="linkage" id="linkage">Linkage/Phylogenetics</a></div>
<a href="/apps/" style="font-size:12px">back to top</a><br />
<p></p>
<div class="subheading"><a href="AdmixTools.html">AdmixTools</a> (7.0.2) </div><p class="appDesc">ADMIXTOOLS is a software package that supports formal tests of whether admixture occurred, and makes it possible to infer admixture proportions and dates. </p>
<div class="subheading"><a href="admixture.html">admixture</a> (1.3.0) </div><p class="appDesc">ADMIXTURE is a software tool for maximum likelihood estimation of individual ancestries from multilocus SNP genotype datasets. It uses the same statistical model as STRUCTURE but calculates estimates much more rapidly using a fast numerical optimization algorithm. </p>
<div class="subheading"><a href="arcashla.html">arcashla</a> (0.5.0) </div><p class="appDesc">high resolution HLA typing from RNA seq</p>
<div class="subheading"><a href="Beagle.html">Beagle</a> (5.4_22Jul22) </div><p class="appDesc">Beagle is a package for imputing genotypes, inferring haplotype phase, and performing genetic association analysis. BEAGLE is designed to analyze large-scale data sets with hundreds of thousands of markers genotyped on thousands of samples.</p>
<div class="subheading"><a href="https://beagle-dev.github.io/html/beagle_8h.html">beagle-lib</a> (4.0) </div><p class="appDesc">BEAGLE is a high-performance library that can perform the core calculations at the heart of most Bayesian and Maximum Likelihood phylogenetics packages. It can make use of highly-parallel processors such as those in graphics cards (GPUs) found in many PCs.
<br>
module name: <tt>beagle-lib</tt></p>
<div class="subheading"><a href="BEAST.html">BEAST</a> (1.10.4,2.6.2) </div><p class="appDesc">BEAST (Bayesian Evolutionary Analysis Sampling Trees) is a cross-platform program for Bayesian MCMC analysis of molecular sequences.</p>
<div class="subheading"><a href="biobakery_workflows.html">biobakery_workflows</a> (3.1) </div><p class="appDesc">bioBakery is a meta’omic analysis environment and collection of individual software
tools with the capacity to process raw shotgun sequencing data into actionable microbial community
feature profiles, summary reports, and publication-ready figures.
It includes a collection of preconfigured analysis modules also joined into workflows for reproducibility.
Each individual module has been developed to perform a particular task, e.g. quantitative taxonomic
profiling or statistical analysis.</p>
<div class="subheading"><a href="cd-hit.html">CD-HIT</a> (4.6.8) </div><p class="appDesc">CD-HIT is a very widely used program for clustering and comparing protein or nucleotide sequences.</p>
<div class="subheading"><a href="eigensoft.html">eigensoft</a> (8.0.0) </div><p class="appDesc">The EIGENSOFT package combines functionality from population genetics methods and EIGENSTRAT stratification correction method.</p>
<div class="subheading"><a href="FastQTL.html">FastQTL</a> (2.184) </div><p class="appDesc">In order to discover quantitative trait loci (QTLs),
multi-dimensional genomic datasets combining
DNA-seq and ChiP-/RNA-seq require methods that rapidly correlate tens of thousands of
molecular phenotypes with millions of genetic variants while appropriately controlling for multiple testing. FastQTL implements a popular cis-QTL mapping strategy
in a user- and cluster-friendly tool. FastQTL also proposes an efficient
permutation procedure to control for multiple testing.
</p>
<div class="subheading"><a href="FastTree.html">FastTree</a> (2.1.11) </div><p class="appDesc">FastTree infers approximately-maximum-likelihood phylogenetic trees from alignments of nucleotide or protein sequences. FastTree can handle alignments with up to a million of sequences in a reasonable amount of time and memory. For large alignments, FastTree is 100-1,000 times faster than PhyML 3.0 or RAxML 7. </p>
<div class="subheading"><a href="GCTA.html">GCTA</a> (1.94.0beta0) </div><p class="appDesc">GCTA (Genome-wide Complex Trait Analysis) is designed to estimate the proportion of phenotypic variance explained by genome- or chromosome-wide SNPs for complex traits.</p>
<div class="subheading"><a href="gtdb-tk.html">gtdb-tk</a> (2.3.2) </div><p class="appDesc">GTDB-Tk is a software toolkit for assigning objective taxonomic classifications to bacterial and archaeal genomes based on the Genome Database Taxonomy</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/gubbins.html">gubbins</a> (2.3.4) </div><p class="appDesc">Gubbins is an algorithm that iteratively identifies loci containing elevated densities of base substitions while concurrently constructing a phylogeny based on the putative point mutations outside of these regions.
</p>
<div class="subheading"><a href="hyphy.html">hyphy</a> (2.5.29) </div><p class="appDesc">HyPhy (Hypothesis Testing using Phylogenies) is an open-source software package for the analysis of genetic sequences (in particular the inference of natural selection) using techniques in phylogenetics, molecular evolution, and machine learning.</p>
<div class="subheading"><a href="iqtree.html">iqtree</a> (2.2.0.5) </div><p class="appDesc">Efficient phylogenomic software by maximum likelihood</p>
<div class="subheading"><a href="king.html">king</a> (2.2.7) </div><p class="appDesc">KING is a toolset to explore genotype data from a genome-wide association study (GWAS) or a sequencing project. KING can be used to check family relationship and flag pedigree errors by estimating kinship coefficients and inferring IBD segments for all pairwise relationships.</p>
<div class="subheading"><a href="LTSOFT.html">LTSOFT</a> (4.0) </div><p class="appDesc">The LTSOFT application implements a new approach
to using information from known associated variants
when conducting disease association studies.
The approach is based in part on the classical technique
of liability threshold modeling and performs estimation
of model parameters for each known variant
while accounting for the published disease prevalence from
the epidemiological literature.</p>
<div class="subheading"><a href="Madeline.html">Madeline2</a> (2.0) </div><p class="appDesc">The Madeline 2.0 Pedigree Drawing Engine (PDE) is a
pedigree drawing program for use in linkage and family-based
association studies. The program is designed to handle large and
complex pedigrees with an emphasis on readability and aesthetics.</p>
<div class="subheading"><a href="MEGA.html">MEGA</a> (11.0.10) </div><p class="appDesc">MEGA, Molecular Evolutionary Genetics Analysis, is a software suite for analyzing DNA and protein sequence data from species and populations</p>
<div class="subheading"><a href="mega2.html">mega2</a> (6.0.0) </div><p class="appDesc">Mega2 is a data-handling program for facilitating genetic linkage and association analyses.</p>
<div class="subheading"><a href="merlin.html">merlin</a> (1.1.2) </div><p class="appDesc">MERLIN uses sparse trees to represent gene flow in pedigrees and is one of the fastest pedigree analysis packages around.</p>
<div class="subheading"><a href="mothur.html">mothur</a> (1.48.0) </div><p class="appDesc">mothur is a tool for analyzing 16S rRNA gene sequences generated on multiple platforms as part of microbial ecology projects.</p>
<div class="subheading"><a href="mrbayes.html">MrBayes</a> (3.2.7) </div><p class="appDesc">MrBayes is a program for Bayesian inference and model choice across a wide range of phylogenetic and evolutionary models. MrBayes uses Markov chain Monte Carlo (MCMC) methods to estimate the posterior distribution of model parameters.</p>
<div class="subheading"><a href="ohana.html">ohana</a> (current) </div><p class="appDesc">Ohana is a suite of software for analyzing population structure and admixture history using unsupervised learning methods.</p>
<div class="subheading"><a href="phase.html">phase</a> (2.1.1) </div><p class="appDesc">infers haplotypes from population genotype data</p>
<div class="subheading"><a href="Phylip.html">Phylip</a> (3.698) </div><p class="appDesc">Phylip is a package of programs for inferring phylogenies (evolutionary trees). Includes methods for parsimony, distance matrix and likelihood methods.

</p>
<div class="subheading"><a href="pplacer.html">pplacer</a> (1.1) </div><p class="appDesc">Pplacer places query sequences on a fixed reference phylogenetic tree to maximize phylogenetic likelihood or posterior probability according to a reference alignment. Pplacer is designed to be fast, to give useful information about uncertainty, and to offer advanced visualization and downstream analysis. </p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/PRIMUS.html">PRIMUS</a> (1.9.0) </div><p class="appDesc">PRIMUS, Pedigree Reconstruction and Identification of a Maximum Unrelated Set, is used to read genome-wide IBD estimates and identify a maximum unrelated set and pedigree reconstruction.</p>
<div class="subheading"><a href="prsice.html">PRSice</a> (2.3.3) </div><p class="appDesc">PRSice is a Polygenic Risk Score software for calculating, applying, evaluating and plotting the results of polygenic risk scores (PRS) analyses. </p>
<div class="subheading"><a href="QIIME.html">QIIME</a> (2023.5) </div><p class="appDesc">QIIME is an open source software package for comparison and analysis of microbial communities, primarily based on high-throughput amplicon sequencing data (such as SSU rRNA) generated on a variety of platforms, but also supporting analysis of other types of data (such as shotgun metagenomic data). </p>
<div class="subheading"><a href="raxml.html">RAxML</a> (8.2.12) </div><p class="appDesc">RAxML (randomized axelerated maximum likelihood) is a sequential and parallel program for inference of large phylogenies with maximum likelihood (ML). </p>
<div class="subheading"><a href="raxml-ng.html">raxml-ng</a> (1.2.0) </div><p class="appDesc">RAxML-NG is a phylogenetic tree inference tool which uses maximum-likelihood (ML) optimality criterion. Its search heuristic is based on iteratively performing a series of Subtree Pruning and Regrafting (SPR) moves, which allows to quickly navigate to the best-known ML tree. Successor to raxml.</p>
<div class="subheading"><a href="revbayes.html">RevBayes</a> (1.2.1) </div><p class="appDesc">Bayesian phylogenetic inference using probabilistic graphical models and an interpreted language</p>
<div class="subheading"><a href="scDRS.html">scDRS</a> (1.02) </div><p class="appDesc">The scDRS application implements an approach that links scRNA-seq
with polygenic disease risk at single-cell resolution, independent of annotated cell types.
scDRS identifies cells exhibiting excess expression across disease-associated genes
implicated by genome-wide association studies (GWASs).
Genes whose expression was correlated with the scDRS score across cells
(reflecting coexpression with GWAS disease-associated genes)
were strongly enriched for gold-standard drug target and Mendelian disease genes.</p>
<div class="subheading"><a href="shapeit.html">shapeit</a> (5.1.0) </div><p class="appDesc">SHAPEIT is a fast and accurate haplotype inference software</p>
<div class="subheading"><a href="SMR.html">SMR</a> (0.702; 1.03; 1.3.1) </div><p class="appDesc">SMR integrates summary-level data from GWAS with data from expression quantitative trait locus (eQTL) studies to identify genes whose expression levels are associated with a complex trait because of pleiotropy. It implements  methods to test for pleiotropic association between the expression level of a gene and a complex trait of interest using summary-level data from GWAS and expression quantitative trait loci (eQTL) studies (Zhu et al. 2016 Nat Genet).</p>
<div class="subheading"><a href="solar.html">solar</a> (9.0.1) </div><p class="appDesc">SOLAR-Eclipse is an extensive, flexible software package for genetic variance components analysis, including linkage analysis, quantitative genetic analysis, SNP association analysis (QTN and QTLD), and covariate screening.

</p>
<div class="subheading"><a href="sumtrees.html">sumtrees</a> (4.5.2) </div><p class="appDesc">Summarize non-parameteric bootstrap or Bayesian posterior probability support for splits or clades on phylogenetic trees.</p>
<div class="subheading"><a href="TensorQTL.html">TensorQTL</a> (1.0.7) </div><p class="appDesc">ensoorQTL leverages general-purpose libraries and graphics processing units
(GPUs) to achieve high efficiency of computations at low costR. Using PyTorch or TensorFlow
it allows > 200-fold decreases in runtime and ~ 5–10-fold reductions in cost when
running on GPUs relative to CPUs.</p>
<div class="subheading"><a href="treemix.html">treemix</a> (1.13) </div><p class="appDesc">TreeMix is a method for inferring the patterns of population splits and mixtures in the history of a set of populations.
</p>
<div class="subheading"><a href="treetime.html">treetime</a> (0.11.1) </div><p class="appDesc">TreeTime provides routines for ancestral sequence reconstruction and inference of molecular-clock phylogenies.</p>
<div class="heading"><a name="massspec" id="massspec">Mass Spectrometry</a></div>
<a href="/apps/" style="font-size:12px">back to top</a><br />
<p></p>
<div class="subheading">diann (1.8.1) </div><p class="appDesc">DIA-NN - a universal software for data-independent acquisition (DIA) proteomics data processing</p>
<div class="subheading"><a href="fragpipe.html">fragpipe</a> (20.0) </div><p class="appDesc">FragPipe is a Java Graphical User Interface (GUI) and CLI workflow tool for a suite of computational tools enabling comprehensive analysis of mass spectrometry-based proteomics data. It is powered by MSFragger.</p>
<div class="subheading"><a href="ionquant.html">ionquant</a> (1.9.8) </div><p class="appDesc">IonQuant is a fast and comprehensive tool for MS1 precursor intensity-based quantification for timsTOF PASEF DDA and non-timsTOF (e.g., Orbitrap) data. It enables label-free quantification with false discovery (FDR) controlled match-between-runs (MBR). It can also be used for quantification in labelling-based experiments such as those involving SILAC, dimethyl, or similar labelling strategies. IonQuant is available as part of FragPipe (recommended option), but can also be run as a command-line tool.</p>
<div class="subheading"><a href="https://biospec.nih.gov/">Mascot</a> (2.8) </div><p class="appDesc">The <a href="http://www.matrixscience.com/">Mascot search engine</a> uses mass spectrometry data to identify proteins from primary sequence databases. Mascot searches can be run directly on the NIH Mascot server at <a href="https://biospec.nih.gov">https://biospec.nih.gov</a>, or by using the Mascot daemon on your own desktop PC.</p>
<div class="subheading"><a href="maxquant.html">maxquant</a> (2.4.9.0) </div><p class="appDesc">MaxQuant is a quantitative proteomics software package designed for analyzing large mass-spectrometric data sets. It is specifically aimed at high-resolution MS data. Several labeling techniques as well as label-free quantification are supported.</p>
<div class="subheading"><a href="msfragger.html">msfragger</a> (3.8) </div><p class="appDesc">An ultrafast database search tool for peptide identification in mass spectrometry-based proteomics.</p>
<div class="subheading"><a href="philosopher.html">philosopher</a> (4.2.2;4.8.1) </div><p class="appDesc">Philosopher is fast, easy-to-use, scalable, and versatile data analysis software for mass spectrometry-based proteomics. Philosopher is dependency-free and can analyze both traditional database searches and open searches for post-translational modification (PTM) discovery.</p>
<div class="heading"><a name="math" id="math">Mathematical/Statistics</a></div>
<a href="/apps/" style="font-size:12px">back to top</a><br />
<p></p>
<div class="subheading"><a href="ANTs.html">ANTs</a> (2.4.2) </div><p class="appDesc">Advanced Normalization Tools (ANTs) extracts information from complex datasets that include imaging. Paired with ANTsR (answer), ANTs is useful for managing, interpreting and visualizing multidimensional data. </p>
<div class="subheading"><a href="cmdstan.html">cmdstan</a> (2.30.1) </div><p class="appDesc">Command line interface to stan</p>
<div class="subheading"><a href="Comsol.html">Comsol</a> (6.1.0.346) </div><p class="appDesc">The COMSOL Multiphysics engineering simulation software environment facilitates all steps in the modeling process − defining your geometry, meshing, specifying your physics, solving, and then visualizing your results.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/dentist.html">dentist</a> (1.1.0.0) </div><p class="appDesc">DENTIST (Detecting Errors iN analyses of summary staTISTics) is a quality control (QC) tool for summary-level data from genome-wide association studies (GWASs). It leverages the difference between the observed GWAS test-statistic of a variant and its predicted value (using the neighbouring variants and linkage equilibrium (LD) data from a reference panel) to remove problematic variants.</p>
<div class="subheading"><a href="FEBio.html">FEBio</a> (v4.0) </div><p class="appDesc">FEBio software suite implement a nonlinear implicit finite element (FE) framework,
designed specifically for analysis in computational solid biomechanics.
FEBio offers modeling scenarios, constitutive models, and boundary conditions,
which are relevant to numerous applications in biomechanics.
The open-source FEBio software is written in C++,
with particular attention to scalar and parallel performance on modern computer architectures.</p>
<div class="subheading"><a href="GAUSS.html">GAUSS</a> (10) </div><p class="appDesc">The GAUSS Mathematical and Statistical System is an easy-to-use data analysis environment based on the fast and powerful GAUSS Matrix Programming Language designed for computationally intensive tasks. </p>
<div class="subheading"><a href="gurobi.html">gurobi</a> (9.0.0) </div><p class="appDesc">Gurobi is a mathematical optimization solver. It is a commercial product developed by gurobi.com. On Biowulf, Gurobi is licensed for use by the members of the CDSL_Gurobi_users group only. It is installed in /data/CDSL_Gurobi_users and is not accessible by any other users. A token license server, running on Biowulf, manages the Gurobi license.

</p>
<div class="subheading"><a href="idl.html">IDL/ENVI</a> (8.8/5.6) </div><p class="appDesc">IDL and ENVI are a complete computing environment for the interactive analysis and visualization of data. IDL integrates an array-oriented language with mathematical analysis and graphical display techniques. ENVI is designed for extracting information from geospatial and medical imagery.</p>
<div class="subheading"><a href="lifelines.html">lifelines</a> (0.27.4) </div><p class="appDesc">Lifelines is a complete survival analysis library, written in pure Python.
It has the benefits of easy installation, internal plotting methods, simple and intuitive API
handles right, left and interval censored data. It also contains the most popular parametric,
semi-parametric and non-parametric models.</p>
<div class="subheading"><a href="mathematica.html">Mathematica</a> (13.3.0) </div><p class="appDesc">Mathematica is an interactive system for doing mathematical computation. It performs numerical, symbolic and graphical computations, and incorporates a high-level programming language.</p>
<div class="subheading"><a href="Matlab.html">Matlab</a> (2023a) </div><p class="appDesc">MATLAB is an interactive software package for scientific and engineering numeric computation. MATLAB integrates numerical analysis, matrix computation, signal processing, and graphics in an environment where problems and solutions are expressed just as they are written mathematically.</p>
<div class="subheading"><a href="mcl.html">MCL</a> (14-137) </div><p class="appDesc">MCL implements Markov cluster algorithm. Among its applications
is the assignment of proteins into families based on precomputed sequence
similarity information. This approach
does not suffer from the problems that normally hinder other protein sequence
clustering algorithms, such as the presence of multi-domain proteins,
promiscuous domains and fragmented proteins.</p>
<div class="subheading"><a href="Meep.html">Meep</a> (1.17.1) </div><p class="appDesc">Meep (or MEEP) is a free finite-difference time-domain (FDTD) simulation software package developed at MIT to model electromagnetic systems, along with the MPB eigenmode package. </p>
<div class="subheading"><a href="octave.html">Octave</a> (5.1.0) </div><p class="appDesc">GNU Octave is a high-level language, primarily intended for numerical computations. It provides a convenient command line interface for solving linear and nonlinear problems numerically, and for performing other numerical experiments using a language that is mostly compatible with Matlab.</p>
<div class="subheading"><a href="PEER.html">PEER</a> (1.3) </div><p class="appDesc">PEER stands for probabilistic estimation of expression residuals. It
is a collection of Bayesian approaches to infer hidden determinants and
their effects from gene expression profiles using factor analysis methods.</p>
<div class="subheading"><a href="R.html">R</a> (4.3.0) </div><p class="appDesc">R (the R Project) is a language and environment for statistical computing and graphics. R is similar to S, and provides a wide variety of statistical and graphical techniques (linear and nonlinear modelling, statistical tests, time series analysis, classification, clustering, ...).</p>
<div class="subheading"><a href="RStudio.html">Rstudio</a> (2023.03.1-446) </div><p class="appDesc">RStudio is a set of integrated tools designed to help you be more productive with R. It includes a console, syntax-highlighting editor that supports direct code execution, as well as tools for plotting, history, debugging and workspace management.</p>
<div class="subheading"><a href="SAS.html">SAS</a> (9.4M7) </div><p class="appDesc">Base SAS provides a scalable, integrated software environment specially designed for data access, transformation and reporting.</p>
<div class="heading"><a name="model" id="model">Molecular Modeling/Graphics</a></div>
<a href="/apps/" style="font-size:12px">back to top</a><br />
<p></p>
<div class="subheading"><a href="AutodockCrankprep.html">AutodockCrankprep</a> (1.0) </div><p class="appDesc">"AutoDock CrankPep or ADCP is an AutoDock docking engine specialized for docking peptides. 
It combines technology form the protein folding filed with an efficient representation of
a rigid receptor as affinity grids to fold the peptide in the context of the energy landscape
created by the receptor."</p>
<div class="subheading"><a href="blender.html">blender</a> (2.82) </div><p class="appDesc">Blender is the free and open source 3D creation suite.  Blender on Biowulf is meant for command-line rendering.</p>
<div class="subheading"><a href="chap.html">chap</a> (0.9.1) </div><p class="appDesc"> CHAP is a tool for the functional annotation of ion channel structures</p>
<div class="subheading"><a href="Chimera.html">Chimera</a> (1.16.0) </div><p class="appDesc">Chimera is a  highly extensible program for interactive visualization and analysis of molecular structures and related data, including density maps, supramolecular assemblies, sequence alignments, docking results, trajectories, and conformational ensembles.</p>
<div class="subheading"><a href="chromeister.html">chromeister</a> (1.5.a) </div><p class="appDesc">Chromeister is an ultra fast, heuristic approach to detect conserved signals in extremely large pairwise genome comparisons.</p>
<div class="subheading"><a href="Coot.html">Coot</a> (0.9.8.92) </div><p class="appDesc">Coot is for macromolecular model building, model completion and validation, particularly suitable for protein modelling using X-ray data.</p>
<div class="subheading"><a href="Cytoscape.html">Cytoscape</a> (3.9.1) </div><p class="appDesc">Cytoscape is an open source software platform for visualizing molecular interaction networks and biological pathways and integrating these networks with annotations, gene expression profiles and other state data.</p>
<div class="subheading">iqmol (2.14.0) </div><p class="appDesc">free open-source molecular editor and visualization package. It offers a range of features including a molecular editor, surface generation (orbitals and densities) and animations (vibrational modes and reaction pathways).</p>
<div class="subheading"><a href="lammps.html">lammps</a> (29Oct20) </div><p class="appDesc">LAMMPS is a classical molecular dynamics code, and an acronym for Large-scale Atomic/Molecular Massively Parallel Simulator. It runs on a variety of different computer systems, including single processor systems, distributed-memory machines with MPI, and GPU and Xeon Phi systems. LAMMPS is open source software, released under the GNU General Public License.</p>
<div class="subheading"><a href="mosaics.html">mosaics</a> (1.0.0) </div><p class="appDesc">A collection of tools for characterizing membrane structure and dynamics within simulated trajectories of molecular systems. </p>
<div class="subheading"><a href="OpenBabel.html">OpenBabel</a> (3.1.1) </div><p class="appDesc">Open Babel is a chemical toolbox designed to speak the many languages of chemical data.</p>
<div class="subheading"><a href="pdb2pqr.html">pdb2pqr</a> (3.6.1) </div><p class="appDesc">Automates many of the common tasks of preparing structures for continuum solvation calculations as well as many other types of biomolecular structure modeling</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/PyMOL.html">posefilter</a> (619eca51) </div><p class="appDesc">PoseFilter is a PyMOL plugin and assists in the analysis of docked ligands through identification of unique oligomeric poses by utilizing RMSD and interaction fingerprint analysis methods.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/psi4.html">Psi4</a> (1.6.1) </div><p class="appDesc">Psi4 is an ab-initio electronic structure code that supports various methods for calculating energies and gradients of molecular systems.</p>
<div class="subheading"><a href="rosetta.html">Rosetta</a> (2023.26) </div><p class="appDesc">The Rosetta++ software suite can perform de novo protein structure predictions, identify low free energy sequences for target protein backbones, predict the structure of a protein-protein complex from the individual structures of the monomer components, incorporate NMR data into the basic Rosetta protocol to accelerate the process of NMR structure prediction, and more...</p>
<div class="subheading"><a href="schrodinger/index.html">Schrodinger</a> (2023.1) </div><p class="appDesc">A limited number of Schrödinger applications are available on the Biowulf cluster through the Molecular Modeling Interest Group. Most are available through the Maestro GUI.</p>
<div class="subheading"><a href="/apps/vmd.html">vmd</a> (1.9.3) </div><p class="appDesc">VMD is a molecular visualization program for displaying, animating, and analyzing large biomolecular systems using 3-D graphics and built-in scripting. To use, type vmd at the prompt.</p>
<div class="heading"><a name="seqanal" id="seqanal">Sequence Analysis</a></div>
<a href="/apps/" style="font-size:12px">back to top</a><br />
<p></p>
<div class="subheading"><a href="advntr.html">advntr</a> (1.4.1) </div><p class="appDesc">a tool for genotyping Variable Number Tandem Repeats (VNTR) from sequence data</p>
<div class="subheading">agat (1.2.0) </div><p class="appDesc">Another Gtf/Gff Analysis Toolkit</p>
<div class="subheading"><a href="alleleCount.html">alleleCount</a> (4.2.1) </div><p class="appDesc">Calculates genotype frequencies of a SNPMatrix. This component tests each SNP for its Hardy-Weinberg equilibrium. If there are NA values, the frequencies of missing value per sample in the input file are calculated. </p>
<div class="subheading"><a href="ANNOgesic.html">ANNOgesic</a> (1.0.22) </div><p class="appDesc">Processing and integrating RNA-Seq data in order to
generate high-resolution annotations is challenging, time consuming and requires
numerous different steps. ANNOgesic is a powerful and modular
pipeline that provides the required analyses and simplifies
RNA-Seq-based bacterial and archaeal genome annotation. It predicts and
annotates numerous features, including small non-coding RNAs, with high
precision.</p>
<div class="subheading"><a href="antismash.html">antiSMASH</a> (7.0.0) </div><p class="appDesc">antiSMASH allows the rapid genome-wide identification, annotation and analysis of secondary metabolite biosynthesis gene clusters in bacterial and fungal genomes.</p>
<div class="subheading"><a href="anvio.html">anvio</a> (7.1) </div><p class="appDesc">Anvi’o is an open-source, community-driven analysis and visualization platform for microbial ‘omics. It brings together many aspects of today’s cutting-edge strategies including genomics, metagenomics, metatranscriptomics, pangenomics, metapangenomics, phylogenomics, and microbial population genetics in an integrated and easy-to-use fashion through extensive interactive visualization capabilities.</p>
<div class="subheading"><a href="arcashla.html">arcashla</a> (0.5.0) </div><p class="appDesc">high resolution HLA typing from RNA seq</p>
<div class="subheading"><a href="arriba.html">arriba</a> (2.3.0) </div><p class="appDesc">Arriba identifies gene fusions in RNA-Seq data.  It also can detect other structural variants in genomic data, such as intron duplications and gene truncations.</p>
<div class="subheading"><a href="augustus.html">augustus</a> (3.4.0) </div><p class="appDesc">AUGUSTUS is a program that predicts genes in eukaryotic genomic sequences. </p>
<div class="subheading">bakta (1.8.1) </div><p class="appDesc">Rapid & standardized annotation of bacterial genomes, MAGs & plasmids</p>
<div class="subheading"><a href="bartender.html">Bartender</a> (1.1) </div><p class="appDesc">Bartender is an accurate clustering algorithm to detect barcodes
and their abundances from raw next-generation sequencing data.
In contrast with existing methods that cluster based on sequence similarity alone,
Bartender uses a modified two-sample proportion test that also considers cluster size. This modification
results in higher accuracy and lower rates of under- and over-clustering artifacts.</p>
<div class="subheading"><a href="biobakery_workflows.html">biobakery_workflows</a> (3.1) </div><p class="appDesc">bioBakery is a meta’omic analysis environment and collection of individual software
tools with the capacity to process raw shotgun sequencing data into actionable microbial community
feature profiles, summary reports, and publication-ready figures.
It includes a collection of preconfigured analysis modules also joined into workflows for reproducibility.
Each individual module has been developed to perform a particular task, e.g. quantitative taxonomic
profiling or statistical analysis.</p>
<div class="subheading"><a href="Blast.html">Blast</a> (2.13.0+) </div><p class="appDesc">NCBI's well-known sequence database searching program which compares a nucleotide or protein query sequence against all sequences in a database.</p>
<div class="subheading"><a href="blat.html">blat</a> (3.5) </div><p class="appDesc">BLAT is a DNA/Protein Sequence Analysis program that is designed to quickly find sequences of 95% and greater similarity of length 40 bases or more. </p>
<div class="subheading"><a href="BOLT-LMM.html">BOLT-LMM</a> (2.4.1) </div><p class="appDesc">The  BOLT-LMM  algorithm  computes  statistics  for  testing  association  between  phenotype  and
genotypes using a linear mixed model (LMM)</p>
<div class="subheading"><a href="bracken.html">bracken</a> (2.8) </div><p class="appDesc">Bracken is a companion program to Kraken 1, KrakenUniq, or Kraken 2 While Kraken classifies reads to multiple levels in the taxonomic tree, Bracken allows estimation of abundance at a single level using those classifications (e.g. Bracken can estimate abundance of species within a sample).</p>
<div class="subheading"><a href="bwa-mem2.html">bwa-mem2</a> (2.2.1) </div><p class="appDesc">The next version of the bwa-mem algorithm in bwa. </p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/cactus.html">cactus</a> (2.6.4) </div><p class="appDesc">Cactus is a reference-free whole-genome multiple alignment program.</p>
<div class="subheading"><a href="CADD.html">CADD</a> (1.6.post1) </div><p class="appDesc">CADD (Combined Annotation Dependent Depletion) is a tool for scoring the deleteriousness of single nucleotide variants as well as insertion/deletions variants in the human genome. Currently, it supports the builds: GRCh37/hg19 and GRCh38/hg38.</p>
<div class="subheading"><a href="caviar.html">CAVIAR</a> (2.2) </div><p class="appDesc">CAVIAR (CAusal Variants Identication in Associated Regions) is a statistical framework that quantifies the probability of each variant to be causal while allowing with arbitrary number of causal variants</p>
<div class="subheading"><a href="cell2location.html">cell2location</a> (0.1.3) </div><p class="appDesc">Cell2location: Comprehensive mapping of tissue cell architecture via integrated single cell and spatial transcriptomics (cell2location model)</p>
<div class="subheading"><a href="cellsnp-lite.html">cellsnp-lite</a> (1.2.2) </div><p class="appDesc">Efficient genotyping bi-allelic SNPs on single cells</p>
<div class="subheading">cgpbigwig (1.6.0) </div><p class="appDesc">BigWig manpulation tools using libBigWig and htslib</p>
<div class="subheading"><a href="checkm2.html">checkm2</a> (1.0.2) </div><p class="appDesc">Rapid assessment of genome bin quality using machine learning.</p>
<div class="subheading"><a href="chess.html">CHESS</a> (0.3.7) </div><p class="appDesc">The CHESS (Comparison of Hi-C Experiments using Structural Similarity) application
implements an algorithm for the comparison of chromatin contact maps and automatic
differential feature extraction.</p>
<div class="subheading"><a href="chipseq_pipeline.html">chipseq_pipeline</a> (2.1.6) </div><p class="appDesc">AQUAS Transcription Factor and Histone ChIP-Seq processing pipeline. The AQUAS pipeline is based off the ENCODE (phase-3) transcription factor and histone ChIP-seq pipeline specifications (by Anshul Kundaje) </p>
<div class="subheading"><a href="circleseq.html">CIRCLE-seq</a> (1.1) </div><p class="appDesc">CIRCLE-seq is a tool ifor analysis of circularization for in vitro reporting of cleavage effects by sequencing.
It is a highly sensitive, sequencing-efficient in vitro screening strategy
that outperforms existing cell-based or biochemical approaches
for identifying CRIScrisPR–Cas9 genome-wide off-target mutations.
It can be practiced using widely accessible next-generation sequencing technology
and does not require reference genome sequences.
Importantly, CIRCLEcircle-seq can be used to identify off-target mutations
associated with cell-type-specific single-nucleotide polymorphisms,
demonstrating the feasibility
and importance of generating personalized specificity profiles.</p>
<div class="subheading"><a href="circos.html">circos</a> (0.69-9) </div><p class="appDesc">Circos is a program for the generation of publication-quality, circularly composited renditions of genomic data and related annotations. Circos is particularly suited for visualizing alignments, conservation and intra and inter-chromosomal relationships. Also, Circos is useful to visualize any type of information that benefits from a circular layout. Thus, although it has been designed for the field of genomics, it is sufficiently flexible to be used in other data domains.</p>
<div class="subheading"><a href="Clair3.html">Clair3</a> (0.1-r6) </div><p class="appDesc">Clair3 is a small variant caller for Illumina, PacBio and ONT long reads. Compare to PEPPER (r0.4), Clair3 (v0.1) shows a better SNP F1-score with ≤30-fold of ONT data (precisionFDA Truth Challenge V2), and a better Indel F1-score, while runs generally four times faster.   </p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/cloops2.html">cloops2</a> (0.0.4) </div><p class="appDesc">cLoops2 is an enhanced and elegant flexible peak/loop/domain -calling and analysis tool for 1D/3D genomic data. </p>
<div class="subheading"><a href="ClustalW.html">clustalo</a> (1.2.4) </div><p class="appDesc">Clustal-Omega is a general purpose multiple sequence alignment (MSA)
program for proteins and DNA/RNA. It produces high quality MSAs and is
capable of handling data-sets of hundreds of thousands of sequences in
reasonable time.</p>
<div class="subheading"><a href="ClustalW.html">ClustalW</a> (2.1) </div><p class="appDesc">ClustalW is a general-purpose multiple alignment program for DNA or protein sequences.</p>
<div class="subheading"><a href="conifer.html">conifer</a> (0.2.2) </div><p class="appDesc">CoNIFER (copy number inference from exome reads) uses exome sequencing data to find copy number variants (CNVs) and genotype the copy-number of duplicated genes. It can reliably be used to discover disruptive genic CNVs missed by standard approaches and should have broad
application in human genetic studies of disease.</p>
<div class="subheading"><a href="cooltools.html">cooltools</a> (0.5.4) </div><p class="appDesc">Cooltools is a suite of computational tools that enables flexible, scalable, and
reproducible analysis of high-resolution contact frequency data. Cooltools leverages the widely-adopted
cooler format which handles storage and access for high-resolution datasets. Cooltools provides a paired
command line interface and Python application programming interface, which respectively
facilitate workflows on high-performance computing clusters and in interactive analysis environments.</p>
<div class="subheading"><a href="coverageMaster.html">coverageMaster</a> (20220706) </div><p class="appDesc">CoverageMaster (CoM) is a copy number variation (CNV) calling algorithm i
ased on depth-of-coverage maps designed to detect CNVs of any size i
n exome [whole exome sequencing (WES)] and genome [whole genome sequencing (WGS)] data.
The core of the algorithm is the compression of sequencing coverage data
in a multiscale Wavelet space and the analysis through an iterative Hidden Markov Model.</p>
<div class="subheading"><a href="CutRunTools.html">CutRunTools</a> (20200629 ) </div><p class="appDesc">CutRunTools is a flexible, general pipeline for facilitating the identification of
chromatin-associated protein binding and genomic footprinting analysis from antibody-targeted
CutRun primary cleavage data. CutRunTools extracts endonuclease cut site information from
sequences of short read fragments and produces single-locus binding estimates, aggregate motif
footprints, and informative visualizations to support the high-resolution mapping capability of
CutRun.</p>
<div class="subheading"><a href="CytoSig.html">CytoSig</a> (0.1) </div><p class="appDesc">CytoSig is a data-driven infrastructure hosted by the National Cancer Institute.
CytoSig includes both a database of target genes modulated by cytokines and
a predictive model of cytokine signaling activity and regulatory cascade from transcriptomic profiles.</p>
<div class="subheading"><a href="cytotrace.html">cytotrace</a> (0.3.3) </div><p class="appDesc">CytoTRACE (Cellular (Cyto) Trajectory Reconstruction Analysis using gene Counts and Expression) is a computational method
that predicts the differentiation state of cells from single-cell RNA-sequencing data.
CytoTRACE leverages a simple, yet robust, determinant of developmental potential—the number of detectably expressed genes per cell, or gene counts. We have validated CytoTRACE on ~150K single-cell transcriptomes spanning 315 cell phenotypes, 52 lineages, 14 tissue types, 9 scRNA-seq platforms, and 5 species.</p>
<div class="subheading"><a href="decifer.html">decifer</a> (2.1.3) </div><p class="appDesc">DeCiFer is an algorithm that simultaneously selects mutation multiplicities and clusters somatic single-nucleotide variants (SNVs) by their corresponding descendant cell fractions (DCF), a statistic that quantifies the proportion of cells which acquired the SNV or whose ancestors acquired the SNV. DCF is related to the commonly used cancer cell fraction (CCF) but further accounts for SNVs which are lost due to deleterious somatic copy-number aberrations (CNAs), identifying clusters of SNVs which occur in the same phylogenetic branch of tumour evolution.</p>
<div class="subheading"><a href="deeploc.html">deeploc</a> (2.0) </div><p class="appDesc">DeepLoc 2.0 predicts the subcellular localization(s) of eukaryotic proteins. It is is able to predict one or more localizations for any given protein.</p>
<div class="subheading"><a href="deepsea.html">deepsea</a> (0.94c) </div><p class="appDesc">DeepSEA is a deep learning-based algorithmic framework for predicting the chromatin effects of sequence alterations with single nucleotide sensitivity. DeepSEA can accurately predict the epigenetic state of a sequence, including transcription factors binding, DNase I sensitivities and histone marks in multiple cell types, and further utilize this capability to predict the chromatin effects of sequence variants and prioritize regulatory variants.</p>
<div class="subheading"><a href="denovogear.html">denovogear</a> (1.1.1) </div><p class="appDesc">DeNovoGear is az software for analyzing de novo mutations from familial and somatic tissue sequencing data.
It uses likelihood-based error modeling to reduce the false positive rate of mutation discovery in exome analysis
and fragment information to identify the parental origin of germ-line mutations. DeNovoGear has been used on human
whole-genome sequencing data to produce a set of predicted de novo insertion and/or deletion (indel) mutations.</p>
<div class="subheading"><a href="diamond.html">diamond</a> (2.0.15) </div><p class="appDesc">DIAMOND is a new high-throughput program for aligning DNA reads or protein sequences against a protein reference database such as NR, at up to 20,000 times the speed of BLAST, with high sensitivity. </p>
<div class="subheading"><a href="https://hpcwebapps.cit.nih.gov/dnaworks/">DNAWorks</a> (3.2.4) </div><p class="appDesc">DNAWorks is a computer program that automates the design of oligonucleotides for gene synthesis by PCR-based gene assembly. The program requires simple input information: an amino acid sequence of the target protein or a DNA sequence, and a desired annealing temperature. It is a web-based tool available at <a href="https://hpcwebapps.cit.nih.gov/dnaworks/">https://hpcwebapps.cit.nih.gov/dnaworks/</a>.</p>
<div class="subheading"><a href="drep.html">drep</a> (3.2.2) </div><p class="appDesc">dRep is a python program for rapidly comparing large numbers of genomes. dRep can also "de-replicate" a genome set by identifying groups of highly similar genomes and choosing the best representative genome for each genome set.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/duphold.html">duphold</a> (0.2.3) </div><p class="appDesc">Duphold: scalable, depth-based annotation and curation of high-confidence structural variant calls</p>
<div class="subheading"><a href="dyno.html">dyno</a> (20220906) </div><p class="appDesc">dyno is a meta package that installs several other packages
from the dynvers (https://github.com/dynverse). It comprises a set of R packages
to construct and interpret single-cell trajectories.</p>
<div class="subheading"><a href="Eagle.html">Eagle</a> (2.4) </div><p class="appDesc">Eagle performs a reference-based haplotype phasing. It attains high accuracy across a broad range of cohort sizes by efficiently leveraging information from large external reference panels (such as the Haplotype Reference  onsortium; HRC)
using a new data structure based on the positional Burrows-Wheeler transform.</p>
<div class="subheading"><a href="epic2.html">epic2</a> (0.0.52) </div><p class="appDesc">epic2 is an ultraperformant reimplementation of SICER. It focuses on speed, low memory overhead and ease of use.</p>
<div class="subheading"><a href="exonerate.html">exonerate</a> (2.4.0) </div><p class="appDesc">Exonerate is a generic tool for pairwise sequence comparison. It allows you to align sequences using a many alignment models, either exhaustive dynamic programming or a variety of heuristics.</p>
<div class="subheading"><a href="https://github.com/Illumina/ExpansionHunter/blob/master/docs/01_Introduction.md">ExpansionHunter</a> (5.0.0) </div><p class="appDesc">Expansion Hunter: a tool for estimating repeat sizes. There are a number of regions in the human genome consisting of repetitions of short unit sequence (commonly a trimer). Such repeat regions can expand to a size much larger than the read length and thereby cause a disease. Expansion Hunter aims to estimate sizes of such repeats by performing a targeted search through a BAM/CRAM file for reads that span, flank, and are fully contained in each repeat.</p>
<div class="subheading"><a href="https://github.com/Illumina/ExpansionHunterDenovo/blob/master/documentation/00_Introduction.md">ExpansionHunterDenovo</a> (0.9.0) </div><p class="appDesc">ExpansionHunter Denovo (EHdn) is a suite of tools for detecting novel expansions of short tandem repeats (STRs). EHdn is intended for analysis of a collection of BAM/CRAM files containing alignments of short (100-200bp) reads.</p>
<div class="subheading"><a href="fastp.html">fastp</a> (0.23.2) </div><p class="appDesc">A tool designed to provide fast all-in-one preprocessing for FastQ files. This tool is developed in C++ with multithreading supported to afford high performance.</p>
<div class="subheading"><a href="fastq_demux.html">fastq_demux</a> (20230713) </div><p class="appDesc">fastq_demux is a simple program to demultiplex a FASTQ file or a pair of FASTQ files based on the barcodes present in the FASTQ headers.</p>
<div class="subheading"><a href="FitHiChIP.html">FitHiChIP</a> (9.1; 10.0; 11.0) </div><p class="appDesc">FitHiChIP is a computational method for identifying chromatin contacts
among regulatory regions such as enhancers and promoters from HiChIP/PLAC-seq data.
FitHiChIP jointly models the non-uniform coverage and genomic distance scaling of HiChIP data,
captures previously validated enhancer interactions for several genes
including MYC and TP53, and recovers contacts genome-wide
that are supported by ChIA-PET, promoter capture Hi-C and Hi-C data.</p>
<div class="subheading"><a href="flair.html">FLAIR</a> (1.6.1) </div><p class="appDesc">FLAIR (Full-Length Alternative Isoform analysis of RNA) is a workflow
leveraging the full-length transcript sequencing data that nanopore affords.
It uses multiple alignment steps and splice site filters
to increase confidence in the set of isoforms defined from noisy data.</p>
<div class="subheading">focus (0.802) </div><p class="appDesc">FOCUS (Fine-mapping Of CaUsal gene Sets) is software to fine-map transcriptome-wide association study statistics at genomic risk regions. The software takes as input summary GWAS data along with eQTL weights and outputs a credible set of genes to explain observed genomic risk.</p>
<div class="subheading"><a href="freepsi.html">freepsi</a> (0.3) </div><p class="appDesc">FreePSI is a new method for genome-wide percent spliced in (PSI) estimation that requires neither a reference transcriptome (hence, transcriptome-free) nor the mapping of RNA-seq reads (hence, alignment-free). The first attribute allows FreePSI to work effectively when a high quality reference transcriptome is unavailable and the second not only helps make FreePSI more efficient, it also eliminates the necessity of dealing with multi-reads.</p>
<div class="subheading"><a href="funannotate.html">funannotate</a> (1.8.15) </div><p class="appDesc">Funannotate is a genome prediction, annotation, and comparison software package.
It was originally written to annotate fungal genomes (small eukaryotes ~ 30 Mb genomes),
but has evolved over time to accomodate larger genomes.</p>
<div class="subheading"><a href="FuSeq.html">FuSeq</a> (1.1.4) </div><p class="appDesc">FuSeq is a software for discovering fusion genes from paired-end RNA sequencing data. It implements a fast and accurate method to discover fusion genes based on quasi-mapping to quickly map the reads, extract initial candidates from split reads and fusion equivalence classes of mapped reads, and finally apply multiple filters and statistical tests to get the final candidates.</p>
<div class="subheading"><a href="fusioninspector.html">fusioninspector</a> (2.8.0) </div><p class="appDesc">In silico Validation of Fusion Transcript Predictions</p>
<div class="subheading"><a href="GEM.html">GEM1</a> (1.4.3) </div><p class="appDesc">GEM (Gene-Environment interaction analysis for Millions of samples)
is a software program for large-scale gene-environment interaction testing
in samples from unrelated individuals.
It enables genome-wide association studies in up to millions of samples
while allowing for multiple exposures, control for genotype-covariate interactions, and robust inference.</p>
<div class="subheading"><a href="genometools.html">genometools</a> (1.6.2) </div><p class="appDesc">collection of bioinformatic tools</p>
<div class="subheading"><a href="gfatools.html">gfatools</a> (0.5) </div><p class="appDesc">gfatools is a set of tools for manipulating sequence graphs in the GFA or the rGFA format. It has implemented parsing, subgraph and conversion to FASTA/BED.</p>
<div class="subheading"><a href="gffcompare.html">gffcompare</a> (0.12.6) </div><p class="appDesc">gffcompare can be used to compare and evaluate the accuracy of RNA-Seq transcript assemblers (Cufflinks, Stringtie). It can collapse (merge) duplicate transcripts from multiple GTF/GFF3 files (e.g. resulted from assembly of different samples) and classify transcripts from one or multiple GTF/GFF3 files as they relate to reference transcripts provided in a annotation file (also in GTF/GFF3 format).</p>
<div class="subheading"><a href="GimmeMotifs.html">GimmeMotifs</a> (0.17.2) </div><p class="appDesc">GimmeMotifs is a pipeline for transcription factor motif analysis written in Python.
It incorporates an ensemble of computational tools to predict
motifs de novo from ChIP-sequencing data. Similar
redundant motifs are compared using the weighted information
content similarity score and clustered using an iterative
procedure. A comprehensive output report is generated with several
different evaluation metrics to compare and evaluate the results.
</p>
<div class="subheading"><a href="GLIMPSE.html">GLIMPSE</a> (1.1.1) </div><p class="appDesc">GLIMPSE is a phasing and imputation method for large-scale low-coverage sequencing studies.
It perform accurate imputed genotype calls and outperforms SNP arrays.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/glnexus.html">glnexus</a> (1.4.1) </div><p class="appDesc">Joint variant calling for large cohort sequencing</p>
<div class="subheading"><a href="graphmap.html">graphmap</a> (0.5.2) </div><p class="appDesc">GraphMap is a highly sensitive and accurate mapper for long, error-prone reads.
It offers a number of valuable features, such as mapping position agnostic to alignment parameters,
high sensitivity and precision, handling circular genomes, meaningful mapping quality,
various alignment strategies, and more.</p>
<div class="subheading"><a href="gtdb-tk.html">gtdb-tk</a> (2.3.2) </div><p class="appDesc">GTDB-Tk is a software toolkit for assigning objective taxonomic classifications to bacterial and archaeal genomes based on the Genome Database Taxonomy</p>
<div class="subheading"><a href="gutsmash.html">gutSMASH</a> (0-20210610) </div><p class="appDesc">gutSMASH is a tool that has been developed to systematically evaluate the metabolic potential of anaerobic bacteria in the gut by predicting both known and novel anaerobic metabolic gene clusters from the gut microbiome.</p>
<div class="subheading"><a href="Hap-IBD.html">Hap-IBD</a> (20221201) </div><p class="appDesc">The hap-ibd program detects identity-by-descent (IBD) segments and homozygosity-by-descent (HBD) segments
in phased genotype data. The hap-ibd program can analyze data sets with hundreds of thousands of samples.</p>
<div class="subheading"><a href="HemTools.html">HemTools</a> (20230512) </div><p class="appDesc">HemTools is a collection of NGS pipelines and bioinformatic analysis tools. It includes tools for data visualization, motif analysis, integrative analysis, bioinformatica analysis, differential analysis, CRISPR analysis, and more.</p>
<div class="subheading"><a href="hhsuite.html">hhsuite</a> (3.3.0) </div><p class="appDesc">The HH-suite is an open-source software package for sensitive protein sequence searching based on the pairwise alignment of hidden Markov models (HMMs).</p>
<div class="subheading"><a href="hicmaptools.html">hicmaptools</a> (20230303) </div><p class="appDesc">hicmaptools is a command line tool to access HiC maps. The complete program provides multi-query modes and analysis tools.</p>
<div class="subheading"><a href="hipstr.html">hipstr</a> (0.7) </div><p class="appDesc">Tool for genotyping short tandom repeats from Illumina sequencing data</p>
<div class="subheading"><a href="hmmer.html">HMMER</a> (3.3.2) </div><p class="appDesc">HMMER is used for searching sequence databases for homologs of protein sequences, and for making protein sequence alignments. It implements methods using probabilistic models called "profile hidden Markov models" (profile HMMs).

Compared to BLAST, FASTA, and other sequence alignment and database search tools based on older scoring methodology, HMMER aims to be significantly more accurate and more able to detect remote homologs because of the strength of its underlying mathematical models.</p>
<div class="subheading"><a href="HTGTSrep.html">HTGTSrep</a> (9fe74ff) </div><p class="appDesc">A pipeline for comprehensive analysis of HTGTS-Rep-seq.</p>
<div class="subheading"><a href="iDiffIR.html">iDiffIR</a> (20220121) </div><p class="appDesc">iDiffIR is a tool for identifying differential IR from RNA-seq data.
It accepts any sorted, indexed BAM file for single- or paired-end reads.</p>
<div class="subheading"><a href="igblast.html">igblast</a> (1.21.0) </div><p class="appDesc">IgBlast is a sequence analysis tool for immunoglobulin variable domains. </p>
<div class="subheading"><a href="infernal.html">infernal</a> (1.1.4) </div><p class="appDesc">Package for searching DNA sequence databases for RNA structure and sequence similarities</p>
<div class="subheading"><a href="interproscan.html">interproscan</a> (5.63-95.0) </div><p class="appDesc">InterProScan is the software package that allows sequences (protein and nucleic) to be scanned against InterPro's signatures. Signatures are predictive models, provided by several different databases, that make up the InterPro consortium.</p>
<div class="subheading"><a href="InterVar.html">InterVar</a> (2.1.2, 2.1.3) </div><p class="appDesc">In 2015, the American College of Medical Genetics and Genomics (ACMG) and the Association for Molecular Pathology (AMP) published
updated standards and guidelines for the clinical interpretation of sequence variants with respect to human diseases on the basis
of 28 criteria. However, variability between individual interpreters can be extensive because of reasons such as the different understandings of these guidelines and the lack of standard algorithms for implementing them, yet computational tools for semi-automated variant interpretation are not available. To address these problems, InterVar implements these criteria to help human reviewers interpret the clinical significance of variants. InterVar can take a pre-annotated or VCF file as input and generate automated interpretation on 18 criteria.</p>
<div class="subheading"><a href="intogen.html">intogen</a> (2023.1) </div><p class="appDesc">Identifies cancer genes and pinpoints their putative mechanism of action across tumor types</p>
<div class="subheading"><a href="iVar.html">iVar</a> (1.3.1) </div><p class="appDesc">iVar is a computational package that contains functions broadly useful for viral amplicon-based sequencing. Additional tools for metagenomic sequencing are actively being incorporated into iVar. While each of these functions can be accomplished using existing tools, iVar contains an intersection of functionality from multiple tools that are required to call iSNVs and consensus sequences from viral sequencing data across multiple replicates.</p>
<div class="subheading"><a href="jellyfish.html">jellyfish</a> (2.3.0) </div><p class="appDesc">Jellyfish is a tool for fast, memory-efficient counting of k-mers in DNA.</p>
<div class="subheading"><a href="kallisto.html">kallisto</a> (0.48.0) </div><p class="appDesc">kallisto is a program for quantifying abundances of transcripts from RNA-Seq data, or more generally of target sequences using high-throughput sequencing reads. It is based on the novel idea of pseudoalignment for rapidly determining the compatibility of reads with targets, without the need for alignment. </p>
<div class="subheading"><a href="last.html">LAST</a> (1454) </div><p class="appDesc">LAST is designed for moderately large data (e.g. genomes, DNA reads, proteomes). It's especially geared toward:
    <ul>
	<li>Finding rearrangements and recombinations (<tt>last-split</tt>)</li>  
	<li>Finding DNA-versus-protein related regions, especially protein fossils.</li>
	<li>Unusual data, e.g. AT-rich DNA, because it can fit parameters to the data and calculate significance.</li>
    <li>Sensitive DNA-DNA search, due to fitting, sensitive seeding, and calculating significance.</li>
</ul></p>
<div class="subheading"><a href="LASTZ.html">LASTZ</a> (1.04.03) </div><p class="appDesc">LASTZ is a tool for (1) aligning two DNA sequences, and (2) inferring appropriate scoring parameters automatically. LASTZ is a drop-in replacement for BLASTZ, and is backward compatible with BLASTZ's command-line syntax.</p>
<div class="subheading"><a href="ldsc.html">ldsc</a> (1.0.1-20200724) </div><p class="appDesc">ldsc is a command line tool for estimating heritability and genetic correlation from GWAS summary statistics. ldsc also computes LD Scores.</p>
<div class="subheading"><a href="Liger2LiGer.html">liger2liger</a> (20230901) </div><p class="appDesc">Liger2LiGer is a Nanopore chimera splitting/detection tool.
Automated end-to-end chimera detection and dataset evaluation starting from fastq.</p>
<div class="subheading"><a href="LJA.html">LJA</a> (0.1) </div><p class="appDesc">The La Jolla Assembler (LJA) is a tool for assemblies of long and accurate reads. It reduces the error rate in
these reads by three orders of magnitude (making them nearly error-free) and constructs the de Bruijn
graph for large genomes and large k-mer sizes. Since the de Bruijn graph constructed for a fixed k-mer
size is typically either too tangled or too fragmented, LJA uses a new concept of a multiplex de Bruijn
graph with varying k-mer sizes.</p>
<div class="subheading"><a href="https://ecogenomic.org/m-tools">m-tools</a> (20210208) </div><p class="appDesc">A selection of software developed at the Australian Centre for Ecogenomics to aid in the analysis of metagenomic datasets: unitem, refinem, checkm, graftm, groopm, bamm, finishm, singlem, orfm, and coverm</p>
<div class="subheading">ma-focus (0.802) </div><p class="appDesc">FOCUS (Fine-mapping Of CaUsal gene Sets) is software to fine-map transcriptome-wide association study statistics at genomic risk regions. The software takes as input summary GWAS data along with eQTL weights and outputs a credible set of genes to explain observed genomic risk.</p>
<div class="subheading"><a href="magma.html">magma</a> (1.10) </div><p class="appDesc">MAGMA is a tool for gene analysis and generalized gene-set analysis of GWAS data. It can be used to analyse both raw genotype data as well as summary SNP p-values from a previous GWAS or meta-analysis.</p>
<div class="subheading"><a href="mantis.html">mantis</a> (1.0.5) </div><p class="appDesc">Microsatellite Analysis for Normal-Tumor InStability is a program developed for detecting microsatellite instability from paired-end BAM files. </p>
<div class="subheading">maps (1.1.0) </div><p class="appDesc">a set of multiple scripts used to analyze PLAC-Seq and HiChIP data.</p>
<div class="subheading"><a href="MashMap.html">MashMap</a> (2.0) </div><p class="appDesc">MashMap is an approximate algorithm for computing local alignment boundaries between long
DNA sequences. Given a minimum alignment length and an identity threshold, it computes the
desired alignment boundaries and identity estimates using kmer-based statistics,
and maintains sufficient probabilistic guarantees on the output sensitivity.</p>
<div class="subheading"><a href="https://melt.igs.umaryland.edu/manual.php">MELT</a> (2.2.2) </div><p class="appDesc">MELT is an application for identifying mobile elements in genomic data</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/Meme.html">meme</a> (5.4.1) </div><p class="appDesc">MEME is used to discover motifs in groups of DNA/protein sequences or databases.</p>
<div class="subheading"><a href="MINTIE.html">MINTIE</a> (0.4.2) </div><p class="appDesc">MINTIE is a tool for identifying novel, rare transcriptional variants in cancer RNA-seq data. MINTIE detects
gene fusions, transcribed structural variants, novel splice variants and complex variants, and annotates all novel transcriptional variants.</p>
<div class="subheading"><a href="mmseqs.html">mmseqs</a> (2-13-45111-219-gaabc78c) </div><p class="appDesc">MMseqs2 (Many-against-Many sequence searching) is a software suite to search and cluster huge protein and nucleotide sequence sets.</p>
<div class="subheading"><a href="MoChA.html">MoChA</a> (1.10.2; 1.11; 1.16) </div><p class="appDesc">MoChA is a bcftools extension to call mosaic chromosomal alterations
starting from phased VCF files
with either B Allele Frequency (BAF) and Log R Ratio (LRR) or allelic depth (AD)</p>
<div class="subheading"><a href="msisensor-pro.html">msisensor-pro</a> (1.2.0) </div><p class="appDesc">evaluates Microsatellite Instability (MSI) for cancer patients with next generation sequencing data.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/mummer.html">MUMmer</a> (4.0.0rc1) </div><p class="appDesc">Mummer is a system for aligning entire genomes extremely rapidly.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/muscle.html">MUSCLE</a> (5.1) </div><p class="appDesc">Fast Multiple Sequence Alignment program. </p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/mutsig2cv.html">mutsig2cv</a> (3.11) </div><p class="appDesc">mutsig2cv analyzes somatic point mutations discovered in DNA sequencing, identifying genes mutated more often than expected by chance.
</p>
<div class="subheading"><a href="ncbi-toolkit.html">ncbi-toolkit</a> (25.0.0) </div><p class="appDesc">The NCBI C++ Toolkit is a set of executables and libraries for a multitude of sequence analysis functions.</p>
<div class="subheading"><a href="nested.html">nested</a> (2.0.0) </div><p class="appDesc">nested (now also called TE-greedy) is software to analyze nested LTR transposable elements
in DNA sequences, such as reference genomes. </p>
<div class="subheading"><a href="netOglyc.html">netOglyc</a> (4.0) </div><p class="appDesc">NetOglyc produces neural network predictions of mucin type GalNAc O-glycosylation sites in mammalian proteins.</p>
<div class="subheading"><a href="Nirvana.html">Nirvana</a> (3.18.1) </div><p class="appDesc">Nirvana provides clinical-grade annotation of genomic variants (SNVs, MNVs, insertions, deletions, indels, and SVs (including CNVs). It can be run as a stand-alone package or integrated into larger software tools that require variant annotation.</p>
<div class="subheading">Octopus (0.7.4) </div><p class="appDesc">Octopus is a mapping-based variant caller that implements several calling models within a unified haplotype-aware framework. Octopus takes inspiration from particle filtering by constructing a tree of haplotypes and dynamically pruning and extending the tree based on haplotype posterior probabilities in a sequential manner. This allows octopus to implicitly consider all possible haplotypes at a given loci in reasonable time.</p>
<div class="subheading"><a href="oncodriveCLUSTL.html">oncodriveCLUSTL</a> (1.1.1) </div><p class="appDesc">OncodriveCLUSTL is a sequence-based clustering algorithm to detect significant clustering signals
across genomic regions.
It is based based on a local background model
derived from the simulation of mutations
accounting for the composition of trior penta-nucleotide context substitutions observed in the cohort under study.</p>
<div class="subheading"><a href="oncodriveFML.html">oncodriveFML</a> (2.2.0) </div><p class="appDesc">OncodriveFML is a method designed to analyze the pattern of somatic mutations
across tumors in both coding and non-coding genomic regions
to identify signals of positive selection, and therefore, their involvement in tumorigenesis.</p>
<div class="subheading"><a href="OpenCRAVAT.html">OpenCRAVAT</a> (1.7.0; 2.2.5) </div><p class="appDesc">OpenCRAVAT is  a new open source, scalable decision support system for
variant and gene prioritization. It includses a modular resource catalog
to maximize community and developer involvement, and as a result the catalog
is being actively developed and growing every month. Resources made available via
the store are well-suited for analysis of cancer, as well as Mendelian and complex
diseases.</p>
<div class="subheading"><a href="optitype.html">OptiType</a> (1.3.5) </div><p class="appDesc">OptiType is a HLA genotyping algorithm based on integer linear programming, capable of producing accurate 4-digit HLA genotyping predictions from NGS data by simultaneously selecting all major and minor HLA Class I alleles.</p>
<div class="subheading"><a href="ORFfinder.html">ORFfinder</a> (0.4.3) </div><p class="appDesc">ORF finder searches for open reading frames (ORFs) in the DNA sequence you enter. The program returns the range of each ORF, along with its protein translation. Use ORF finder to search newly sequenced DNA for potential protein encoding segments, verify predicted protein using newly developed SMART BLAST or regular BLASTP.</p>
<div class="subheading"><a href="orthofinder.html">OrthoFinder</a> (2.5.4) </div><p class="appDesc">OrthoFinder is an accurate and comprehensive platform for comparative genomics.
It finds orthogroups and orthologs, infers rooted gene trees for all orthogroups
and identifies all of the gene duplication events in those gene trees.</p>
<div class="subheading"><a href="PAINTOR.html">PAINTOR</a> (3.0) </div><p class="appDesc">PAINTOR (Probabilistic Annotation INtegraTOR) is a probabilistic framework that integrates association strength with genomic functional annotation data to improve accuracy in selecting plausible causal variants for functional validation.</p>
<div class="subheading"><a href="pangenome.html">pangenome</a> (1.0.0) </div><p class="appDesc">nf-core/pangenome is a bioinformatics best-practise analysis pipeline for the rendering of a collection of sequences into a pangenome graph.</p>
<div class="subheading"><a href="pangolin.html">pangolin</a> (2.3.6) </div><p class="appDesc">Phylogenetic Assignment of Named Global Outbreak LINeages.  PANGOLIN is a system for identifying phylogenetic COVID lineages that contribute most to active spread. 
</p>
<div class="subheading"><a href="parsnp.html">parsnp</a> (1.7.4) </div><p class="appDesc">Parsnp is a command-line-tool for efficient microbial core genome alignment and SNP detection.
Parsnp was designed to work in tandem with Gingr, a flexible platform for visualizing genome alignments and phylogenetic trees</p>
<div class="subheading"><a href="patric.html">patric</a> (1.035) </div><p class="appDesc">PATRIC is an integration of different types of data and software tools that support research on bacterial pathogens.  For users that wish command-line access to PATRIC there are the P3-scripts. They are intended to run on locally, going over the network to access the services provided by PATRIC.</p>
<div class="subheading"><a href="PEPPER_deepvariant.html">PEPPER_deepvariant</a> (0.7) </div><p class="appDesc">PEPPER-Margin-DeepVariant is a haplotype-aware variant calling pipeline for processing third-generation
nanopore sequence data. This pipeline is also applicable to PacBio HiFi data,
providing an efficient solution with superior performance over the current WhatsHap-DeepVariant standard.</p>
<div class="subheading"><a href="phasebook.html">phasebook</a> (1.0.0) </div><p class="appDesc">phasebook is a novel approach for reconstructing the haplotypes of diploid genomes from long reads de novo, that is without the need for a reference genome.</p>
<div class="subheading"><a href="phaser.html">phaser</a> (1.1.1) </div><p class="appDesc">phasing and Allele Specific Expression from RNA-seq. Performs haplotype phasing using read alignments in BAM format from both DNA and RNA based assays, and provides measures of haplotypic expression for RNA based assays.</p>
<div class="subheading"><a href="phylowgs.html">phylowgs</a> (20181105) </div><p class="appDesc">This Python/C++ code is the accompanying software for the paper PhyloWGS: Reconstructing subclonal composition and evolution from whole-genome sequencing of tumors, with authors Amit G. Deshwar, Shankar Vembu, Christina K. Yung, Gun Ho Jang, Lincoln Stein, and Quaid Morris.</p>
<div class="subheading">popscle (0.1) </div><p class="appDesc">A suite of population scale analysis tools for single-cell genomics data including implementation of Demuxlet/Freemuxlet methods and auxiliary tools </p>
<div class="subheading"><a href="PRANK.html">PRANK</a> (150803) </div><p class="appDesc">PRANK is a probabilistic multiple alignment program for DNA, codon and amino-acid sequences. PRANK is based on a novel algorithm that treats insertions correctly and avoids over-estimation of the number of deletion events.</p>
<div class="subheading"><a href="preseq.html">preseq</a> (3.1.2) </div><p class="appDesc">predicting library complexity and genome coverage in high-throughput sequencing</p>
<div class="subheading"><a href="primer3.html">primer3</a> (2.6.1) </div><p class="appDesc">Primer3 is a program for designing PCR primers, hybridization probes, and sequencing primers.</p>
<div class="subheading"><a href="prokka.html">prokka</a> (1.14.6) </div><p class="appDesc">Prokka is a software tool for the rapid annotation of prokaryotic genomes.</p>
<div class="subheading"><a href="PSIPRED.html">PSIPRED</a> (4.0) </div><p class="appDesc">PSIPRED is a  simple and accurate secondary structure prediction method, incorporating two feed-forward neural networks which perform an analysis on output obtained from PSI-BLAST (Position Specific Iterated - BLAST).</p>
<div class="subheading"><a href="pysamstats.html">pysamstats</a> (1.1.2) </div><p class="appDesc">Pysamstats is a fast Python and command-line utility for extracting simple statistics against genome positions based on sequence alignments from a SAM or BAM file.</p>
<div class="subheading"><a href="QTLtools.html">QTLtools</a> (1.3.1) </div><p class="appDesc">A tool set for molecular QTL discovery and analysis. It allows to go from the raw sequence data to collection of molecular Quantitative Trait Loci (QTLs) in few easy-to-perform steps.</p>
<div class="subheading"><a href="randfold.html">randfold</a> (2.0.1) </div><p class="appDesc">RandFold computes the probability that, for a given RNA sequence, the Minimum Free Energy (MFE) of the secondary structure is different from a distribution of MFE computed with random sequences.

</p>
<div class="subheading"><a href="regenie.html">regenie</a> (3.0.3) </div><p class="appDesc">regenie is a C++ program for whole genome regression modelling of large genome-wide association studies.  It is developed and supported by a team of scientists at the Regeneron Genetics Center. regenie employs the BGEN library. </p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/remora.html">remora</a> (2.1.1) </div><p class="appDesc">Modified base model training and application.</p>
<div class="subheading"><a href="RGI.html">RGI</a> (6.0.2) </div><p class="appDesc">RGI (Resistance Gene Identifier) is a robust antimicrobial resistance (AMR) gene predicting tool.
It is based on newly curated Comprehensive Antibiotic Research Database (CARD) and allows detection
detect AMR genes from thirteen genomes of Pseudomonas strains.</p>
<div class="subheading"><a href="rMATS.html">rMATS </a> (4.1.2; 4.0.2) </div><p class="appDesc">MATS is a computational tool to detect differential alternative splicing events from RNA-Seq data. </p>
<div class="subheading"><a href="rmblast.html">rmblast</a> (2.14.0) </div><p class="appDesc">RMBlast is a RepeatMasker-compatible version of the standard NCBI blastn program. RMBlast supports RepeatMasker searches by adding a few necessary features to the stock NCBI blastn program.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/trinotate.html#deps">RNAmmer</a> (1.2) </div><p class="appDesc">RNAmmer predicts ribosomal RNA genes in full genome sequences by utilising two levels of Hidden Markov Models: An initial spotter model searches both strands. The spotter model is constructed from highly conserved loci within a structural alignment of known rRNA sequences. Once the spotter model detects an approximate position of a gene, flanking regions are extracted and parsed to the full model which matches the entire gene. </p>
<div class="subheading"><a href="roary.html">roary</a> (3.13.0) </div><p class="appDesc">Roary is a high speed stand alone pan genome pipeline, which takes annotated assemblies in GFF3 format (produced by Prokka) and calculates the pan genome.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/ROSE.html">ROSE</a> (1.3.1) </div><p class="appDesc">ROSE (Rank Ordering of Super-Enhancers) is tool for 
(1) creating stitched enhancers, and  
(2) separating super-enhancers from typical enhancers. 
given sequencing data (.bam) and a file of previously identified constituent enhancers (.gff)</p>
<div class="subheading"><a href="SAIGE.html">SAIGE</a> (1.1.9) </div><p class="appDesc">R package for large-scale genetic association studies.</p>
<div class="subheading"><a href="scanpy.html">scanpy</a> (1.8.1) </div><p class="appDesc">Scanpy is a scalable toolkit for analyzing single-cell gene expression data. It includes preprocessing, visualization, clustering, pseudotime and trajectory inference and differential expression testing. The Python-based implementation efficiently deals with datasets of more than one million cells.</p>
<div class="subheading"><a href="scChromHMM.html">scChromHMM</a> (20220610) </div><p class="appDesc">scChromHMM provides a suite of tools for rapid processing of single-cell histone modification data to perform chromatin states analysis of the genome within each single-cell. It is an extention of bulk ChromHMM framework, which consumes the HMM model learned from ChromHMM and perform chromatin state analysis by running forward-backward algorithm for each single-cell.</p>
<div class="subheading"><a href="scDRS.html">scDRS</a> (1.02) </div><p class="appDesc">The scDRS application implements an approach that links scRNA-seq
with polygenic disease risk at single-cell resolution, independent of annotated cell types.
scDRS identifies cells exhibiting excess expression across disease-associated genes
implicated by genome-wide association studies (GWASs).
Genes whose expression was correlated with the scDRS score across cells
(reflecting coexpression with GWAS disease-associated genes)
were strongly enriched for gold-standard drug target and Mendelian disease genes.</p>
<div class="subheading"><a href="scvelo.html">scvelo</a> (0.2.3) </div><p class="appDesc">scVelo is a method to describe the rate of gene expression change for an individual gene at a given time point based on the ratio of its spliced and unspliced messenger RNA (mRNA).
It avoids errors in the velocity estimates by solving the full transcriptional dynamics of splicing kinetics using a likelihood-based dynamical model.
This generalizes RNA velocity to systems with transient cell states, which are common in development and in response to perturbations.</p>
<div class="subheading"><a href="SEACR.html">SEACR</a> (1.3) </div><p class="appDesc">SEACR is intended to call peaks and enriched regions from sparse Cleavage Under Targets and Release Using Nuclease (CUT&RUN) or chromatin profiling data in which background is dominated by "zeroes" (i.e. regions with no read coverage). </p>
<div class="subheading"><a href="selscan.html">selscan</a> (1.3.0; 2.0.0) </div><p class="appDesc">selscan is a tool for haplotype-based scans to detect natural selection, which are useful to identify recent or ongoing positive selection in genomes. It is an efficient multithreaded application that implements Extended Haplotype Homozygosity (EHH), Integrated Haplotype Score (iHS), and Cross-population EHH (XPEHH). selscan accepts phased genotypes in multiple formats, including TPED.</p>
<div class="subheading"><a href="seq2hla.html">seq2hla</a> (2.3) </div><p class="appDesc">seq2HLA computationally determines human leukocyte antigen (HLA) genotypes of a sample using RNA-Seq sequencing reads.</p>
<div class="subheading"><a href="sequencetubemap.html">sequencetubemap</a> (2023.8) </div><p class="appDesc">A JavaScript module for the visualization of genomic sequence graphs. It automatically generates a "tube map"-like visualization of sequence graphs which have been created with vg. </p>
<div class="subheading">shimmer (0.2) </div><p class="appDesc">Shimmer is a software package for the characterization of genetic differences between two very similar samples, e.g., a tumor sample and its matched normal tissue sample.</p>
<div class="subheading"><a href="https://services.healthtech.dtu.dk/service.php?SignalP-6.0">signalp</a> (6.0g) </div><p class="appDesc">SignalP predicts the presence  and  location of signal peptide  cleavage sites in  amino acid  sequences from different organisms: Gram-positive bacteria, Gram-negative bacteria, and eukaryotes.  The method incorporates a prediction of cleavage sites and  a signal peptide/non-signal peptide prediction based on
a combination of several artificial neural networks.</p>
<div class="subheading"><a href="slivar.html">slivar</a> (0.3.0) </div><p class="appDesc">slivar is a set of command-line tools that enables rapid querying and filtering of VCF files. It facilitates operations on trios and groups and allows arbitrary expressions using simple javascript.</p>
<div class="subheading">smcpp (1.15.4) </div><p class="appDesc">SMC++ is a program for estimating the size history of populations from whole genome sequence data.</p>
<div class="subheading"><a href="smoove.html">smoove</a> (0.2.7) </div><p class="appDesc">smoove simplifies and speeds calling and genotyping SVs for short reads.
It also improves specificity by removing many spurious alignment signals
that are indicative of low-level noise and often contribute to spurious calls.</p>
<div class="subheading"><a href="SnapATAC2.html">SnapATAC2</a> (2.1.2) </div><p class="appDesc">SnapATAC is a software package for analyzing scATAC-seq datasets.
Conventional assays to map regulatory elements via open chromatin analysis of primary tissues
is hindered by sample heterogeneity.
SnapATAC dissects cellular heterogeneity in an unbiased manner and map the trajectories of cellular states.</p>
<div class="subheading">splam (1.0.9) </div><p class="appDesc">Splice junction recognition model based on a deep residual convolutional neural network to assess splice junctions.</p>
<div class="subheading"><a href="straw.html">straw</a> (1.3.1) </div><p class="appDesc">Straw is library which allows rapid streaming of contact data from .hic files.</p>
<div class="subheading"><a href="stripenn.html">stripenn</a> (1.1.50; 1.1.65.15) </div><p class="appDesc">Stripenn is a command line interface python package developed for detection of atchitectural stripes
from chromatin conformation capture (3C) data. It implements an algorithm rooted in computer vision
for demarcation and quantification of the architectural stripes. Stripenn was demonstrated to outperform
existing methods, be applicable in the context of analysis of B and T lymphocytes, and to allow
examination of the role of sequence variation on the architectural stripes by studying the conservation
of these features in inbred strains of mice.</p>
<div class="subheading"><a href="SynthDNM.html">SynthDNM</a> (0.1.3) </div><p class="appDesc">SynthDNM is a random-forest based classifier that can be readily adapted
to new sequencing or variant-calling pipelines
by applying a flexible approach to constructing simulated
training examples from real data. The optimized SynthDNM classifiers predict de novo SNPs and indels
with robust accuracy across multiple methods of variant calling.</p>
<div class="subheading"><a href="tandem-genotypes.html">tandem-genotypes</a> (1.9.0) </div><p class="appDesc">tandem-genotypes finds changes in length of tandem repeats, from "long" DNA reads aligned to a genome.</p>
<div class="subheading"><a href="tantan.html">tantan</a> (40) </div><p class="appDesc">A tool to mask low complexity and short period tandem repeats</p>
<div class="subheading"><a href="TElocal.html">TElocal</a> (1.1.1) </div><p class="appDesc">TElocal: a tool that utilizes both uniquely and ambiguously mapped reads to quantify transposable element expression at the locus level.
</p>
<div class="subheading"><a href="telomerehunter.html">TelomereHunter</a> (1.1.0) </div><p class="appDesc">TelomereHunter is a software for the detailed characterization of telomere maintenance mechanism footprints in the genome. The tool is implemented for the analysis of large cancer genome cohorts and provides a variety of diagnostic diagrams as well as machine-readable output for subsequent analysis.</p>
<div class="subheading">TMHMM (2.0c) </div><p class="appDesc">TMHMM predicts transmembrane helices in proteins. </p>
<div class="subheading"><a href="toga.html">toga</a> (1.1.2) </div><p class="appDesc">TOGA is a new method that integrates gene annotation, inferring orthologs and classifying genes as intact or lost. TOGA implements a novel machine learning based paradigm to infer orthologous genes between related species and to accurately distinguish orthologs from paralogs or processed pseudogenes. This tutorial explains how to get started using TOGA. It shows how to install and execute TOGA, and how to handle possible issues that may occur. </p>
<div class="subheading"><a href="tombo.html">tombo</a> (1.5.1) </div><p class="appDesc">a suite of tools primarily for the identification of modified nucleotides from nanopore sequencing data. </p>
<div class="subheading"><a href="trf.html">TRF</a> (4.09) </div><p class="appDesc">A tandem repeat in DNA is two or more adjacent, approximate copies of a pattern of nucleotides. Tandem Repeats Finder is a program to locate and display tandem repeats in DNA sequences.</p>
<div class="subheading"><a href="trgt.html">trgt</a> (0.4.0) </div><p class="appDesc">TRGT is a tool for targeted genotyping of tandem repeats from PacBio HiFi data. In addition to the basic size genotyping, TRGT profiles sequence composition, mosaicism, and CpG methylation of each analyzed repeat. TRGT comes with a companion tool TRVZ for visualization of reads overlapping the repeats.</p>
<div class="subheading"><a href="trimAl.html">trimAl</a> (1.2rev59) </div><p class="appDesc">trimAl is a tool for the automated removal of spurious sequences or poorly aligned regions from a multiple sequence alignment.
It can consider several parameters, alone or in multiple combinations,
in order to select the most-reliable positions in the alignment.
These include the proportion of sequences with a gap, the level of residue similarity and,
if several alignments for the same set of sequences are provided, the consistency level of columns among alignments.
Moreover, trimAl is able to manually select a set of columns to be removed from the alignment.</p>
<div class="subheading">triodenovo (0.06) </div><p class="appDesc">triodenovo is an application for calling de novo mutations in trios for NGS data.</p>
<div class="subheading">trust4 (1.0.8) </div><p class="appDesc">Tcr Receptor Utilities for Solid Tissue (TRUST) is a computational tool to analyze TCR and BCR sequences using unselected RNA sequencing data, profiled from solid tissues, including tumors. TRUST4 performs de novo assembly on
V, J, C genes including the hypervariable complementarity-determining region 3 (CDR3) and reports consensus of BCR/TCR sequences. TRUST4 then realigns the contigs to IMGT reference gene sequences to report the corresponding information. TRUST4 supports both single-end and paired-end sequencing data with any read length.
</p>
<div class="subheading"><a href="ultra.html">ultra</a> (0.1) </div><p class="appDesc">uLTRA implements an alignment method for long RNA sequencing reads
based on a novel two-pass collinear chaining algorithm. uLTRA is guided by a database of exon annotations,
but it can also be used as a wrapper around minimap2 to align reads outside annotated regions.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/unicycler.html">unicycler</a> (0.5.0) </div><p class="appDesc">Unicycler is an assembly pipeline for bacterial genomes. It can assemble Illumina-only read sets where it functions as a SPAdes-optimiser. It can also assembly long-read-only sets (PacBio or Nanopore) where it runs a miniasm+Racon pipeline. </p>
<div class="subheading"><a href="uropa.html">uropa</a> (4.0.2) </div><p class="appDesc">UROPA is a command line based tool for genomic region annotation</p>
<div class="subheading"><a href="usearch.html">usearch</a> (11.0.667) </div><p class="appDesc">USEARCH is a unique sequence analysis tool with thousands of users world-wide. USEARCH offers search and clustering algorithms that are often orders of magnitude faster than BLAST. </p>
<div class="subheading"><a href="VADR.html">VADR</a> (1.3) </div><p class="appDesc">VADR stands for Viral Annotation DefineR. It is a suite of tools for classifying and analyzing sequences
homologous to a set of reference models of viral genomes or gene families. It has been mainly tested for analysis of Norovirus, Dengue, and SARS-CoV-2 virus sequences
in preparation for submission to the GenBank database.</p>
<div class="subheading"><a href="VarScan.html">VarScan</a> (2.4.6) </div><p class="appDesc">A platform-independent, technology-independent software tool for identifying SNPs and indels in massively parallel sequencing of individual and pooled samples.</p>
<div class="subheading"><a href="VCF-kit.html">VCF-kit</a> (0.2.9) </div><p class="appDesc">VCF-kit is a collection of utility tools for processing and analyzing the VCF (variant call format) files, including primer generation for variant validation, dendrogram production,genotype imputation from sequence data in linkage studies, and additional tools to be used by statistical and population geneticists.</p>
<div class="subheading"><a href="vcontact2.html">vcontact2</a> (0.11.3) </div><p class="appDesc">vConTACT2 is a tool to perform guilt-by-contig-association classification of viral genomic sequence data. It's designed to cluster and provide taxonomic context of metagenomic sequencing data.</p>
<div class="subheading"><a href="velocyto.html">velocyto</a> (0.17) </div><p class="appDesc">Velocyto is a library for the analysis of RNA velocity. It includes a command line tool and an analysis pipeline.</p>
<div class="subheading"><a href="viennarna.html">viennarna</a> (2.5.1) </div><p class="appDesc">RNA Secondary Structure Prediction and Comparison</p>
<div class="subheading"><a href="VirSorter2.html">VirSorter2</a> (2.2.3) </div><p class="appDesc">VirSorter2 is a DNA and RNA virus identification tool. It leverages genome-informed
database advances across a collection of customized automatic classifiers
to improve the accuracy and range of virus sequence detection.</p>
<div class="subheading"><a href="weblogo.html">weblogo</a> (3.6) </div><p class="appDesc">contains seqlogo utility to create sequence logo summarizing sequence alignments</p>
<div class="subheading"><a href="xengsort.html">xengsort</a> (1.1.0) </div><p class="appDesc">A fast xenograft read sorter based on space-efficient k-mer hashing.</p>
<div class="subheading"><a href="xenome.html">xenome</a> (1.0.1) </div><p class="appDesc">xenome is a tool for classifying reads from xenograft source.</p>
<div class="subheading"><a href="xHLA.html">xHLA</a> (2018-04-04) </div><p class="appDesc">The HLA gene complex on human chromosome 6 is one of the
most polymorphic regions in the human genome and contributes
in large part to the diversity of the immune system. Accurate typing of HLA genes with short-read sequencing data has historically been difficult due to the sequence similarity between the polymorphic alleles. xHLA iteratively refines the mapping results at the amino acid level to
achieve high typing accuracy for both class I and II HLA genes.</p>
<div class="subheading"><a href="xtail.html">xtail</a> (1.1.5) </div><p class="appDesc">Xtail is an analysis pipeline tailored for ribosome profiling data
that comprehensively and accurately identifies differentially translated genes in pairwise comparisons.
Applied on simulated and real datasets, Xtail exhibits high sensitivity with minimal false-positive rates,
outperforming existing methods in the accuracy of quantifying differential translations.</p>
<div class="subheading"><a href="xTea.html">xTea</a> (0.1.9; 1.0.0) </div><p class="appDesc">xTea (x-Transposable element analyzer), is a tool for identifying TE insertions
in whole-genome sequencing data. Whereas existing methods are mostly designed for shortread
data, xTea can be applied to both short-read and long-read data. xTea outperforms other short read-based methods
for both germline and somatic TE insertion discovery.</p>
<div class="heading"><a name="structbio" id="structbio">Structural Biology</a></div>
<a href="/apps/" style="font-size:12px">back to top</a><br />
<p></p>
<div class="subheading"><a href="alphafold2.html">alphafold2</a> (2.3.2) </div><p class="appDesc">This package provides an implementation of the protein structure inference pipeline of AlphaFold v2.0.</p>
<div class="subheading"><a href="alphalink.html">alphalink</a> (1.0) </div><p class="appDesc">AlphaLink predicts protein structures using deep learning given a sequence and a set of experimental contacts. It extends OpenFold with crosslinking MS data or other experimental distance restraint by explicitly incorporating them in the OpenFold architecture.</p>
<div class="subheading"><a href="alphapulldown.html">AlphaPulldown</a> (0.30.7) </div><p class="appDesc">AlphaPulldown is a Python package that streamlines protein-protein interaction screens
and high-throughput modelling of higher-order oligomers using AlphaFold-Multimer. It provides
a convenient command-line interface, a variety of confidence scores and a graphical analysis tool.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/AutodockVina.html">Autodock-GPU</a> (1.5.3) </div><p class="appDesc">Autodock-GPU performs docking calculations, and processes ligand-receptor poses in parallel over multiple compute units on GPUs.</p>
<div class="subheading"><a href="braker.html">braker</a> (3) </div><p class="appDesc">A pipeline for fully automated prediction of protein coding gene structures with GeneMark-ES/ET and AUGUSTUS in novel eukaryotic genomes.</p>
<div class="subheading"><a href="CCP4.html">CCP4</a> (8.0.016) </div><p class="appDesc">CCP4 is a  suite of programs for protein crystallography and structural biology.</p>
<div class="subheading"><a href="colabfold.html">colabfold</a> (1.5.2) </div><p class="appDesc">ColabFold batch scripts</p>
<div class="subheading"><a href="cryoDRGN.html">cryoDRGN</a> (1.1.0) </div><p class="appDesc">CryoDRGN is an algorithm that leverages the representation power of deep neural networks
to directly reconstruct continuous distributions of 3D density maps
and map per-particle heterogeneity of single-particle cryo-EM datasets.
It contains interactive tools to visualize a dataset’s distribution of per-particle
variability, generate density maps for exploratory analysis, extract particle
subsets for use with other tools and generate trajectories to visualize molecular motions.</p>
<div class="subheading"><a href="csd.html">CSD</a> (2022) </div><p class="appDesc">The Cambridge Structural Database is the world repository of small molecule crystal structures. 

</p>
<div class="subheading"><a href="dali.html">Dali</a> (5.1) </div><p class="appDesc">The three-dimensional co-ordinates of each protein are used to calculate residue - residue distance matrices.</p>
<div class="subheading"><a href="deepab.html">deepab</a> (1.2) </div><p class="appDesc">DeepAb is antibody structure prediction using interpretable deep learning.</p>
<div class="subheading"><a href="deepmedic.html">deepmedic</a> (0.8.4) </div><p class="appDesc">This project aims to offer easy access to Deep Learning for segmentation of structures of interest in biomedical 3D scans. It is a system that allows the easy creation of a 3D Convolutional Neural Network, which can be trained to detect and segment structures if corresponding ground truth labels are provided for training. The system processes NIFTI images, making its use straightforward for many biomedical tasks.</p>
<div class="subheading"><a href="DeepMM.html">DeepMM</a> (20220830) </div><p class="appDesc">DeepMM implements fully automated de novo structure modeling method, MAINMAST, which builds three-dimensional models of a protein from a near-atomic resolution EM map. The method directly traces the protein’s main-chain and identifies Cα positions as tree-graph structures in the EM map.</p>
<div class="subheading"><a href="DSSP.html">DSSP</a> (2.3.0) </div><p class="appDesc">The DSSP program was designed by Wolfgang Kabsch and Chris Sander to standardize secondary structure assignment. DSSP is a database of secondary structure assignments (and much more) for all protein entries in the Protein Data Bank (PDB). DSSP is also the program that calculates DSSP entries from PDB entries. DSSP does not predict secondary structure.</p>
<div class="subheading"><a href="esm.html">esm</a> (2.0.0-74-g2b36991) </div><p class="appDesc">Code and pre-trained weights for Transformer protein language models from the Meta Fundamental AI Research Protein Team (FAIR)</p>
<div class="subheading"><a href="foldseek.html">foldseek</a> (5-53465f0) </div><p class="appDesc">Fast structural similarity search</p>
<div class="subheading"><a href="gautomatch.html">gautomatch</a> (0.56_cuda-9.1) </div><p class="appDesc">Fully automatic acccurate, convenient and extremely fast particle picking for EM</p>
<div class="subheading"><a href="lammps.html">lammps</a> (29Oct20) </div><p class="appDesc">LAMMPS is a classical molecular dynamics code, and an acronym for Large-scale Atomic/Molecular Massively Parallel Simulator. It runs on a variety of different computer systems, including single processor systems, distributed-memory machines with MPI, and GPU and Xeon Phi systems. LAMMPS is open source software, released under the GNU General Public License.</p>
<div class="subheading"><a href="mdtraj.html">mdtraj</a> (1.9.7) </div><p class="appDesc">MDTraj is a python library that allows users to manipulate molecular dynamics (MD) trajectories and perform a variety of analyses, including fast RMSD, solvent accessible surface area, hydrogen bonding, etc.</p>
<div class="subheading"><a href="model-angelo.html">model-angelo</a> (1.0.9) </div><p class="appDesc">ModelAngelo is an automatic atomic model building program for cryo-EM maps.</p>
<div class="subheading">mustache (1.0.1) </div><p class="appDesc">Mustache is a tool for identifying chromatin loops from HiC and MicroC contact maps</p>
<div class="subheading"><a href="naccess.html">naccess</a> (2.1.1) </div><p class="appDesc">The naccess program calculates the atomic accessible surface defined by rolling a probe of given size around a van der Waals surface.</p>
<div class="subheading"><a href="novactf.html">novactf</a> (1.0) </div><p class="appDesc">NovaCTF is a freeware for 3D-CTF correction for electron microscopy.</p>
<div class="subheading"><a href="OmegaFold.html">OmegaFold</a> (1.1.0) </div><p class="appDesc">OmegaFold is the first computational method to successfully predict high-resolution protein structure
from a single primary sequence alone.
Using a new combination of a protein language model
that allows us to make predictions from single sequences
and a geometry-inspired transformer model trained on protein structures,
OmegaFold outperforms RoseTTAFold
and achieves similar prediction accuracy to AlphaFold2 on recently released structures.</p>
<div class="subheading"><a href="pf_refinement.html">pf_refinement</a> (20Nov19) </div><p class="appDesc">Protofilament Refinement is a software package to further refine microtubule structures, by aligning individual protofilaments rather than full microtubule segements.</p>
<div class="subheading"><a href="Phenix.html">Phenix</a> (1.20.1-4487) </div><p class="appDesc">PHENIX is a software suite for the automated determination of macromolecular structures using X-ray crystallography and other methods.</p>
<div class="subheading"><a href="pydockrmsd.html">pydockrmsd</a> (1.0.0) </div><p class="appDesc">DockRMSD is capable of deterministically identifying the minimum symmetry-corrected RMSD and is able to do so without significant loss of computational efficiency compared to other methods.</p>
<div class="subheading"><a href="pyem.html">pyem</a> (230821) </div><p class="appDesc">UCSF pyem is a collection of Python modules and command-line utilities for electron microscopy of biological samples.</p>
<div class="subheading"><a href="PyMOL.html">PyMOL</a> (2.6.0) </div><p class="appDesc">A comprehensive molecular visualization product for rendering and animating 3D molecular structures. </p>
<div class="subheading"><a href="PyRosetta.html">PyRosetta</a> (360.py3.10) </div><p class="appDesc">PyRosetta is an interactive Python-based interface to the powerful Rosetta molecular modeling suite. It enables users to design their own custom molecular modeling algorithms using Rosetta sampling methods and energy functions.</p>
<div class="subheading"><a href="ReLeaSE.html">ReLeaSE</a> (20220825) </div><p class="appDesc">ReLeaSE (Reinforcement Learning for Structural Evolution) is an application for de-novo Drug Design
based on Reinforcement Learning. It integrates two deep neural networks: generative and
predictive, that are trained separately but are used jointly to generate novel targeted chemical libraries.
ReLeaSE uses simple representation of molecules by their simplified molecular input line entry specification (SMILES)          strings only.</p>
<div class="subheading"><a href="RFdiffusion.html">RFdiffusion</a> (1.1.0) </div><p class="appDesc">Rosetta Fold (RF) dissusion is an open source method for structure generation, with or without conditional information
(a motif, target etc). It can perform motif scaffolding, unconditional protein generation, and other tasks.</p>
<div class="subheading"><a href="rosettafold.html">RoseTTAFold</a> (1.1.0) </div><p class="appDesc">Accurate prediction of protein structures and interactions using a 3-track network, , in which information at the 1D sequence level, the 2D distance map level, and the 3D coordinate level is successively transformed and integrated.

</p>
<div class="subheading"><a href="rosettafoldna.html">rosettafoldna</a> (0.1) </div><p class="appDesc">Accurate prediction of nucleic acid and protein-nucleic acid complexes using RoseTTAFoldNA</p>
<div class="subheading"><a href="schrodinger/index.html">Schrodinger</a> (2023.1) </div><p class="appDesc">A limited number of Schrödinger applications are available on the Biowulf cluster through the Molecular Modeling Interest Group. Most are available through the Maestro GUI.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/Scipion/index.html">Scipion</a> (3.0.12) </div><p class="appDesc">Scipion is an image processing framework to obtain 3D models of macromolecular complexes using Electron Microscopy (3DEM). It integrates several software packages and presents an unified interface for both biologists and developers. Scipion allows to execute workflows combining different software tools, while taking care of formats and conversions. Additionally, all steps are tracked and can be reproduced later on.</p>
<div class="subheading"><a href="sidesplitter.html">sidesplitter</a> (1.2) </div><p class="appDesc">Sidesplitter reduces over-fitting in both idealised and experimental settings, while maintaining
independence between the two sides of a split refinement.  It can improve the final resolution in refinements of structures prone to severe over-fitting, such as membrane proteins in detergent micelles.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/subtom.html">subtom</a> (1.1.6-32f731b) </div><p class="appDesc">Subtom is a pipeline for subvolume alignment and averaging of electron cryo-tomography data.</p>
<div class="subheading"><a href="/apps/vmd.html">vmd</a> (1.9.3) </div><p class="appDesc">VMD is a molecular visualization program for displaying, animating, and analyzing large biomolecular systems using 3-D graphics and built-in scripting. To use, type vmd at the prompt.</p>
<div class="subheading"><a href="Xplor-NIH/index.html">Xplor-NIH</a> (3.5) </div><p class="appDesc">Xplor-NIH is a structure determination program which builds on the X-PLOR v3.851 program, including additional tools developed at the NIH.</p>
<div class="subheading"><a href="ZDOCK.html">ZDOCK</a> (3.0.2) </div><p class="appDesc">ZDOCK predicts protein-docking models, and uses a fast Fourier transform to search all possible binding modes for proteins, evaluating based on shape complementarity, desolvation energy, and electrostatics.</p>
<div class="heading"><a name="sysbiol" id="sysbiol">Systems Biology</a></div>
<a href="/apps/" style="font-size:12px">back to top</a><br />
<p></p>
<div class="subheading">bakta (1.8.1) </div><p class="appDesc">Rapid & standardized annotation of bacterial genomes, MAGs & plasmids</p>
<div class="subheading"><a href="bionetgen.html">bionetgen</a> (0.7.9) </div><p class="appDesc">BioNetGen is software for the specification and simulation of rule-based models of biochemical systems, including signal transduction, metabolic, and genetic regulatory networks.  BioNetGen is presently a mixture of Perl, C++, and Python.</p>
<div class="subheading"><a href="cellphonedb.html">cellphonedb</a> (3.1.0) </div><p class="appDesc">A publicly available repository of curated receptors, ligands and their interactions. Subunit architecture is included for both ligands and receptors, representing heteromeric complexes accurately.</p>
<div class="subheading"><a href="CytoSig.html">CytoSig</a> (0.1) </div><p class="appDesc">CytoSig is a data-driven infrastructure hosted by the National Cancer Institute.
CytoSig includes both a database of target genes modulated by cytokines and
a predictive model of cytokine signaling activity and regulatory cascade from transcriptomic profiles.</p>
<div class="subheading"><a href="dyno.html">dyno</a> (20220906) </div><p class="appDesc">dyno is a meta package that installs several other packages
from the dynvers (https://github.com/dynverse). It comprises a set of R packages
to construct and interpret single-cell trajectories.</p>
<div class="subheading"><a href="eggNOGmapper.html">eggNOGmapper</a> (2.1.6) </div><p class="appDesc">eggNOGmapper is a tool for functional annotation of large sets of sequences based on fast orthology assignments using precomputed clusters and phylogenies from the eggNOG  database. Orthology assignment is ideally suited for functional inference. However, predicting orthology is computationally intensive at large scale, and most
other pipelines are relatively inaccessible (e.g., new assignments only available through database updates), so less precise homology-based functional transfer was previously the default for (meta-)genome annotation.</p>
<div class="subheading"><a href="GCN_Cancer.html">GCN_Cancer</a> (20221105) </div><p class="appDesc">The GCN_Cancer application employs graph convolutional network (GCN) models
to classify the gene expression data samples from The Cancer Genonme Atlas (TCAG)
as 33 designated tumor types or as normal. It has been trained on  10,340
cancer samples and 731 normal tissue samples from TCGA dataset.</p>
<div class="subheading"><a href="https://www.gsea-msigdb.org/gsea/index.jsp">gsea</a> (4.3.2) </div><p class="appDesc">Gene Set Enrichment Analysis (GSEA) is a computational method that determines whether an a priori defined set of genes shows statistically significant, concordant differences between two biological states (e.g. phenotypes).</p>
<div class="subheading"><a href="MAKER.html">MAKER</a> (3.01.03) </div><p class="appDesc">MAKER is a portable and easily configurable genome annotation pipeline.
Its purpose is to allow smaller eukaryotic and prokaryotic genome projects
to independently annotate their genomes and to create genome databases.
MAKER identifies repeats, aligns ESTs and proteins to a genome,
produces ab-initio gene predictions and automatically synthesizes these data
into gene annotations having evidence-based quality values.</p>
<div class="subheading"><a href="Neuron.html">Neuron</a> (7.7) </div><p class="appDesc">NEURON is a simulation environment for modeling individual neurons and networks of neurons. It provides tools for conveniently building, managing, and using models in a way that is numerically sound and computationally efficient. It is particularly well-suited to problems that are closely linked to experimental data, especially those that involve cells with complex anatomical and biophysical properties. </p>
<div class="subheading"><a href="robot.html">ROBOT</a> (1.9.4) </div><p class="appDesc">ROBOT is a command-line tool and library for automating ontology development tasks,
with a focus on Open Biological and Biomedical Ontologies (OBO).
It can be used as a command-line tool or as a library for any language on the Java Virtual Machine.</p>
<div class="subheading"><a href="scvelo.html">scvelo</a> (0.2.3) </div><p class="appDesc">scVelo is a method to describe the rate of gene expression change for an individual gene at a given time point based on the ratio of its spliced and unspliced messenger RNA (mRNA).
It avoids errors in the velocity estimates by solving the full transcriptional dynamics of splicing kinetics using a likelihood-based dynamical model.
This generalizes RNA velocity to systems with transient cell states, which are common in development and in response to perturbations.</p>
<div class="subheading"><a href="vdjtools.html">vdjtools</a> (1.2.1) </div><p class="appDesc">A comprehensive analysis framework for T-cell and B-cell repertoire sequencing data.</p>
<div class="heading"><a name="utils" id="utils">Utilities</a></div>
<a href="/apps/" style="font-size:12px">back to top</a><br />
<p></p>
<div class="subheading"><a href="apptainer.html">apptainer</a> (1.1.6) </div><p class="appDesc">Apptainer allows you to build and run Linux containers with emphasis on use in HPC. Apptainer is the Linux Foundation variant of and successor to the widely popular Singularity.</p>
<div class="subheading"><a href="aria2.html">aria2</a> (1.36.0) </div><p class="appDesc">multiprotocol download utility
<br>Type 'module load aria2'then 'aria2c --help' for more info.</p>
<div class="subheading"><a href="https://asciinema.org/">asciinema</a> (2.2.0) </div><p class="appDesc">asciinema [as-kee-nuh-muh] is a free and open source solution for recording terminal sessions and sharing them.
<br>Type 'module load asciinema' then 'asciinema' to run.</p>
<div class="subheading"><a href="/docs/transfer.html">Aspera</a> (3.7..4) </div><p class="appDesc">High-speed fasp-powered file transfers. Mostly used to download data from NCBI, which has an Aspera server. See <a href="/docs/transfer.html">the data transfer page</a> for details.</p>
<div class="subheading"><a href="https://www.gnu.org/software/autoconf/">autoconf</a> (2.71) </div><p class="appDesc">Autoconf is an extensible package of M4 macros that produce shell scripts to automatically configure software source code packages.</p>
<div class="subheading">AWS (1.25.49) </div><p class="appDesc">Command-line tools for Amazon Web Services. Use 
'module load python; aws -help' to see the command-line help, or <a href="http://aws.amazon.com/cli/">http://aws.amazon.com/cli/</a>.</p>
<div class="subheading">azcopy (10.11.0) </div><p class="appDesc">a command-line utility to copy blobs or files to or from Azure storage</p>
<div class="subheading">bakta (1.8.1) </div><p class="appDesc">Rapid & standardized annotation of bacterial genomes, MAGs & plasmids</p>
<div class="subheading">bbcp (15.02.03.01.1) </div><p class="appDesc">Secure and fast copy utility</p>
<div class="subheading"><a href="BGEN.html">BGEN</a> (20230420) </div><p class="appDesc">The BGEN library contains a reference implementation of the BGEN format, written in C++. The library can be used as the basis for BGEN support in other software.</p>
<div class="subheading"><a href="circos.html">circos</a> (0.69-9) </div><p class="appDesc">Circos is a program for the generation of publication-quality, circularly composited renditions of genomic data and related annotations. Circos is particularly suited for visualizing alignments, conservation and intra and inter-chromosomal relationships. Also, Circos is useful to visualize any type of information that benefits from a circular layout. Thus, although it has been designed for the field of genomics, it is sufficiently flexible to be used in other data domains.</p>
<div class="subheading"><a href="http://www.gnu.org/software/coreutils/manual/">coreutils</a> (9.1) </div><p class="appDesc">The GNU Core Utilities are the basic file, shell and text manipulation utilities of the GNU operating system.
</p>
<div class="subheading"><a href="https://www.coherentpdf.com/cpdfmanual.pdf">cpdf</a> (2.5.1) </div><p class="appDesc">Coherent PDF tools</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/cryolo.html">cryolo</a> (1.9.2) </div><p class="appDesc">Automated particle picker for cryo-EM</p>
<div class="subheading">crypt4gh (0.4.1) </div><p class="appDesc">Rust implementation for the Crypt4GH encryption format.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/datalad.html">datalad</a> (0.13.0rc2) </div><p class="appDesc">Datalad is a tool for uploading and downloading public up-t-to-date neuroimaging datasets.</p>
<div class="subheading"><a href="https://www.gnu.org/software/datamash/">datamash</a> (1.7) </div><p class="appDesc">GNU datamash is a command-line program which performs basic numeric,textual and statistical operations on input textual data files.</p>
<div class="subheading"><a href="DNAnexus.html">DNAnexus</a> (0.352.1) </div><p class="appDesc">DNAnexus is a cloud-based commercial solution for next-generation sequence analysis and visualization. It has a command-line interface (CLI) which can be used to log in to the DNAnexus platform, upload and navigate data, and launch analyses. </p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/DNAnexus.html">dxda</a> (0.6.0) </div><p class="appDesc">CLI tool to manage the download of large quantities of files from DNAnexus</p>
<div class="subheading"><a href="edirect.html">EDirect</a> (20.5.20230926) </div><p class="appDesc">Entrez Direct (EDirect) is an advanced method for accessing the NCBI's set of interconnected databases (publication, sequence, structure, gene, variation, expression, etc.) from a UNIX terminal window.</p>
<div class="subheading"><a href="Editors.html">Editors</a></div><p class="appDesc">Text Editors available on the systems</p>
<div class="subheading"><a href="gdc-client.html">gdc-client</a> (1.6.1) </div><p class="appDesc">The GDC Data Transfer Tool provides an optimized method of transferring data to and from the GDC, and enables resumption of interrupted transfers.</p>
<div class="subheading"><a href="https://github.com/prasmussen/gdrive">gdrive</a> (2.1.0) </div><p class="appDesc">gdrive is a command line utility for interacting with Google Drive.</p>
<div class="subheading"><a href="https://www.ghostscript.com/">Ghostscript</a> (9.22) </div><p class="appDesc">Ghostscript is an interpreter for the PostScript language and for PDF.</p>
<div class="subheading"><a href="https://hpc.nih.gov/docs/globus/globus_cron.php#cron">globus-timer-cli</a> (0.2.9) </div><p class="appDesc">Command-line client for setting up and monitoring recurring or scheduled Globus transfers</p>
<div class="subheading">gnuplot (5.4.0) </div><p class="appDesc">Gnuplot is a portable command-line driven graphing utility  to visualize mathematical functions and data interactively,  and can support many non-interactive uses such as web scripting. <br>Type 'gnuplot' to run, or 'module avail gnuplot' to see other available versions.</p>
<div class="subheading">google-cloud-sdk (397.0.0) </div><p class="appDesc">Google Cloud SDK is a set of tools that you can use to manage resources and applications hosted on Google Cloud Platform. These include the gcloud, gsutil, and bq command line tools. See docs at <a href="https://cloud.google.com/sdk/docs/how-to">https://cloud.google.com/sdk/docs/how-to</a>. 
<br>Type 'module load  google-cloud-sdk' to use on Biowulf. </p>
<div class="subheading">Grace (5.1.25) </div><p class="appDesc"><A href="http://plasma-gate.weizmann.ac.il/Grace/">Grace</a> is a WYSIWYG 2D plotting tool for the X-Window system. It is a successor to Xmgr. 
<br>Type 'module load grace', then 'xmgrace' or 'gracebat' to run.</p>
<div class="subheading"><a href="https://graphviz.gitlab.io/">graphviz</a> (2.40) </div><p class="appDesc"><a href="https://graphviz.gitlab.io/">Graphviz</a> is open source graph visualization software. Graph visualization is a way of representing structural information as diagrams of abstract graphs and networks. It has important applications in networking, bioinformatics,  software engineering, database and web design, machine learning, and in visual interfaces for other technical domains. </p>
<div class="subheading">groff (1.22.3) </div><p class="appDesc">Groff (GNU troff) is a typesetting system that reads plain text mixed with formatting commands and produces formatted output. Output may be PostScript or PDF, html, or ASCII/UTF8 for display at the terminal. Formatting commands may be either low-level typesetting requests (“primitives”) or macros from a supplied set. Users may also write their own macros. All three may be combined. </p>
<div class="subheading"><a href="https://github.com/NanoComp/h5utils">h5utils</a> (1.13.1) </div><p class="appDesc">h5utils is a set of utilities for visualization and conversion of scientific data in the free, portable HDF5 format. Type 'module load h5utils' to access the executables (e.g. h5topng)

</p>
<div class="subheading">ImageMagick (7.1.0) </div><p class="appDesc"><a href='http://www.imagemagick.org/'>ImageMagick</a> is a software suite to create, edit, compose, or convert bitmap images. It can read and write images in a variety of formats. <br>The ImageMagick tools are available by default (e.g. type 'convert') or use 'module load ImageMagick' to load the latest versions. </p>
<div class="subheading">inkscape (1.3) </div><p class="appDesc">Inkscape is a Free and open source vector graphics editor</p>
<div class="subheading"><a href="JAX.html">JAX</a> (0.3.7) </div><p class="appDesc">JAX is a python library that brings Autograd and XLA (Accelerated Linear Algebra)  together
for high-performance machine learning research.
JAX uses XLA to compile and run your NumPy programs on GPUs.
Compilation happens under the hood by default, with library calls getting just-in-time compiled and executed. But JAX also lets you just-in-time compile your own Python functions into XLA-optimized kernels using a one-function API, jit</p>
<div class="subheading"><a href="https://github.com/jpmens/jo/blob/master/jo.md">jo</a> (1.6) </div><p class="appDesc">A small utility to create JSON objects from command line arguments.</p>
<div class="subheading"><a href="https://stedolan.github.io/jq/manual">jq</a> (1.6) </div><p class="appDesc">Command line json processor</p>
<div class="subheading"><a href="jupyter.html">jupyter</a> (5.0.0) </div><p class="appDesc">Jupyter is an open-source web application that allows you to create and share documents that contain live code, equations, visualizations and explanatory text. Uses include: data cleaning and transformation, numerical simulation, statistical modeling, machine learning and much more.</p>
<div class="subheading"><a href="https://github.com/attractivechaos/k8">k8</a> (0.2.5) </div><p class="appDesc">javascript shell based on v8, but with fasta/fastq parser and the ability to open files on the filesystem</p>
<div class="subheading"><a href="kronatools.html">kronatools</a> (2.8) </div><p class="appDesc">Krona allows hierarchical data to be explored with zooming, multi-layered pie charts. Krona charts can be created using an Excel template or KronaTools, which includes support for several bioinformatics tools and raw data formats. The interactive charts are self-contained and can be viewed with any modern web browser.</p>
<div class="subheading"><a href="https://github.com/kjn/lbzip2">lbzip2</a> (2.5) </div><p class="appDesc">lbzip2 is a multi-threaded compression utility with support for bzip2 compressed file format.</p>
<div class="subheading"><a href="longshot.html">longshot</a> (0.4.5) </div><p class="appDesc">Longshot is a variant calling tool for diploid genomes using long error prone reads such as Pacific Biosciences (PacBio) SMRT and Oxford Nanopore Technologies (ONT). It takes as input an aligned BAM file and outputs a phased VCF file with variants and haplotype information. It can also output haplotype-separated BAM files that can be used for downstream analysis. Currently, it only calls single nucleotide variants (SNVs).</p>
<div class="subheading"><a href="https://midnight-commander.org">mc</a> (4.8.22) </div><p class="appDesc">GNU Midnight Commander is a visual file manager, with a feature rich full-screen text mode application that allows you to copy, move and delete files and whole directory trees, search for files and run commands in the subshell. Type <b><tt>module load mc</tt></b> and then the command <b><tt>mc</b></tt> to get started.</p>
<div class="subheading"><a href="megit.html">megit</a> (0.4.0) </div><p class="appDesc">Minimal Eclipe GIT. A graphical front end for git based off the eclipse IDE.</p>
<div class="subheading"><a href="MySQL.html">MySQL</a> (8.0.34) </div><p class="appDesc">MySQL is an open-source relational database management system.</p>
<div class="subheading"><a href="nda-tools.html">nda-tools</a> (0.2.12,0.2.16,0.2.25) </div><p class="appDesc">In order to submit data to the National Institute of Mental Health Data Archives (NDA),
users must validate their data to ensure it complies with the required format.
This is done using the NDA validation tool, vtcmd.                                                                Additionally, users can package and download data from NDA as well, using the downloadcmd tool.</p>
<div class="subheading"><a href="https://neovim.io/">neovim</a> (0.9.2) </div><p class="appDesc">Neovim is a refactor, and sometimes redactor, in the tradition of Vim (which itself derives from Stevie). It is not a rewrite but a continuation and extension of Vim.</p>
<div class="subheading"><a href="http://netpbm.sourceforge.net/doc/">netpbm</a> (10.86.33) </div><p class="appDesc">Netpbm is a toolkit for manipulation of graphic images, including conversion of images between a variety of different formats. There are over 300 separate tools in the package including converters for about 100 graphics formats. Examples of the sort of image manipulation we're talking about are: Shrinking an image by 10%; Cutting the top half off of an image; Making a mirror image; Creating a sequence of images that fade from one image to another.</p>
<div class="subheading"><a href="https://nvchecker.readthedocs.io/en/latest/">nvchecker</a> (2.9) </div><p class="appDesc">nvchecker (short for new version checker) is for checking if a new version of some software has been released.</p>
<div class="subheading"><a href="/docs/transfer.html">OpenNeuro_cli</a> (4.14.1) </div><p class="appDesc">This command-line tool allows you to upload and download OpenNeuro.org datasets without a browser.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/parallel.html">parallel</a> (20231022) </div><p class="appDesc"><a href="http://www.gnu.org/software/parallel/">GNU parallel</a> is a shell tool for executing jobs in parallel using one or more computers.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/paraview.html">paraview</a> (5.11.1) </div><p class="appDesc">ParaView is an open-source, multi-platform data analysis and visualization application.</p>
<div class="subheading"><a href="http://nixos.org/patchelf.html">patchelf</a> (0.17) </div><p class="appDesc">patchelf is a small utility to modify the dynamic linker and RPATH of ELF executables.</p>
<div class="subheading"><a href="patric.html">patric</a> (1.035) </div><p class="appDesc">PATRIC is an integration of different types of data and software tools that support research on bacterial pathogens.  For users that wish command-line access to PATRIC there are the P3-scripts. They are intended to run on locally, going over the network to access the services provided by PATRIC.</p>
<div class="subheading"><a href="pdf2svg.html">pdf2svg</a> (0.2.3) </div><p class="appDesc">A simple PDF to SVG converter using the Poppler and Cairo libraries.</p>
<div class="subheading">petsc (3.19.1) </div><p class="appDesc">PETSc (Portable, Extensible Toolkit for Scientific Computation) is a suite of data structures and routines for the scalable (parallel) solution of scientific applications modeled by partial differential equations. It employs the MPI standard for all message-passing communication.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/pigz.html">pigz</a> (2.7) </div><p class="appDesc">pigz (parallel implementation of gzip) is a fully functional replacement for gzip that exploits multiple processors and multiple cores to the hilt when compressing data. </p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/pilon.html">pilon</a> (1.24) </div><p class="appDesc">Pilon is a software tool which can be used to:
1) Automatically improve draft assemblies
2) Find variation among strains, including large event detection</p>
<div class="subheading"><a href="https://www.postgresql.org/">postgresql</a> (15.2) </div><p class="appDesc">PostgreSQL is a powerful, open source object-relational database system with over 35 years of active development that has earned it a strong reputation for reliability, feature robustness, and performance.</p>
<div class="subheading"><a href="POVRay/index.html">POVRay</a> (3.7) </div><p class="appDesc">POVRAY (Persistence of Vision RAYtracer) is a high-quality tool for creating three-dimensional graphics. Raytraced images are publication-quality and 'photo-realistic', but are computationally expensive so that large images can take many hours to create.</p>
<div class="subheading">pre-commit (3.5.0) </div><p class="appDesc">A framework for managing and maintaining multi-language pre-commit hooks</p>
<div class="subheading"><a href="pyega3.html">pyega3</a> (5.0.2) </div><p class="appDesc">A download client for the European Genome-phenome Archive (EGA). The EGA is designed to be a repository for all types of sequence and genotype experiments, including case-control, population, and family studies.</p>
<div class="subheading"><a href="https://qpdf.readthedocs.io/en/stable/index.html">qpdf</a> (1.11.1) </div><p class="appDesc">QPDF is a command-line tool and C++ library that performs content-preserving transformations on PDF files</p>
<div class="subheading"><a href="https://quarto.org/">quarto</a> (1.3.450) </div><p class="appDesc">Quarto is an open-source scientific and technical publishing system built on Pandoc</p>
<div class="subheading"><a href="rclone.html">rclone</a> (1.62.2) </div><p class="appDesc">Rclone is a utility for synchronizing directories on a file-based storage system (e.g. /home or /data) with an object store such as Amazon S3. It uses the S3 protocol, and it can be used with the HPC object storage system.</p>
<div class="subheading"><a href="https://rdfind.pauldreik.se/">rdfind</a> (1.6.0) </div><p class="appDesc">rdfind is a program that finds duplicate files. It is useful for compressing backup directories or just finding duplicate files. It compares files based on their content, NOT on their file names.  After typing <tt><b>module load rdfind</b></tt>, type <tt><b>man rdfind</b></tt> for more information.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/rfmix.html">rfmix</a> (2.0) </div><p class="appDesc">Local Ancestry and Admixture Inference</p>
<div class="subheading"><a href="rstudio-server.html">rstudio-server</a> (2023.06.0-421) </div><p class="appDesc">RStudio Server is a web-based R IDE similar to RStudio Desktop.</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/sb_cli.html">sb_cli</a> (0.25.0) </div><p class="appDesc">Use the Seven Bridges Command Line Interface (SB CLI) to programmatically access and automate your interaction with the Platform via the API. The CLI is called by a simple command: sb.</p>
<div class="subheading"><a href="https://www.gnu.org/software/screen/">screen</a> (4.9.1) </div><p class="appDesc">Screen is a full-screen window manager that multiplexes a physical terminal between several processes, typically interactive shells.</p>
<div class="subheading"><a href="singularity.html">singularity</a> (4.0.1) </div><p class="appDesc">Singularity is a container platform focused on supporting ``Mobility of Compute``. It allows users to emulate, and share custom Linux environments allowing for the creation of self-contained development stacks.   </p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/snp-sites.html">snp-sites</a> (2.4.1) </div><p class="appDesc">Rapidly extracts SNPs from a multi-FASTA alignment</p>
<div class="subheading"><a href="spark.html">spark</a> (3.2.2) </div><p class="appDesc">Apache Spark is a fast and general engine for large-scale data processing. It is commonly used as an in-memory alternative to Hadoop MapReduce.</p>
<div class="subheading"><a href="sqlite.html">SQLite</a> (3.38.5) </div><p class="appDesc"><a href='http://www.sqlite.org/'>SQLite</a> is a software library that implements a self-contained, serverless, zero-configuration, transactional SQL database engine.</p>
<div class="subheading"><a href="stow.html">stow</a> (2.3.1) </div><p class="appDesc">GNU Stow is a symlink farm manager which takes distinct packages of software and/or data located in separate directories on the filesystem, and makes them appear to be installed in the same place.</p>
<div class="subheading"><a href="swarm.html">swarm</a> (23.2.1) </div><p class="appDesc">Swarm is a script designed to simplify submitting a group of commands to the Biowulf cluster.</p>
<div class="subheading"><a href="http://www.swig.org/">swig</a> (4.1.1) </div><p class="appDesc">SWIG is a software development tool that connects programs written in C and C++ with a variety of high-level programming languages.</p>
<div class="subheading"><a href="synapseclient.html">synapseclient</a> (3.1.1) </div><p class="appDesc">The synapseclient package provides an interface to Synapse, a collaborative workspace for reproducible, data intensive research projects</p>
<div class="subheading">Tex (2022) </div><p class="appDesc">Tex + Latex + associated packages for high-quality text formatting. </p>
<div class="subheading">texinfo (6.8) </div><p class="appDesc">Texinfo is the official documentation format of the GNU project</p>
<div class="subheading"><a href="https://github.com/tmux">tmux</a> (3.3a) </div><p class="appDesc">tmux is a terminal multiplexer.
<br>Type 'module load tmux' to load the module, then 'tmux --help'</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/ukbb.html">UKBB</a> (0.1) </div><p class="appDesc">UK Biobank tools for downloading and processing UKBB data. Type 'module load ukbb' to access them. </p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/vartrix.html">vartrix</a> (1.1.22) </div><p class="appDesc">VarTrix is a software tool for extracting single cell variant information from 10x Genomics single cell data.</p>
<div class="subheading"><a href="vcf2db.html">vcf2db</a> (2020.09.14) </div><p class="appDesc">vcf2db creates a gemini-compatible database from a VCF.</p>
<div class="subheading"><a href="https://www.vim.org/docs.php">vim</a> (9.0) </div><p class="appDesc">a text editor that is upwards compatible to Vi. It can be used to edit all kinds of plain text. It is especially useful for editing programs with syntactical coloring. </p>
<div class="subheading"><a href="https://visidata.org">visidata</a> (2.11.1) </div><p class="appDesc">VisiData is an interactive multitool for tabular data</p>
<div class="subheading"><a href="https://hpc.nih.gov/apps/whatshap.html">whatshap</a> (1.1) </div><p class="appDesc">WhatsHap is a software for phasing genomic variants using DNA sequencing reads, also called read-based phasing or haplotype assembly. It is especially suitable for long reads, but works also well with short reads.</p>
<div class="subheading"><a href="https://github.com/asciimoo/wuzz">wuzz</a> (0.5.0) </div><p class="appDesc">Interactive cli tool for HTTP inspection</p>
<div class="subheading"><a href="https://github.com/astrand/xclip">xclip</a> (0.13) </div><p class="appDesc">xclip is a command line utility that is designed to run on any system with an X11 implementation. It provides an interface to X selections ("the clipboard") from the command line. It can read data from standard in or a file and place it in an X selection for pasting into other X applications. xclip can also print an X selection to standard out, which can then be redirected to a file or another program.</p>
<div class="subheading"><a href="https://www.xpdfreader.com/">xpdf</a> (4.04) </div><p class="appDesc">Xpdf is a free PDF viewer and toolkit, including a text extractor, image converter, HTML converter, and more. Most of the tools are available as open source.</p>
<div class="subheading"><a href="https://github.com/facebook/zstd/wiki">zstd</a> (1.5.5) </div><p class="appDesc">Zstandard, or zstd as short version, is a fast lossless compression algorithm, targeting real-time compression scenarios at zlib-level and better compression ratios.</p>
<div class="heading"><a name="workflow" id="workflow">Workflow Managers</a></div>
<a href="/apps/" style="font-size:12px">back to top</a><br />
<p></p>
<div class="subheading">caper (2.1.3) </div><p class="appDesc">A Python-based wrapper for the Cromwell pipeline system used by ENCODE pipelines. Includes the croo and qc2tsv commands for parsing Cromwell and quality control output.</p>
<div class="subheading"><a href="cromwell.html">cromwell</a> (84) </div><p class="appDesc">A Workflow Management System geared towards scientific workflows. </p>
<div class="subheading"><a href="nextflow.html">nextflow</a> (23.10.0) </div><p class="appDesc">Data-driven computational pipelines</p>
<div class="subheading"><a href="snakemake.html">snakemake</a> (7.32.3) </div><p class="appDesc">Snakemake aims to reduce the complexity of creating workflows by providing a fast and comfortable execution environment, together with a clean and modern domain specific specification language (DSL) in python style. It is well suited for bioinformatic workflows.</p>

<!-- End content area - do not edit below this line -->

<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>

