<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'iqtree on Biowulf';</script>
<div class="title">iqtree on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
Efficient software for phylogenomic inference
<blockquote>
The IQ-TREE software was created as the successor of IQPNNI and TREE-PUZZLE (thus the name IQ-TREE).
IQ-TREE was motivated by the rapid accumulation of phylogenomic data, leading to a need for efficient
phylogenomic software that can handle a large amount of data and provide more complex models of
sequence evolution. To this end, IQ-TREE can utilize multicore computers and distributed parallel
computing to speed up the analysis. IQ-TREE automatically performs checkpointing to resume an
interrupted analysis.
</blockquote>
</p>

<h3>References:</h3>
<ul>

  <li>
    B.Q. Minh, H.A. Schmidt, O. Chernomor, D. Schrempf, M.D. Woodhams, A. von Haeseler, R. Lanfear (2020)
    <a href="https://doi.org/10.1093/molbev/msaa015"><b><u>IQ-TREE 2: New models and efficient methods for
    phylogenetic inference in the genomic era.</u></b></a>
    <em>Mol. Biol. Evol., 37:1530-1534.</em>
  </li>
  <li>
    L.-T. Nguyen, H.A. Schmidt, A. von Haeseler, B.Q. Minh (2015)
    <a href="https://doi.org/10.1093/molbev/msu300"><b><u>IQ-TREE: A fast and effective stochastic algorithm
    for estimating maximum likelihood phylogenies</u></b></a>
    <em>Mol. Biol. Evol., 32:268-274.</em>
  </li>

</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="http://www.iqtree.org/">iqtree Main Site</a></li>
<li><a href="https://github.com/iqtree/iqtree2">iqtree Github</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>iqtree</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
<li>Multithreaded application
<li>Example files in <tt>$IQTREE_TEST_DATA</tt>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=6g -c 8</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load iqtree</b>

[user@cn3144 ~]$ <b>cp $IQTREE_TEST_DATA/* .</b>

[user@cn3144 ~]$ <b>iqtree2 -s example.phy -nt $SLURM_CPUS_PER_TASK</b>
IQ-TREE multicore version 2.1.2 COVID-edition for Linux 64-bit built Oct 22 2020
Developed by Bui Quang Minh, James Barbetti, Nguyen Lam Tung,
Olga Chernomor, Heiko Schmidt, Dominik Schrempf, Michael Woodhams.

Host:    cn0852 (AVX512, FMA3, 377 GB RAM)
Command: iqtree2 -s example.phy -nt 8
Seed:    830607 (Using SPRNG - Scalable Parallel Random Number Generator)
Time:    Mon Oct 26 10:33:27 2020
Kernel:  AVX+FMA - 8 threads (8 CPU cores detected)

Reading alignment file example.phy ... Phylip format detected
Alignment most likely contains DNA/RNA sequences
Alignment has 17 sequences with 1998 columns, 1152 distinct patterns
1009 parsimony-informative, 303 singleton sites, 686 constant sites
[...]
Total number of iterations: 102
CPU time used for tree search: 34.761 sec (0h:0m:34s)
Wall-clock time used for tree search: 4.757 sec (0h:0m:4s)
Total CPU time used: 44.325 sec (0h:0m:44s)
Total wall-clock time used: 6.151 sec (0h:0m:6s)

Analysis results written to:
  IQ-TREE report:                example.phy.iqtree
  Maximum-likelihood tree:       example.phy.treefile
  Likelihood distances:          example.phy.mldist
  Screen log file:               example.phy.log

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. iqtree.sh). For example:</p>

<pre class="term">
#!/bin/bash
module load iqtree
iqtree2 -s example.phy -nt $SLURM_CPUS_PER_TASK
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=8 --mem=6g iqtree.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. iqtree.swarm). For example:</p>

<pre class="term">
iqtree2 -s example1.phy -nt $SLURM_CPUS_PER_TASK
iqtree2 -s example2.phy -nt $SLURM_CPUS_PER_TASK
iqtree2 -s example3.phy -nt $SLURM_CPUS_PER_TASK
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f iqtree.swarm -g 6 -t 8 --module iqtree</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module iqtree</tt> <td>Loads the iqtree module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
