<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'iva on Biowulf';</script>
<div class="title">iva on Biowulf</div>

<table width=25% align=right>
    <tr>
        <td>
            <div class="toc">
                <div class="tocHeading">Quick Links</div>
                <div class="tocItem"><A href="#doc">Documentation</a></div>
                <div class="tocItem"><a href="#notes">Notes</a></div>
                <div class="tocItem"><a href="#int">Interactive job </a></div>
                <div class="tocItem"><a href="#sbatch">Batch job </a></div>
                <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
            </div>
        </td>
    </tr>
</table>

<p> Integrated virus assembler (IVA) is a de novo assembler designed for virus
genomes without repeat sequences, using Illumina read pairs from mixed
populations at extremely high and variable depth.</p>

<h3>References:</h3>
<ul>
    <li>Martin Hunt <em>et al.</em>. <em>IVA: accurate de novo assembly of 
        RNA virus genomes</em>. Bioinformatics 2015(31): 2374-2376.
    <a href="http://www.ncbi.nlm.nih.gov/pubmed/25725497">Pubmed</a>&nbsp;|&nbsp;
    <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4495290/">PMC</a>&nbsp;|&nbsp;
    <a href="http://bioinformatics.oxfordjournals.org/content/31/14/2374">Journal</a>
    </li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="http://sanger-pathogens.github.io/iva/">Home page</a></li>
    <li><a href="https://github.com/sanger-pathogens/iva/wiki">Wiki</a></li>
    <li><a href="https://github.com/sanger-pathogens/iva">GitHub</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: iva (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>iva is a multithreaded application. Please match the number of allocated CPUs to the
         number of threads</li>
    <li>some iva versions mis-report their version</li>
    <li>some samtools errors may be spurious</li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>

<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the built in test data
set</p>

<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=2 --mem=6g</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144]$ <b>module load iva</b>
[user@cn3144]$ <b>iva --test --threads=2 --trimmomatic=$TRIMMOJAR test</b>
Running iva in test mode...
Copied input test files into here: PREFIX/test
Current working directory: PREFIX/test
Running iva on the test data with the command:
/usr/local/lib/python3.6/dist-packages/iva-1.0.9-py3.6.egg/EGG-INFO/scripts/iva --threads 2 \
    --trimmomatic /Trimmomatic-0.38/trimmomatic-0.38.jar --pcr_primers hiv_pcr_primers.fa -f reads_1.fq.gz -r reads_2.fq.gz iva.out
Finished running iva
Looks OK. Final output contigs file is: PREFIX/test/iva.out/contigs.fasta
[user@cn3144]$ <b>tree test</b>
test
|-- [user    672]  hiv_pcr_primers.fa
|-- [user   9.7K]  iva_contigs_no_trimmomatic.fasta
|-- [user   9.0K]  iva_contigs_with_trimmomatic.fasta
|-- [user   4.0K]  iva.out
|   |-- [user   3.1K]  adapters.fasta
|   |-- [user   9.8K]  contigs.fasta
|   `-- [user    369]  info.txt
|-- [user   3.6M]  reads_1.fq.gz
|-- [user   4.4M]  reads_2.fq.gz
`-- [user   9.0K]  reference.fasta
</pre>


<p>Run IVA with trimmomatic on the data copied to test by the automated 
test</p>

<pre class="term">
[user@cn3144]$ <b>cd test</b>
[user@cn3144]$ <b>iva --threads 2 --pcr_primers hiv_pcr_primers.fa \
  -f reads_1.fq.gz -r reads_2.fq.gz \
  --trimmomatic $TRIMMOJAR \
  iva.out2 </b> 

[user@cn3144]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. iva.sh), which uses the input file 'iva.in'. For example:</p>

<pre class="term">
#! /bin/bash

function fail {
  echo &gt;2 "$@"
  exit 1
}

module load iva/1.0.11 || fail "could not load iva module"
iva --threads $SLURM_CPUS_PER_TASK \
  -f read1.fq.gz -r read2.fq.gz \
  --trimmomatic=$TRIMMOJAR \
  sample.iva || fail "iva return non-zero exit status"
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=6 iva.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. iva.swarm). For example:</p>

<pre class="term">
iva -r sample1_r1.fq.gz -f sample1_r2.fq.gz --trimmomatic=$TRIMMOJAR --threads $SLURM_CPUS_PER_TASK sample1_out
iva -r sample2_r1.fq.gz -f sample2_r2.fq.gz --trimmomatic=$TRIMMOJAR --threads $SLURM_CPUS_PER_TASK sample2_out
iva -r sample3_r1.fq.gz -f sample3_r2.fq.gz --trimmomatic=$TRIMMOJAR --threads $SLURM_CPUS_PER_TASK sample3_out
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f iva.swarm -g 10 -t 4 --module iva/1.0.3</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>-t # <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td>--module iva <td>Loads the iva module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
