<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'iVar: iVar: a tool for viral amplicon-based sequencing.';</script>
<div class="title">iVar: iVar: a tool for viral amplicon-based sequencing.</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
iVar is a computational package that contains functions broadly useful for viral amplicon-based sequencing. Additional tools for metagenomic sequencing are actively being incorporated into iVar. While each of these functions can be accomplished using existing tools, iVar contains an intersection of functionality from multiple tools that are required to call iSNVs and consensus sequences from viral sequencing data across multiple replicates.
</p>

<h3>References:</h3>
<ul>
<li>Nathan D. Grubaugh, Karthik Gangavarapu, Joshua Quick, Nathaniel L. Matteson, Jaqueline Goes De Jesus, Bradley J. Main, Amanda L. Tan, Lauren M. Paul, Doug E. Brackney, Saran Grewal, Nikos Gurfield, Koen K. A. Van Rompay, Sharon Isern, Scott F. Michael, Lark L. Coffey, Nicholas J. Loman and Kristian G. Andersen  <br />
<i> An amplicon-based sequencing framework for accurately measuring intrahost virus diversity using PrimalSeq and iVar.</i><br />
<a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1618-7">Genome Biology </a> (2019), <b>5</b>(8). https://doi.org/10.1186/s13059-018-1618-7 <br/>
</li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/andersen-lab/ivar">iVar Github page</a></li>
<li><a href="https://andersen-lab.github.io/ivar/html/manualpage.html">iVar manual</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>iVar</tt> (see <a href="https://hpc.nih.gov/apps/modules.html">the modules page</a> for more information)
<li>Unusual environment variables set 
  <ul>
    <li><b>IVAR_HOME</b>&nbsp; IVAR installation directory</li>
    <li><b>IVAR_BIN </b>&nbsp; &nbsp; &nbsp; IVAR executable directory</li>
    <li><b>IVAR_SRC </b>&nbsp; &nbsp; a folder containing the source code</li>
    <li><b>IVAR_DATA</b>&nbsp; &nbsp; a folder containing sample data</li>
    <li><b>IVAR_HMM </b>&nbsp; &nbsp; a folder containing sample Hidden Markov Models</li>
  </ul>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive </b>
[user@cn4471 ~]$ <b>module load ivar </b>
[+] Loading samtools 1.9  ...
[+] Loading ivar 1.3.1  ...
</pre>
Copy sample data to your current folder:
<pre class="term">
[user@cn4471 ~]$ <b>cp -r $IVAR_DATA/* .</b>
[user@cn4471 ~]$ <b>cp test_amplicon.sorted.bam test.bam </b>
</pre>
Preprocess the data with samtools:
[user@cn4471 ~]$ <b>samtools sort -o test.sorted.bam test.bam </b>
[user@cn4471 ~]$ <b>samtools index test.sorted.bam</b>
</pre>
iVar can be used as follows:
<pre class="term">
[user@cn4471 ~]$<b> ivar -h</b>
Usage:  ivar [command &lt;trim|callvariants|filtervariants|consensus|getmasked|removereads|version|help&gt;]

        Command       Description
           trim       Trim reads in aligned BAM file
       variants       Call variants from aligned BAM file
 filtervariants       Filter variants across replicates
      consensus       Call consensus from aligned BAM file
      getmasked       Detect primer mismatches and get primer indices for the amplicon to be masked
    removereads       Remove reads from trimmed BAM file
        version       Show version information

To view detailed usage for each command, type <b>ivar &lt;command&gt;</b>, for example:
[user@cn4471 ~]$<b> ivar trim -h </b>
Usage: ivar trim -i &lt;input.bam&gt; -b &lt;primers.bed&gt; -p &lt;prefix&gt; [-m &lt;min-length&gt;] [-q &lt;min-quality&gt;] [-s &lt;sliding-window-width&gt;]
Input Options    Description
           -i    (Required) Sorted bam file, with aligned reads, to trim primers and quality
           -b    (Required) BED file with primer sequences and positions
           -m    Minimum length of read to retain after trimming (Default: 30)
           -q    Minimum quality threshold for sliding window to pass (Default: 20)
           -s    Width of sliding window (Default: 4)
           -e    Include reads with no primers. By default, reads with no primers are excluded
Output Options   Description
           -p    (Required) Prefix for the output BAM file
</pre>
Perform the trimming on sample data:
<pre class="term">
[user@cn4471 ~]$<b> cp test_isize.bed test_primers.bed </b>
[user@cn4471 ~]$<b> ivar trim -b test_primers.bed -p test.trimmed -i test.bam -q 15 -m 50 -s 4 </b>
Found 218 primers in BED file
Amplicons detected:

Number of references in file: 1
NC_045512.2
Using Region: NC_045512.2

Found 8 mapped reads
Found 0 unmapped reads
Sorted By Coordinate
-------
Processed 10% reads ...
Processed 20% reads ...
Processed 30% reads ...
Processed 40% reads ...

-------
Results:
Primer Name     Read Count
nCoV-2019_1_LEFT        0
nCoV-2019_1_RIGHT       0
nCoV-2019_2_LEFT        0
nCoV-2019_2_RIGHT       0
nCoV-2019_3_LEFT        0
nCoV-2019_3_RIGHT       0
nCoV-2019_4_LEFT        0
nCoV-2019_4_RIGHT       0
...
nCoV-2019_96_LEFT       0
nCoV-2019_96_RIGHT      0
nCoV-2019_97_LEFT       0
nCoV-2019_97_RIGHT      0
nCoV-2019_98_LEFT       0
nCoV-2019_98_RIGHT      0

Trimmed primers from 100% (8) of reads.
0% (0) of reads were quality trimmed below the minimum length of 50 bp and were not written to file.
0% (0) of reads that started outside of primer regions were not written to file
50% (4) of reads had their insert size smaller than their read length
</pre>
End the interactive session:
<pre class="term">
[user@cn4471 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
