<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'KAT on Biowulf';</script>
<div class="title">KAT on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
    KAT (K-mer Analysis Toolkit) provides a suite of tools that, through the use of k-mer counts, help the user address or identify issues such as determining sequencing completeness for assembly, assessing sequencing bias, identifying contaminants, validating genomic assemblies and filtering content. KAT is geared primarily to work with high-coverage genomic reads from Illumina devices, although can work with any fasta or fastq sequence file.
</p>

<h3>References:</h3>
<ul>

    <li>
	Daniel Mapleson, Gonzalo Garcia Accinelli, George Kettleborough, Jonathan Wright, and Bernardo J. Clavijo.
	KAT: A K-mer Analysis Toolkit to quality control NGS datasets and genome assemblies.
	Bioinformatics, 2016. doi: 
	<a href="https://doi.org/10.1093/bioinformatics/btw663"><b><u>10.1093/bioinformatics/btw663</u></b></a>
  </li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="https://www.earlham.ac.uk/kat-tools">KAT Main Site</a></li>
    <li><a href="https://kat.readthedocs.io/en/latest/">Documentation</a></li>
    <li><a href="https://github.com/TGAC/KAT">KAT on GitHub</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>

<!-- 
     <p class="NXblock">This application requires a <a href="/docs/connect.html#nx">graphical connection using NX</a></p>
-->
<ul>
    <li>Module Name: <tt>kat</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
</ul>
<!-- 
	<li>Multithreaded/singlethreaded/MPI...
	<li>This application produces HTML reports. You can use <a href="/docs/hpcdrive.html">hpcdrive</a> to view these reports on your local workstation.</li>
<li>Environment variables set <!--for ones users should be aware of ->
  <ul>
    <li><tt>KAT_HOME</tt></li>
  </ul>
  <li>Example files in <tt>???</tt>
      <li>Reference data in <tt>/fdb/KAT/</tt>
  </ul>
  -->
  <P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load kat</b>

[user@cn3144 ~]$ <b>kat hist /fdb/app_testdata/fastq/H_sapiens/hg100_1m_pe?.fq.gz </b>

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. KAT.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load kat
kat hist /fdb/app_testdata/fastq/H_sapiens/hg100_1m_pe?.fq.gz
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] KAT.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. kat.swarm). For example:</p>

<pre class="term">
kat hist sample1.fastq
kat hist sample2.fastq
kat hist sample3.fastq
kat hist sample4.fastq
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f kat.swarm [-g #] [-t #] --module kat</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module kat</tt> <td>Loads the KAT module for each subjob in the swarm 
</table>
















<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
