<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'KING on Biowulf';</script>
<div class="title">KING on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
KING is a toolset to explore genotype data from a genome-wide association study (GWAS) or a sequencing project. KING can be used to check family relationship and flag pedigree errors by estimating kinship coefficients and inferring IBD segments for all pairwise relationships. Unrelated pairs can be precisely separated from close relatives with no false positives, with accuracy up to 3rd- or 4th-degree (depending on array or WGS) for --related and --ibdseg analyses, and up to 2nd-degree for --kinship analysis.
</p>

<h3>References:</h3>
<ul>
<li>
  Manichaikul A, Mychaleckyj JC, Rich SS, Daly K, Sale M, Chen WM.
  <b><a href="https://www.ncbi.nlm.nih.gov/pubmed/20926424">Robust relationship inference in genome-wide association studies.</a></b>
  <u>Bioinformatics.</u> 2010 Nov 15;26(22):2867-73.
</li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="http://people.virginia.edu/~wc9c/KING/">KING Main Site</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>king</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
<li>Singlethreaded
<li>Environment variables set <!--for ones users should be aware of -->
  <ul>
    <li><tt>KING_EXAMPLES</tt> -- example files for KING</li>
  </ul>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load king</b>

[user@cn3144 ~]$ <b>cp $KING_EXAMPLES/* .</b>
[user@cn3144 ~]$ <b>king -b ex.bed --related</b>
[user@cn3144 ~]$ <b>king -b ex.bed --fam ex.fam --bim ex.bim --related</b>

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. king.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load king
king -b my.bed --duplicate --cpus $SLURM_CPUS_PER_TASK
king -b my.bed --related --cpus $SLURM_CPUS_PER_TASK
king -b my.bed --related --degree --cpus $SLURM_CPUS_PER_TASK
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] king.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. king.swarm). For example:</p>

<pre class="term">
king -b test1.bed --fam test1.fam --bim test1.bim --related
king -b test2.bed --fam test2.fam --bim test2.bim --related
king -b test3.bed --fam test3.fam --bim test3.bim --related
king -b test4.bed --fam test4.fam --bim test4.bim --related
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f king.swarm [-g #] [-t #] --module king</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module king</tt> <td>Loads the king module for each subjob in the swarm 
</table>

<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
