<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'kneaddata on Biowulf';</script>
<div class="title">kneaddata on Biowulf</div>

<table width=25% align=right>
    <tr>
        <td>
            <div class="toc">
                <div class="tocHeading">Quick Links</div>
                <div class="tocItem"><A href="#doc">Documentation</a></div>
                <div class="tocItem"><a href="#notes">Notes</a></div>
                <div class="tocItem"><a href="#int">Interactive job </a></div>
                <div class="tocItem"><a href="#sbatch">Batch job </a></div>
                <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
            </div>
        </td>
    </tr>
</table>

<p>
From the KneadData user manual:
</p>

<blockquote style="width: 70%"> 
    KneadData is a tool designed to perform quality control on
    metagenomic and metatranscriptomic sequencing data, especially data from
    microbiome experiments. In these experiments, samples are typically taken
    from a host in hopes of learning something about the microbial community on
    the host. However, sequencing data from such experiments will often contain
    a high ratio of host to bacterial reads. This tool aims to perform
    principled in silico separation of bacterial reads from these "contaminant"
    reads, be they from the host, from bacterial 16S sequences, or other
    user-defined sources. Additionally, KneadData can be used for other
    filtering tasks. For example, if one is trying to clean data derived from a
    human sequencing experiment, KneadData can be used to separate the human
    and the non-human reads.  
</blockquote>

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="https://huttenhower.sph.harvard.edu/kneaddata/">Manual</a> </li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: kneaddata (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>kneaddata is a multithreaded/multiprocess application. Make sure to match the number 
        of cpus requested with the number of threads.</li>
    <li>Example files in <code>$KNEADDATA_TEST_DATA</code></li>
    <li>Reference data in /fdb/kneaddata/</li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>

<p>Allocate an interactive session with <a href="/docs/userguide.html#int">sinteractive</a>
and use as shown below. In this case we will use test data that is 
an artificial mixture of 1M human exome reads and 1M environmental metagenomic 
reads. The 50% human reads is treated as an artificial contamination and removed:</p>

<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=12g --cpus-per-task=8 --gres=lscratch:10</b>
salloc.exe: Pending job allocation 33247354
salloc.exe: job 33247354 queued and waiting for resources
salloc.exe: job 33247354 has been allocated resources
salloc.exe: Granted job allocation 33247354
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job
srun: error: x11: no local DISPLAY defined, skipping

[user@nc3144]$ <b>module load kneaddata</b>
[user@nc3144]$ <b>cd /lscratch/${SLURM_JOB_ID}</b>
[user@nc3144]$ # copy test data
[user@nc3144]$ <b>cp $KNEADDATA_TEST_DATA/* .</b>
[user@nc3144]$ # the test data is 50% human, 50% environmental metagenomic data
[user@nc3144]$ # and the read names are labelled accordingly
[user@nc3144]$ <b>zcat test_R1.fastq.gz \
                  | awk '/@meta/ {m++} /@human/ {h++} END {printf("human: %i\nmeta:  %i\n", h, m)}'</b>
human: 1000000
meta:  1000000
[user@nc3144]$ <b># run kneaddata on the paired end test data</b>
[user@nc3144]$ <b>kneaddata -i test_R1.fastq.gz -i test_R2.fastq.gz \
                  --reference-db $KNEADDATA_DB/Homo_sapiens_Bowtie2_v0.1/Homo_sapiens \
                  --output-prefix test --output test_out \
                  -p 2 -t 4 --run-fastqc-end </b>
[...snip...]
Final output files created: 
/lscratch/33247583/test_out/test_paired_1.fastq
/lscratch/33247583/test_out/test_paired_2.fastq
/lscratch/33247583/test_out/test_unmatched_1.fastq
/lscratch/33247583/test_out/test_unmatched_2.fastq

[user@nc3144]$ # check the composition of human/metagenome reads in the cleand data
[user@nc3144]$ <b>cat test_out/test_paired_1.fastq \
                  | awk '/@meta/ {m++} /@human/ {h++} END {printf("human: %i\nmeta:  %i\n", h, m)}'</b>
human: 18641
meta:  951886
[user@nc3144]$ <b>exit</b>
[user@biowulf]$</span>
</pre>

<p>So the 50% artificial contamination with human reads was reduced to 2%.</p>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. kneaddata.sh) similar to the following example:</p>

<pre class="term">
#! /bin/bash

module load kneaddata/0.7.0 || exit 1
cd /lscratch/${SLURM_JOB_ID} || exit 1

if [[ ! -e test_R1.fastq.gz ]]; then
    cp $KNEADDATA_TEST_DATA/* .
fi
rm -rf test_out

kneaddata -i test_R1.fastq.gz -i test_R2.fastq.gz \
  --reference-db $KNEADDATA_DB/Homo_sapiens_Bowtie2_v0.1/Homo_sapiens \
  --output-prefix test --output test_out \
  -p 2 -t 4 --run-fastqc-end

cp -r test_out /data/$USER/important_project
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=8 --mem=12g --gres=lscratch:10 kneaddata.sh</pre>


<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. kneaddata.swarm). For example:</p>

<pre class="term">
kneaddata -i sample1_R1.fastq.gz -i sample1_R2.fastq.gz \
  --reference-db $KNEADDATA_DB/Homo_sapiens_Bowtie2_v0.1/Homo_sapiens \
  --output-prefix sample1 --output sample1_out -p 2 -t 4 --run-fastqc-end
kneaddata -i sample2_R1.fastq.gz -i sample2_R2.fastq.gz \
  --reference-db $KNEADDATA_DB/Homo_sapiens_Bowtie2_v0.1/Homo_sapiens \
  --output-prefix sample2 --output sample2_out -p 2 -t 4 --run-fastqc-end
kneaddata -i sample3_R1.fastq.gz -i sample2_R2.fastq.gz \
  --reference-db $KNEADDATA_DB/Homo_sapiens_Bowtie2_v0.1/Homo_sapiens \
  --output-prefix sample3 --output sample3_out -p 2 -t 4 --run-fastqc-end
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f kneaddata.swarm -g 12 -t 8 --module kneaddata/0.7.0</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>-t # <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td>--module kneaddata <td>Loads the kneaddata module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
