<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'kraken on Biowulf';</script>
<div class="title">kraken on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>Kraken is a system for assigning taxonomic labels to short DNA sequences,
usually obtained through metagenomic studies. It uses exact matches of kmers
against a reference database.</p>

<h3>References:</h3>
<ul>
    <li>Derrick E Wood and Steven L Salzberg.<em>Kraken: ultrafast metagenomic
        sequence classification using exact alignments.</em> Genome Biology
    15(2014), R46.
    <a href="http://www.ncbi.nlm.nih.gov/pubmed/24580807">Pubmed</a>&nbsp;|&nbsp;
    <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4053813/">PMC</a>&nbsp;|&nbsp;
    <a href="http://www.genomebiology.com/2014/15/3/R46">Journal</a>
    </li>
</ul>

<p class="alert" ;="" width="940px">
Do not run Kraken with the database in the default <code>/fdb</code> location or a shared <code>/data</code>
 directory. It <i>must</i> be run with the database in <code>/lscratch</code> or <code>/dev/shm</code>.
</p>

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="http://ccb.jhu.edu/software/kraken/">Home page</a></li>
    <li><a href="https://github.com/DerrickWood/kraken">GitHub</a></li>
    <li><a href="http://ccb.jhu.edu/software/kraken/MANUAL.html">Manual</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: kraken (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>kraken is a multithreaded application</li>
    <li>Example data can be found in <code>$KRAKEN_TEST_DATA</code></li>
    <li>Kraken databases can be found in <b><code>/fdb/kraken/</code></b> which is
    also the value of <code>$KRAKEN_DB_PATH</code>. The standard database
        contains genomes of archaea, bacteria, and viruses from NCBI.</li>
    <li>kraken has undergone a number of changes with version 2 (default) including a different database format.
      kraken2-compatible databases in <code>/fdb/kraken</code> end in <code>_kraken2</code>.</li>
    <li><span style="background-color: #ffff99">To run Kraken version 2 efficiently, 
      you should allocate at least as much memory as the size of the database and copy the database to
      <code>/lscratch</code>. Shrunken databases with
      subsets of kmers are provided to accomodate running with smaller memory requirements.</span> 
      The current standard Kraken 2 database is about 60GB in size.</li>
    <li><span style="background-color: #ffff99">Classification should <strong>not</strong> make direct use
    of the databases in <code>/fdb/kraken</code> since this can cause some severe file system strain.
    Database should either be copied to memory or lscratch.</span></li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. In
the following sample session we will use kraken to build a custom database and run a classification
with the custom data base. Shown below is the same example for kraken and kraken2. kraken2 differs from
kraken 1 in several important ways.</p>

<script>
  $(document).ready(function() {
      $('ul.tabs li').unbind('click');
      $('ul.tabs li').click(function(){
        var tab_id = $(this).attr('data-tab');

        $(this).parent().children().removeClass('current');
        $(this).parent().children().each(function () {
            $("#"+$(this).attr('data-tab')).removeClass('current');
        });

        $(this).addClass('current');
        $("#"+tab_id).addClass('current');
    });
  });
</script>

<div class="tabs-container">
    <ul class="tabs">
    <li data-tab="tab-1">Version 1</li>
    <li class="current" data-tab="tab-2">Version 2</li>
    </ul>
    <div id="tab-1" class="tab-content">

<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=24g --cpus-per-task=4 --gres=lscratch:100</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load kraken/1.1</b>

###
### build a custom kraken database with some 400+ genomes
###
[user@cn3144 ~]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144 ~]$ <b>mkdir fa</b>
[user@cn3144 ~]$ <b>cd fa</b>
[user@cn3144 ~]$ <b>wget --quiet --input-file $KRAKEN_TEST_DATA/genomes_for_custom_db.urls</b>
[user@cn3144 ~]$ <b>gunzip *.gz</b>
[user@cn3144 ~]$ <b>ls -lh | head -5</b>
total 1.3G
-rw-r--r-- 1 user group 3.9M May 22  2016 GCA_000006745.1_ASM674v1_genomic.fna
-rw-r--r-- 1 user group 2.1M May 26  2016 GCA_000006885.1_ASM688v1_genomic.fna
-rw-r--r-- 1 user group 2.0M May 27  2016 GCA_000007045.1_ASM704v1_genomic.fna
-rw-r--r-- 1 user group 5.3M Aug  4  2014 GCA_000007825.1_ASM782v1_genomic.fna

[user@cn3144 ~]$ <b>cd ..</b>
[user@cn3144 ~]$ <b>kraken-build --download-taxonomy --db custom_db</b>
[user@cn3144 ~]$ <b>for f in fa/*.fna; do
                       kraken-build --add-to-library $f --db custom_db
                    done</b>
Added "fa/GCA_000006745.1_ASM674v1_genomic.fna" to library (custom_db)
Added "fa/GCA_000006885.1_ASM688v1_genomic.fna" to library (custom_db)
[...snip...]
Added "fa/GCA_900129335.1_PP_HGAG_QV30_2SC_T4_genomic.fna" to library (custom_db)
Added "fa/GCA_900185995.1_BK1071_genomic.fna" to library (custom_db)

[user@cn3144 ~]$ <b>kraken-build --build --threads $SLURM_CPUS_PER_TASK --db custom_db</b>
Kraken build set to minimize disk writes.
Creating k-mer set (step 1 of 6)...
Found jellyfish v1.1.11
Hash size not specified, using '253231360'
K-mer set created. [6m11.633s]
Skipping step 2, no database reduction requested.
Sorting k-mer set (step 3 of 6)...
K-mer set sorted. [2m20.480s]
Skipping step 4, GI number to seqID map now obsolete.
Creating seqID to taxID map (step 5 of 6)...
721 sequences mapped to taxa. [3m48.166s]
Setting LCAs in database (step 6 of 6)...
Finished processing 721 sequences
Database LCAs set. [19m6.065s]
Database construction complete. [Total: 31m26.381s]

[user@cn3144 ~]$ <b>kraken-build --clean --db custom_db</b>
[user@cn3144 ~]$ <b>du -sh custom_db</b>
10G     custom_db/
[user@cn3144 ~]$ <b>tree custom_db</b>
custom_db/
|-- [user    14K]  accmap_file.tmp
|-- [user   8.0G]  database.idx
|-- [user   1.7G]  database.kdb
`-- [user   4.0K]  taxonomy
    |-- [user   142M]  names.dmp
    `-- [user   109M]  nodes.dmp

[user@cn3144 ~]$ <b>cp -r custom_db /data/$USER/kraken_databases</b>

###
### Classification
###
[user@cn3144 ~]$ <b>cp -r $KRAKEN_TEST_DATA/accuracy .</b>
[user@cn3144 ~]$ <b>cd accuracy</b>
[user@cn3144 ~]$ <b>ls -lh</b>
total 7.5M
-rw-rw-r-- 1 user group 2.2K Aug 14 09:55 ACCURACY.README
-rw-rw-r-- 1 user group 1.2M Aug 14 09:55 HiSeq_accuracy.fa
-rw-rw-r-- 1 user group 1.8M Aug 14 09:55 MiSeq_accuracy.fa
-rw-rw-r-- 1 user group 4.6M Aug 14 09:55 simBA5_accuracy.fa
-rw-rw-r-- 1 user group  396 Aug 14 09:55 taxonomyDocSum.xslt
</pre>

<p>Now we will use the in memory temp file system <code>/dev/shm</code>
to store the database for classification. Note that unlike <code>lscratch</code>,
<code>/dev/shm</code> is not automatically cleaned at exit and care has to
be taken to clean up explicitly at the end of a job.</p>

<pre class="term">
[user@cn3144 ~]$ <b>cp -r custom_db /dev/shm</b>
[user@cn3144 ~]$ <b>kraken --db /dev/shm/custom_db --fasta-input \
                        --output HiSeq.kraken HiSeq_accuracy.fa</b>
10000 sequences (0.92 Mbp) processed in 0.422s (1421.8 Kseq/m, 131.15 Mbp/m).
  8140 sequences classified (81.40%)
  1860 sequences unclassified (18.60%)

[user@cn3144 ~]$ <b>head -5 HiSeq.kraken</b>
C       A_hydrophila_HiSeq.922  380703  101     644:22 380703:1 644:30 0:18
C       A_hydrophila_HiSeq.1263 644     101     1236:3 644:1 2:1 644:8 1236:8 644:50
C       A_hydrophila_HiSeq.2905 1448139 101     644:8 1448139:51 0:1 644:11
C       A_hydrophila_HiSeq.4866 1448139 101     644:61 1448139:10
C       A_hydrophila_HiSeq.7009 644     101     0:13 644:14 0:31 644:13

### CLEANUP
[user@cn3144 ~]$ <b>rm -rf /dev/shm/custom_db</b>
</pre>

<p>The same approach of copying to <code>/dev/shm</code> can be taken with the databases
from <code>/fdb/kraken</code> as long as they fit. Note that <code>/dev/shm</code> as
a whole is limited to 50% of memory and per job by the memory allocation. For example,
using a reduced size 'minikraken' database obtained from the kraken maintainers:</p>

<pre class="term">
[user@cn3144 ~]$ <b>cp -r /fdb/kraken/20180220_standard_8GB /dev/shm</b>
[user@cn3144 ~]$ <b>kraken --db /dev/shm/20180220_standard_8GB --fasta-input \
                        --output HiSeq.kraken.2 HiSeq_accuracy.fa</b>
10000 sequences (0.92 Mbp) processed in 0.602s (997.0 Kseq/m, 91.96 Mbp/m).
  7523 sequences classified (75.23%)
  2477 sequences unclassified (24.77%)
</pre>

<p>Column three is the predicted taxid. Let's get the name and rank of the
unique taxids from NCBI eutils and compare them to the predicted taxa:<p/>

<pre class="term">
[user@cn3144 ~]$ <b>cut -f3 HiSeq.kraken.2 | sort -u | grep -wv 0 > taxid.uniq</b>
[user@cn3144 ~]$ <b>esum="https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi"</b>
[user@cn3144 ~]$ <b>for taxid in $(cat taxid.uniq); do
                        curl -s -G "${esum}?db=taxonomy&id=${taxid}" | xsltproc taxonomyDocSum.xslt -
                    done 2> /dev/null > taxa.uniq</b>
[user@cn3144 ~]$ <b>join -1 3 -2 1 -t '|' \
                        &lt;(sort -k3,3 HiSeq.kraken.2 | tr '\t' '|') \
                        &lt;(paste -d '|' taxid.uniq taxa.uniq | sort -t'|' -k1,1) \
                    | awk -F'|' '{printf("%s prediction: %s(%s)\n", $3, $6, $7)}' \
                    | sort > HiSeq.kraken.pred</b>
[user@cn3144 ~]$ <b>tail -n -100 HiSeq.kraken.pred | head -n 5</b>
X_axonopodis_HiSeq.875312 prediction: Xanthomonas phaseoli pv. dieffenbachiae LMG 695,()
X_axonopodis_HiSeq.87911 prediction: Xanthomonas citri group,species group()
X_axonopodis_HiSeq.881987 prediction: Xanthomonas,genus()
X_axonopodis_HiSeq.882508 prediction: Xanthomonas phaseoli pv. dieffenbachiae LMG 695,()
X_axonopodis_HiSeq.882724 prediction: Xanthomonas,genus()

### CLEANUP
[user@cn3144 ~]$ <b>rm -rf /dev/shm/20180220_standard_8GB</b>
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>
    </div>
    <!-- END VERSION 1 -->

    <!-- START VERSION 2 -->
    <div id="tab-2" class="tab-content current">

<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=36g --cpus-per-task=4 --gres=lscratch:100</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load kraken/2.1.2</b>

###
### build a custom kraken database with some 400+ genomes
###
[user@cn3144 ~]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144 ~]$ <b>mkdir fa</b>
[user@cn3144 ~]$ <b>cd fa</b>
[user@cn3144 ~]$ <b>wget --quiet --input-file $KRAKEN_TEST_DATA/genomes_for_custom_db.urls</b>
[user@cn3144 ~]$ <b>gunzip *.gz</b>
[user@cn3144 ~]$ <b>ls -lh | head -5</b>
total 1.3G
-rw-r--r-- 1 user group 3.9M May 22  2016 GCA_000006745.1_ASM674v1_genomic.fna
-rw-r--r-- 1 user group 2.1M May 26  2016 GCA_000006885.1_ASM688v1_genomic.fna
-rw-r--r-- 1 user group 2.0M May 27  2016 GCA_000007045.1_ASM704v1_genomic.fna
-rw-r--r-- 1 user group 5.3M Aug  4  2014 GCA_000007825.1_ASM782v1_genomic.fna

[user@cn3144 ~]$ <b>cd ..</b>
[user@cn3144 ~]$ <b>kraken2-build --download-taxonomy --db custom_db</b>
[user@cn3144 ~]$ <b>for f in fa/*.fna; do
                       kraken2-build --add-to-library $f --db custom_db
                    done</b>
Added "fa/GCA_000006745.1_ASM674v1_genomic.fna" to library (custom_db)
Added "fa/GCA_000006885.1_ASM688v1_genomic.fna" to library (custom_db)
[...snip...]
Added "fa/GCA_900129335.1_PP_HGAG_QV30_2SC_T4_genomic.fna" to library (custom_db)
Added "fa/GCA_900185995.1_BK1071_genomic.fna" to library (custom_db)

# larger databases would need more threads to build
[user@cn3144 ~]$ <b>kraken2-build --build --threads 2 --db custom_db</b>
Creating sequence ID to taxonomy ID map (step 1)...
Found 721/721 targets, searched through 203908352 accession IDs, search complete.
Sequence ID to taxonomy ID map complete. [18.874s]
Estimating required capacity (step 2)...
Estimated hash table requirement: 258445896 bytes
Capacity estimation complete. [22.086s]
Building database files (step 3)...
Taxonomy parsed and converted.
CHT created with 8 bits reserved for taxid.
Completed processing of 721 sequences, 1343692302 bp
Writing data to disk...  complete.
Database files completed. [1m7.904s]
Database construction complete. [Total: 1m48.906s]

[user@cn3144 ~]$ <b>kraken2-build --clean --db custom_db</b>
Database disk usage: 34G
After cleaning, database uses 247M
[user@cn3144 ~]$ <b>tree custom_db</b>
custom_db/
|-- [user   246M]  hash.k2d
|-- [user     48]  opts.k2d
|-- [user    16K]  taxo.k2d
`-- [user    16K]  unmapped.txt


[user@cn3144 ~]$ <b>cp -r custom_db /data/$USER/kraken_databases</b>

###
### Classification
###
[user@cn3144 ~]$ <b>cp -r $KRAKEN_TEST_DATA/accuracy .</b>
[user@cn3144 ~]$ <b>cd accuracy</b>
[user@cn3144 ~]$ <b>ls -lh</b>
total 7.5M
-rw-rw-r-- 1 user group 2.2K Aug 14 09:55 ACCURACY.README
-rw-rw-r-- 1 user group 1.2M Aug 14 09:55 HiSeq_accuracy.fa
-rw-rw-r-- 1 user group 1.8M Aug 14 09:55 MiSeq_accuracy.fa
-rw-rw-r-- 1 user group 4.6M Aug 14 09:55 simBA5_accuracy.fa
-rw-rw-r-- 1 user group  396 Aug 14 09:55 taxonomyDocSum.xslt

[user@cn3144 ~]$ <b>kraken2 --db ../custom_db --output HiSeq.kraken HiSeq_accuracy.fa</b>
Loading database information... done.
10000 sequences (0.92 Mbp) processed in 0.100s (5993.0 Kseq/m, 552.81 Mbp/m).
  8381 sequences classified (83.81%)
  1619 sequences unclassified (16.19%)


[user@cn3144 ~]$ <b>head -5 HiSeq.kraken</b>
C       A_hydrophila_HiSeq.922  644     101     644:56 0:3 644:4 0:4
C       A_hydrophila_HiSeq.1263 644     101     2:2 644:5 1236:3 2:4 1236:4 644:24 1236:5 644:1 1236:5 644:14
C       A_hydrophila_HiSeq.2905 1448139 101     644:7 1448139:28 644:5 1448139:5 644:4 1448139:10 644:8
C       A_hydrophila_HiSeq.4866 1448139 101     644:60 9
C       A_hydrophila_HiSeq.7009 644     101     0:5 644:19 0:3 644:9 0:18 644:13

[user@cn3144 ~]$ <b>cp -r /fdb/kraken/20210223_standard_16GB_kraken2 ..</b>
[user@cn3144 ~]$ <b>kraken2 --db ../20210223_standard_16GB_kraken2 \
                        --output HiSeq.kraken.2 HiSeq_accuracy.fa</b>
Loading database information... done.
10000 sequences (0.92 Mbp) processed in 0.148s (4062.8 Kseq/m, 374.76 Mbp/m).
  9033 sequences classified (90.33%)
  967 sequences unclassified (9.67%)
</pre>

<p>Column three is the predicted taxid. Let's get the name and rank of the
unique taxids from NCBI eutils and compare them to the predicted taxa:<p/>

<pre class="term">
[user@cn3144 ~]$ <b>cut -f3 HiSeq.kraken.2 | sort -u | grep -wv 0 > taxid.uniq</b>
[user@cn3144 ~]$ <b>esum="https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi"</b>
[user@cn3144 ~]$ <b>for taxid in $(cat taxid.uniq); do
                        curl -s -G "${esum}?db=taxonomy&id=${taxid}" | xsltproc taxonomyDocSum.xslt -
                    done 2> /dev/null > taxa.uniq</b>
[user@cn3144 ~]$ <b>join -1 3 -2 1 -t '|' \
                        &lt;(sort -k3,3 HiSeq.kraken.2 | tr '\t' '|') \
                        &lt;(paste -d '|' taxid.uniq taxa.uniq | sort -t'|' -k1,1) \
                    | awk -F'|' '{printf("%s prediction: %s(%s)\n", $3, $6, $7)}' \
                    | sort > HiSeq.kraken.pred</b>
[user@cn3144 ~]$ <b>tail -n -100 HiSeq.kraken.pred | head -n 5</b>
X_axonopodis_HiSeq.875312 prediction: Xanthomonas phaseoli pv. dieffenbachiae LMG 695,()
X_axonopodis_HiSeq.87911 prediction: Xanthomonas citri group,species group()
X_axonopodis_HiSeq.881987 prediction: Xanthomonas,genus()
X_axonopodis_HiSeq.882508 prediction: Xanthomonas phaseoli pv. dieffenbachiae LMG 695,()
X_axonopodis_HiSeq.882724 prediction: Xanthomonas,genus()

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>
    </div>
</div><!-- versions div -->

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. kraken.sh) which copies a reduced size standard database
to <code>/lscratch</code> (or <code>/dev/shm</code> for version 1) and runs several fastq samples against a local copy of the database.</p>
<div class="tabs-container">
  <ul class="tabs">
  <li data-tab="btab-1">Version 1</li>
  <li class="current" data-tab="btab-2">Version 2</li>
  </ul>
  <div id="btab-1" class="tab-content">

<pre class="term">
#! /bin/bash
# USAGE: kraken.sh dbname infile [infile ...]

module load kraken/1.1
db=$1
dbname=$(basename $1)
shift
[[ ${1:-none} != "none" ]] || exit 1

# make sure the database in /dev/shm is cleaned up
trap 'rm -rf /dev/shm/${dbname}' EXIT

[[ -d ${db} ]] &amp;&amp; cp -r $db /dev/shm || exit 1

for infile in "$@"; do
    [[ -f ${infile} ]] \
    &amp;&amp; kraken --db /dev/shm/${dbname} --threads ${SLURM_CPUS_PER_TASK} \
        --fastq-input --gzip-compressed --output ${infile}.kraken
done
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">
sbatch --cpus-per-task=16 --mem=38g kraken.sh \
    /fdb/kraken/20171019_standard_32GB sample0[0-9].fastq.gz
</pre>
</div>
<div id="btab-2" class="tab-content current">

<pre class="term">
#! /bin/bash
# USAGE: kraken2.sh dbname infile [infile ...]

module load kraken/2.1.2
db=$1
dbname=$(basename $1)
shift
[[ ${1:-none} != "none" ]] || exit 1
    
[[ -d ${db} ]] && cp -r $db /lscratch/$SLURM_JOB_ID || exit 1

for infile in "$@"; do
    [[ -f ${infile} ]] \
    && kraken2 --db /lscratch/$SLURM_JOB_ID/${dbname} --threads ${SLURM_CPUS_PER_TASK} \
        --output ${infile}.kraken ${infile}
done
</pre>

  <p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

  <pre class="term">
  sbatch --cpus-per-task=16 --mem=72g kraken2.sh \
      /fdb/kraken/20220803_standard_kraken2 sample0[0-9].fastq.gz
  </pre>
  </div>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. kraken2.swarm) that uses the script above to process several
fastq files in each job</p>

<pre class="term">
kraken2.sh /fdb/kraken/20220803_standard_kraken2 sample0[0-9].fastq.gz
kraken2.sh /fdb/kraken/20220803_standard_kraken2 sample1[0-9].fastq.gz
kraken2.sh /fdb/kraken/20220803_standard_kraken2 sample2[0-9].fastq.gz
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f kraken2.swarm -g 72 -t 16 --module kraken/2.1.2</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>-t # <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td>--module kraken <td>Loads the kraken module for each subjob in the swarm
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
