<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'LDSC on Biowulf';</script>
<div class="title">LDSC on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
LDSC is a command line tool for estimating heritability and genetic correlation from GWAS summary statistics. LDSC also computes LD Scores.
</p>

<h3>References:</h3>

The following is from <a href="https://github.com/bulik/ldsc#citation">https://github.com/bulik/ldsc#citation</a>

<p>
<uL>
<li>
If you use the software or the LD Score regression intercept, please cite
<p>
Bulik-Sullivan, et al. LD Score Regression Distinguishes Confounding from Polygenicity in Genome-Wide Association Studies. Nature Genetics, 2015.
</li>
<li>
For genetic correlation, please also cite
<p>
Bulik-Sullivan, et al. An Atlas of Genetic Correlations across Human Diseases and Traits. bioRxiv doi: http://dx.doi.org/10.1101/014498
</li>
<li>
For partitioned heritability, please also cite
<p>
Finucane, HK, et al. Partitioning Heritability by Functional Category using GWAS Summary Statistics. bioRxiv doi: http://dx.doi.org/10.1101/014241
</li>
<li>
If you find the fact that LD Score regression approximates HE regression to be conceptually useful, please cite
<p>
Bulik-Sullivan, Brendan. Relationship between LD Score and Haseman-Elston, bioRxiv doi http://dx.doi.org/10.1101/018283
</li>
<li>
For LD Hub, please cite
<p>
Zheng, et al. LD Hub: a centralized database and web interface to perform LD score regression that maximizes the potential of summary level GWAS data for SNP heritability and genetic correlation analysis. Bioinformatics (2016) https://doi.org/10.1093/bioinformatics/btw613
</li>
</p>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/bulik/ldsc">LDSC Home Page</a></li>
<li><a href="https://github.com/bulik/ldsc/wiki">LDSC Documentation</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: <tt>ldsc</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
	<!-- 
<li>Environment Variables:
  <ul>
      <li><tt>LDSC_HOME</tt></li>
  </ul>
  -->
</ul>
<P>

<p>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load ldsc</b>
[+] Loading python 2.7 ...
[+] Loading ldsc, version 1.0.0-92-gcf1707e...
[user@cn3144 ~]$ <b>wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1kg_eur.tar.bz2</b>
--2017-08-01 17:06:24--  https://data.broadinstitute.org/alkesgroup/LDSCORE/1kg_eur.tar.bz2
Resolving dtn01-e0... 10.1.200.237
Connecting to dtn01-e0|10.1.200.237|:3128... connected.
Proxy request sent, awaiting response... 200 OK
Length: 1134910 (1.1M) [application/x-bzip2]
Saving to: “1kg_eur.tar.bz2”

100%[==========================================================================================================>] 1,134,910    833K/s   in 1.3s    

2017-08-01 17:06:30 (833 KB/s) - “1kg_eur.tar.bz2” saved [1134910/1134910]

[user@cn3144 ~]$ <b>tar -xf 1kg_eur.tar.bz2</b>
[user@cn3144 ~]$ <b>cd 1kg_eur/</b>
[user@cn3144 ~]$ <b>ldsc.py --bfile 22 --l2 --ld-wind-cm 1 --out 22</b>
*********************************************************************
* LD Score Regression (LDSC)
* Version 1.0.0
* (C) 2014-2015 Brendan Bulik-Sullivan and Hilary Finucane
* Broad Institute of MIT and Harvard / MIT Department of Mathematics
* GNU General Public License v3
*********************************************************************
Call: 
./ldsc.py \
--ld-wind-cm 1.0 \
--out 22 \
--bfile 22 \
--l2  

Beginning analysis at Tue Aug  1 17:12:04 2017
Read list of 19156 SNPs from 22.bim
Read list of 379 individuals from 22.fam
Reading genotypes from 22.bed
After filtering, 19156 SNPs remain
Estimating LD Score.
Writing LD Scores for 19156 SNPs to 22.l2.ldscore.gz

Summary of LD Scores in 22.l2.ldscore.gz
         MAF        L2
mean  0.2323   18.5353
std   0.1453   16.1039
min   0.0013    0.0657
25%   0.1042    7.8392
50%   0.2243   13.4837
75%   0.3549   22.9722
max   0.5000  109.7163

MAF/LD Score Correlation Matrix
        MAF      L2
MAF  1.0000  0.2749
L2   0.2749  1.0000
Analysis finished at Tue Aug  1 17:12:08 2017
Total time elapsed: 3.31s
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. ldsc.sh). For example:</p>

<pre class="term">
#!/bin/sh
module load ldsc

wget https://data.broadinstitute.org/alkesgroup/LDSCORE/1kg_eur.tar.bz2
tar -xf 1kg_eur.tar.bz2
cd 1kg_eur/
ldsc.py --bfile 22 --l2 --ld-wind-cm 1 --out 22
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] ldsc.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. ldsc.swarm). For example:</p>

<pre class="term">
cd /data/$USER/ldsc/set1 && ldsc.py --bfile 22 --l2 --ld-wind-cm 1 --out 22
cd /data/$USER/ldsc/set2 && ldsc.py --bfile 22 --l2 --ld-wind-cm 1 --out 22
cd /data/$USER/ldsc/set3 && ldsc.py --bfile 22 --l2 --ld-wind-cm 1 --out 22
cd /data/$USER/ldsc/set4 && ldsc.py --bfile 22 --l2 --ld-wind-cm 1 --out 22
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f ldsc.swarm [-g #] [-t #] --module ldsc</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>-t # <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td>--module ldsc <td>Loads the TEMPLATE module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
