<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'mageck-vispr on Biowulf';</script>
<div class="title">mageck-vispr on Biowulf</div>

<table width=25% align=right>
    <tr>
        <td>
            <div class="toc">
                <div class="tocHeading">Quick Links</div>
                <div class="tocItem"><A href="#doc">Documentation</a></div>
                <div class="tocItem"><a href="#notes">Notes</a></div>
                <div class="tocItem"><a href="#int">Interactive job </a></div>
                <div class="tocItem"><a href="#sbatch">Batch job </a></div>
            </div>
        </td>
    </tr>
</table>

<p>
The mageck-vispr toolkit is used to QC, analyse, and visualize results from CRISPR/Cas9
screens. It includes the
<ul>
<li>mageck algorithm for identifying essential genes from CRISPER/Cas9 screens</li>
<li>vispr visualization platform for mageck results</li>
</ul>
</p>

<p>The mageck workflow is implemented as a snakemake pipeline and runs automatically.
Vispr on the other hand is essentially a web application that will run a temporary server on
a compute node and the user will connect to it using a browser on his/her own computer
through an ssh tunnel.</p>

<h3>References:</h3>
<ul>
    <li>W. Li, H. Xu, T. Xiao, L. Cong, M. Love, F. Zhang, R. Irizarry, J. Liu, 
    M. Brown and S. Liu. <em>MAGeCK enables robust identification of essential
        genes from genome-scale CRISPR/Cas9 knockout screens</em>. Genome
    Biology 2014, 15:554.
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/25476604">PubMed</a>&nbsp;|&nbsp;
    <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4290824/">PMC</a>&nbsp;|&nbsp;
    <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0554-4">Journal</a>
    </li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="https://bitbucket.org/liulab/vispr">visper repo</a> </li>
    <li><a href="https://bitbucket.org/liulab/mageck-vispr/overview">mageck repo</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: mageck-vispr (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>Example files in <code>$MAGECK_VISPR_TEST_DATA</code></li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an interactive session with <a href="/docs/userguide.html#int">sinteractive</a>
and use as shown below. See our <a href="https://hpc.nih.gov/docs/tunneling/">tunneling page</a>
for more details on setting up ssh tunnels.</p>

<pre class="term">
biowulf$ <b>sinteractive --cpus-per-task=8 --mem=16g --tunnel</b>
salloc.exe: Pending job allocation 31864544
salloc.exe: job 31864544 queued and waiting for resources
salloc.exe: job 31864544 has been allocated resources
salloc.exe: Granted job allocation 31864544
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn2692 are ready for job
srun: error: x11: no local DISPLAY defined, skipping

Created 1 generic SSH tunnel(s) from this compute node to
biowulf for your use at port numbers defined
in the $PORTn ($PORT1, ...) environment variables.


Please create a SSH tunnel from your workstation to these ports on biowulf.
On Linux/MacOS, open a terminal and run:

    ssh  -L 46763:localhost:46763 biowulf.nih.gov

For Windows instructions, see https://hpc.nih.gov/docs/tunneling

[user@cn3144]$ <b>module load mageck-vispr</b>
</pre>

<p>Copy test data and run mageck pipeline</p>
<pre class="term">
[user@cn3144]$ <b>cp -r ${MAGECK_VISPR_TEST_DATA}/esc-testdata .</b>
[user@cn3144]$ <b>cd esc-testdata</b>
[user@cn3144]$ <b>mageck-vispr init ./test_workflow --reads \
    reads/ERR376998.subsample.fastq \
    reads/ERR376999.subsample.fastq \
    reads/ERR377000.subsample.fastq</b>
[user@cn3144]$ <b>tree test_workflow</b>
  test_workflow/
  |-- [user   2.9K]  config.yaml
  |-- [user   7.3K]  README.txt
  `-- [user   5.8K]  Snakefile
  
  0 directories, 3 files

[user@cn3144]$ <b>cd test_workflow</b>
</pre>
<p>Before running the workflow it is necessary to edit the automatically
generated config file. The generated file contains many comments. Here is
the edited file with the comments stripped for readability:</p>

<pre class="term">
library: ../yusa_library.csv
species: mus_musculus
assembly: mm10

targets:
    genes: true
sgrnas:
    update-efficiency: false
    trim-5: AUTO
    len: AUTO

samples:
    esc1:
        - ../reads/ERR376999.subsample.fastq
    esc2:
        - ../reads/ERR377000.subsample.fastq
    plasmid:
        - ../reads/ERR376998.subsample.fastq

experiments:
    "ESC-MLE":
        designmatrix: ../designmatrix.txt

# the following setting is required for 0.5.6
correct_cnv: false
</pre>

<p>Once the config file has been modified to reflect the experimental
design, run the pipeline. Note that snakemake is used to run this locally, not
by submitting tasks as cluster jobs. Note that the snakemake installation
used for mageck-vispr has been renamed to <code>mageck-vispr-snakemake</code>
to avoid interfering with the general use snakemake.</p>

<pre class="term">
[user@cn3144]$ <b>mageck-vispr-snakemake --cores=$SLURM_CPUS_PER_TASK</b>
</pre>

<p>Next, start the vispr server for visualization. Note that sinteractive sets the
<code>$PORT1</code> variable to the port selected for tunneling.</p>
<pre class="term">
[user@cn3144]$ <b>cd test_workflow</b>
[user@cn3144]$ <b>vispr server --port $PORT1 --host=localhost results/ESC-MLE.vispr.yaml</b>
Loading data.
Starting server.

Open:  go to localhost:39335 in your browser.
Note: Safari and Internet Explorer are currently unsupported.
Close: hit Ctrl-C in this terminal.

</pre>

<p>On your local workstation, create an ssh tunnel to biowulf as describe in our
<a href="https://hpc.nih.gov/docs/tunneling/">tunneling documentation</a>.
</p>

<div style='width:70%; margin:0 auto;'>
    <img src="/images/mageck_vispr_screenshot.png" alt="vispr web app" width="600px"/>
</div>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. mageck-vispr.sh), which uses the input file 'mageck-vispr.in'. For example:</p>

<p>Create a batch script for an existing config file similar to the following example:</p>
<pre class="term">
#! /bin/bash
# this file is mageck.batch

module load mageck-vispr/0.5.4 || exit 1
cd /path/to/workdir

mageck-vispr-snakemake --cores=$SLURM_CPUS_PER_TASK
</pre>

<p>Submit to the queue with <a href="/docs/userguide.html">sbatch</a>:</p>
<pre class="term">
biowulf$ <b>sbatch --cpus-per-task=8 --mem=16g mageck.batch</b>
</pre>






<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
