<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'mothur on Biowulf';</script>
<div class="title">mothur on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>mothur is a tool for analyzing 16S rRNA gene sequences generated on multiple platforms as part of microbial ecology projects.</p>

<h3>References:</h3>
<ul>
  <li>Schloss PD1, Westcott SL, Ryabin T, Hall JR, Hartmann M, Hollister EB, Lesniewski RA, Oakley BB, Parks DH, Robinson CJ, Sahl JW, Stres B, Thallinger GG, Van Horn DJ, Weber CF. <a href="https://www.ncbi.nlm.nih.gov/pubmed/19801464"><u>Introducing mothur: open-source, platform-independent, community-supported software for describing and comparing microbial communities.</u> <em>Appl Environ Microbiol. (2009)</em></a></li>
  <li>Westcott SL, Schloss PD. <a href="https://www.ncbi.nlm.nih.gov/pubmed/26664811"><u>De novo clustering methods outperform reference-based methods for assigning 16S rRNA gene sequences to operational taxonomic units.</u>  <em>PeerJ. (2015)</em></a></li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
  <li><a href="http://www.mothur.org/wiki/Main_Page">mothur Main Site</a></li>
  <li><a href="http://www.mothur.org/wiki/MiSeq_SOP">mothur MiSeq_SOP Tutorial</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>mothur</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
<li>Multithreaded
<li>Environment variables set <!--for ones users should be aware of -->
  <ul>
    <li><tt>MOTHUR_HOME</tt></li>
    <li><tt>MOTHUR_EXAMPLES</tt></li>
  </ul>
<li>Example files in <tt>$MOTHUR_EXAMPLES</tt>
</ul>

<p class="alert"> (April 2021) In Mothur version 1.45.2, users attempting classification or alignment with BLAST-based, such as with <tt>classify.seqs</tt> will likely encounter issues where Mothur is unable to find BLAST binaries. We recommend using kmer classification instead while we work to resolve the problem.
</p>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=8 --mem=8g --gres=lscratch:10</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144 ~]$ <b>module load mothur</b>
[user@cn3144 ~]$ <b>tar zxf $MOTHUR_EXAMPLES/MiSeq_SOP.tgz</b>
[user@cn3144 ~]$ <b>cd MiSeq_SOP</b>
[user@cn3144 ~]$ <b>mothur stability.batch</b>
Linux version

Using ReadLine,Boost,HDF5,GSL
mothur v.1.45.2
Last updated: 4/9/21
by
Patrick D. Schloss

Department of Microbiology & Immunology

University of Michigan
http://www.mothur.org

When using, please cite:
Schloss, P.D., et al., Introducing mothur: Open-source, platform-independent, community-supported software for describing and comparing microbial communities. Appl Environ Microbiol, 2009. 75(23):7537-41.

Distributed under the GNU General Public License

Type 'help()' for information on the commands that are available

For questions and analysis support, please visit our forum at https://forum.mothur.org

Type 'quit()' to exit program

[NOTE]: Setting random seed to 19760620.

Batch Mode

Setting environment variable PROCS to $SLURM_CPUS_PER_TASK

mothur > pcr.seqs(fasta=silva.bacteria.fasta, start=11894, end=25319, keepdots=F, processors=8)

Using 8 processors.

...

mothur > quit()


It took 163 seconds to run 23 commands from stability.batch batch file.

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. mothur.sh). For example:</p>

<pre class="term">#----- This file is MiSeq_SOP.sh -----#
#!/bin/bash

# Set the environment
module load mothur

# Untar the example data files
tar xzf $MOTHUR_EXAMPLES/MiSeq_SOP.tgz
cd MiSeq_SOP

# Run the batch script
mothur stability.batch
</pre>

<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=8 --mem=4g --job-name=MiSeq_SOP --output=MiSeq_SOP.out MiSeq_SOP.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. mothur.swarm). For example:</p>

<pre class="term">
cd run1; mothur run1.batch
cd run2; mothur run2.batch
cd run3; mothur run3.batch
cd run4; mothur run4.batch
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f mothur.swarm [-g #] [-t #] --module mothur</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module mothur</tt> <td>Loads the mothur module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
