<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'nanopolish on Biowulf';</script>
<div class="title">nanopolish on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
nanopolish is a software package for signal-level analysis of Oxford Nanopore sequencing data. Nanopolish can calculate an improved consensus sequence for a draft genome assembly, detect base modifications, call SNPs and indels with respect to a reference genome and more 
</p>

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://nanopolish.readthedocs.io/en/latest/index.html">nanopolish Main Site</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
    <li>Module Name: <tt>nanopolish</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
	<li>Multithreaded app (use -t option; not available in all commands)
	<li>Environment variables set <!--for ones users should be aware of -->
  <ul>
    <li><tt>NANOPOLISH_TESTDATA</tt></li>
  </ul>
<li>Example files in <tt>/usr/local/apps/nanopolish/TEST_DATA</tt>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --mem 4g</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load nanopolish</b>

[user@cn3144 ~]$ <b>cp $NANOPOLISH_TESTDATA/* .</b>

[user@cn3144 ~]$ <b>tar -xf ecoli_2kb_region.tar.gz</b>

[user@cn3144 ~]$ <b>cd ecoli_2kb_region/</b>

[user@cn3144 ~]$ <b>nanopolish index -d fast5_files/ reads.fasta</b>
[readdb] indexing fast5_files/
[readdb] num reads: 112, num reads with path to fast5: 112

[user@cn3144 ~]$ <b>nanopolish</b>
error: no command provided
usage: nanopolish [command] [options]
  valid commands: 
    --help
    --version
    call-methylation
    eventalign
    extract
    getmodel
    help
    index
    methyltrain
    phase-reads
    polya
    scorereads
    variants
    vcf2fasta
  for help on given command, type nanopolish command --help

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. nanopolish.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load nanopolish
nanopolish index -d fast5_files/ reads.fasta</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=2 --mem=4g nanopolish.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. nanopolish.swarm). For example:</p>

<pre class="term">
nanopolish index -d fast5_files1/ reads1.fasta
nanopolish index -d fast5_files2/ reads2.fasta
nanopolish index -d fast5_files3/ reads3.fasta
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f nanopolish.swarm -g 4 -t 2 --module nanopolish</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module nanopolish</tt> <td>Loads the nanopolish module for each subjob in the swarm 
</table>
















<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
