<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'NGMLR on Biowulf';</script>
<div class="title">NGMLR on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
CoNvex Gap-cost alignMents for Long Reads (ngmlr) is a long-read mapper designed to sensitively align PacBilo or Oxford Nanopore to (large) reference genomes. It was designed to quickly and correctly align the reads, including those spanning (complex) structural variations. Ngmlr uses an SV aware k-mer search to find approximate mapping locations for a read and then a banded Smith-Waterman alignment algorithm to compute the final alignment. Ngmlr uses a convex gap cost model that penalizes gap extensions for longer gaps less than for shorter ones to compute precise alignments. The gap model allows ngmlr to account for both the sequencing error and real genomic variations at the same time and makes it especially effective at more precisely identifying the position of breakpoints stemming from structural variations. The k-mer search helps to detect and split reads that cannot be aligned linearly, enabling ngmlr to reliably align reads to a wide range of different structural variations including nested SVs (e.g. inversions flanked by deletions).
</p>

<h3>References:</h3>
<ul>
<li>Accurate detection of complex structural variations using single-molecule sequencing.
Fritz J. Sedlazeck, Philipp Rescheneder, Moritz Smolka, Han Fang, Maria Nattestad, Arndt von Haeseler & Michael C. Schatz.
Nature Methods
volume 15, 461â€“468 (2018).
<a href="https://www.nature.com/articles/s41592-018-0001-7">[html]</a></li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/philres/ngmlr">NGMLR Main Site</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>ngmlr</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
<li>Example files in <tt>/fdb/ngmlr</tt>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task 4</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load ngmlr</b>

[user@cn3144 ~]$ <b>ngmlr -t $SLURM_CPUS_PER_TASK -r reference.fasta -q reads.fastq -o test.sam </b>

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. ngmlr.sh). For example:</p>

<pre class="term">
#!/bin/sh
set -e
module load ngmlr

test -n "$SLURM_CPUS_PER_TASK" || SLURM_CPUS_PER_TASK=2

ngmlr -t $SLURM_CPUS_PER_TASK -r reference.fasta -q reads.fastq -o test.sam 
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] ngmlr.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. ngmlr.swarm). For example:</p>

<pre class="term">
test -n "$SLURM_CPUS_PER_TASK" || SLURM_CPUS_PER_TASK=2; ngmlr -t $SLURM_CPUS_PER_TASK -r reference.fasta -q sample1.fastq -o sample1.sam
test -n "$SLURM_CPUS_PER_TASK" || SLURM_CPUS_PER_TASK=2; ngmlr -t $SLURM_CPUS_PER_TASK -r reference.fasta -q sample2.fastq -o sample2.sam
test -n "$SLURM_CPUS_PER_TASK" || SLURM_CPUS_PER_TASK=2; ngmlr -t $SLURM_CPUS_PER_TASK -r reference.fasta -q sample3.fastq -o sample3.sam
test -n "$SLURM_CPUS_PER_TASK" || SLURM_CPUS_PER_TASK=2; ngmlr -t $SLURM_CPUS_PER_TASK -r reference.fasta -q sample4.fastq -o sample4.sam
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f ngmlr.swarm [-g #] [-t #] --module ngmlr</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module ngmlr</tt> <td>Loads the NGMLR module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
