<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'OpenCRAVAT: a platform for the annotation of human genetic variation';</script>
<div class="title"><b>OpenCRAVAT: a platform for the annotation of human genetic variation</b></div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
OpenCRAVAT is  a new open source, scalable decision support system for
variant and gene prioritization. It includses a modular resource catalog 
to maximize community and developer involvement, and as a result the catalog
is being actively developed and growing every month. Resources made available via
the store are well-suited for analysis of cancer, as well as Mendelian and complex
diseases.
</p>
OB
<h3>References:</h3>
<ul>
<li>Kymberleigh A. Pagel, Rick Kim, Kyle Moad, Ben Busby, Lily Zheng, Collin Tokheim,
Michael Ryan,  and Rachel Karchin <br />
<i>Integrated Informatics Analysis of Cancer-Related Variants</i>
<a href="https://ascopubs.org/doi/full/10.1200/CCI.19.00132"> JCO Clinical Cancer Informatics  </a>, 2020, v.4, p.310-317.
</ul>

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/KarchinLab/open-cravat/wiki/quickstart-command-line">OpenCRAVAT quick start guide</a></li>
<li><a href="https://github.com/KarchinLab/open-cravat/wiki">OpenCRAVAT Github wiki page</a></li>
<li><a href="https://bio.tools/OpenCRAVAT"> OpenCRAVAT Home page</a></li>
<li><a href="https://github.com/KarchinLab/open-cravat/wiki/5.-GUI-usage">Instructions for the GUI usage</a></li>
<li><a href="https://www.youtube.com/watch?v=5Y4U1ocMs4U&ab_channel=OpenCRAVAT">OpenCRAVAT NCI Webinar 9/2021</a></li>
</ul>
<div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>OpenCRAVAT</tt> (see <a href="https://hpc.nih.gov/apps/modules.html">the modules page</a> for more information)
<li>Unusual environment variables set 
  <ul>
    <li><b>OC_HOME</b>&nbsp; OpenCRAVAT installation directory</li>
    <li><b>OC_BIN </b>&nbsp; &nbsp; &nbsp; OpenCRAVAT executable directory</li>
    <li><b>OC_SRC </b>&nbsp; &nbsp; OpenCRAVAT source code directory</li>
  </ul>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive  --mem=16g  --gres=gpu:p100:1,lscratch:10 -c4</b>
[user@cn2389 ~]$ <b>module load OpenCRAVAT </b>
[+] Loading annovar 2020-06-08 on cn2389
[+] Loading OpenCRAVAT 2.3.1  ...
</pre>
In order to annotate and interpret variants, OpenCRAVAT (OC) makes use of a database comprising chanks of data called "modules" (not to be confused with the Biowulf modules). The currenly installed modules OC modules are available in the folder: $OC_MODULES. <br /><br />
To get started with OpenCRAVAT, create an example input file in your current  working directory by using the commands:
<pre class="term">
[user@cn2389 ~]$ <b>cd /data/$USER</b>
[user@cn2389 ~]$ <b>mkdir OpenCRAVAT && cd OpenCRAVAT</b>
[user@cn2389 ~]$ <b>oc new example-input . </b>
</pre>
The latter command will create a file "example_input" in your current directory:
<pre class="term">
[user@cn2389 ~]$ <b>cat example_input | wc -l </b>
373
[user@cn2389 ~]$ <b> head -n 20 example_input</b>
chr10   121593817       -       A       T       s0
chr10   2987654 +       T       A       s1
chr10   43077259        +       A       T       s2
chr10   8055656 +       A       T       s3
chr10   87864470        +       A       T       s4
chr10   87864486        +       A       -       s0
chr10   87864486        +       AA      -       s1
chr10   87894027        +       -       CG      s2
chr10   87894027        +       -       CT      s3
chr1    100719861       +       A       T       s4
chr1    10100   +       C       T       s0
chr1    110340653       +       CGGCTTT -       s1
chr11   108227625       +       A       T       s2
chr11   113789394       +       G       A       s3
chr1    111762684       +       G       A       s4
chr11   119206418       +       A       T       s0
chr1    114713881       +       TGGTC   -       s1
chr1    114713881       +       TGGTCTC -       s2
chr1    114716160       -       A       T       s3
chr11   1584916 +       -       GCC     s4
</pre>
Run OpenCRAVAT on the test input file:
<pre class="term">
[user@cn2389 ~]$ <b>oc run ./example_input -l hg38</b>
Input file(s): /vf/users/user/OpenCRAVAT/example_input
Genome assembly: hg38
Running converter...
        Converter (converter)           finished in 2.055s
Running gene mapper...                  finished in 21.534s
Running annotators...
        annotator(s) finished in 1.412s
Running aggregator...
        Variants                        finished in 0.174s
        Genes                           finished in 0.101s
        Samples                         finished in 0.192s
        Tags                            finished in 0.295s
Indexing
        variant base__chrom     finished in 0.009s
        variant base__coding    finished in 0.010s
        variant base__so        finished in 0.009s
Running postaggregators...
        Tag Sampler (tagsampler)        finished in 0.122s
Finished normally. Runtime: 26.077s
</pre>
Once the job is finished, the following files wil be created: <br /><br />
<pre class="term">
example_input.log<br />
example_input.sqlite<br />
example_input.err<br />
</pre>
In particular, file example_input.sqlite is the sqlite database with the results.
This sqlite database can be opened in the OpenCRAVAT web viewer as follows:
<pre class="term">
[user@cn2389 ~]$ <b>oc gui example_input.sqlite</b>

   ____                   __________  ___ _    _____  ______
  / __ \____  ___  ____  / ____/ __ \/   | |  / /   |/_  __/
 / / / / __ \/ _ \/ __ \/ /   / /_/ / /| | | / / /| | / /
/ /_/ / /_/ /  __/ / / / /___/ _, _/ ___ | |/ / ___ |/ /
\____/ .___/\___/_/ /_/\____/_/ |_/_/  |_|___/_/  |_/_/
    /_/

OpenCRAVAT is served at 0.0.0.0:8080
(To quit: Press Ctrl-C or Ctrl-Break if run on a Terminal or Windows, or click "Cancel" and then "Quit" if run through OpenCRAVAT app on Mac OS)
START /usr/bin/exo-open --launch WebBrowser "http://0.0.0.0:8080/result/index.html?dbpath=/vf/users/user/OpenCRAVAT/example_input.sqlite"
...
</pre>
On your local system, open a new window and type:
<pre class="term">
<b>ssh -t -L 8080:localhost:8080 biowulf.nih.gov "ssh -L 8080:localhost:8080 cn2389"</b>
</pre>
Navigate a browser on your local system to the URL: localhost:8080. <br /><br />
<img src="opencravat/opencravat_gui.png" width="800" border="0" alt="">
<br /> <br />
Exit an interactive session:
<pre class="term">
[user@cn2389 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>
<P>

<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
