<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'pb-CpG-tools on Biowulf';</script>
<div class="title">pb-CpG-tools on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
A collection of tools for the analysis of CpG/5mC data from PacBio HiFi reads aligned to a reference genome (e.g., an aligned BAM). To use these tools, the HiFi reads should already contain 5mC base modification tags, generated on-instrument or by using primrose. The aligned BAM should also be sorted and indexed.
</p>

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/PacificBiosciences/pb-CpG-tools">pb-CpG-tools Github</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
  <li>Module Name: <tt>pb-cpg-tools</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
	<li>Multithreaded</li>
<li>Environment variables set <!--for ones users should be aware of -->
  <ul>
    <li><tt>CPG_TEST_DATA</tt></li>
    <li><tt>CPG_PILEUP_MODEL</tt></li>
  </ul>
<li>Test data files in <tt>$CPG_TEST_DATA</tt></li>
<li>The models distributed with the git repository can be copied from <tt>$CPG_PILEUP_MODEL</tt> directory</li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=32G -c8 --gres=lscratch:20</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load pb-cpg-tools</b>

[user@cn3144 ~]$ <b>cd /lscratch/$SLURM_JOB_ID</b>

[user@cn3144 ~]$ <b>cp $CPG_TEST_DATA/*bam* .</b>

[user@cn3144 ~]$ <b>aligned_bam_to_cpg_scores \
                --bam HG002.GRCh38.haplotagged.truncated.bam \
                --output-prefix test \
                --model $CPG_PILEUP_MODEL/pileup_calling_model.v1.tflite \
                --threads $SLURM_CPUS_PER_TASK</b>

[2023-07-24][14:31:38][aligned_bam_to_cpg_scores][INFO] Starting aligned_bam_to_cpg_scores
[2023-07-24][14:31:38][aligned_bam_to_cpg_scores][INFO] cmdline: aligned_bam_to_cpg_scores --bam HG002.GRCh38.haplotagged.truncated.bam --output-prefix test --model /usr/local/apps/pb-cpg-tools/2.3.1/models/pileup_calling_model.v1.tflite --threads 8
[2023-07-24][14:31:38][aligned_bam_to_cpg_scores][INFO] Running on 8 threads
[2023-07-24][14:31:38][aligned_bam_to_cpg_scores][INFO] Processing alignment file 'HG002.GRCh38.haplotagged.truncated.bam'
[2023-07-24][14:32:16][aligned_bam_to_cpg_scores][INFO] Finished processing alignment files.
[2023-07-24][14:32:16][aligned_bam_to_cpg_scores][INFO] Writing hap2 site methylation to bed file: 'test.hap2.bed'
[2023-07-24][14:32:16][aligned_bam_to_cpg_scores][INFO] Writing hap1 site methylation to bed file: 'test.hap1.bed'
[2023-07-24][14:32:16][aligned_bam_to_cpg_scores][INFO] Writing combined site methylation to bed file: 'test.combined.bed'
[2023-07-24][14:32:16][aligned_bam_to_cpg_scores][INFO] Writing combined site methylation to bigwig file: 'test.combined.bw'
[2023-07-24][14:32:16][aligned_bam_to_cpg_scores][INFO] Writing hap2 site methylation to bigwig file: 'test.hap2.bw'
[2023-07-24][14:32:16][aligned_bam_to_cpg_scores][INFO] Writing hap1 site methylation to bigwig file: 'test.hap1.bw'
[2023-07-24][14:32:17][aligned_bam_to_cpg_scores][INFO] aligned_bam_to_cpg_scores completed. Total Runtime: 00:00:39.426

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. pb-cpg-tools.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load pb-cpg-tools
cd /lscratch/$SLURM_JOB_ID
cp $CPG_TEST_DATA/*bam* .
aligned_bam_to_cpg_scores \
            --bam HG002.GRCh38.haplotagged.truncated.bam \
            --output-prefix test \
            --model $CPG_PILEUP_MODEL/pileup_calling_model.v1.tflite \
            --threads $SLURM_CPUS_PER_TASK
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --mem=32g --cpus-per-task=8 --gres=lscratch:20  pb-cpg-tools.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. pb-cpg-tools.swarm). For example:</p>

<pre class="term">
aligned_bam_to_cpg_scores --bam input1.bam --output-prefix out1 --model v1.tflite --threads $SLURM_CPUS_PER_TASK
aligned_bam_to_cpg_scores --bam input2.bam --output-prefix out2 --model v1.tflite --threads $SLURM_CPUS_PER_TASK
aligned_bam_to_cpg_scores --bam input3.bam --output-prefix out3 --model v1.tflite --threads $SLURM_CPUS_PER_TASK
aligned_bam_to_cpg_scores --bam input4.bam --output-prefix out4 --model v1.tflite --threads $SLURM_CPUS_PER_TASK
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f pb-cpg-tools.swarm [-g #] [-t #] [--gres=lscratch:#] --module pb-cpg-tools</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i></tt></td><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)</td></tr>
  <tr><td><tt>-t <i>#</i></tt></td><td>Number of threads/CPUs required for each process (1 line in the swarm command file).</td></tr>
  <tr><td><tt>--gres=lscratch:#</tt></td><td>lscratch amount in GB allocated for each process (1 line in the swarm command file).</td></tr>
  <tr><td><tt>--module pb-cpg-tools</tt></td><td>Loads the pb-cpg-tools module for each subjob in the swarm</td></tr>
</table>
















<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
