<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'PBSuite on Biowulf';</script>
<div class="title">PBSuite on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
The PBSuite contains two projects created for analysis of Pacific Biosciences long-read sequencing data: PBHoney and PBJelly.
</p>
<p>
PBHoney is an implementation of two variant-identification approaches designed to exploit the high mappability of long reads (i.e., greater than 10,000 bp). PBHoney considers both intra-read discordance and soft-clipped tails of long reads to identify structural variants.
</p>
<p>
PBJelly is a highly automated pipeline that aligns long sequencing reads (such as PacBio RS reads or long 454 reads in fasta format) to high-confidence draft assembles. PBJelly fills or reduces as many captured gaps as possible to produce upgraded draft genomes.
</p>

<h3>References:</h3>
<ul>
<li>English AC, Richards S, Han Y, Wang M, Vee V, et al. (2012) Mind the Gap: Upgrading Genomes with Pacific Biosciences RS Long-Read Sequencing Technology. PLOS ONE 7(11): e47768. <a href="https://doi.org/10.1371/journal.pone.0047768">doi:10.1371/journal.pone.0047768</a></li>

<li>English A.C., Salerno W.J. & Reid J.G. (2014) PBHoney: identifying genomic variants via long-read discordance and interrupted mapping. BMC Bioinformatics 15, 180. <a href="https://doi.org/10.1186/1471-2105-15-180">doi:10.1186/1471-2105-15-180</a></li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://sourceforge.net/projects/pb-jelly/">PBSuite Homepage</a></li>
<li>$PBSUITE_HOME/docs</li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
    <li>Module Name: <tt>pbsuite</tt> or <tt>pbjelly</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
<li>Environment variables set <!--for ones users should be aware of -->
  <ul>
    <li><tt>PBSUITE_HOME</tt></li>
  </ul>
<li>Example files in <tt>$PBSUITE_HOME/docs/honeyExample/</tt> and <tt>$PBSUITE_HOME/docs/jellyExample/</tt>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample sessions (user input in <b>bold</b>):</p>

<h3>PBHoney</h3>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load pbsuite</b>
[user@cn3144 ~]$ <b>cp $PBSUITE_HOME/docs/honeyExample/* .</b>
[user@cn3144 ~]$ <b>sh workflow.sh</b>
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<h3>PBJelly</h3>
<P>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load pbsuite</b>
[user@cn3144 ~]$ <b>cp -r $PBSUITE_HOME/docs/jellyExample/* .</b>
[user@cn3144 ~]$ <b>sed -i "s|/__PATH__/_TO_/jellyExample|$PWD|g" Protocol.xml</b>
[user@cn3144 ~]$ <b>for stage in setup mapping support extraction assembly output
do
    Jelly.py $stage Protocol.xml
done
</b>
[user@cn3144 ~]$ <b>summarizeAssembly.py jelly.out.fasta </b><i># check your results</i>
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. pbjelly.sh). For example:</p>

<pre class="term">
#!/bin/sh
set -e
module load pbsuite

for stage in setup mapping support extraction assembly output
do
    Jelly.py $stage Protocol.xml
done
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] pbjelly.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. pbjelly.swarm). For example:</p>

<pre class="term">
for stage in setup mapping support extraction assembly output; do Jelly.py $stage protocol1.xml || exit 1; done
for stage in setup mapping support extraction assembly output; do Jelly.py $stage protocol2.xml || exit 1; done
for stage in setup mapping support extraction assembly output; do Jelly.py $stage protocol3.xml || exit 1; done
for stage in setup mapping support extraction assembly output; do Jelly.py $stage protocol4.xml || exit 1; done
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f pbjelly.swarm [-g #] [-t #] --module pbsuite</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module pbsuite</tt> <td>Loads the pbsuite module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
