<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Pear on HPC';</script>
    <div class="title">Pear on HPC</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</td></tr></table>

    <p> <strong>PEAR</strong>Â is an ultrafast, memory-efficient and highly accurate 
      pair-end read merger. It is fully parallelized and can run with as low as 
      just a few kilobytes of memory. </p>
    <p>PEAR evaluates all possible paired-end read overlaps and without requiring 
      the target fragment size as input. In addition, it implements a statistical 
      test for minimizing false-positive results. </p>
    <h3><a Name="doc"></a></h3>
<div class="heading">Documentation</div>
    <ul>
      <li><a href="https://sco.h-its.org/exelixis/web/software/pear/doc.html">https://sco.h-its.org/exelixis/web/software/pear/doc.html</a></li>
    </ul>

<a Name="notes"></a>
    <div class="heading"><a name="notes" id="notes"></a>Important Notes</div>
    <ul>
      <li>Module Name: <tt>pear</tt> (see <a href="/apps/modules.html">the modules 
        page</a> for more information) </li>
      <li>Multithreaded</li>
    </ul>
<P>

<a Name="int"></a></p><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>    
    <pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=4</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <strong>module load pear</strong>
[user@cn3144 ~]$ <strong>pear -f input_R1.fastq.gz -r input_R2.fastq.gz -o output -j $SLURM_CPUS_PER_TASK</strong>
 ____  _____    _    ____
|  _ \| ____|  / \  |  _ \
| |_) |  _|   / _ \ | |_) |
|  __/| |___ / ___ \|  _ <
|_|   |_____/_/   \_\_| \_\

PEAR v0.9.11 [Nov 5, 2017]

Citation - PEAR: a fast and accurate Illumina Paired-End reAd mergeR
Zhang et al (2014) Bioinformatics 30(5): 614-620 | doi:10.1093/bioinformatics/btt593

Forward reads file.................: input_R1.fastq.gz
Reverse reads file.................: input_R2.fastq.gz
PHRED..............................: 33
Using empirical frequencies........: YES
Statistical method.................: OES
Maximum assembly length............: 999999
Minimum assembly length............: 50
p-value............................: 0.010000
Quality score threshold (trimming).: 0
Minimum read size after trimming...: 1
Maximal ratio of uncalled bases....: 1.000000
Minimum overlap....................: 10
Scoring method.....................: Scaled score
Threads............................: 4

Allocating memory..................: 200,000,000 bytes
Computing empirical frequencies....: DONE
  A: 0.256699
  C: 0.237672
  G: 0.250363
  T: 0.255267
  693857 uncalled bases
Assemblying reads: 100%

Assembled reads ...................: 9,010,229 / 10,000,000 (90.102%)
Discarded reads ...................: 0 / 10,000,000 (0.000%)
Not assembled reads ...............: 989,771 / 10,000,000 (9.898%)
Assembled reads file...............: output.assembled.fastq
Discarded reads file...............: output.discarded.fastq
Unassembled forward reads file.....: output.unassembled.forward.fastq
Unassembled reverse reads file.....: output.fastq.unassembled.reverse.fastq

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a></p><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
    <p>Create a batch input file (e.g. batch.sh). For example:</p>

    <pre class="term">
#!/bin/bash
set -e
module load pear
pear -f input1.fastq.gz -r input2.fastq.gz -o output -j $SLURM_CPUS_PER_TASK</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

    <pre class="term">sbatch --cpus-per-task=4 batch.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

    <p>Create a swarmfile (e.g. job.swarm). For example:</p>

<pre class="term">
cd dir1; pear -f S1_R1.fastq.gz -r S1_R2.fastq.gz -o S1 -j $SLURM_CPUS_PER_TASK
cd dir2; pear -f S2_R1.fastq.gz -r S2_R2.fastq.gz -o S2 -j $SLURM_CPUS_PER_TASK
cd dir3; pear -f S3_R1.fastq.gz -r S3_R2.fastq.gz -o S3 -j $SLURM_CPUS_PER_TASK
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

    <pre class="term">swarm -f job.swarm -t 4 --module pear</pre>
where
    <table width="63%" border=0>
      <tr>
        <td width="18%"><tt>-t <i>#</i></tt> </td>
        <td width="82%">Number of threads/CPUs required for each process (1 line 
          in the swarm command file). </td>
      </tr>
      <tr> 
        <td><tt>--module </tt></td>
        <td>Loads the module for each subjob in the swarm </td>
      </tr>
    </table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
