<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'peddy on Biowulf';</script>
<div class="title">peddy on Biowulf</div>

<table width=25% align=right>
    <tr>
        <td>
            <div class="toc">
                <div class="tocHeading">Quick Links</div>
                <div class="tocItem"><A href="#doc">Documentation</a></div>
                <div class="tocItem"><a href="#notes">Notes</a></div>
                <div class="tocItem"><a href="#int">Interactive job </a></div>
                <div class="tocItem"><a href="#sbatch">Batch job </a></div>
                <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
            </div>
        </td>
    </tr>
</table>

<div style="width: 70%;">
<p>
peddy is used to compare sex and familial relationships given in a PED file
with those inferred from a VCF file. This is done by sampling 25000 sites plus chrX
from the VCF file to estimate relatedness, heterozygosity, sex and ancestry. It uses
data from the thousand genome project.
</p>

<h3>References:</h3>
<ul>
    <li>Brent S. Pedersen, Aaron R. Quinlan. <em>Whoâ€™s Who? Detecting and Resolving Sample
        Anomalies in Human DNA Sequencing Studies with Peddy</em>. Am. J. Hum. Genetics 2017,
    3:406-413.
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/28190455">PubMed</a>&nbsp;|&nbsp;
    <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5339084/">PMC</a>&nbsp;|&nbsp;
    <a href="https://www.cell.com/action/showFullTextImages?pii=S0002-9297(17)30017-4">Journal</a>
    </li>
</ul>

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="https://github.com/brentp/peddy">GitHub</a></li>
</ul>
</div>



<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: peddy (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>peddy is a multithreaded application. Make sure to match the number of cpus requested with the
    number of threads.</li>
    <li>Example files in <code>$PEDDY_TEST_DATA</code></li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=2 --gres=lscratch:10</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144]$ <b>module load peddy</b>
[user@cn3144]$ <b>cp -r $PEDDY_TEST_DATA/data .</b>
[user@cn3144]$ <b>peddy -p $SLURM_CPUS_PER_TASK --plot --prefix ceph-1463 \
                     data/ceph1463.peddy.vcf.gz data/ceph1463.ped</b>
2018-05-21 08:14:22 cn4242 peddy.cli[35381] INFO Running Peddy version 0.4.8
2018-05-21 08:14:23 cn4242 peddy.cli[35381] INFO ped_check
2022-10-13 21:04:54 cn4291 peddy.peddy[83025] INFO plotting
2022-10-13 21:04:55 cn4291 peddy.cli[83025] INFO ran in 2.0 seconds
2022-10-13 21:04:55 cn4291 peddy.cli[83025] INFO het_check
2022-10-13 21:04:56 cn4291 peddy.pca[83025] INFO loaded and subsetted thousand-genomes genotypes (shape: (2504, 2724)) in 0.3 seconds
2022-10-13 21:04:56 cn4291 peddy.pca[83025] INFO ran randomized PCA on thousand-genomes samples at 2724 sites in 0.5 seconds
2022-10-13 21:04:57 cn4291 peddy.pca[83025] INFO Projected thousand-genomes genotypes and sample genotypes and predicted ancestry via SVM in 0.1 seconds
2022-10-13 21:04:57 cn4291 peddy.cli[83025] INFO ran in 2.7 seconds
2022-10-13 21:04:57 cn4291 peddy.cli[83025] INFO sex_check
2022-10-13 21:04:57 cn4291 peddy.peddy[83025] INFO sex-check: 0 skipped / 814 kept
2022-10-13 21:04:57 cn4291 peddy.cli[83025] INFO ran in 0.1 seconds

[user@cn3144]$ <b>ls -lh</b>
-rw-r--r-- 1 user group 169K May 21 08:14 ceph-1463.background_pca.json
-rw-r--r-- 1 user group 2.0K May 21 08:14 ceph-1463.het_check.csv
-rw-r--r-- 1 user group  18K May 21 08:14 ceph-1463.het_check.png
-rw-r--r-- 1 user group 211K May 21 08:14 ceph-1463.html
-rw-r--r-- 1 user group 118K May 21 08:14 ceph-1463.pca_check.png
-rw-r--r-- 1 user group  13K May 21 08:14 ceph-1463.ped_check.csv
-rw-r--r-- 1 user group 108K May 21 08:14 ceph-1463.ped_check.png
-rw-r--r-- 1 user group   96 May 21 08:14 ceph-1463.ped_check.rel-difference.csv
-rw-r--r-- 1 user group 1.7K May 21 08:14 ceph-1463.peddy.ped
-rw-r--r-- 1 user group  835 May 21 08:14 ceph-1463.sex_check.csv
-rw-r--r-- 1 user group  25K May 21 08:14 ceph-1463.sex_check.png

[user@cn3144]$

[user@cn3144]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf]$
</pre>

<p>peddy creates several plots, data tables, and a sumamry report in html. This includes, for example,
the following check on ancestry showing reported ancestry in the pedigree overlayed on a PCA of background
genomes.</p>

<div style="width: 50%; margin: 0 auto">
      <img src="/images/peddy_pca_check.png" width="100%"/>
</div>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>

<p>Create a batch input file (e.g. peddy.sh), which uses the data in <code>$PEDDY_TEST_DATA</code>. 
For example:</p>

<pre class="term">
#! /bin/bash
# this file is peddy.batch
module load peddy/0.4.8 || exit 1

td=/usr/local/apps/peddy/TEST_DATA/data
cp -r $PEDDY_TEST_DATA/data .
peddy -p $SLURM_CPUS_PER_TASK --plot --prefix ceph-1463 \
    data/ceph1463.peddy.vcf.gz data/ceph1463.ped
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=4 --mem=4g peddy.sh</pre>


<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. peddy.swarm). For example:</p>

<pre class="term">
peddy -p $SLURM_CPUS_PER_TASK --plot --prefix fam1 fam1/fam1.vcf.gz fam1/fam1.ped
peddy -p $SLURM_CPUS_PER_TASK --plot --prefix fam2 fam2/fam2.vcf.gz fam2/fam2.ped
peddy -p $SLURM_CPUS_PER_TASK --plot --prefix fam3 fam3/fam3.vcf.gz fam3/fam3.ped
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f peddy.swarm -g 4 -t 4 --module peddy/0.4.8</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>-t # <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td>--module peddy <td>Loads the peddy module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
