<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'pomoxis on Biowulf';</script>
<div class="title">pomoxis on Biowulf</div>

<table width=25% align=right>
    <tr>
        <td>
            <div class="toc">
                <div class="tocHeading">Quick Links</div>
                <div class="tocItem"><A href="#doc">Documentation</a></div>
                <div class="tocItem"><a href="#notes">Notes</a></div>
                <div class="tocItem"><a href="#int">Interactive job </a></div>
                <div class="tocItem"><a href="#sbatch">Batch job </a></div>
                <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
            </div>
        </td>
    </tr>
</table>

<p>
Pomoxis contains convenience wrappers around nanopore tools.
</p>



<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li>pomoxis on <a href="https://github.com/nanoporetech/pomoxis">GitHub</a></li>
<li>pomoxis <a href="https://nanoporetech.github.io/pomoxis/index.html">Documentation</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: pomoxis (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>Some pomoxis tools are multithreaded. Please match the number of allocated CPUs and the number of threads</li>
    <li>Example files in <code>$POMOXIS_TEST_DATA</code></li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --gres=lscratch:50 --cpus-per-task=6 --mem=12g</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144]$ <b>ml pomoxis</b>
[+] Loading minimap2, version 2.17...
[+] Loading miniasm 0.3.r179  ...
[+] Loading samtools 1.9  ...
[+] Loading racon 1.3.2  ...
[+] Loading seqkit  0.10.2
[+] Loading porechop  0.2.4
[+] Loading pomoxis  0.2.3

[user@cn3144]$ <b>cp -L $POMOXIS_TEST_DATA/* .</b>
[user@cn3144]$ <b>ls -lh</b>
total 906M
-rw-r----- 1 user staff 4.5M Sep 12 09:12 NC_000913.3.fasta
-rw-r----- 1 user staff 901M Sep 12 09:12 R9_Ecoli_K12_MG1655_lambda_MinKNOW_0.51.1.62.all.fasta
</pre>

<p>The ecoli reads used in these examples were obtained from the 
<a href="http://lab.loman.net/2016/07/30/nanopore-r9-data-release/">Loman Labs</a>. The 
<code>mini_align</code> script uses minimap2 to align reads and runs some common post-alignment
processing steps like sorting and aligning the resulting bam file.</p>
<pre class="term">

[user@cn3144]$ <b>mini_align -r NC_000913.3.fasta \
                     -i R9_Ecoli_K12_MG1655_lambda_MinKNOW_0.51.1.62.all.fasta \
                     -t $SLURM_CPUS_PER_TASK -p ecoli</b>
Constructing minimap index.
[M::mm_idx_gen::0.178*1.01] collected minimizers
[M::mm_idx_gen::0.220*1.38] sorted minimizers
[M::main::0.274*1.31] loaded/built the index for 1 target sequence(s)
[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1
[M::mm_idx_stat::0.286*1.29] distinct minimizers: 838542 (98.18% are singletons); average occurrences: 1.034; average spacing: 5.352
[M::main] Version: 2.17-r941
[M::main] CMD: minimap2 -I 16G -x map-ont -d NC_000913.3.fasta.mmi NC_000913.3.fasta
[M::main] Real time: 0.292 sec; CPU: 0.376 sec; Peak RSS: 0.049 GB
[samfaipath] build FASTA index...
[M::main::0.070*1.02] loaded/built the index for 1 target sequence(s)
[M::mm_mapopt_update::0.086*1.01] mid_occ = 12
[M::mm_idx_stat] kmer size: 15; skip: 10; is_hpc: 0; #seq: 1
[M::mm_idx_stat::0.097*1.01] distinct minimizers: 838542 (98.18% are singletons); average occurrences: 1.034; average spacing: 5.352
[M::worker_pipeline::128.218*5.37] mapped 72602 sequences
[M::worker_pipeline::203.551*5.30] mapped 59810 sequences
[M::main] Version: 2.17-r941
[M::main] CMD: minimap2 -x map-ont -t 6 -a NC_000913.3.fasta.mmi R9_Ecoli_K12_MG1655_lambda_MinKNOW_0.51.1.62.all.fasta
[M::main] Real time: 203.564 sec; CPU: 1078.051 sec; Peak RSS: 3.832 GB
[bam_sort_core] merging from 0 files and 6 in-memory blocks...
</pre>

<p>De novo assembly with miniasm</p>
<pre class="term">
[user@cn3144]$ <b>mini_assemble -i R9_Ecoli_K12_MG1655_lambda_MinKNOW_0.51.1.62.all.fasta \
                     -o ecoli_assm -t $SLURM_CPUS_PER_TASK  -m 1 -c</b>
...much output...
[user@cn3144]$ <b>assess_assembly -r NC_000913.3.fasta \
                     -i  ecoli_assm/reads_final.fa -t $SLURM_CPUS_PER_TASK</b>
...
  name     mean     q10      q50      q90
 err_ont  2.033%   1.783%   1.925%   2.332%
 err_bal  2.052%   1.797%   1.942%   2.357%
    iden  0.320%   0.257%   0.296%   0.344%
     del  0.821%   0.721%   0.788%   0.913%
     ins  0.913%   0.764%   0.858%   1.079%

#  Q Scores
  name     mean      q10      q50      q90
 err_ont  16.92    17.49    17.16    16.32
 err_bal  16.88    17.46    17.12    16.28
    iden  24.95    25.89    25.29    24.64
     del  20.85    21.42    21.03    20.40
     ins  20.40    21.17    20.67    19.67


[user@cn3144]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. pomoxis.sh), which uses the input file 'pomoxis.in'. For example:</p>

<pre class="term">
#!/bin/bash
wd=$PWD
module load pomoxis/0.2.3 || exit 1
cd /lscratch/$SLURM_JOB_ID
cp -L $POMOXIS_TEST_DATA/R9_Ecoli_K12_MG1655_lambda_MinKNOW_0.51.1.62.all.fasta .
mini_assemble -i R9_Ecoli_K12_MG1655_lambda_MinKNOW_0.51.1.62.all.fasta \
    -o ecoli_assm -t $SLURM_CPUS_PER_TASK  -m 1 -c
mv ecoli_assm $wd
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=6 --gres=lscratch:50 --mem=10g pomoxis.sh</pre>


<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. pomoxis.swarm). For example:</p>

<pre class="term">
mini_align -r NC_000913.3.fasta -i expt1.fastq -t $SLURM_CPUS_PER_TASK -p expt1
mini_align -r NC_000913.3.fasta -i expt2.fastq -t $SLURM_CPUS_PER_TASK -p expt2
mini_align -r NC_000913.3.fasta -i expt3.fastq -t $SLURM_CPUS_PER_TASK -p expt3
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f pomoxis.swarm -g 10 -t 6 --module pomoxis/0.2.3</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>-t # <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td>--module pomoxis <td>Loads the pomoxis module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
