<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Primer3 on Biowulf';</script>
<div class="title">Primer3 on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
Primer3 is a program for designing PCR primers.
Primer3 can also design hybridization probes and sequencing primers.
</p>

<h3>References:</h3>
<ul>

  <li>
    Untergasser A, Cutcutache I, Koressaar T, Ye J, Faircloth BC, Remm M and Rozen SG.
<a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3424584/"><u><b>Primer3--new capabilities and interfaces.</b></u></a>
<em>Nucleic Acids Res. 2012 Aug 1;40(15):e115</em>
  </li>

  <li>
    Koressaar T and Remm M.
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/17379693"><b><u>Enhancements and modifications of primer design program Primer3.</u></b></a>
    <em>Bioinformatics 2007;23(10):1289-1291.</em>
  </li>

  <li>
     Koressaar T, Lepamets M, Kaplinski L, Raime K, Andreson R and Remm M.
     <a href="https://www.ncbi.nlm.nih.gov/pubmed/29360956"><b><u>Primer3_masker: integrating masking of template sequence with primer design software.</u></b></a>
     <em>Bioinformatics 2018;34(11):1937-1938.</em>
</ul>

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://primer3.org/">Primer3 Main Site</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>

<ul>
    <li>Module Name: <tt>primer3</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
	<li>This application produces HTML reports. You can use <a href="/docs/hpcdrive.html">hpcdrive</a> to view these reports on your local workstation.</li>
<!--
<li>Environment variables set
  <ul>
    <li><tt>TEMPLATE_HOME</tt></li>
  </ul>
<li>Example files in <tt>???</tt>
<li>Reference data in <tt>/fdb/TEMPLATE/</tt>
-->
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>

<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load primer3</b>
</pre>

<p>
Following the <a href="https://primer3.org/manual.html#example">Primer3 manual's example</a>, create the file <tt>example</tt> with the following contents:
<pre class="term">
SEQUENCE_ID=example
SEQUENCE_TEMPLATE=GTAGTCAGTAGACNATGACNACTGACGATGCAGACNACACACACACACACAGCACACAGGTATTAGTGGGCCATTCGATCCCGACCCAAATCGATAGCTACGATGACG
SEQUENCE_TARGET=37,21
PRIMER_TASK=generic
PRIMER_PICK_LEFT_PRIMER=1
PRIMER_PICK_INTERNAL_OLIGO=1
PRIMER_PICK_RIGHT_PRIMER=1
PRIMER_OPT_SIZE=18
PRIMER_MIN_SIZE=15
PRIMER_MAX_SIZE=21
PRIMER_MAX_NS_ACCEPTED=1
PRIMER_PRODUCT_SIZE_RANGE=75-100
P3_FILE_FLAG=1
SEQUENCE_INTERNAL_EXCLUDED_REGION=37,21
PRIMER_EXPLAIN_FLAG=1
=
</pre>

<p>
Then run primer3 and show the output files:

<pre class="term">
[user@cn3144 ~]$ <b>primer3_core < example </b>

[user@cn3144 ~]$ <b>ls</b>
example
example.for
example.int
example.rev

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. primer3.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load primer3
primer3_core < example
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] primer3.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. primer3.swarm). For example:</p>

<pre class="term">
primer3_core < seq1
primer3_core < seq2
primer3_core < seq3
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f primer3.swarm [-g #] [-t #] --module primer3</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module primer3</tt> <td>Loads the primer3 module for each subjob in the swarm 
</table>
















<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
