<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'prokka on Biowulf';</script>
<div class="title">prokka on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>Prokka is a pipeline for rapidly annotating prokaryotic genomes. It produces
GFF3, GBK and SQN files that are ready for editing in Sequin and ultimately
submitted to Genbank/DDJB/ENA.</p>

<h3>References:</h3>
<ul>
    <li>T. Seeman. <em>Prokka: rapid prokaryotic genome annotation.</em>
    Bioinformatics 2014, 30:2068-2069.
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/24642063">Pubmed</a> &nbsp;|&nbsp;
    PMC &nbsp;|&nbsp;
    <a href="http://bioinformatics.oxfordjournals.org/content/30/14/2068.long">Journal</a>
    </li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="http://www.vicbioinformatics.com/software.prokka.shtml">Home page</a> </li>
    <li><a href="https://github.com/tseemann/prokka/blob/master/README.md">Manual</a></li>
    <li><a href="https://github.com/microgenomics/tutorials/blob/master/pangenome.md">Tutorial</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: prokka (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>prokka is a multithreaded application</li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=2g --cpus-per-task=4</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$
node$ <b>module load prokka</b>
node$ <b>prokka --listdb</b>
[16:11:09] Looking for databases in: /opt/anaconda/bin/../db
[16:11:09] * Kingdoms: Archaea Bacteria Mitochondria Viruses
[16:11:09] * Genera: Enterococcus Staphylococcus
[16:11:09] * HMMs: HAMAP
[16:11:09] * CMs: Bacteria Viruses
node$ <b>cp /usr/local/apps/prokka/TEST_DATA/GCA_000021185.1_ASM2118v1_genomic.fna .</b>
node$ <b>prokka --cpus 4 --force \
    --kingdom Bacteria \
    --outdir prokka_GCA_000021185 \
    --genus Listeria \
    --locustag GCA_000021185 GCA_000021185.1_ASM2118v1_genomic.fna</b>
[...snip...]

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. prokka.sh), which uses the input file 'prokka.in'. For example:</p>

<pre class="term">
#! /bin/bash

function die {
    echo "$@" &gt;&amp;2
    exit 1
}

module load prokka/1.13 || die "Could not load prokka module"
cp /usr/local/apps/prokka/TEST_DATA/GCA_000021185.1_ASM2118v1_genomic.fna . \
    || die "Could not find test data"

prokka --cpus ${SLURM_CPUS_PER_TASK} --force \
    --kingdom Bacteria \
    --outdir prokka_GCA_000021185 \
    --genus Listeria \
    --locustag GCA_000021185 GCA_000021185.1_ASM2118v1_genomic.fna
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=6 --mem=3g --time=10 prokka.sh</pre>

<p>This should create the following output directory:</p>
<pre class="term">
./prokka_GCA_000021185
|-- GCA_000021185_10252016.err
|-- GCA_000021185_10252016.faa
|-- GCA_000021185_10252016.ffn
|-- GCA_000021185_10252016.fna
|-- GCA_000021185_10252016.fsa
|-- GCA_000021185_10252016.gbk
|-- GCA_000021185_10252016.gff
|-- GCA_000021185_10252016.log
|-- GCA_000021185_10252016.sqn
|-- GCA_000021185_10252016.tbl
`-- GCA_000021185_10252016.txt
</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Copy the example data</p>
<pre class="term">
biowulf$ <b>cp -r /usr/local/apps/prokka/TEST_DATA .</b>
</pre>

<p>Create a swarmfile (e.g. prokka.swarm). For example:</p>

<pre class="term">
prokka --cpus ${SLURM_CPUS_PER_TASK} --force --kingdom Bacteria --outdir prokka_GCA_000008285 \
    --genus Listeria --locustag GCA_000008285 TEST_DATA/GCA_000008285.1_ASM828v1_genomic.fna
prokka --cpus ${SLURM_CPUS_PER_TASK} --force --kingdom Bacteria --outdir prokka_GCA_000021185 \
    --genus Listeria --locustag GCA_000021185 TEST_DATA/GCA_000021185.1_ASM2118v1_genomic.fna
prokka --cpus ${SLURM_CPUS_PER_TASK} --force --kingdom Bacteria --outdir prokka_GCA_000026705 \
    --genus Listeria --locustag GCA_000026705 TEST_DATA/GCA_000026705.1_ASM2670v1_genomic.fna
prokka --cpus ${SLURM_CPUS_PER_TASK} --force --kingdom Bacteria --outdir prokka_GCA_000168635 \
    --genus Listeria --locustag GCA_000168635 TEST_DATA/GCA_000168635.2_ASM16863v2_genomic.fna
prokka --cpus ${SLURM_CPUS_PER_TASK} --force --kingdom Bacteria --outdir prokka_GCA_000168815 \
    --genus Listeria --locustag GCA_000168815 TEST_DATA/GCA_000168815.1_ASM16881v1_genomic.fna
prokka --cpus ${SLURM_CPUS_PER_TASK} --force --kingdom Bacteria --outdir prokka_GCA_000196035 \
    --genus Listeria --locustag GCA_000196035 TEST_DATA/GCA_000196035.1_ASM19603v1_genomic.fna
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f prokka.swarm -g 2 -t 6 --module prokka/1.13</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>-t # <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td>--module prokka <td>Loads the prokka module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
