<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'RagTag on Biowulf';</script>
<div class="title">RagTag on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>RagTag is a collection of software tools for scaffolding and improving modern genome assemblies.</p>

<p>Tasks include:</p>

<ul>
  <li>Homology-based misassembly correction</li>
  <li>Homology-based assembly scaffolding and patching</li>
  <li>Scaffold merging</li>
</ul>

<p>Ragtag also provides command line utilities for working with common genome assembly file formats.</p>

<h3>References:</h3>
<ul>
  <li>
    Alonge, Michael, et al. 
    <a href="https://pubmed.ncbi.nlm.nih.gov/31661016/"><b><u>RaGOO: fast and accurate reference-guided scaffolding of draft genomes.</u></b></a>
    <em>Genome biology 20.1 (2019): 1-17.</em>
  </li>
</ul>

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/malonge/RagTag/wiki">RagTag Main Site</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>

<ul>
    <li>Module Name: <tt>ragtag</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
    <li>singlethreaded
    <li>Reference data in <tt>/data/genome/fasta/</tt>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load ragtag</b>

[user@cn3144 ~]$ <b>ragtag.py --help</b>

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. ragtag.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load ragtag
ragtag.py correct ref.fasta query.fasta
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] ragtag.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. ragtag.swarm). For example:</p>

<pre class="term">
ragtag.py correct ref.fasta query1.fasta
ragtag.py correct ref.fasta query2.fasta
ragtag.py correct ref.fasta query3.fasta
ragtag.py correct ref.fasta query4.fasta
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f ragtag.swarm [-g #] [-t #] --module ragtag</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module ragtag</tt> <td>Loads the ragtag module for each subjob in the swarm 
</table>
















<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
