<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'RAREMETAL on Biowulf';</script>
<div class="title">RAREMETAL on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
RAREMETAL is a program that facilitates the meta-analysis of rare variants from genotype arrays or sequencing.
</p>

<h3>Reference:</h3>
<ul>
<li><a href="https://academic.oup.com/bioinformatics/article/30/19/2828/2422168">Feng, Shuang, et al. "RAREMETAL: fast and powerful meta-analysis for rare variants." <i>Bioinformatics</i> 30.19 (2014): 2828-2829.</a></li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://genome.sph.umich.edu/wiki/RAREMETAL">RAREMETAL Main Site</a></li>
<li><a href="https://github.com/statgen/raremetal">RAREMETAL GitHub repository</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
    <li>Module Name: <tt>raremetal</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
    <li>Environment variables set <!--for ones users should be aware of -->
    <ul>
        <li><tt>RAREMETAL_HOME</tt></li>
    </ul>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf ~]$ <b>sinteractive --cpus-per-task=4 --mem=8G --gres=lscratch:10</b>
salloc.exe: Pending job allocation 42848391
salloc.exe: job 42848391 queued and waiting for resources
salloc.exe: job 42848391 has been allocated resources
salloc.exe: Granted job allocation 42848391
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3101 are ready for job
srun: error: x11: no local DISPLAY defined, skipping

[user@cn3101 ~]$ <b>cd /lscratch/$SLURM_JOB_ID</b>

[user@cn3101 42848391]$ <b>cp -r /fdb/raremetal/4.15.1/raremetal_tutorial .</b>

[user@cn3101 42848391]$ <b>cd raremetal_tutorial/</b>

[user@cn3101 raremetal_tutorial]$ <b>module load raremetal samtools</b>
[+] Loading raremetal  4.15.1  on cn3101
[+] Loading singularity  3.4.2  on cn3101
[+] Loading samtools 1.9  ...

[user@cn3101 raremetal_tutorial]$ <b>raremetalworker  --ped example1.ped --dat example1.dat --vcf example1.vcf.gz \
     --traitName QT1 --inverseNormal --makeResiduals --kinSave --kinGeno \
     --prefix STUDY1</b>

RAREMETALWORKER 4.15.1 -- A Forerunner of RareMetal
          (c) 2012-2016 Shuang Feng, Dajiang Liu, Sai Chen, Goncalo Abecasis
[...snip...]

[user@cn3101 raremetal_tutorial]$ <b>raremetalworker  --ped example2.ped --dat example2.dat --vcf example2.vcf.gz \
     --traitName QT1 --inverseNormal --makeResiduals --kinSave --kinGeno \
     --prefix STUDY2</b>

RAREMETALWORKER 4.15.1 -- A Forerunner of RareMetal
          (c) 2012-2016 Shuang Feng, Dajiang Liu, Sai Chen, Goncalo Abecass
[...snip...]

[user@cn3101 raremetal_tutorial]$ <b>bgzip STUDY1.QT1.singlevar.score.txt</b>

[user@cn3101 raremetal_tutorial]$ <b>tabix -c "#" -s 1 -b 2 -e 2 STUDY1.QT1.singlevar.score.txt.gz</b>

[user@cn3101 raremetal_tutorial]$ <b>bgzip STUDY1.QT1.singlevar.cov.txt</b>

[user@cn3101 raremetal_tutorial]$ <b>tabix -c "#" -s 1 -b 2 -e 2 STUDY1.QT1.singlevar.cov.txt.gz</b>

[user@cn3101 raremetal_tutorial]$ <b>bgzip STUDY2.QT1.singlevar.score.txt</b>

[user@cn3101 raremetal_tutorial]$ <b>tabix -c "#" -s 1 -b 2 -e 2 STUDY2.QT1.singlevar.score.txt.gz</b>

[user@cn3101 raremetal_tutorial]$ <b>bgzip STUDY2.QT1.singlevar.cov.txt</b>

[user@cn3101 raremetal_tutorial]$ <b>tabix -c "#" -s 1 -b 2 -e 2 STUDY2.QT1.singlevar.cov.txt.gz</b>

[user@cn3101 raremetal_tutorial]$ <b>cat >summaryfiles<<"EOF"
STUDY1.QT1.singlevar.score.txt.gz
STUDY2.QT1.singlevar.score.txt.gz
EOF</b>

[user@cn3101 raremetal_tutorial]$ <b>cat >covfiles<<"EOF"
STUDY1.QT1.singlevar.cov.txt.gz
STUDY2.QT1.singlevar.cov.txt.gz
EOF</b>

[user@cn3101 raremetal_tutorial]$ <b>raremetal --summaryFiles summaryfiles --covFiles covfiles \
    --groupFile group.file --SKAT --burden --MB --VT --longOutput \
    --tabulateHits --hitsCutoff 1e-05 --prefix COMBINED.QT1 --hwe 1.0e-05 \
    --callRate 0.95 -</b>

RAREMETAL 4.15.1 -- A Tool for Rare Variants Meta-Analyses for Quantitative Traits
          (c) 2012-2017 Shuang Feng, Dajiang Liu, Sai Chen, Goncalo Abecasis

[...snip...]

[user@cn3101 raremetal_tutorial]$ <b>ls</b>
COMBINED.QT1.meta.burden.results      example2.vcf.gz
COMBINED.QT1.meta.MB.results          example2.vcf.gz.tbi
COMBINED.QT1.meta.plots.pdf           group.file
COMBINED.QT1.meta.singlevar.results   STUDY1.Empirical.Kinship.gz
COMBINED.QT1.meta.SKAT_.results       STUDY1.QT1.additive.plots.pdf
COMBINED.QT1.meta.tophits.burden.tbl  STUDY1.QT1.singlevar.cov.txt.gz
COMBINED.QT1.meta.tophits.MB.tbl      STUDY1.QT1.singlevar.cov.txt.gz.tbi
COMBINED.QT1.meta.tophits.SKAT.tbl    STUDY1.QT1.singlevar.score.txt.gz
COMBINED.QT1.meta.tophits.VT.tbl      STUDY1.QT1.singlevar.score.txt.gz.tbi
COMBINED.QT1.meta.VT_.results         STUDY1.singlevar.log
COMBINED.QT1.raremetal.log            STUDY2.Empirical.Kinship.gz
command_to_use                        STUDY2.QT1.additive.plots.pdf
covfiles                              STUDY2.QT1.singlevar.cov.txt.gz
example1.dat                          STUDY2.QT1.singlevar.cov.txt.gz.tbi
example1.ped                          STUDY2.QT1.singlevar.score.txt.gz
example1.vcf.gz                       STUDY2.QT1.singlevar.score.txt.gz.tbi
example1.vcf.gz.tbi                   STUDY2.singlevar.log
example2.dat                          summaryfiles
example2.ped

[user@cn3101 raremetal_tutorial]$ <b>exit</b>
exit
salloc.exe: Relinquishing job allocation 42848391

[user@biowulf ~]$

</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. raremetal.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load raremetal
raremetal --summaryFiles summaryfiles --covFiles covfiles \
    --groupFile group.file --SKAT --burden --MB --VT --longOutput \
    --tabulateHits --hitsCutoff 1e-05 --prefix COMBINED.QT1 --hwe 1.0e-05 \
    --callRate 0.95 -
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] raremetal.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. raremetal.swarm). For example:</p>

<pre class="term">
raremetal --summaryFiles summaryfiles --covFiles covfiles \
    --groupFile group.file --SKAT --burden --MB --VT --longOutput \
    --tabulateHits --hitsCutoff 1e-05 --prefix COMBINED.QT1 --hwe 1.0e-05 \
    --callRate 0.95 -
raremetal --summaryFiles summaryfiles --covFiles covfiles \
    --groupFile group.file --SKAT --burden --MB --VT --longOutput \
    --tabulateHits --hitsCutoff 1e-05 --prefix COMBINED.QT2 --hwe 1.0e-05 \
    --callRate 0.95 -
raremetal --summaryFiles summaryfiles --covFiles covfiles \
    --groupFile group.file --SKAT --burden --MB --VT --longOutput \
    --tabulateHits --hitsCutoff 1e-05 --prefix COMBINED.QT3 --hwe 1.0e-05 \
    --callRate 0.95 -
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f raremetal.swarm [-g #] [-t #] --module raremetal</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module raremetal</tt> <td>Loads the RAREMETAL module for each subjob in the swarm 
</table>
















<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
