<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'RFmix on Biowulf';</script>
<div class="title">RFmix on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
RFMIX is a program to identify the ancestry of genomic segments using random forest discriminative machine learning methods combined with a conditional random field model of the linear chromosome.
</p>

<h3>Web site</h3>
<ul>
    <li><a href="https://github.com/slowkoni/rfmix">Home page</a></li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/slowkoni/rfmix/blob/master/MANUAL.md">RFmix Documentation</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>rfmix</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 12345678
salloc.exe: job 12345678 queued and waiting for resources
salloc.exe: job 12345678 has been allocated resources
salloc.exe: Granted job allocation 12345678
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn1234 are ready for job

[user@cn1234 ~]$ <b>module load rfmix</b>

[user@cn1234 ~]$ <b>rfmix --help</b>
RFMIX v2.03-r0 - Local Ancestry and Admixture Inference
(c) 2016, 2017 Mark Koni Hamilton Wright
Bustamante Lab - Stanford University School of Medicine
Based on concepts developed in RFMIX v1 by Brian Keith Maples, et al.

This version is licensed for non-commercial academic research use only
For commercial licensing, please contact cdbadmin@stanford.edu

--- For use in scientific publications please cite original publication ---
Brian Maples, Simon Gravel, Eimear E. Kenny, and Carlos D. Bustamante (2013).
RFMix: A Discriminative Modeling Approach for Rapid and Robust Local-Ancestry
Inference. Am. J. Hum. Genet. 93, 278-288

Summary of command line options - see manual for details

   -f <string>, --query-file=<string>	(required)
      VCF file with samples to analyze                      (required)
   -r <string>, --reference-file=<string>	(required)
      VCF file with reference individuals                   (required)
   -m <string>, --sample-map=<string>	(required)
      Reference panel sample population classification map  (required)
   -g <string>, --genetic-map=<string>	(required)
      Genetic map file                                      (required)
   -o <string>, --output-basename=<string>	(required)
      Basename (prefix) for output files                    (required)
   --chromosome=<string>	(required)
      Execute only on specified chromosome                  (required)

   -c <float>, --crf-spacing=<float>
      Conditional Random Field spacing (# of SNPs)
   -s <float>, --rf-window-size=<float>
      Random forest window size (class estimation window size)
   -w <float>, --crf-weight=<float>
      Weight of observation term relative to transition term in conditional random field
   -G <float>, --generations=<float>
      Average number of generations since expected admixture
   -e <int>, --em-iterations=<int>
      Maximum number of EM iterations
   --reanalyze-reference
      In EM, analyze local ancestry of the reference panel and reclassify it

   -n <int>, --node-size=<int>
      Terminal node size for random forest trees
   -t <int>, --trees=<int>
      Number of tree in random forest to estimate population class probability
   --max-missing=<float>
      Maximum proportion of missing data allowed to include a SNP
   -b <int>, --bootstrap-mode=<int>
      Specify random forest bootstrap mode as integer code (see manual)
   --rf-minimum-snps=<int>
      With genetic sized rf windows, include at least this many SNPs regardless of span
   --analyze-range=<string>
      Physical position range, specified as <start pos>-<end pos>, in Mbp (decimal allowed)

   --debug=<flag>
      Turn on any debugging output
   --n-threads=<int>
      Force number of simultaneous thread for parallel execution
   --random-seed=<string>
      Seed value for random number generation (integer)
	(maybe specified in hexadecimal by preceeding with 0x), or the string
	"clock" to seed with the current system time.

[user@cn1234 ~]$ <b> vartrix --bam test_dna.bam --cell-barcodes dna_barcodes.tsv --fasta test_dna.fa --out-matrix test_dna.out --vcf test_dna.vcf</b>

[user@cn1234 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 12345678
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. vartrix_job.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load vartrix
vartrix --bam test_dna.bam \
   --cell-barcodes dna_barcodes.tsv \
   --fasta test_dna.fa \
   --out-matrix test_dna.out \
   --vcf test_dna.vcf
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] vartrix_job.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. vartrix_jobs.swarm). For example:</p>

<pre class="term">
vartrix --bam test_dna1.bam \ 
   --cell-barcodes dna_barcodes1.tsv \
   --fasta test_dna1.fa \
   --out-matrix test_dna1.out \
   --vcf test_dna1.vcf
vartrix --bam test_dna2.bam \
   --cell-barcodes dna_barcodes2.tsv \
   --fasta test_dna2.fa \
   --out-matrix test_dna2.out \
   --vcf test_dna2.vcf
vartrix --bam test_dna3.bam \
   --cell-barcodes dna_barcodes3.tsv \
   --fasta test_dna3.fa \
   --out-matrix test_dna3.out \
   --vcf test_dna3.vcf
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f vartrix_jobs.swarm [-g #] [-t #] --module vartrix</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module vartrix</tt> <td>Loads the vartrix module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
