<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'sailfish on Biowulf';</script>
<div class="title">sailfish on Biowulf</div>

<table width=25% align=right>
    <tr>
        <td>
            <div class="toc">
                <div class="tocHeading">Quick Links</div>
                <div class="tocItem"><A href="#doc">Documentation</a></div>
                <div class="tocItem"><a href="#notes">Notes</a></div>
                <div class="tocItem"><a href="#int">Interactive job </a></div>
                <div class="tocItem"><a href="#sbatch">Batch job </a></div>
                <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
            </div>
        </td>
    </tr>
</table>

<p>
Sailfish quantifies the expression of a given set of transcripts using NGS reads. 
It is run in two stages: (1) The indexing step is run once per set of transcripts
(2) The quantification step is run once for each sample.
</p>

<h3>References:</h3>
<ul>
    <li>R. Patro, S. M. Mount, and C. Kingsford. <em>Sailfish enables 
        alignment-free isoform quantification from RNA-seq reads using 
        lightweight algorithms</em>. Nature Biotechnology 2014, 32:462-464.
    <a href="http://www.ncbi.nlm.nih.gov/pubmed/24752080">Pubmed</a>&nbsp;|&nbsp;
    <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4077321/">PMC</a>&nbsp;|&nbsp;
    <a href="http://www.nature.com/nbt/journal/v32/n5/full/nbt.2862.html">Journal</a>
    </li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="https://github.com/kingsfordgroup/sailfish">GitHub</a> </li>
    <li><a href="http://sailfish.readthedocs.io/en/master/index.html">Manual</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: sailfish (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>sailfish is a multithreaded application. Make sure to match the number of cpus requested with
    the number of threads (-p).</li>
    <li>Example files in <code>SAILFISH_TEST_DATA</code></li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=8g --cpus-per-task=4 --gres=lscratch:20</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144]$ <b>module load sailfish</b>
[user@cn3144]$ <b>zcat $SAILFISH_TEST_DATA/gencode.vM9.transcripts.fa.gz &gt; M9.fa</b>
[user@cn3144]$ <b>sailfish index -t M9.fa -o M9.idx -p $SLURM_CPUS_PER_TASK</b>
[user@cn3144]$ <b>cp $SAILFISH_TEST_DATA/ENCFF138LJO.fastq.gz .</b>
[user@cn3144]$ <b>sailfish quant -i M9.idx -r &lt;(zcat ENCFF138LJO.fastq.gz) --libType U \
                      -o quant -p $SLURM_CPUS_PER_TASK</b>

[user@cn3144]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. sailfish.sh), which uses the input file 'sailfish.in'. For example:</p>

<pre class="term">
#! /bin/bash

module load sailfish/0.10.0 || exit 1
wd=$PWD
cd /lscratch/$SLURM_JOB_ID || exit 1

# get transcriptome from the example directory
# This is usually done once - not for each job. Only included
# here to show all steps involved in sailfish quantitation.
zcat $SAILFISH_TEST_DATA/gencode.vM9.transcripts.fa.gz \
    &gt; gencode.vM9.transcripts.fa

# index the transcripts
sailfish index -t gencode.vM9.transcripts.fa -o gencode.vM9.idx \
    -p $SLURM_CPUS_PER_TASK

# quantify the transcripts
cp $SAILFISH_TEST_DATA/ENCFF138LJO.fastq.gz .
sailfish quant -i gencode.vM9.idx -l U \
    -r &lt;(zcat ENCFF138LJO.fastq.gz) \
    -o quant -p $SLURM_CPUS_PER_TASK
cp -r quant $wd
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=8 --mem=8g --gres=lscratch:16 sailfish.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. sailfish.swarm). For example:</p>

<pre class="term">
sailfish quant -i gencode.vM9.idx -l U -r &lt;(zcat sample1.fq.gz) \
    -o quant_sample1 -p $SLURM_CPUS_PER_TASK
sailfish quant -i gencode.vM9.idx -l U -r &lt;(zcat sample2.fq.gz) \
    -o quant_sample2 -p $SLURM_CPUS_PER_TASK
sailfish quant -i gencode.vM9.idx -l U -r &lt;(zcat sample3.fq.gz) \
    -o quant_sample3 -p $SLURM_CPUS_PER_TASK
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f sailfish.swarm -g 8 -t 8 --module sailfish/0.10.0</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>-t # <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td>--module sailfish <td>Loads the sailfish module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
