<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'salmonte on Biowulf';</script>
<div class="title">salmonte on Biowulf</div>

<table width=25% align=right>
    <tr>
        <td>
            <div class="toc">
                <div class="tocHeading">Quick Links</div>
                <div class="tocItem"><A href="#doc">Documentation</a></div>
                <div class="tocItem"><a href="#notes">Notes</a></div>
                <div class="tocItem"><a href="#int">Interactive job </a></div>
                <div class="tocItem"><a href="#sbatch">Batch job </a></div>
            </div>
        </td>
    </tr>
</table>

<p> SalmonTE is an ultra-Fast and Scalable Quantification Pipeline of Transpose
Element (TE) Abundances. It is based on snakemake, salmon and R. Note that SalmonTE
packages its own version of salmon.</p>

<h3>References:</h3>
<ul>
<li>H. H. Jeong, H. K. Yalamanchili, C. Guo C, J. M. Shulman, and Z. Liu
<em>An ultra-fast and scalable quantification pipeline for transposable elements from next generation sequencing data.</em> Pac Symp Biocomput. 23:168-179 (2018).
<a href="https://www.ncbi.nlm.nih.gov/pubmed/29218879">PubMed</a>&nbsp;|&nbsp;
<a href="https://www.worldscientific.com/doi/abs/10.1142/9789813235533_0016">Journal</a>
</li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li>SalmonTE on <a href="https://github.com/LiuzLab/SalmonTE">GitHub</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
<li>Module Name: salmonte (see <a href="/apps/modules.html">the modules page</a>
for more information)</li>
<li><code>SalmonTE.py quant</code> is multithreaded. Please match the number of
threads to the number of allocated CPUs</li>
<li>Example files in <code>$SALMONTE_TEST_DATA</code></li>
<li>Reference data in /fdb/salmonte/</li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=6g --cpus-per-task=4 --gres=lscratch:10</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144]$ <b>module load salmonte</b>
[user@cn3144]$ <b>cp -r ${SALMONTE_TEST_DATA:-none}/data .</b>
[user@cn3144]$ <b>ls -lh data</b>
total 5.0M
-rw-rw-r-- 1 user group 634K Nov 11 10:13 CTRL_1_R1.fastq
-rw-rw-r-- 1 user group 634K Nov 11 10:13 CTRL_1_R2.fastq
-rw-rw-r-- 1 user group 634K Nov 11 10:13 CTRL_2_R1.fastq
-rw-rw-r-- 1 user group 634K Nov 11 10:13 CTRL_2_R2.fastq
-rw-rw-r-- 1 user group 634K Nov 11 10:13 TARDBP_1_R1.fastq
-rw-rw-r-- 1 user group 634K Nov 11 10:13 TARDBP_1_R2.fastq
-rw-rw-r-- 1 user group 634K Nov 11 10:13 TARDBP_2_R1.fastq
-rw-rw-r-- 1 user group 634K Nov 11 10:13 TARDBP_2_R2.fastq
[user@cn3144]$ <b>SalmonTE.py quant  --reference=hs --outpath=quant_out \
                    --num_threads=$SLURM_CPUS_PER_TASK --exprtype=count data</b>
2019-11-11 10:19:30,550 Starting quantification mode
2019-11-11 10:19:30,550 Collecting FASTQ files...
2019-11-11 10:19:30,553 The input dataset is considered as a paired-ends dataset.
2019-11-11 10:19:30,553 Collected 4 FASTQ files.
2019-11-11 10:19:30,553 Quantification has been finished.
2019-11-11 10:19:30,553 Running Salmon using Snakemake
...
[user@cn3144]$ <b>ls -lh quant_out</b>
total 68K
-rw-rw-r-- 1 user group  23K Nov 11 10:19 clades.csv
-rw-rw-r-- 1 user group   63 Nov 11 10:19 condition.csv
drwxrwxr-x 5 user group 4.0K Nov 11 10:19 CTRL_1
drwxrwxr-x 5 user group 4.0K Nov 11 10:19 CTRL_2
-rw-rw-r-- 1 user group  17K Nov 11 10:19 EXPR.csv
-rw-rw-r-- 1 user group  161 Nov 11 10:19 MAPPING_INFO.csv
drwxrwxr-x 5 user group 4.0K Nov 11 10:19 TARDBP_1
drwxrwxr-x 5 user group 4.0K Nov 11 10:19 TARDBP_2
</pre>

<p>Notes:</p>
<ol>
    <li>SalmonTE tries to determine if data is paired end by comparing sequence
        ids. This can fail with certain id lines like, for example, those
        produced by the sra-toolkit. In other words, it's a fragile
        autodetection and fastq files may have to be reformatted to make it
        work correctly.</li>
    <li>SalmonTE supports compressed fastq files</li>
</ol>

<p>Before running the differential expression test, it is necessary
to update the file quant_out/condition.csv to include your experimental
conditions.</p>

<pre class="term">
[user@cn3144]$ <b>mv quant_out/condition.csv quant_out/condition.csv.orig</b>
[user@cn3144]$ <b>cat &lt;&lt;EOF &gt; quant_out/condition.csv</b>
SampleID,condition
TARDBP_1,treatment
CTRL_1,control
TARDBP_2,treatment
CTRL_2,control
<b>EOF</b>
[user@cn3144]$ ### or just edit the condition.csv file with your favorite text editor
[user@cn3144]$ <b>SalmonTE.py test --inpath=quant_out --outpath=test_out \
                    --tabletype=csv --figtype=png --analysis_type=DE \
                    --conditions=control,treatment</b>
[user@cn3144]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. salmonte.sh), which uses the input file 'salmonte.in'. For example:</p>

<pre class="term">
#!/bin/bash
module load salmonte/0.4 || exit 1
SalmonTE.py quant  --reference=hs --outpath=all_quant_out \
    --num_threads=$SLURM_CPUS_PER_TASK --exprtype=count data</b>
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=8 --mem=10g salmonte.sh</pre>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
