<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'samtools on Biowulf';</script>
<div class="title">samtools on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>Samtools is a suite of applications for processing high throughput sequencing data:</p>
<ul>
    <li><em><code>samtools</code></em> is used for working with SAM, BAM, and CRAM files containing aligned
      sequences. It can also be used to index fasta files.</li>
    <li><em><code>bcftools</code></em> is used for working with BCF2, VCF, and gVCF files containing
      variant calls.</li>
    <li><em><code>bgzip</code></em> is a block compression/decompression utility. <code>bgzip</code> compressed
      files can be decompressed with <code>gunzip</code>. However, they are in fact made up
      of a number of compressed blocks which facilitates indexing and random access.</li>
    <li><em><code>tabix</code></em> is used to index <code>bgzip</code> compressed tabular data. It can create <code>.tbi</code> and
      <code>.csi</code> format indices.</li>
    <li><em><code>wgsim</code></em> is used to simulate NGS reads.</li>
    <li>various conversion utilities (<code>bowtie2sam</code>, <code>novo2sam</code>, <code>blast2sam</code>, <code>psl2sam</code>, ...)</li>
    <li><em><code>htslib</code></em> is a library for reading and writing the formats mentioned above. <code>samtools</code>
      and <code>bcftools</code> are based on <code>htslib</code></li>
</ul>

<p>Each of the main tools and file formats have their own manpages
(e. g. <code>man samtools</code> or <code>man sam</code>).</p>
</p>

<h3>Notes on releases</h3>
<p>For more detailed release notes see the GitHub release pages for
<a href="https://github.com/samtools/htslib/releases">htslib</a> and
<a href="https://github.com/samtools/samtools/releases">samtools</a>.</p>

<ul>
<li>There have been a number of visible changes with the transition to
the &gt; 1.0 versions based on <code>htslib</code>.</li>
<li>Accessing files via https in addition to http and ftp became available in samtools 1.3. Use
URLs like <code>http://...</code>, <code>https://...</code>, or
<code>ftp://...</code></li>
<li>Accessing files stored in Amazon S3 became available in samtools
1.3. Use pseudo URLs in the format <code>s3://bucket/...</code></li>
<li>Accessing files stored in Google cloud storage became available in
samtools 1.4. Use pseudo URLs in the format <code>gs://bucket/...</code></li>
<li>Accessing files in the HPC object store is possible since version 1.5 with
a local patch.</li>
<li>Version 1.10 is currently not able to access the HPC object store</li>
</ul>

<h3>References:</h3>
<ul>
<li>Heng Li et al. <em>The Sequence alignment/map (SAM) format and SAMtools.</em> 
  Bioinformatics 2009, 25:2078-2079. 
  <a href="http://www.ncbi.nlm.nih.gov/pubmed/19505943">Pubmed</a>&nbsp;|&nbsp;
  <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2723002/">PMC</a>&nbsp;|&nbsp;
  <a href="http://bioinformatics.oxfordjournals.org/content/25/16/2078.long">Journal</a></li>
<li>Heng Li. <em>A statistical framework for SNP calling, mutation discovery, 
  association mapping and population genetical parameter estimation from 
  sequencing data</em>. 
  Bioinformatics 2011, 27:2987-2993.
  <a href="http://www.ncbi.nlm.nih.gov/pubmed/21903627">Pubmed</a>&nbsp;|&nbsp;
  <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3198575/">PMC</a>&nbsp;|&nbsp;
  <a href="http://bioinformatics.oxfordjournals.org/content/27/21/2987.long">Journal</a></li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="http://www.htslib.org/">Home page</a></li>
<li><a href="http://www.htslib.org/doc/#manuals">Manual</a></li>
<li><a href="https://github.com/samtools/">GitHub</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: samtools (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>Samtools is a multithreaded application</li>
    <li>Example data can be found in <code>$SAMTOOLS_TEST_DATA</code></li>
</ul>

<p class="alert">If the memory per thread is set to a low value for example
by not providing a unit [K/M/G], a potentially <strong>very large</strong>
number of temporary files will be created. Please take care to properly set
this value. Starting with version 1.4 there is a minimum amount of
memory per thread. Check the manpage for details.</p>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=8g --cpus-per-task=4 --gres=lscratch:30</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load samtools</b>
[user@cn3144 ~]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144 ~]$ <b>cp $SAMTOOLS_TEST_DATA/* .</b>
[user@cn3144 ~]$ <b>ls -l</b>
-rw-r--r-- 1 user group 45M Jan 18 16:15 mm10_500k_pe.bam
[user@cn3144 ~]$ # show the header
[user@cn3144 ~]$ <b>samtools view -H mm10_500k_pe.bam</b>
@SQ     SN:chr10        LN:130694993
@SQ     SN:chr11        LN:122082543
@SQ     SN:chr12        LN:120129022
@SQ     SN:chr13        LN:120421639
...snip...

[user@cn3144 ~]$ # sort using lscratch for temp files
[user@cn3144 ~]$ <b>samtools sort -O bam -o mm10_500k_pe_sorted.bam \
                       -T /lscratch/$SLURM_JOB_ID/mm10_500k_pe \
                       -@4 -m 1800M mm10_500k_pe.bam</b>
[user@cn3144 ~]$ # index the sorted file
[user@cn3144 ~]$ <b>samtools index mm10_500k_pe_sorted.bam</b>

[user@cn3144 ~]$
</pre>

<h3>Accessing data stored on Amazon AWS and Google cloud storage</h3>
<p>Access public data via https or on Amazon S3 and Google cloud storage
(GCS). Only version 1.4 and newer support both protocols:</p>

<pre class="term">
[user@cn3144 ~]$ <b>module load samtools</b>
[user@cn3144 ~]$ <b>samtools view \
https://storage.googleapis.com/genomics-public-data/platinum-genomes/bam/NA12877_S1.bam \
chr20:100000-100001 | head -5</b>
[M::test_and_fetch] downloading file 'https://storage.googleapis.com/genomics-public-data/platinum-genom es/bam/NA12877_S1.bam.bai' to local directory
...snip...
[user@cn3144 ~]$ <b>export HTS_S3_V2=1</b>  ### this is only required for 1.10 for the public buckets
[user@cn3144 ~]$ <b>samtools view \
s3://1000genomes/phase1/data/NA12878/exome_alignment/NA12878.mapped.illumina.mosaik.CEU.exome.20110411.bam \
20:100000-100001</b>
[M::test_and_fetch] downloading file 's3://1000genomes/phase1/data/NA12878/exome_alignment/NA12878.mappe d.illumina.mosaik.CEU.exome.20110411.bam.bai' to local directory
...snip...
[user@cn3144 ~]$ <b>unset GCS_OAUTH_TOKEN</b>
[user@cn3144 ~]$ <b>samtools view \
gs://genomics-public-data/platinum-genomes/bam/NA12877_S1.bam \
chr20:100000-100001 | head -5</b>
...snip...
</pre>

<p>Note that if you have the environment variable
<code>GCS_OAUTH_TOKEN</code> set the last command will fail with a
permissions error. This variable is, however, needed to access data in
your private GCS buckets.</p>

<p>Now, upload data to your private GCS bucket and access it with
samtools. This assumes that you have GCS set up and the google-cloud-sdk
set up and initiated. In the example below, the bucket is named 
<code>test-2050be34</code>. One complication is that the python
environments required for Google cloud SDK also contain a samtools
version which may be older than 1.4 and may not work. If you get
'Protocol not supported' errors check that a samtools &gt;= 1.4 is 
first on your path.</p>

<pre class="term">
[user@cn3144 ~]$ <b>module load google-cloud-sdk</b>
[user@cn3144 ~]$ # upload some data to your bucket
[user@cn3144 ~]$ <b>gsutil cp gcat_set_053.bam* gs://test-2050be34/</b>
[user@cn3144 ~]$ # create a application token with gcloud. This should only be
      # necessary once; this involves a sign-in on the web
[user@cn3144 ~]$ <b>gcloud auth application-default login</b>
[user@cn3144 ~]$ # save token to environment variable for samtools.
[user@cn3144 ~]$ <b>export GCS_OAUTH_TOKEN=$(gcloud auth application-default print-access-token)</b>
[user@cn3144 ~]$ # make sure the python/bin/samtools does not interfere
[user@cn3144 ~]$ <b>module unload google-cloud-sdk</b>
[user@cn3144 ~]$ <b>samtools view gs://test-2050be34/gcat_set_053.bam chr20:100000-200001 | head -5</b>
</pre>

<p>For S3, samtools will try to obtain tokens from either the URL,
environment variables, or ~/.aws/credentials or ~/.s3cfg to access
private buckets. Profiles can be used in which case URLs take the format
<code>s3://profile@bucket/...</code>. 

<a name="hpc_obj"></a><h3>Accessing data on the HPC object store</h3>
<p class="alert">Currently not available for 1.10</p>
<p>Samtools starting with version 1.5 can now access data in our local
object store with some preparation. First, let's create or modify a 
<code>~/.s3cfg</code> file to be able to access our object store.
Note that the settings are kept in a named profile (obj) rather than the default 
profile. This helps to avoid interfering with the commands above. However,
you may choose to make the local object store the default, in which case the
profile name does not have to be included in the s3 URLs. The important
settings are <code>access_key, secret_key, host_base, and url_mode</code>.</p>

<pre class="term">
node$ <b>cat &gt;&gt; $HOME/.s3cfg &lt;&lt;EOF</b>
[obj]
access_key = [...your key...]
secret_key = [...your secret...]
host_base = os2access1
host_bucket = os2access1/$USER
url_mode = path
proxy_host = dtn03-e0
proxy_port = 3128
use_https = False

encrypt = False
server_side_encryption = False
human_readable_sizes = True
website_endpoint = http://os2access1/$USER
<b>EOF</b>
</pre>

<p>Now copy a bam file and it's index to the object store</p>
<pre class="term">
[user@cn3144 ~]$ <b>obj_put gcat_set_053.bam</b>
[user@cn3144 ~]$ <b>obj_put gcat_set_053.bam.bai</b>
</pre>

<p>And access them directly with samtools. Note that the <code>obj@</code> is
only required if your HPC object store settings are not the default in
<code>~/.s3cfg</code>. Note also that for now the <code>+http</code>
is required to force samtools to use http. This may change in the future.</p>
<pre class="term">
[user@cn3144 ~]$ <b>samtools view -H s3+http://obj@user/gcat_set_053.bam</b>
@HD     VN:1.3  SO:coordinate
@SQ     SN:chrM LN:16571
@SQ     SN:chr1 LN:249250621
@SQ     SN:chr2 LN:243199373
@SQ     SN:chr3 LN:198022430
[...snip...]
[user@cn3144 ~]$ <b>samtools view s3+http://obj@user/gcat_set_053.bam chr1:2000000-2000100</b>
11V6WR1:111:D1375ACXX:1:2304:6028:48213 99      chr1    1999930 60      100M    = 
[...snip...]
</pre>

<p>End the interactive session</p>
<pre class="term">
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. samtools.sh), which uses the input file 'samtools.in'. For example:</p>

<pre class="term">
#!/bin/bash
set -e

module load samtools 
module load gnuplot 
cd /data/$USER/test_data
samtools stats bam/gcat_set_025.bam &gt; bam/gcat_set_025.bam.stats
plot-bamstats -p  bam/gcat_set_025_stats/ bam/gcat_set_025.bam.stats
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] samtools.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. samtools.swarm). For example:</p>

<pre class="term">
samtools rmdup bam/read1_250k_sorted.bam bam/read1_250k_sorted_rmdup.bam
samtools rmdup bam/read1_500k_sorted.bam bam/read1_500k_sorted_rmdup.bam
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f samtools.swarm [-g #] [-t #] --module samtools</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>-t # <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td>--module samtools <td>Loads the samtools module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
