<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'selscan: haplotype based scans for selection ';</script>
<div class="title"><b>selscan: haplotype based scans for selection
 </b></div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
selscan is a tool for haplotype-based scans to detect natural selection, 
which are useful to identify recent or ongoing positive selection in genomes.
It is an efficient multithreaded application that implements Extended Haplotype Homozygosity (EHH), 
Integrated Haplotype Score (iHS), and Cross-population EHH (XPEHH). 
selscan accepts phased genotypes in multiple formats, including TPED.
</p>

<h3>References:</h3>
<ul>
<li>Zachary A. Szpiech and Ryan D. Hernandez <br />
<i>selscan: An Efficient Multithreaded Program to Perform EHH-Based Scans for Positive Selection</i><br />
<a href="https://academic.oup.com/mbe/article/31/10/2824/1012603"> Molecular Biology and Evolution</a> 2014, <b>31</b>
(10):2824–2827 doi:10.1093/molbev/msu211 <br/>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/szpiech/selscan">selscan Github page</a></li>
<li><a href="https://github.com/szpiech/selscan/blob/master/manual/selscan-manual.pdf">selscan User Manual</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>selscan</tt> (see <a href="https://hpc.nih.gov/apps/modules.html">the modules page</a> for more information)
<li>Unusual environment variables set 
  <ul>
    <li><b>SELSCAN_HOME</b>&nbsp; installation directory</li>
    <li><b>SELSCAN_BIN </b>&nbsp; &nbsp; &nbsp; executable directory</li>
    <li><b>SELSCAN_SRC </b>&nbsp; &nbsp; &nbsp; source code directory</li>
    <li><b>SELSCAN_DATA</b>&nbsp; sample data and checkpoints directory
  </ul>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=10g --gres=lscratch:10 -c8</b>
[user@cn3101 ~]$<b>module load selscan  </b>
[+] Loading selscan  1.3.0    
</pre>
The available executables are:
<pre class="term">
[user@cn3101]$ <b>ls $SELSCAN_BIN </b>
norm  selscan 
</pre>
In particular, the command line options of the executable selscan are as follows:                  
<pre class="term">
[user@cn3101]$ <b>selscan --help</b>
selscan v1.3.0

selscan v1.3.0 -- a program to calculate EHH-based scans for positive selection in genomes.
Source code and binaries can be found at <https://www.github.com/szpiech/selscan>.

selscan currently implements EHH, iHS, XP-EHH, and nSL.

Citations:

selscan: ZA Szpiech and RD Hernandez (2014) MBE 31: 2824-2827.
iHH12: R Torres et al. (2018) PLoS Genetics 15: e1007898.
       N Garud et al. (2015) PLoS Genetics 11: 1–32.
nSL: A Ferrer-Admetlla et al. (2014) MBE 31: 1275-1291.
XP-nSL: Szpiech et al. (2020) bioRxiv doi:
        https://doi.org/10.1101/2020.05.19.104380.
XP-EHH: PC Sabeti et al. (2007) Nature 449: 913–918.
        K Wagh et al. (2012) PloS ONE 7: e44751.
iHS: BF Voight et al. (2006) PLoS Biology 4: e72.
EHH: PC Sabeti et al. (2002) Nature 419: 832–837.

To calculate EHH:

./selscan --ehh <locusID> --vcf <vcf> --map <mapfile> --out <outfile>

To calculate iHS:

./selscan --ihs --vcf <vcf> --map <mapfile> --out <outfile>

To calculate nSL:

./selscan --nsl --vcf <vcf> --out <outfile>

To calculate XP-nSL:

./selscan --xpnsl --vcf <vcf> --vcf-ref <vcf> --out <outfile>

To calculate iHH12:

./selscan --ihh12 --vcf <vcf> --map <mapfile> --out <outfile>

To calculate XP-EHH:

./selscan --xpehh --vcf <vcf> --vcf-ref <vcf> --map <mapfile> --out <outfile>

----------Command Line Arguments----------

--alt <bool>: Set this flag to calculate homozygosity based on the sum of the
        squared haplotype frequencies in the observed data instead of using
        binomial coefficients.
        Default: false

--cutoff <double>: The EHH decay cutoff.
        Default: 0.05

--ehh <string>: Calculate EHH of the '1' and '0' haplotypes at the specified
        locus. Output: <physical dist> <genetic dist> <'1' EHH> <'0' EHH>
        Default: __NO_LOCUS__

--ehh-win <int>: When calculating EHH, this is the length of the window in bp
        in each direction from the query locus.
        Default: 100000

--gap-scale <int>: Gap scale parameter in bp. If a gap is encountered between
        two snps > GAP_SCALE and < MAX_GAP, then the genetic distance is
        scaled by GAP_SCALE/GAP.
        Default: 20000

--hap <string>: A hapfile with one row per haplotype, and one column per
        variant. Variants should be coded 0/1
        Default: __hapfile1

--help <bool>: Prints this help dialog.
        Default: false

--ihh12 <bool>: Set this flag to calculate iHH12.
        Default: false

--ihs <bool>: Set this flag to calculate iHS.
        Default: false

--ihs-detail <bool>: Set this flag to write out left and right iHH scores for '1' and '0' in addition to iHS.
        Default: false

--keep-low-freq <bool>: Include low frequency variants in the construction of your haplotypes.
        Default: false

--maf <double>: If a site has a MAF below this value, the program will not use
        it as a core snp.
        Default: 0.05

--map <string>: A mapfile with one row per variant site.
        Formatted <chr#> <locusID> <genetic pos> <physical pos>.
        Default: __mapfile

--max-extend <int>: The maximum distance an EHH decay curve is allowed to extend from the core.
        Set <= 0 for no restriction.
        Default: 1000000

--max-extend-nsl <int>: The maximum distance an nSL haplotype is allowed to extend from the core.
        Set <= 0 for no restriction.
        Default: 100

--max-gap <int>: Maximum allowed gap in bp between two snps.
        Default: 200000

--nsl <bool>: Set this flag to calculate nSL.
        Default: false

--out <string>: The basename for all output files.
        Default: outfile

--pi <bool>: Set this flag to calculate mean pairwise sequence difference in a sliding window.
        Default: false

--pi-win <int>: Sliding window size in bp for calculating pi.
        Default: 100

--pmap <bool>: Use physical map instead of a genetic map.
        Default: false

--ref <string>: A hapfile with one row per haplotype, and one column per
        variant. Variants should be coded 0/1. This is the 'reference'
        population for XP-EHH calculations.  Ignored otherwise.
        Default: __hapfile2

--skip-low-freq <bool>: **This flag is now on by default. If you want to include low frequency variants
in the construction of your haplotypes please use the --keep-low-freq flag.
        Default: false

--threads <int>: The number of threads to spawn during the calculation.
        Partitions loci across threads.
        Default: 1

--tped <string>: A TPED file containing haplotype and map data.
        Variants should be coded 0/1
        Default: __hapfile1

--tped-ref <string>: A TPED file containing haplotype and map data.
        Variants should be coded 0/1. This is the 'reference'
        population for XP-EHH calculations and should contain the same number
        of loci as the query population. Ignored otherwise.
        Default: __hapfile2

--trunc-ok <bool>: If an EHH decay reaches the end of a sequence before reaching the cutoff,
        integrate the curve anyway (iHS and XPEHH only).
        Normal function is to disregard the score for that core.
        Default: false

--vcf <string>: A VCF file containing haplotype data.
        A map file must be specified with --map.
        Default: __hapfile1

--vcf-ref <string>: A VCF file containing haplotype and map data.
        Variants should be coded 0/1. This is the 'reference'
        population for XP-EHH calculations and should contain the same number
        of loci as the query population. Ignored otherwise.
        Default: __hapfile2

--wagh <bool>: Set this flag to calculate XP-EHH using definition of EHH which
        separates core SNP alleles in the denominator.
        Default: false

--xpehh <bool>: Set this flag to calculate XP-EHH.
        Default: false

--xpnsl <bool>: Set this flag to calculate XP-nSL.
        Default: false
</pre>
To perform training of the predictor network using this executable, copy sample data to the current folder:
<pre class="term">
[user@cn3101]$ <b>cp $SELSCAN_DATA/* .</b>
</pre>
A sample command to run selscan:
<pre class="term">
[user@cn3101]$ <b>selscan --ehh Locus1 --hap example2.pop2.hap --map example2.map --out my_output</b>
selscan v1.3.0
Opening example2.pop2.hap...
Loading 125 haplotypes and 12920 loci...
Opening example2.map...
Loading map data for 12920 loci
Found Locus1 in data.
--skip-low-freq set. Removing all variants &lt; 0.05.
Removed 8039 low frequency variants.
</pre>
The command the following output files:
<pre class="term">
my_output.ehh.Locus1.log
my_output.ehh.Locus1.out
my_output.ehh.Locus1.out.anc.colormap
my_output.ehh.Locus1.out.der.colormap
</pre>
End the interactive session:
<pre class="term">
[user@cn3101 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
