<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Shapeit on HPC';</script>
    <div class="title">Shapeit on HPC</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</td></tr></table>    <p> <strong>SHAPEIT</strong> is a fast and accurate method for <a href="http://en.wikipedia.org/wiki/Haplotype_estimation">estimation 
      of haplotypes</a> (aka phasing) from genotype or sequencing data.<br />
      <br />
      <strong>SHAPEIT</strong> has several notable features:</p>
    <ul>
      <li> Linear complexity with the number of SNPs and conditioning haplotypes. 
      </li>
      <li>Whole chromosome GWAS scale datasets can be phased in a single run.</li>
      <li>Phasing individuals with any level of relatedness</li>
      <li>Phasing is multi-threaded to tailor computational times to your resources</li>
      <li>Handles X chromosomes phasing</li>
      <li>Phasing using a reference panel (eg.1,000 Genomes) to aid phasing</li>
      <li>Ideal for pre-phasing imputation together with IMPUTE2</li>
    </ul>
<h3>References:</h3>
    <ul>
      <li><strong>SHAPEIT</strong> has primarily been developed by <a href="http://funpopgen.unige.ch/members/olivier-delaneau">Dr 
        Olivier Delaneau</a> through a collaborative project between the research 
        groups of <a href="http://gba.cnam.fr/index_eng.php?index=jfz">Prof Jean-Francois 
        Zagury </a>at <a href="http://www.cnam.fr/">CNAM</a> and <a href="http://www.stats.ox.ac.uk/~marchini/">Prof 
        Jonathan Marchini </a>at <a href="http://www.ox.ac.uk/">Oxford</a>. Funding 
        for this project has been received from several sources : <a href="http://www.cnam.fr/">CNAM</a>, 
        <a href="http://www.peptinov.fr/">Peptinov</a>, <a href="http://www.mrc.ac.uk/index.htm">MRC</a>, 
        <a href="http://www.leverhulme.ac.uk/">Leverhulme</a>, <a href="http://www.wellcome.ac.uk/">The 
        Wellcome Trust</a>.</li>
    </ul>


<a Name="doc"></a><div class="heading">Documentation</div>
    <ul>
      <li><a href="https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.html">https://mathgen.stats.ox.ac.uk/genetics_software/shapeit/shapeit.html</a></li>
    </ul>

<a Name="notes"></a>
    <div class="heading"><a name="notes" id="notes"></a>Important Notes</div>
    <ul>
      <li>Module Name: <tt>shapeit</tt> (see <a href="/apps/modules.html">the 
        modules page</a> for more information) </li>
      <li>Multithreaded</li>
      <li>Example files in <tt>/usr/local/apps/shapeit/example</tt></li>
    </ul>
<P>

<a Name="int"></a></p><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>    
    <pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=4</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <strong>module load shapeit
</strong>[user@cn3144 ~]$ <strong>shapeit --input-bed gwas.bed gwas.bim gwas.fam --input-map genetic_map.txt --output-max gwas.phased.haps gwas.phased.sample --thread $SLURM_CPUS_PER_TASK</strong>

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a></p><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
    <p>Create a batch input file (e.g. batch.sh). For example:</p>

    <pre class="term">
#!/bin/bash
set -e
module load shapeit
shapeit --input-bed gwas.bed gwas.bim gwas.fam --input-map genetic_map.txt --output-max gwas.phased.haps gwas.phased.sample --thread $SLURM_CPUS_PER_TASK</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

    <pre class="term">sbatch --cpus-per-task=4 [--mem=#] batch.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

    <p>Create a swarmfile (e.g. job.swarm). For example:</p>

<pre class="term">
cd dir1; shapeit --input-bed gwas.bed gwas.bim gwas.fam --input-map genetic_map.txt --output-max gwas.phased.haps gwas.phased.sample --thread $SLURM_CPUS_PER_TASK
cd dir2; shapeit --input-bed gwas.bed gwas.bim gwas.fam --input-map genetic_map.txt --output-max gwas.phased.haps gwas.phased.sample --thread $SLURM_CPUS_PER_TASK
cd dir3; shapeit --input-bed gwas.bed gwas.bim gwas.fam --input-map genetic_map.txt --output-max gwas.phased.haps gwas.phased.sample --thread $SLURM_CPUS_PER_TASK
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

    <pre class="term">swarm -f job.swarm [-g #] -t 4 --module shapeit</pre>
where
    <table width="63%" border=0>
      <tr><td width=18%><tt>-g <i>#</i> </tt></td><td width="82%">Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  </td></tr><tr><td><tt>-t <i>#</i></tt> </td><td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  </td></tr><tr>
        <td><tt>--module </tt></td>
        <td>Loads the module for each subjob in the swarm </td>
      </tr></table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
