<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'shasta on Biowulf';</script>
<div class="title">shasta on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
The goal of the Shasta long read assembler is to rapidly produce accurate assembled sequence using DNA reads generated by Oxford Nanopore flow cells as input.
</p>
<p>
Computational methods used by the Shasta assembler include:
</p>
<p>
 <li> &nbsp  Using a run-length representation of the read sequence. This makes the assembly process more resilient to errors in homopolymer repeat counts, which are the most common type of errors in Oxford Nanopore reads.
</li>
 <li>&nbsp   Using in some phases of the computation a representation of the read sequence based on markers, a fixed subset of short k-mers (k â‰ˆ 10).
</li>
</p>
<p>
<ul>
  
</p>
</ul>

<h3>References:</h3>
<ul>
<li>
Shafin K et.al.<em>Nanopore sequencing and the Shasta toolkit enable efficient de novo assembly of eleven human genomes.</em>
Nat Biotechnol. 2020 Sep;38(9):1044-1053. doi: 10.1038/s41587-020-0503-6. Epub 2020 May 4.
    <a href="https://www.ncbi.nlm.nih.gov/pubmed/32686750">PubMed</a>&nbsp;|&nbsp;
    <a href="https://www.nature.com/articles/s41587-020-0503-6">Journal</a>
</li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li>shasta Main Site:<a href="https://github.com/chanzuckerberg/shasta">Main Site</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
    <li>Module Name: <tt>shasta</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
	<pre class="term">
	shasta-Linux-0.11.1 --help</pre>
<li>Environment variables set <!--for ones users should be aware of -->
  <ul>
    <li>$SHASTA_TEST_DATA
  </ul>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=10 --mem=10G</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load shasta</b>
[user@cn3144 ~]$ <b>cp $SHASTA_TEST_DATA/TinyTest.fasta.gz .</b>
[user@cn3144 ~]$ <b>gunzip TinyTest.fasta.gz</b>
[user@cn3144 ~]$ <b>shasta-Linux-0.11.1 \
		--input TinyTest.fasta \
		--config Nanopore-Oct2021 \
		--threads=8 \
		--assemblyDirectory shasta_test</b>
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. shasta.sh). For example:</p>

<pre class="term">
#!/bin/bash
#SBATCH --cpus-per-task=10
#SBATCH --mem=10G
#SBATCH --time=2:00:00
#SBATCH --partition=norm

set -e
module load shasta
shasta-Linux-0.11.1 \
                --input TinyTest.fasta \
                --config Nanopore-Oct2021 \
                --threads=8 \
                --assemblyDirectory shasta_test
</pre>
<p> Submit the job:
<pre class="term">sbatch shasta.sh</pre>



<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

    <p>Create a swarmfile (e.g. job.swarm). For example:</p>

    <pre class="term">

cd dir1; shasta-Linux-0.11.1 --input TinyTest1.fasta --config Nanopore-Oct2021 --threads=8 --assemblyDirectory test1
cd dir2; shasta-Linux-0.11.1 --input TinyTest2.fasta --config Nanopore-Oct2021 --threads=8 --assemblyDirectory test2

    </pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

    <pre class="term">swarm -f job.swarm [-g #] --module shasta</pre>
where
    <table width="63%" border=0>
      <tr>
        <td width=18%><tt>-g <i>#</i> </tt></td>
        <td width="82%">Number of Gigabytes of memory required for each process
          (1 line in the swarm command file) </td>
      </tr>
      <tr>
        <td><tt>--module </tt></td>
        <td>Loads the module for each subjob in the swarm </td>
      </tr>
    </table>














<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
