<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'SIFT (5.x) on Biowulf';</script>
<div class="title">SIFT (5.x) on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
      </div>
</table>

<p>
SIFT predicts whether an amino acid substitution affects protein function based on sequence homology and the physical properties of amino acids. SIFT can be applied to naturally occurring nonsynonymous polymorphisms and laboratory-induced missense mutations.
</p>

<h3>References:</h3>
<ul>
  <li>Sim NL, Kumar P, Hu J, Henikoff S, Schneider G, Ng PC. <ul><a href="https://www.ncbi.nlm.nih.gov/pubmed/22689647">SIFT web server: predicting effects of amino acid substitutions on proteins.</a></ul><em>Nucleic Acids Res. 2012 Jul;40</em></li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<p class="alert">The documentation here pertains to SIFT version 5.x.  For version 6.x, <a href="SIFT_6.html">click here</a>.</p>
<ul>
  <li><a href="http://sift.bii.a-star.edu.sg/">SIFT website</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>SIFT</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
<li>Multithreaded
<li>Environment variables set <!--for ones users should be aware of -->
  <ul>
    <li><tt><b>BLIMPS_DIR</b></tt>: directory where blimps is installed
    <li><tt><b>NCBI</b></tt>: directory where NCBI blast is installed
    <li><tt><b>SIFTDB</b></tt>: directory where SIFT reference files are held
    <li><tt><b>SIFTHOME</b></tt>: directory where SIFT is installed
    <li><tt><b>SIFT_SCRATCHDIR</b></tt>: directory where SIFT output is written
  </ul>
<li>Example files in <tt>$SIFT/home</tt></li>
<li>Reference data in <tt>/fdb/SIFT/</tt></li>
</ul>

<p class="alert">SIFT can require a large amount of disk space.  The environment variable <b>$SIFT_SCRATCHDIR</b> is set to
<b>/lscratch/$SLURM_JOB_ID</b> by default, but can be changed.</p>

<pre class="term">export SIFT_SCRATCHDIR=/path/to/new/tmp/area</pre>

<p>SIFT can use the following databases for protein alignment:</p>

<ul>
  <li>/fdb/blastdb/nr -- NCBI non-redundant</li>
  <li>$SIFTDB/uniref90.fa -- UniRef 90</li>
  <li>$SIFTDB/uniprotkb_swissprot.fa -- SwissProt</li>
</ul>


<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ module load SIFT/5.2.2
[user@cn3144 ~]$ cp $SIFTHOME/test/lacI.fasta .
[user@cn3144 ~]$ seqs_chosen_via_median_info.csh lacI.fasta /fdb/blastdb/nr 2.75
[user@cn3144 ~]$ mv $SIFT_SCRATCHDIR/lacI.* .
[user@cn3144 ~]$ cp $SIFTHOME/test/snvs_build37.input .
[user@cn3144 ~]$ SIFT_exome_nssnvs.pl -i snvs_build37.input -d $SIFTDB/Human_db_37 -o $SIFT_SCRATCHDIR -z output.tsv

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. SIFT.sh). For example:</p>

<pre class="term">
#!/bin/bash
module load SIFT/5.2.2
cp $SIFTHOME/test/snvs_build37.input .
SIFT_exome_nssnvs.pl -i snvs_build37.input -d $SIFTDB/Human_db_37 -o $SIFT_SCRATCHDIR -z output.tsv</pre>
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] TEMPLATE.sh</pre>


<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
