<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Slamdunk on Biowulf';</script>
<div class="title">Slamdunk on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
      </div>
</table>

<p>
Streamlining SLAMseq analysis with ultra-high sensitivity: SlamDunk is a novel, fully automated software tool for automated, robust, scalable and reproducible SLAMseq data analysis.
</p>

<h3>References:</h3>
<ul>
<li><a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-019-2849-7">Quantification of experimentally induced nucleotide conversions in high-throughput sequencing datasets</a>, Neumann, T., Herzog, V. A. et al. BMC Bioinformatics, 20(1), 258, 2019.</li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="http://t-neumann.github.io/slamdunk/">Slamdunk documentation</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
    <li>Module Name: <tt>slamdunk</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>Multithreaded</li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. 

<br>Sample session (user input in <b>bold</b>):
</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=8 --mem=20g --gres=lscratch:20</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load slamdunk</b>

[user@cn3144 ~]$ <b>cd /lscratch/$SLURM_JOB_ID</b>

[user@cn3144 ~]$ <b>cp /fdb/genome/hg19/chr_all.fa .</b>

[user@cn3144 ~]$ <b>slamdunk all -r chr_all.fa -b ./exampleBAM.bed \
      -o out -t $SLURM_CPUS_PER_TASK ./exampleBAM.bam</b>
slamdunk all
Running slamDunk map for 1 files (8 threads)
.
Running slamDunk sam2bam for 1 files (8 threads)
.
Creating output directory: out/filter
Running slamDunk filter for 1 files (8 threads)
.
[...]

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. slamdunk.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load slamdunk
cd /lscratch/$SLURM_JOB_ID
cp /fdb/genome/hg19/chr_all.fa .
slamdunk all -r chr_all.fa -b ./exampleBAM.bed \
             -o out -t $SLURM_CPUS_PER_TASK ./exampleBAM.bam
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] [--gres=lscratch:#] slamdunk.sh</pre>


<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
