<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'sniffles on Biowulf';</script>
<div class="title">sniffles on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
Sniffles is a structural variation caller using third generation sequencing (PacBio or Oxford Nanopore). It detects all types of SVs (10bp+) using evidence from split-read alignments, high-mismatch regions, and coverage analysis. Please note the current version of Sniffles requires sorted output from <a href="bwa.html">BWA-MEM</a> (use -M and -x parameter) or <a href="ngmlr.html">NGMLR</a> with the optional SAM attributes enabled
</p>

<h3>References:</h3>
<ul>
<li>Accurate detection of complex structural variations using single-molecule sequencing.
Fritz J. Sedlazeck, Philipp Rescheneder, Moritz Smolka, Han Fang, Maria Nattestad, Arndt von Haeseler & Michael C. Schatz.
Nature Methods
volume 15, 461â€“468 (2018).
<a href="https://www.nature.com/articles/s41592-018-0001-7">[html]</a></li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/fritzsedlazeck/Sniffles">sniffles Main Site</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>sniffles</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
<li>Example files in <tt>/fdb/sniffles</tt>
<li>sniffles uses 4 threads by default.
Specify <tt>--threads</tt> and allocate the corresponding number of threads with your job submission.
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load sniffles</b>
[+] Loading sniffles, version 2.0.2...
[user@cn3144 ~]$ <b>sniffles -i /fdb/sniffles/reads_region.bam -v test.vcf</b>
Estimating parameter...
Max dist between aln events: 4
Max diff in window: 50
Min score ratio: 2
Avg DEL ratio: 0.0432235
Avg INS ratio: 0.0740733
Start parsing... 21
Switch Chr 21
Finalizing  ..
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. sniffles.sh). For example:</p>

<pre class="term">
#!/bin/sh
set -e
module load sniffles
sniffles -i /fdb/sniffles/reads_region.bam -v test.vcf
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] sniffles.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. sniffles.swarm). For example:</p>

<pre class="term">
sniffles -i sample1.bam -v sample1.vcf --threads $SLURM_CPUS_PER_TASK
sniffles -i sample2.bam -v sample2.vcf --threads $SLURM_CPUS_PER_TASK
sniffles -i sample3.bam -v sample3.vcf --threads $SLURM_CPUS_PER_TASK
sniffles -i sample4.bam -v sample4.vcf --threads $SLURM_CPUS_PER_TASK
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f sniffles.swarm [-g #] -t 4 --module sniffles</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module sniffles</tt> <td>Loads the sniffles module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
