<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'SpeedSeq on Biowulf';</script>
<div class="title">SpeedSeq on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
SpeedSeq is a genome analysis platform designed for rapid whole-genome variant detection and interpretation
</p>

<h3>References:</h3>
<ul>
    <li>Chiang C, Layer RM, Faust GG, Lindberg MR, Rose DB, Garrison EP, Marth GT, Quinlan AR, Hall IM. SpeedSeq: ultra-fast personal genome analysis and interpretation. Nature methods. 2015 Aug 10;12(10):966. doi: <a href="https://doi.org/10.1038/nmeth.3505">10.1038/nmeth.3505</a></li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="https://github.com/hall-lab/speedseq">SpeedSeq Main Site</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
    <li>Module Name: <tt>speedseq</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
<li>Environment variables set <!--for ones users should be aware of -->
  <ul>
    <li><tt>SPEEDSEQ_HOME</tt></li>
  </ul>
<li>Example files in <tt>$SPEEDSEQ_HOME/example</tt>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>) based on<br> <tt>$SPEEDSEQ_HOME/example/run_speedseq.sh</tt>:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task 2 --gres lscratch:10</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load speedseq</b>
[+] Loading VEP 95 on cn3092 
[+] Loading singularity  on cn3092 
[+] Loading ROOT 6.13.02  ... 
[+] Loading gcc  7.3.0  ... 
[+] Loading cnvnator  0.3.3 
[+] Loading speedseq, version 0.1.2-20180208-4e60002... 
[user@cn3144 ~]$ <b>speedseq align \
    -o example \
    -M $(expr $SLURM_MEM_PER_NODE / 1000) \
    -t $SLURM_CPUS_PER_TASK \
    -T /lscratch/$SLURM_JOB_ID/speedseq \
    -p \
    -R "@RG\tID:NA12878\tSM:NA12878\tLB:lib1" \
    $SPEEDSEQ_HOME/example/data/human_g1k_v37_20_42220611-42542245.fasta \
    $SPEEDSEQ_HOME/example/data/NA12878.20slice.30X.fastq.gz</b>
[user@cn3144 ~]$ <b>speedseq var \
    -T /lscratch/$SLURM_JOB_ID/speedseq \
    -t $SLURM_CPUS_PER_TASK \
    -o example \
    $SPEEDSEQ_HOME/example/data/human_g1k_v37_20_42220611-42542245.fasta \
    example.bam</b>
[user@cn3144 ~]$ <b>speedseq sv \
    -T /lscratch/$SLURM_JOB_ID/speedseq \
    -o example \
    -B example.bam \
    -S example.splitters.bam \
    -D example.discordants.bam \
    -R $SPEEDSEQ_HOME/example/data/human_g1k_v37_20_42220611-42542245.fasta</b>
[user@cn3144 ~]$ <b>speedseq realign \
    -t $SLURM_CPUS_PER_TASK \
    -T /lscratch/$SLURM_JOB_ID/speedseq \
    -o example.realign \
    -M $(expr $SLURM_MEM_PER_NODE / 1000) \
    $SPEEDSEQ_HOME/example/data/human_g1k_v37_20_42220611-42542245.fasta \
    example.bam</b>
[user@cn3144 ~]$ <b> </b>
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. speedseq.sh). For example:</p>

<pre class="term">
#!/bin/sh
set -e
module load speedseq

test -n "$SLURM_CPUS_PER_TASK" || SLURM_CPUS_PER_TASK=2

speedseq align \
    -o example \
    -M $(expr $SLURM_MEM_PER_NODE / 1000) \
    -t $SLURM_CPUS_PER_TASK \
    -T /lscratch/$SLURM_JOB_ID/speedseq \
    -p \
    -R "@RG\tID:NA12878\tSM:NA12878\tLB:lib1" \
    $SPEEDSEQ_HOME/example/data/human_g1k_v37_20_42220611-42542245.fasta \
    $SPEEDSEQ_HOME/example/data/NA12878.20slice.30X.fastq.gz

speedseq var \
    -T /lscratch/$SLURM_JOB_ID/speedseq \
    -t $SLURM_CPUS_PER_TASK \
    -o example \
    $SPEEDSEQ_HOME/example/data/human_g1k_v37_20_42220611-42542245.fasta \
    example.bam

speedseq sv \
    -T /lscratch/$SLURM_JOB_ID/speedseq \
    -o example \
    -B example.bam \
    -S example.splitters.bam \
    -D example.discordants.bam \
    -R $SPEEDSEQ_HOME/example/data/human_g1k_v37_20_42220611-42542245.fasta

speedseq realign \
    -t $SLURM_CPUS_PER_TASK \
    -T /lscratch/$SLURM_JOB_ID/speedseq \
    -o example.realign \
    -M $(expr $SLURM_MEM_PER_NODE / 1000) \
    $SPEEDSEQ_HOME/example/data/human_g1k_v37_20_42220611-42542245.fasta \
    example.bam
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] --gres lscratch:# speedseq.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. speedseq.swarm). For example:</p>

<pre class="term">
speedseq align \
    -o example1 \
    -M $(expr $SLURM_MEM_PER_NODE / 1000) \
    -t $SLURM_CPUS_PER_TASK \
    -T /lscratch/$SLURM_JOB_ID/speedseq \
    -p \
    -R "@RG\tID:NA12878\tSM:NA12878\tLB:lib1" \
    $SPEEDSEQ_HOME/example/data/human_g1k_v37_20_42220611-42542245.fasta \
    $SPEEDSEQ_HOME/example/data/NA12878.20slice.30X.fastq.gz
speedseq align \
    -o example2 \
    -M $(expr $SLURM_MEM_PER_NODE / 1000) \
    -t $SLURM_CPUS_PER_TASK \
    -T /lscratch/$SLURM_JOB_ID/speedseq \
    -p \
    -R "@RG\tID:NA12878\tSM:NA12878\tLB:lib1" \
    $SPEEDSEQ_HOME/example/data/human_g1k_v37_20_42220611-42542245.fasta \
    $SPEEDSEQ_HOME/example/data/NA12878.20slice.30X.fastq.gz
speedseq align \
    -o example3 \
    -M $(expr $SLURM_MEM_PER_NODE / 1000) \
    -t $SLURM_CPUS_PER_TASK \
    -T /lscratch/$SLURM_JOB_ID/speedseq \
    -p \
    -R "@RG\tID:NA12878\tSM:NA12878\tLB:lib1" \
    $SPEEDSEQ_HOME/example/data/human_g1k_v37_20_42220611-42542245.fasta \
    $SPEEDSEQ_HOME/example/data/NA12878.20slice.30X.fastq.gz
speedseq align \
    -o example4 \
    -M $(expr $SLURM_MEM_PER_NODE / 1000) \
    -t $SLURM_CPUS_PER_TASK \
    -T /lscratch/$SLURM_JOB_ID/speedseq \
    -p \
    -R "@RG\tID:NA12878\tSM:NA12878\tLB:lib1" \
    $SPEEDSEQ_HOME/example/data/human_g1k_v37_20_42220611-42542245.fasta \
    $SPEEDSEQ_HOME/example/data/NA12878.20slice.30X.fastq.gz
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f speedseq.swarm [-g #] -t # --gres lscratch:# --module speedseq</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
      <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
	  <tr><td><tt>--gres lscratch:#</tt></td> <td>Number of Gigabytes of local scratch space to allocate</td></tr>
  <tr><td><tt>--module speedseq</tt> <td>Loads the TEMPLATE module for each subjob in the swarm 
</table>
















<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
