<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'talon on Biowulf';</script>
<div class="title">talon on Biowulf</div>

<table width=25% align=right>
    <tr>
        <td>
            <div class="toc">
                <div class="tocHeading">Quick Links</div>
                <div class="tocItem"><A href="#doc">Documentation</a></div>
                <div class="tocItem"><a href="#notes">Notes</a></div>
                <div class="tocItem"><a href="#int">Interactive job </a></div>
                <div class="tocItem"><a href="#sbatch">Batch job </a></div>
                <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
            </div>
        </td>
    </tr>
</table>

<div style="width: 70%">

<blockquote> TALON is a Python package for identifying and quantifying known
    and novel genes/isoforms in long-read transcriptome data sets. TALON is
    technology-agnostic in that it works from mapped SAM files, allowing data
    from different sequencing platforms (i.e. PacBio and Oxford Nanopore) to be
    analyzed side by side. </blockquote>

    <h3>References:</h3>
    <ul>
        <li>Dana Wyman et al., <em>A technology-agnostic long-read analysis pipeline for transcriptome discovery and quantification</em>. <a href="https://doi.org/10.1101/672931">bioRxiv</a>
    </ul>
</div>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li>talon on <a href="https://github.com/mortazavilab/TALON">GitHub</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: talon (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
<li>Some talon tools are multithreaded. Please match the number of threads with the number
    of allocated CPUs</li>
<li>Example files in <code>$TALON_TEST_DATA</code></li>
<li>Some talon steps may require large amounts of memory. See <a href="https://github.com/mortazavilab/TALON/issues/90">
        this GitHub issue</a> for a discussion on how to reduce memory usage.</li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run through
the steps using 2 replicates of human cardiac atrium tissue runs on a PacBio Sequel II:</p>

<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=6 --mem=16G --gres=lscratch:50</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144]$ <b>module load talon</b>
[user@cn3144]$ <b>cp -L ${TALON_TEST_DATA:-none}/* .</b>
[user@cn3144]$ <b>ls -lh</b>
total 4.6G
-rw-r--r-- 1 user group 172M Apr 20 14:31 ENCFF291EKY.bam
-rw-r--r-- 1 user group 1.7M Apr 20 14:31 ENCFF291EKY.bam.bai
-rw-r--r-- 1 user group 189M Apr 20 14:31 ENCFF613SDS.bam
-rw-r--r-- 1 user group 1.7M Apr 20 14:31 ENCFF613SDS.bam.bai
-rw-r--r-- 1 user group 1.3G Apr 20 14:31 gencode.v35.primary_assembly.annotation.gtf
-rw-r--r-- 1 user group 3.0G Apr 20 14:31 GRCh38.primary_assembly.genome.fa
-rw-r--r-- 1 user group 6.4K Apr 20 14:31 GRCh38.primary_assembly.genome.fa.fai
[user@cn3144]$ <b>gtf=gencode.v35.primary_assembly.annotation.gtf</b>
[user@cn3144]$ <b>genome=GRCh38.primary_assembly.genome.fa</b>
[user@cn3144]$ <b>bam1=ENCFF291EKY.bam</b>
[user@cn3144]$ <b>bam2=ENCFF613SDS.bam</b>
[user@cn3144]$ <b>talon_initialize_database \
        --f $gtf \
        --a gencode_35 \
        --g GRCh38 \
        --o example_talon</b>
chr1
bulk update genes...
bulk update gene_annotations...
bulk update transcripts...
[...snip...]
[user@cn3144]$ <b>mkdir -p labeled tmp</b>
[user@cn3144]$ ### check internal priming sites
[user@cn3144]$ <b>talon_label_reads --f $bam1 \
        --g $genome  \
        --t $SLURM_CPUS_PER_TASK \
        --ar 20 \
        --tmpDir=/lscratch/$SLURM_JOB_ID/tmp \
        --deleteTmp \
        --o labeled/${bam1%.bam}</b>
[ 2021-04-20 17:10:44 ] Started talon_label_reads run.
[ 2021-04-20 17:10:44 ] Splitting SAM by chromosome...
[ 2021-04-20 17:10:44 ] -----Writing chrom files...
[ 2021-04-20 17:10:59 ] Launching parallel jobs...
[ 2021-04-20 17:11:14 ] Pooling output files...
[ 2021-04-20 17:11:27 ] Run complete
[user@cn3144]$ <b>talon_label_reads --f $bam2 \
        --g $genome  \
        --t $SLURM_CPUS_PER_TASK \
        --ar 20 \
        --tmpDir=/lscratch/$SLURM_JOB_ID/tmp \
        --deleteTmp \
        --o labeled/${bam2%.bam}</b>
[...snip...]
[user@cn3144]$ ### run talon annotator
[user@cn3144]$ <b>cat > config.csv &lt;&lt;__EOF__</b>
ex_rep1,GRCh38,PacBio-Sequel2,labeled/${bam1%.bam}_labeled.sam
ex_rep2,GRCh38,PacBio-Sequel2,labeled/${bam2%.bam}_labeled.sam
<b>__EOF__</b>
[user@cn3144]$ <b>talon \
        -t $SLURM_CPUS_PER_TASK \
        --f config.csv \
        --db example_talon.db \
        --build GRCh38 \
        --o example</b>
[user@cn3144]$ ### summarize results
[user@cn3144]$ <b> talon_summarize \
       --db example_talon.db \
       --v \
       --o example</b>
[user@cn3144]$ ### run any other tools and then copy results back to shared space
[user@cn3144]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. talon.sh), which uses the input file 'talon.in'. For example:</p>

<pre class="term">
#! /bin/bash

module load talon/5.0

bam1=ENCFF291EKY.bam
bam2=ENCFF613SDS.bam
gtf=gencode.v35.primary_assembly.annotation.gtf
genome=GRCh38.primary_assembly.genome.fa



cd /lscratch/$SLURM_JOB_ID
cp -L ${TALON_TEST_DATA:-none}/* .
talon_initialize_database \
    --f $gtf \
    --a gencode_35 \
    --g GRCh38 \
    --o example_talon

mkdir -p labeled tmp
talon_label_reads --f $bam1 \
    --g $genome  \
    --t $SLURM_CPUS_PER_TASK \
    --ar 20 \
    --tmpDir=/lscratch/$SLURM_JOB_ID/tmp \
    --deleteTmp \
    --o labeled/${bam1%.bam}
talon_label_reads --f $bam2 \
    --g $genome  \
    --t $SLURM_CPUS_PER_TASK \
    --ar 20 \
    --tmpDir=/lscratch/$SLURM_JOB_ID/tmp \
    --deleteTmp \
    --o labeled/${bam2%.bam}

cat > config.csv &lt;&lt;__EOF__
ex_rep1,GRCh38,PacBio-Sequel2,labeled/${bam1%.bam}_labeled.sam
ex_rep2,GRCh38,PacBio-Sequel2,labeled/${bam2%.bam}_labeled.sam
__EOF__

talon \
    -t $SLURM_CPUS_PER_TASK \
    --f config.csv \
    --db example_talon.db \
    --build GRCh38 \
    --o example

talon_summarize \
   --db example_talon.db \
   --v \
   --o example
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] talon.sh</pre>


<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. talon.swarm). For example:</p>

<pre class="term">
talon_label_reads --f ENCFF291EKY.bam \
    --g GRCh38.primary_assembly.genome.fa  \
    --t $SLURM_CPUS_PER_TASK \
    --ar 20 \
    --tmpDir=/lscratch/$SLURM_JOB_ID \
    --deleteTmp \
    --o labeled/ENCFF291EKY
talon_label_reads --f ENCFF613SDS.bam \
    --g GRCh38.primary_assembly.genome.fa  \
    --t $SLURM_CPUS_PER_TASK \
    --ar 20 \
    --tmpDir=/lscratch/$SLURM_JOB_ID \
    --deleteTmp \
    --o labeled/ENCFF613SDS
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f talon.swarm [-g 10] [-t 6] --gres=lscratch:50 --module talon/5.0</pre>
where
<table border=0>
  <tr><td width=20%>-g # <td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td>-t # <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td>--module talon <td>Loads the talon module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
