<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type="text/javascript" language="JavaScript">document.querySelector('title').textContent = "tandem-genotypes";</script>

<div class="title">tandem-genotypes on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
tandem-genotypes finds changes in length of tandem repeats, from "long" DNA reads aligned to a genome.
</p>

<h3>References:</h3>
<ul>

    <li>
	Mitsuhashi, S., Frith, M.C., Mizuguchi, T. et al.
	<a href="https://doi.org/10.1186/s13059-019-1667-6"><b><u>Tandem-genotypes: robust detection of tandem repeat expansions from long DNA reads.</u></b></a>
	<em>Genome Biol 20, 58 (2019).</em>
  </li>

</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="https://github.com/mcfrith/tandem-genotypes">tandem-genotypes Main Site</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>

<ul>
    <li>Module Name: <tt>tandem-genotypes</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>This application produces plots via the <tt>tandem-genotypes-plot</tt> command. You can use <a href="/docs/hpcdrive.html">hpcdrive</a> to view them on your local workstation.</li>
<li>Environment variables set <!--for ones users should be aware of -->
  <ul>
      <li><tt>TANDEMGENOTYPES_HOME</tt></li>
  </ul>
  <li>Example files in <tt>$TANDEMGENOTYPES_HOME/tests</tt>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load tandem-genotypes</b>

[user@cn3144 ~]$ <b>tandem-genotypes $TANDEMGENOTYPES_HOME/tests/microsat.txt $TANDEMGENOTYPES_HOME/tests/nano.maf</b>
# tandem-genotypes /usr/local/apps/tandem-genotypes/1.9.0/tests/microsat.txt /usr/local/apps/tandem-genotypes/1.9.0/tests/nano.maf
chr22	41994883	41994923	TG	.	.	-7,-6,-4,-3,-1,0,0,2,3,4,8	-16,-14,-12,-11,-11,-11,-9,-8,-7
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. tandem-genotypes.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load tandem-genotypes
tandem-genotypes $TANDEMGENOTYPES_HOME/tests/microsat.txt $TANDEMGENOTYPES_HOME/tests/nano.maf > tg.txt
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] tandem-genotypes.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. tandem-genotypes.swarm). For example:</p>

<pre class="term">
tandem-genotypes $TANDEMGENOTYPES_HOME/tests/microsat.txt $TANDEMGENOTYPES_HOME/tests/sample1.maf > tg1.txt
tandem-genotypes $TANDEMGENOTYPES_HOME/tests/microsat.txt $TANDEMGENOTYPES_HOME/tests/sample2.maf > tg2.txt
tandem-genotypes $TANDEMGENOTYPES_HOME/tests/microsat.txt $TANDEMGENOTYPES_HOME/tests/sample3.maf > tg3.txt
tandem-genotypes $TANDEMGENOTYPES_HOME/tests/microsat.txt $TANDEMGENOTYPES_HOME/tests/sample4.maf > tg4.txt
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f tandem-genotypes.swarm [-g #] [-t #] --module tandem-genotypes</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module tandem-genotypes</tt> <td>Loads the tandem-genotypes module for each subjob in the swarm 
</table>

<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
