<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'TransDecoder on Biowulf';</script>
<div class="title">TransDecoder on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>TransDecoder identifies candidate coding regions within transcript sequences, such as those generated by de novo RNA-Seq transcript assembly using Trinity, or constructed based on RNA-Seq alignments to the genome using Tophat and Cufflinks.</p>
<p>TransDecoder identifies likely coding sequences based on the following criteria:</p>
<ul>
  <li>a minimum length open reading frame (ORF) is found in a transcript sequence</li>
  <li>a log-likelihood score similar to what is computed by the GeneID software is &gt; 0.</li>
  <li>the above coding score is greatest when the ORF is scored in the 1st reading frame as compared to scores in the other 5 reading frames.</li>
  <li>if a candidate ORF is found fully encapsulated by the coordinates of another candidate ORF, the longer one is reported. However, a single transcript can report multiple ORFs (allowing for operons, chimeras, etc).</li>
  <li><em>optional</em> the putative peptide has a match to a Pfam domain above the noise cutoff score.</li>
</ul>

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
  <li>TransDecoder Main Site: <a href="http://transdecoder.github.io/">TransDecoder at GitHub</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>TransDecoder</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
<li>Multithreaded/singlethreaded/MPI...
<li>Environment variables set <!--for ones users should be aware of -->
  <ul>
    <li><tt>TRANSDECODER_HOME</tt></li>
    <li><tt>TRANSDECODER_EXAMPLES</tt></li>
  </ul>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ module load TransDecoder
[user@cn3144 ~]$ cp $TRANSDECODER_EXAMPLES/simple_transcriptome_target/* .
[user@cn3144 ~]$ ./runMe.sh

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. TransDecoder.sh). For example:</p>

<pre class="term">
#!/bin/bash
module load TransDecoder
cufflinks_gtf_genome_to_cdna_fasta.pl transcripts.gtf test.genome.fasta &gt; transcripts.fasta
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] TransDecoder.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. TransDecoder.swarm). For example:</p>

<pre class="term">
TransDecoder.LongOrfs -t target_transcripts1.fasta
TransDecoder.LongOrfs -t target_transcripts2.fasta
TransDecoder.LongOrfs -t target_transcripts3.fasta
TransDecoder.LongOrfs -t target_transcripts4.fasta
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f TransDecoder.swarm [-g #] [-t #] --module TransDecoder</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module TransDecoder</tt> <td>Loads the TransDecoder module for each subjob in the swarm 
</table>

<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
