<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'trimAl: a tool for automated alignment trimming';</script>
<div class="title">trimAl: a tool for automated alignment trimming</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
trimAl is a tool for the automated removal of spurious sequences or poorly aligned regions from a multiple sequence alignment.
It can consider several parameters, alone or in multiple combinations, 
in order to select the most-reliable positions in the alignment. 
These include the proportion of sequences with a gap, the level of residue similarity and, 
if several alignments for the same set of sequences are provided, the consistency level of columns among alignments. 
Moreover, trimAl is able to manually select a set of columns to be removed from the alignment. 
</p>

<h3>References:</h3>
<ul>
<li>Salvador Capella-Gutiérrez, José M. Silla-Martínez, Toni Gabaldón <br />
<i>trimAl: a tool for automated alignment trimming in large-scale phylogenetic analyses </i> <br />
<a href="https://academic.oup.com/bioinformatics/article/25/15/1972/213148">Bioinformatics, Volume 25, Issue 15, 1 August 2009, Pages 1972–1973, https://doi.org/10.1093/bioinformatics/btp348 </a> 

</li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="http://trimal.cgenomics.org/">trimAl home page</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>trimAl</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
<li>Unusual environment variables set
  <ul>
    <li><b>TRIMAL_HOME</b>&nbsp; installation directory</li>
    <li><b>TRIMAL_BIN</b>&nbsp; executable   directory</li>
  </ul>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive -c 16 --mem 45g --gres=lscratch:20</b>
[user@cn3144 ~]$ <b>module load trimAl</b>
[+] Loading singularity  3.8.5-1  on cn3144
[+] Loading trimAl  1.2rev59
[user@cn3144 ~]$ <b> trimal </b>

trimAl 1.2rev59. Copyright (C) 2009. Salvador Capella-Gutierrez and Toni Gabaldón.

trimAl webpage: http://trimal.cgenomics.org

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, the last available version.

Please cite:    Salvador Capella-Gutierrez, Jose M. Silla-Martinez and
                Toni Gabaldon. trimAl: a tool for automated alignment
                trimming (2009).

Basic usage
        trimal -in &lt;inputfile&gt; -out &lt;outputfile&gt; -(other options).

Common options (for a complete list please see the User Guide or visit http://trimal.cgenomics.org):

    -h                       Print this information and show some examples.
    --version                Print the trimAl version.

    -in &lt;inputfile&gt;          Input file in several formats (clustal, fasta, NBRF/PIR, nexus, phylip3.2, phylip).
    -compareset &lt;inputfile&gt;  Input list of paths for the files containing the alignments to compare.
    -matrix &lt;inpufile&gt;       Input file for user-defined similarity matrix (default is Blosum62).

    -out &lt;outputfile&gt;        Output alignment in the same input format (default stdout). (default input format)
    -htmlout &lt;outputfile&gt;    Get a summary of trimal's work in an HTML file.

    -clustal                 Output file in CLUSTAL format
    -fasta                   Output file in FASTA format
    -nbrf                    Output file in NBRF/PIR format
    -nexus                   Output file in NEXUS format
    -mega                    Output file in MEGA format
    -phylip3.2               Output file in PHYLIP3.2 format
    -phylip                  Output file in PHYLIP/PHYLIP4 format

    -complementary           Get the complementary alignment.
    -colnumbering            Get the relationship between the columns in the old and new alignment.

    -select { n,l,m-k }      Selection of columns to be removed from the alignment. (see User Guide).
    -gt -gapthreshold &lt;n&gt;    1 - (fraction of sequences with a gap allowed).
    -st -simthreshold &lt;n&gt;    Minimum average similarity allowed.
    -ct -conthreshold &lt;n&gt;    Minimum consistency value allowed.
    -cons &lt;n&gt;                Minimum percentage of the positions in the original alignment to conserve.

    -nogaps                  Remove all positions with gaps in the alignment.
    -noallgaps               Remove columns composed only by gaps.

    -gappyout                Use automated selection on "gappyout" mode. This method only uses information based on gaps' distribution. (see User Guide).
    -strict                  Use automated selection on "strict" mode. (see User Guide).
    -strictplus              Use automated selection on "strictplus" mode. (see User Guide).
                             (Optimized for Neighbour Joining phylogenetic tree reconstruction).

    -automated1              Use a heuristic selection of the automatic method based on similarity statistics. (see User Guide).
                             (Optimized for Maximum Likelihood phylogenetic tree reconstruction).

    -resoverlap              Minimum overlap of a positions with other positions in the column to be considered a "good position". (see User Guide).
    -seqoverlap              Minimum percentage of "good positions" that a sequence must have in order to be conserved. (see User Guide).

    -w &lt;n&gt;                   (half) Window size, score of position i is the average of the window (i - n) to (i + n).
    -gw &lt;n&gt;                  (half) Window size only applies to statistics/methods based on Gaps.
    -sw &lt;n&gt;                  (half) Window size only applies to statistics/methods based on Similarity.
    -cw &lt;n&gt;                  (half) Window size only applies to statistics/methods based on Consistency.

    -sgc                     Print gap percentage count for columns in the input alignment.
    -sgt                     Print accumulated gap percentage count.
    -scc                     Print conservation values for columns in the input alignment.
    -sct                     Print accumulated conservation values count.
    -sfc                     Print compare values for columns in the selected alignment from compare files method.
    -sft                     Print accumulated compare values count for the selected alignment from compare files method.
    -sident                  Print identity statistics for all sequences in the alignemnt. (see User Guide).
[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>



<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
