<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'trinity on Biowulf';</script>
<div class="title">trinity on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
      </div>
</table>

<p> Trinity represents a novel method for the efficient and robust de novo
reconstruction of transcriptomes from RNA-seq data. Trinity combines three
independent software modules: Inchworm, Chrysalis, and Butterfly, applied
sequentially to process large volumes of RNA-seq reads. Trinity partitions the
sequence data into many individual de Bruijn graphs, each representing the
transcriptional complexity at at a given gene or locus, and then processes each
graph independently to extract full-length splicing isoforms and to tease apart
transcripts derived from paralogous genes. Briefly, the process works like so:</p>

<ul> 
    <li>Inchworm assembles the RNA-seq data into the unique sequences of
    transcripts, often generating full-length transcripts for a dominant
    isoform, but then reports just the unique portions of alternatively spliced
    transcripts.</li>

    <li>Chrysalis clusters the Inchworm contigs into clusters and constructs
    complete de Bruijn graphs for each cluster. Each cluster represents the
    full transcriptonal complexity for a given gene (or sets of genes that
    share sequences in common). Chrysalis then partitions the full read set
    among these disjoint graphs.</li>

    <li>Butterfly then processes the individual graphs in parallel, tracing the
    paths that reads and pairs of reads take within the graph, ultimately
    reporting full-length transcripts for alternatively spliced isoforms, and
    teasing apart transcripts that corresponds to paralogous genes.</li>
</ul>

<p> Trinity was developed at the Broad Institute &amp; the Hebrew University of
Jerusalem.
</p>

<p>In addition to these core functions, Trinity also incudes scripts to 
do in silico normalization, transcript quantitation, differential expression,
and other downstream analyses.</p>

<p> Trinotate, the comprehensive annotation suite designed for automatic
functional annotation of transcriptomes, particularly de novo assembled
transcriptomes, from model or non-model organisms, is also available.</p>

<h3>References:</h3>
<ul>
    <li>Manfred G. Grabherr et al. <em>Trinity: reconstructing a full-length 
        transcriptome without a genome from RNA-Seq data</em>.  Nature Biotechnology 
    2011, 29:644-652.
    <a href="http://www.ncbi.nlm.nih.gov/pubmed/21572440">Pubmed</a>&nbsp;|&nbsp;
    <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3571712/">PMC</a>&nbsp;|&nbsp;
    <a href="http://www.nature.com/nbt/journal/v29/n7/abs/nbt.1883.html">Journal</a>
    </li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
    <li><a href="https://github.com/trinityrnaseq/trinityrnaseq">GitHub</a> </li>
    <li><a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki">Manual</a></li>
</ul>

<div class="heading">Important Notes</div>

<ul>
    <li>Module Name: trinity (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>Trinity is a multithreaded application. Cluster mode is not supported on biowulf any longer</li> 
    <li>For some subset of trinity functions like, for example, expression estimation additional
    modules may have to be loaded.</li>
</ul>

<p class="alert">Trinity creates a lot of temporary files and for efficiency we highly
recomment that it be run from <strong>lscratch</strong> as shown in the examples
below. This is especially true for swarms of trinity runs which can result in
severe stress on the shared file systems if lscratch is not used.</p> 

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=6 --gres=lscratch:150 --mem=20g</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144]$ <b>module load trinity</b>
[user@cn3144]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144]$ <b>cp -pr $TRINITY_TEST_DATA/test_Trinity_Assembly .</b>
[user@cn3144]$ <b>cd test_Trinity_Assembly</b>
[user@cn3144]$ <b>Trinity --seqType fq --max_memory 2G \
              --left reads.left.fq.gz \
              --right reads.right.fq.gz \
              --SS_lib_type RF \
              --CPU 4</b>
[...snip...]

[user@cn3144]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>

<p>
A Trinity run can be devided into two phases - (I) one phase that uses more memory
but is not easily parallelized and (II) a phase that uses less memory but can be
parallelized. Below is a trace of the memory usage and the number of active
threads for a genome based Trinity (2.4.0) assembly of ~15M normalized paired end human reads 
running on a single node with 10 allocated CPUs:
</p>

<div style="width:75%; margin:0 auto;">
  <img src="/images/trinity_profile_singlenode.png" alt="Trinity memory/thread profile" width="100%">
</div>

<p>While Trinity can parallelize Phase II across multiple nodes, we currently
do not support this capability on biowulf.</p>

<p>To run Trinity on a single node, create a batch script similar to the
following example.</p> 

<pre class="term">
#! /bin/bash
# this file is trinity.sh
function die() {
    echo "$@" &gt;&amp;2
    exit 1
}
module load trinity/2.14.0 || die "Could not load trinity module"
[[ -d /lscratch/$SLURM_JOB_ID ]] || die "no lscratch allocated"

inbam=$1
mkdir /lscratch/$SLURM_JOB_ID/in
mkdir /lscratch/$SLURM_JOB_ID/out
cp $inbam /lscratch/$SLURM_JOB_ID/in
bam=/lscratch/$SLURM_JOB_ID/in/$(basename $inbam)
out=/lscratch/$SLURM_JOB_ID/out

Trinity --genome_guided_bam $bam \
    --SS_lib_type RF \
    --output  $out \
    --genome_guided_max_intron 10000 \
    --max_memory 28G \
    --CPU 12
mv $out/Trinity-GG.fasta /data/$USER/trinity_out/$(basename $inbam .bam)-Trinity-GG.fasta
</pre>



<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">
biowulf$ <b>sbatch --mem=30g --cpus-per-task=12 --gres=lscratch:150 trinity.sh /data/$USER/trinity_in/sample.bam</b>
</pre>





<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
