<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'tRNAscan-SE on Biowulf';</script>
<div class="title">tRNAscan-SE on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
 tRNAscan-SE 2.0 has advanced the state-of-the-art methodology in tRNA gene detection and functional prediction, captured by rich new content of the companion Genomic tRNA Database</p>

<h3>References:</h3>
<ul>
<li>Lowe, T.M. and Chan, P.P. (2016) tRNAscan-SE On-line: Search and Contextual Analysis of Transfer RNA Genes. Nucl. Acids Res. 44: W54-57.</li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="http://lowelab.ucsc.edu/tRNAscan-SE/">tRNAscan-SE Main Site</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>trnascan-se</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
<li>Singlethreaded app
<li>Example files in <tt>/usr/local/apps/trnascan-se/TEST_DATA</tt>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load trnascna-se</b>
[+] Loading trnascan-se  2.0.0  on cn0166 

[user@cn3144 ~]$ <b>cp /usr/local/apps/trnascan-se/TEST_DATA/ExampleSequences.fa .</b>

[user@cn3144 ~]$ <b>tRNAscan-SE ExampleSequences.fa</b>

tRNAscan-SE v.2.0 (December 2017) - scan sequences for transfer RNAs
Copyright (C) 2017 Patricia Chan and Todd Lowe
                   University of California Santa Cruz
Freely distributed under the GNU General Public License (GPLv3)

------------------------------------------------------------
Sequence file(s) to search:        ExampleSequences.fa
Search Mode:                       Eukaryotic
Results written to:                Standard output
Output format:                     Tabular
Searching with:                    Infernal First Pass->Infernal
Isotype-specific model scan:       Yes
Covariance model:                  /usr/local/apps/trnascan-se/2.0.0/lib/tRNAscan-SE/models/TRNAinf-euk.cm
                                   /usr/local/apps/trnascan-se/2.0.0/lib/tRNAscan-SE/models/TRNAinf-euk-SeC.cm
Infernal first pass cutoff score:  10

Temporary directory:               /tmp
------------------------------------------------------------

Status: Phase I: Searching for tRNAs with HMM-enabled Infernal
Status: Phase II: Infernal verification of candidate tRNAs detected with first-pass scan
Sequence		tRNA	Bounds	tRNA	Anti	Intron Bounds	Inf	      
Name    	tRNA #	Begin	End	Type	Codon	Begin	End	Score	Note
--------	------	-----	------	----	-----	-----	----	------	------
MySeq1  	1	13 	85 	Thr	TGT	0	0	78.0	
MySeq2  	1	6  	79 	Arg	TCT	0	0	75.1	
MySeq3  	1	14 	114	Ser	CGA	51	69	71.8	
MySeq4  	1	6  	88 	Leu	AAG	0	0	65.0	
MySeq5  	1	3  	89 	SeC	TCA	0	0	146.9	
MySeq6  	1	7  	92 	Lys	CTT	0	0	72.1	

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. trnascan-se.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load trnascan-se
tRNAscan-SE ExampleSequences.fa
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch trnascan-se.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. trnascan-se.swarm). For example:</p>

<pre class="term">
tRNAscan-SE ExampleSequences1.fa > out1.txt
tRNAscan-SE ExampleSequences2.fa > out2.txt
tRNAscan-SE ExampleSequences3.fa > out3.txt
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f trnascan-se.swarm --module trnascan-se</pre>
where
<table border=0>
  <tr><td><tt>--module trnascan-se</tt> <td>Loads the trnascan-se module for each subjob in the swarm 
</table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
