<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'uLTRA: a tool for splice alignment of long transcriptomic reads to a genome';</script>
<div class="title"><b>uLTRA: a tool for splice alignment of long transcriptomic reads to a genome</b></div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
uLTRA implements an alignment method for long RNA sequencing reads 
based on a novel two-pass collinear chaining algorithm. uLTRA is guided by a database of exon annotations,
but it can also be used as a wrapper around minimap2 to align reads outside annotated regions.
</p>

<h3>References:</h3>
<ul>
<li>Kristoffer Sahlin and Veli Makinen <br />
Accurate spliced alignment of long RNA sequencing reads </i> <br />
<a href="https://academic.oup.com/bioinformatics/article/37/24/4643/6327681">Bioinformatics </a> <b>37</b>(24), Pages 4643â€“4651 (2021)
</ul>
<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/ksahlin/ultra">uLTRA GitHub page</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
<li>Module Name: <tt>ultra</tt> (see <a href="https://hpc.nih.gov/apps/modules.html">the modules page</a> for more information)
<li>Unusual environment variables set 
  <ul>
    <li><b>ULTRA_HOME</b>&nbsp; installation directory</li>
    <li><b>ULTRA_BIN </b>&nbsp; &nbsp; &nbsp; executable directory</li>
    <li><b>ULTRA_SRC </b>&nbsp; &nbsp; &nbsp; source code directory</li>
    <li><b>ULTRA_DATA</b>&nbsp; sample data directory
  </ul>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --mem=4g</b>
[user@cn0911 ~]$<b>module load ultra   </b>
[+] Loading singularity  3.10.5  on cn0911
[+] Loading Loading ultra  0.1
[user@cn0911 ~]$<b>cp $ULTRA_DATA/* . </b>
[user@cn0911 ~]$<b>uLTRA pipeline SIRV_genes.fasta SIRV_genes_C_170612a.gtf reads.fa outfolder</b>
creating outfolder
total_flanks2: 90
total_flank_size 72816
total_unique_segment_counter 24396
total_segments_bad 25056
bad 460
total parts size: 24856
total exons size: 74187
min_intron: 20
Number of ref seqs in gff: 71
Number of ref seqs in fasta: 7
Filtering reads aligned to unindexed regions with minimap2
Running minimap2...
minimap2 done.
Done filtering. Reads filtered:0
Time for minimap2:0.10440826416015625 seconds.
Processing reads for NAM finding
Completed processing poly-A tails
Time for processing reads:0.006848812103271484 seconds.
RUNNING NAMFINDER USING COMMAND:
namfinder -k 10 -s 10 -l 10 -u 11 -C 500 -L 1000 -t 3 -S outfolder/refs_sequences.fa outfolder/reads_tmp.fa.gz 2> outfolder/namfinder_stderr.1 | gzip -1 --stdout > outfolder/seeds.txt.gz
Time for namfinder to find seeds:0.48410677909851074 seconds.
Starting aligning reads.
Wrote 0 batches of reads.
file_IO: Reading records done. Tot read: 4
file_IO: Written records in producer process: 0
Process: [0, 0, 0, 0, 0, 0, 0, 0]
Number of instances solved with quadratic collinear chainer solution: 0
Number of instances solved with n*log n collinear chainer solution: 0
Process: [0, 0, 0, 0, 0, 0, 0, 0]
Number of instances solved with quadratic collinear chainer solution: 0
Number of instances solved with n*log n collinear chainer solution: 0
Process: [3.8929521968943623, 4, 0, 0, 0, 0, 0, 0]
Number of instances solved with quadratic collinear chainer solution: 4
Number of instances solved with n*log n collinear chainer solution: 0
Wrote 1 batches of reads.
file_IO: Remainig written records after consumer join: 4
Done joining processes.
Time to align reads:0.07692599296569824 seconds.
Total mm2's primary alignments replaced with uLTRA: 2
Total mm2's alignments unmapped but mapped with uLTRA: 0
2 primary alignments had better score with uLTRA.
0 primary alignments had equal score with alternative aligner.
2 primary alignments had slightly better score with alternative aligner (typically ends bonus giving better scoring in ends, which needs to be implemented in uLTRA).
0 primary alignments had significantly better score with alternative aligner.
0 reads were unmapped with ultra but not by alternative aligner.
0 reads were not attempted to be aligned with ultra (unindexed regions), instead alternative aligner was used.
Time for selecting final best alignments (selecting the best of mm2's vs uLTRA's alignments):0.009923696517944336 seconds.
FSM : 4, NO_SPLICE : 0, Insufficient_junction_coverage_unclassified : 0, ISM/NIC_known : 0, NIC_novel : 0, NNC : 0
total alignment coverage: 3.8929521968943623
Deleting temporary files...
removed: outfolder/minimap2.sam
removed: outfolder/minimap2_errors.1
removed: outfolder/uLTRA_batch_0.stderr
removed: outfolder/uLTRA_batch_1.stderr
removed: outfolder/uLTRA_batch_2.stderr
removed: outfolder/reads_after_genomic_filtering.fasta
removed: outfolder/indexed.sam
removed: outfolder/unindexed.sam
removed: outfolder/refs_sequences.fa
Done.
[user@cn0911 ~]$<b>exit</b>
</pre> 
End the interactive session:
<pre class="term">
[user@cn0911 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
