<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Ultraplex on Biowulf';</script>
<div class="title">Ultraplex on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
Ultraplex is an application for demultiplexing and processing FASTQ files. The processing steps include removing low quality bases, removing sequencing adaptors, and separating FASTQ using demultiplexing barcodes. Ultraplex is intended for use with custom library preparation protocols instead of commercial prep kits.
</p>

<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://github.com/ulelab/ultraplex">Ultraplex Github</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>

<p class="NXblock">This application requires a <a href="/docs/userguide.html#local">lscratch allocation</a></p>

<ul>
  <li>Module Name: <tt>ultraplex</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
	<li>Multithreaded</li>
<li>Environment variables set <!--for ones users should be aware of -->
  <ul>
    <li><tt>ULTRAPLEX_TEST_DATA</tt></li>
  </ul>
<li>Test data files in <tt>/usr/local/apps/ultraplex/TEST_DATA</tt>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --gres=lscratch:10</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load ultraplex</b>
[+] Loading singularity  3.8.5-1  on cn0847
[+] Loading ultraplex 1.2.5  ...

[user@cn3144 ~]$ <b>cd /lscratch/$SLURM_JOB_ID</b>

[user@cn3144 ~]$ <b>cp -r $ULTRAPLEX_TEST_DATA tests</b>

[user@cn3144 ~]$ <b>ultraplex -i tests/reads1.fastq.gz \
                              -i2 tests/reads2.fastq.gz \
                              -b tests/barcodes_5_and_3.csv \
                              -d PE -o paired_end</b>
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
@@@@@   @@@@   .@@   @@@@@@@          @@         @@@@@@    (@@@@@        @@@    @@@@@@@         @@    @@@    @
@@@@   @@@@    @@   ,@@@@@@@@@    @@@@@    @@@   @@@@   (   @@@@   @@@   @@    @@@@@@@   @@@@@@@@@@   #   @@@@
@@@   &@@@    @@    @@@@@@@@@%   @@@@@         @@@@(   @    @@@         @@    @@@@@@@         @@@@@     @@@@@@
@@    @@@    @@    @@@@@@@@@@   @@@@@    @@    @@@          @@   .@@@@@@@*   @@@@@@@    @@@@@@@@.   @    @@@@@
@@@       @@@@.        @@@@@   @@@@@&   @@@.   &   &@@@@    @    @@@@@@@@         @          @    @@@@    @@@@
@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@

Namespace(adapter='AGATCGGAAGAGCGGTTCAG', adapter2='AGATCGGAAGAGCGTCGTG', barcodes='tests/barcodes_5_and_3.csv', directory='PE', dont_build_reference=False, final_min_length=20, fiveprimemismatches=1, ignore_no_match=False, ignore_space_warning=False, input_2='tests/reads2.fastq.gz', inputfastq='tests/reads1.fastq.gz', keep_barcode=False, min_trim=3, outputprefix='paired_end', phredquality=30, phredquality_5_prime=0, sbatchcompression=False, threads=4, threeprimemismatches=0, ultra=False)
Demultiplexing...
Demultiplexing complete! 2700 reads processed in 0.0 seconds

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. ultraplex.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load ultraplex
ultraplex -i R1.fastq.gz \
          -i2 R2.fastq.gz \
          -b barcods.csv \
          -d PE \
          -t $SLURM_CPUS_PER_TASK
          -o output_dir
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] --gres=lscratch:# ultraplex.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. ultraplex.swarm). For example:</p>

<pre class="term">
ultraplex -t $SLURM_CPUS_PER_TASK -i L1.R1.fastq.gz -i2 L1.R2.fastq.gz -b barcodes.csv -d PE -o L1
ultraplex -t $SLURM_CPUS_PER_TASK -i L2.R1.fastq.gz -i2 L2.R2.fastq.gz -b barcodes.csv -d PE -o L2
ultraplex -t $SLURM_CPUS_PER_TASK -i L3.R1.fastq.gz -i2 L3.R2.fastq.gz -b barcodes.csv -d PE -o L3
ultraplex -t $SLURM_CPUS_PER_TASK -i L4.R1.fastq.gz -i2 L4.R2.fastq.gz -b barcodes.csv -d PE -o L4
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f ultraplex.swarm [-g #] [-t #] --gres=lscratch:# --module ultraplex</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i></tt></td><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)</td></tr>
  <tr><td><tt>-t <i>#</i></tt></td><td>Number of threads/CPUs required for each process (1 line in the swarm command file).</td></tr>
  <tr><td><tt>--gres=lscratch:#</tt></td><td>lscratch amount in GB allocated for each process (1 line in the swarm command file).</td></tr>
  <tr><td><tt>--module ultraplex</tt></td><td>Loads the Ultraplex module for each subjob in the swarm</td></tr>
</table>
















<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
