<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'Umitools on HPC';</script>
    <div class="title">Umitools on HPC</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</td></tr></table>    <p> Umi-tools are tools for dealing with Unique Molecular Identifiers (UMIs)/Random 
      Molecular Tags (RMTs) and single cell RNA-Seq cell barcodes. Currently there 
      are 6 commands.</p>
    <p>The extract and whitelist commands are used to prepare a fastq containg 
      UMIs +/- cell barcodes for alignment.</p>
    <p>whitelist:<br />
      Builds a whitelist of the 'real' cell barcodes<br />
      This is useful for droplet-based single cell RNA-Seq where the identity 
      of the true cell barcodes is unknown. Whitelist can then be used to filter 
      with extract (see below)</p>
    <p>extract:<br />
      Flexible removal of UMI sequences from fastq reads.<br />
      UMIs are removed and appended to the read name. Any other barcode, for example 
      a library barcode, is left on the read. Can also filter reads by quality 
      or against a whitelist (see above)<br />
      The remaining commands, group, dedup and count/count_tab, are used to identify 
      PCR duplicates using the UMIs and perform different levels of analysis depending 
      on the needs of the user. A number of different UMI deduplication schemes 
      are enabled - The recommended method is directional.</p>
    <p>group:<br />
      Groups PCR duplicates using the same methods available through `dedup`.<br />
      This is useful when you want to manually interrogate the PCR duplicates</p>
    <p>dedup:<br />
      Groups PCR duplicates and deduplicates reads to yield one read per group<br />
      Use this when you want to remove the PCR duplicates prior to any downstream 
      analysis</p>
    <p>count:<br />
      Groups and deduplicates PCR duplicates and counts the unique molecules per 
      gene<br />
      Use this when you want to obtain a matrix with unique molecules per gene. 
      Can also perform per-cell counting for scRNA-Seq.</p>
    <p>count_tab:<br />
      As per count except input is a flatfile</p>
    <h3>&nbsp;</h3>
<a Name="doc"></a><div class="heading">Documentation</div>    <ul>
      <li><a href="https://github.com/CGATOxford/UMI-tools">https://github.com/CGATOxford/UMI-tools</a></li>
    </ul>

<a Name="notes"></a>
    <div class="heading"><a name="notes" id="notes"></a>Important Notes</div>
    <ul>
      <li>Module Name: <tt>umitools</tt> (see <a href="/apps/modules.html">the 
        modules page</a> for more information) </li>
      <li>Example files in /usr/local/apps/umitools/example.fastq.gz</li>
    </ul>
<P>

<a Name="int"></a></p><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. Sample session:</p>
    <pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <strong>module load umitools</strong>
[user@cn3144 ~]$ <strong>cp /usr/local/apps/umitools/example.fastq.gz . </strong>
[user@cn3144 ~]$ <strong>umi_tools extract --stdin=example.fastq.gz --bc-pattern=NNNNNNNNN --log=processed.log --stdout processed.fastq.gz</strong>

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a></p><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
    <p>Create a batch input file (e.g. batch.sh). For example:</p>

    <pre class="term">
#!/bin/bash
set -e
module load umitools
umi_tools extract --stdin=example.fastq.gz --bc-pattern=NNNNNNNNN --log=processed.log --stdout processed.fastq.gz</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

    <pre class="term">sbatch [--mem=#] batch.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

    <p>Create a swarmfile (e.g. job.swarm). For example:</p>

<pre class="term">
cd dir1; umi_tools extract --stdin=example.fastq.gz --bc-pattern=NNNNNNNNN --log=processed.log --stdout processed.fastq.gz
cd dir2; umi_tools extract --stdin=example.fastq.gz --bc-pattern=NNNNNNNNN --log=processed.log --stdout processed.fastq.gz
cd dir3; umi_tools extract --stdin=example.fastq.gz --bc-pattern=NNNNNNNNN --log=processed.log --stdout processed.fastq.gz
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

    <pre class="term">swarm -f job.swarm [-g #] --module umitools</pre>
where
    <table width="63%" border=0>
      <tr>
        <td width=18%><tt>-g <i>#</i> </tt></td>
        <td width="82%">Number of Gigabytes of memory required for each process 
          (1 line in the swarm command file) </td>
      </tr>
      <tr> 
        <td><tt>--module </tt></td>
        <td>Loads the module for each subjob in the swarm </td>
      </tr>
    </table>




<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
