<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'verkko on Biowulf';</script>
<div class="title">verkko on Biowulf</div>

<table width=25% align=right>
    <tr>
        <td>
            <div class="toc">
                <div class="tocHeading">Quick Links</div>
                <div class="tocItem"><A href="#doc">Documentation</a></div>
                <div class="tocItem"><a href="#notes">Notes</a></div>
                <div class="tocItem"><a href="#int">Interactive job </a></div>
                <div class="tocItem"><a href="#sbatch">Batch job </a></div>
            </div>
        </td>
    </tr>
</table>

<p>
From the tool description
</p>
<blockquote>
Verkko is a hybrid genome assembly pipeline developed for telomere-to-telomere
assembly of PacBio HiFi and Oxford Nanopore reads. Verkko is Finnish for net,
mesh and graph.

Verkko uses Canu to correct remaining errors in the HiFi reads, builds a
multiplex de Bruijn graph using MBG, aligns the Oxford Nanopore reads to the
graph using GraphAligner, progressively resolves loops and tangles first with
the HiFi reads then with the aligned Oxford Nanopore reads, and finally creates
contig consensus sequences using Canu's consensus module.
</blockquote>

<h3>References:</h3>
<ul>
<li>M. Rautiainen, S. Nurk, B. P. Walenz, G. A. Logsdon, D. Porubsky, A. Rhie, E. E. Eichler, A. M. Phillippy, S. Koren. 
    <em>Verkko: telomere-to-telomere assembly of diploid chromosomes.</em>
    <a href="https://doi.org/10.1101/2022.06.24.497523">bioRxiv</a> (2022)</li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li>verkko on <a href="https://github.com/marbl/verkko">GitHub</a></li>
</ul>

<div class="heading">Important Notes</div>
<ul>
    <li>Module Name: verkko (see <a href="/apps/modules.html">the modules page</a> for more information)</li>
    <li>verkko uses snakemake to run a complex workflow. For small data sets it can be run in local mode as a single job
        but for realistic size data it is likely to be used in cluster mode with <code>--slurm</code></li>
    <li>Example files in <code>$VERKKO_TEST_DATA</code></li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run verkko in local mode for the smaller test
data. Don't move data to lscratch for slurm mode. Sample session:</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive --cpus-per-task=8 --mem=24g --gres=lscratch:100</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144]$ <b>module load verkko</b>
[user@cn3144]$ <b>cd /lscratch/$SLURM_JOB_ID</b>
[user@cn3144]$ <b>cp ${VERKKO_TEST_DATA:-none}/* .</b>
[user@cn3144]$ <b>ls -lh</b>
total 363M
-rw-r--r-- 1 user group 119M Oct 26 12:07 hifi.fastq.gz
-rw-r--r-- 1 user group 245M Oct 26 12:07 ont.fastq.gz

[user@cn3144]$ <b>verkko -d asm --cleanup --local --local-memory 22 \
    --local-cpus $SLURM_CPUS_PER_TASK --hifi ./hifi.fastq.gz --nano ./ont.fastq.gz</b>
Launching verkko branch  commit
Using snakemake 7.19.1
Building DAG of jobs...
Using shell: /usr/bin/bash
Provided cores: 8
Rules claiming more threads will be scaled down.
Provided resources: mem_gb=22
Job stats:
job                            count    min threads    max threads
---------------------------  -------  -------------  -------------
buildGraph                         1              4              4
[...snip...]
[user@cn3144]$ <b>ls -lh asm</b>
total 36M
drwxr-xr-x 2 user group 4.0K Oct 26 12:16 1-buildGraph
drwxr-xr-x 2 user group 4.0K Oct 26 12:26 2-processGraph
drwxr-xr-x 2 user group 4.0K Oct 26 12:26 3-align
drwxr-xr-x 2 user group 4.0K Oct 26 12:26 4-processONT
drwxr-xr-x 2 user group 4.0K Oct 26 12:26 5-untip
drwxr-xr-x 2 user group 4.0K Oct 26 12:18 6-layoutContigs
drwxr-xr-x 2 user group 4.0K Oct 26 12:26 7-consensus
-rw-r--r-- 1 user group 4.5M Oct 26 12:26 assembly.fasta
-rw-r--r-- 1 user group   40 Oct 26 12:26 assembly.hifi-coverage.csv
-rw-r--r-- 1 user group 3.3M Oct 26 12:26 assembly.homopolymer-compressed.gfa
-rw-r--r-- 1 user group 419K Oct 26 12:26 assembly.homopolymer-compressed.layout
-rw-r--r-- 1 user group  105 Oct 26 12:26 assembly.homopolymer-compressed.noseq.gfa
-rw-r--r-- 1 user group   41 Oct 26 12:26 assembly.ont-coverage.csv
-rw-r--r-- 1 user group    0 Oct 26 12:11 emptyfile
-rw-r--r-- 1 user group  28M Oct 26 12:16 hifi-corrected.fasta.gz
-rwxr-xr-x 1 user group  340 Oct 26 12:11 snakemake.sh
-rw-r--r-- 1 user group 2.9K Oct 26 12:11 verkko.yml

[user@cn3144]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>In this example verkko is run in slurm mode so the main job doesn't need many resources</p>

<pre class="term">
#!/bin/bash
module load verkko/1.1
cp -r "${VERKKO_TEST_DATA:-none}" input
verkko -d asm --cleanup --local --local-memory 10 --slurm \
    --local-cpus $SLURM_CPUS_PER_TASK --hifi input/hifi.fastq.gz --nano input/ont.fastq.gz
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch --cpus-per-task=2 --mem=10g verkko.sh</pre>






<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
