<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'WHATSHAP on Biowulf';</script>
<div class="title">WHATSHAP on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
WhatsHap is an application for phasing genomic variants using DNA sequencing reads.
</p>

<h3>References:</h3>
<ul>
<li><a href="https://doi.org/10.1101/085050">WhatsHap: fast and accurate read-based phasing</a>. Marcel Martin, Murray Patterson, Shilpa Garg, Sarah O. Fischer, Nadia Pisanti, Gunnar W. Klau, Alexander Schoenhuth, Tobias Marschall. bioRxiv 085050</li>
<li><a href="http://dx.doi.org/10.1089/cmb.2014.0157">WhatsHap: Weighted Haplotype Assembly for Future-Generation Sequencing Reads</a>. Murray Patterson, Tobias Marschall, Nadia Pisanti, Leo van Iersel, Leen Stougie, Gunnar W. Klau, Alexander Sch√∂nhuth. Journal of Computational Biology, 22(6), pp. 498-509, 2015.</li>
</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://whatshap.readthedocs.io">Whatshap Main Site</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>
<ul>
    <li>Module Name: <tt>whatshap</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf]$ <b>sinteractive</b>
salloc.exe: Pending job allocation 46116226
salloc.exe: job 46116226 queued and waiting for resources
salloc.exe: job 46116226 has been allocated resources
salloc.exe: Granted job allocation 46116226
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3144 are ready for job

[user@cn3144 ~]$ <b>module load whatshap</b>

[user@cn3144 ~]$ <b>cd /data/$USER/WHATSHAP_TEST</b>

[user@cn3144 ~]$ <b>whatshap phase -o phased.vcf input.vcf pacbio_reads_30x.bam</b>
Working on 1 samples from 1 family
======== Working on chromosome 'contig1'
---- Processing individual SAMPLE
Using maximum coverage per sample of 15X
Number of variants skipped due to missing genotypes: 0
Number of remaining heterozygous variants: 177
Reading alignments for sample 'SAMPLE'and detecting alleles ...
WARNING: Sample 'SAMPLE' not found in any BAM/CRAM file.
Found 0 reads covering 0 variants
Kept 0 reads that cover at least two variants each
Reducing coverage to at most 15X by selecting most informative reads ...
Selected 0 reads covering 0 variants
Best-case phasing would result in 0 non-singleton phased blocks (0 in total)
... after read selection: 0 non-singleton phased blocks (0 in total)
Variants covered by at least one phase-informative read in at least one individual after read selection: 0
Phasing 1 sample by solving the MEC problem ...
MEC cost: 0
No. of phased blocks: 0
======== Writing VCF
Done writing VCF
======== Working on chromosome 'contig2'
---- Processing individual SAMPLE
Using maximum coverage per sample of 15X
Number of variants skipped due to missing genotypes: 0
Number of remaining heterozygous variants: 134
Reading alignments for sample 'SAMPLE'and detecting alleles ...
WARNING: Sample 'SAMPLE' not found in any BAM/CRAM file.
Found 0 reads covering 0 variants
Kept 0 reads that cover at least two variants each
Reducing coverage to at most 15X by selecting most informative reads ...
Selected 0 reads covering 0 variants
Best-case phasing would result in 0 non-singleton phased blocks (0 in total)
... after read selection: 0 non-singleton phased blocks (0 in total)
Variants covered by at least one phase-informative read in at least one individual after read selection: 0
Phasing 1 sample by solving the MEC problem ...
MEC cost: 0
No. of phased blocks: 0
======== Writing VCF
Done writing VCF
======== Working on chromosome 'contig3'
---- Processing individual SAMPLE
Using maximum coverage per sample of 15X
Number of variants skipped due to missing genotypes: 0
Number of remaining heterozygous variants: 165
Reading alignments for sample 'SAMPLE'and detecting alleles ...
WARNING: Sample 'SAMPLE' not found in any BAM/CRAM file.
Found 0 reads covering 0 variants
Kept 0 reads that cover at least two variants each
Reducing coverage to at most 15X by selecting most informative reads ...
Selected 0 reads covering 0 variants
Best-case phasing would result in 0 non-singleton phased blocks (0 in total)
... after read selection: 0 non-singleton phased blocks (0 in total)
Variants covered by at least one phase-informative read in at least one individual after read selection: 0
Phasing 1 sample by solving the MEC problem ...
MEC cost: 0
No. of phased blocks: 0
======== Writing VCF
Done writing VCF

== SUMMARY ==
Maximum memory usage: 0.050 GB
Time spent reading BAM/CRAM:                    0.0 s
Time spent parsing VCF:                         0.1 s
Time spent selecting reads:                     0.0 s
Time spent phasing:                             0.0 s
Time spent writing VCF:                         0.1 s
Time spent finding components:                  0.0 s
Time spent on rest:                             0.0 s
Total elapsed time:                             0.2 s

[user@cn3144 ~]$ <b>exit</b>
salloc.exe: Relinquishing job allocation 46116226
[user@biowulf ~]$
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. whatshap.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
module load whatshap
whatshap phase -o /data/$USER/WHATSHAP_TEST/phased.vcf \
                  /data/$USER/WHATSHAP_TEST/input.vcf \
                  /data/$USER/WHATSHAP_TEST/pacbio_reads_30x.bam
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] whatshap.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. whatshap.swarm). For example:</p>

<pre class="term">
whatshap phase -o /data/$USER/WHATSHAP_TEST/phased_A.vcf \
                  /data/$USER/WHATSHAP_TEST/input_A.vcf \
                  /data/$USER/WHATSHAP_TEST/pacbio_reads_A.bam
whatshap phase -o /data/$USER/WHATSHAP_TEST/phased_B.vcf \
                  /data/$USER/WHATSHAP_TEST/input_B.vcf \
                  /data/$USER/WHATSHAP_TEST/pacbio_reads_B.bam
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f whatshap.swarm [-g #] [-t #] --module whatshap</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module whatshap</tt> <td>Loads the whatshap module for each subjob in the swarm 
</table>
















<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
