<script type="text/javascript" language="JavaScript" src='/js/header.js'></script>
<!-- Start content - do not edit above this line  -->
<script type='text/javascript' language='JavaScript'>document.querySelector('title').textContent = 'xengsort on Biowulf';</script>
<div class="title">xengsort on Biowulf</div>

<table width=25% align=right>
  <tr>
    <td>
      <div class="toc">
        <div class="tocHeading">Quick Links</div>
        <div class="tocItem"><A href="#doc">Documentation</a></div>
        <div class="tocItem"><a href="#notes">Notes</a></div>
        <div class="tocItem"><a href="#int">Interactive job </a></div>
        <div class="tocItem"><a href="#sbatch">Batch job </a></div>
        <div class="tocItem"><a href="#swarm">Swarm of jobs </a></div>
      </div>
</table>

<p>
xengsort is a fast xenograft read sorter based on space-efficient k-mer hashing.
</p>

<h3>References:</h3>
<ul>
  <li>
    <a href="https://www.biorxiv.org/content/biorxiv/early/2020/05/19/2020.05.14.095604.full.pdf">Zentgraf, Jens, and Sven Rahmann. "Fast lightweight accurate xenograft sorting." <i>bioRxiv</i> (2020).</a>
  </li>

</ul>


<a Name="doc"></a><div class="heading">Documentation</div>
<ul>
<li><a href="https://gitlab.com/genomeinformatics/xengsort/">xengsort on GitLab</a></li>
</ul>

<a Name="notes"></a><div class="heading">Important Notes</div>

<ul>
    <li>Module Name: <tt>xengsort</tt> (see <a href="/apps/modules.html">the modules page</a> for more information)
	<li>This application is designed to work with <a href="https://hpc.nih.gov/apps/snakemake.html">snakemake</a>.</li>
</ul>
<P>

<a Name="int"></a><div class="heading">Interactive job</div>
<div class="nudgeblock"><a href="/docs/userguide.html#int">Interactive jobs</a> should be used for debugging, graphics, or applications that cannot be run as batch jobs.</div>
<p>Allocate an <a href="/docs/userguide.html#int">interactive session</a> and run the program. <br>Sample session (user input in <b>bold</b>):</p>
<pre class="term">
[user@biowulf ~]$ <b>sinteractive --cpus-per-task=8 --mem=32g</b>
salloc.exe: Pending job allocation 61524097
salloc.exe: job 61524097 queued and waiting for resources
salloc.exe: job 61524097 has been allocated resources
salloc.exe: Granted job allocation 61524097
salloc.exe: Waiting for resource configuration
salloc.exe: Nodes cn3137 are ready for job
srun: error: x11: no local DISPLAY defined, skipping

[user@cn3137 ~]$ <b>cd /data/$USER</b>

[user@cn3137 user]$ <b>git clone https://gitlab.com/genomeinformatics/xengsort.git</b>
Cloning into 'xengsort'...
remote: Enumerating objects: 226, done.
remote: Counting objects: 100% (226/226), done.
remote: Compressing objects: 100% (103/103), done.
remote: Total 226 (delta 139), reused 183 (delta 117), pack-reused 0
Receiving objects: 100% (226/226), 108.80 KiB | 0 bytes/s, done.
Resolving deltas: 100% (139/139), done.

[user@cn3137 user]$ <b>cd xengsort/</b>

[user@cn3137 xengsort]$ <b>module load xengsort snakemake</b>
[+] Loading xengsort  28762aac  on cn3137
[+] Loading singularity  3.5.3  on cn3137
[+] Loading snakemake  5.19.3

[user@cn3137 xengsort]$ <b>snakemake -j 8</b>
Building DAG of jobs...
Using shell: /usr/bin/bash
Provided cores: 8
Rules claiming more threads will be scaled down.
Job counts:
        count   jobs
        1       all
        1       build_index
        1       classify_mouse_exomes
        2       download_mouse_exomes
        1       download_refs
        6

[Mon Jul 20 14:15:43 2020]
rule download_refs:
    output: ref/Homo_sapiens.GRCh38.dna.toplevel.fa.gz, ref/Mus_musculus.GRCm38.dna.toplevel.fa.gz, ref/Homo_sapiens.GRCh38.cdna.all.fa.gz, ref/Mus_musculus.GRCm38.cdna.all.fa.gz
    jobid: 5


[Mon Jul 20 14:15:43 2020]
rule download_mouse_exomes:
    output: raw/BALBc-M1-normal_1.fq.gz.1
    jobid: 3
    wildcards: filename=BALBc-M1-normal_1.fq.gz.1


[Mon Jul 20 14:15:43 2020]
rule download_mouse_exomes:
    output: raw/BALBc-M1-normal_2.fq.gz.1
    jobid: 4
    wildcards: filename=BALBc-M1-normal_2.fq.gz.1

--2020-07-20 14:15:43--  https://sra-pub-src-1.s3.amazonaws.com/SRR9130497/BALBc-M1-normal_2.fq.gz.1
--2020-07-20 14:15:43--  https://sra-pub-src-1.s3.amazonaws.com/SRR9130497/BALBc-M1-normal_1.fq.gz.1
--2020-07-20 14:15:43--  ftp://ftp.ensembl.org/pub/release-98/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.toplevel.fa.gz
Resolving dtn05-e0 (dtn05-e0)... 10.1.200.241
Connecting to dtn05-e0 (dtn05-e0)|10.1.200.241|:3128... connected.
Proxy request sent, awaiting response... Resolving dtn05-e0 (dtn05-e0)... Resolving dtn05-e0 (dtn05-e0)... 10.1.200.24110.1.200.241
Connecting to dtn05-e0 (dtn05-e0)|10.1.200.241|:3128...
Connecting to dtn05-e0 (dtn05-e0)|10.1.200.241|:3128... connected.
connected.
Proxy request sent, awaiting response... Proxy request sent, awaiting response... 200 OK
Length: 5034316763 (4.7G) [application/x-troff-man]
Saving to: ‘raw/BALBc-M1-normal_2.fq.gz.1’

 0% [                                                             ] 0           --.-K/s              200 OK
Length: 4783409755 (4.5G) [application/x-troff-man]
Saving to: ‘raw/BALBc-M1-normal_1.fq.gz.1’

 0% [                                                             ] 33,550,787  32.2MB/s             200 Gatewaying
Length: 1107654500 (1.0G) [text/plain]
Saving to: ‘ref/Homo_sapiens.GRCh38.dna.toplevel.fa.gz’
[...snip...]
</pre>

<P>
<a Name="sbatch"></a><div class="heading">Batch job</div>
<div class="nudgeblock">Most jobs should be run as <A href="/docs/userguide.html#submit">batch jobs</a>.</div>
<p>Create a batch input file (e.g. xengsort.sh). For example:</p>

<pre class="term">
#!/bin/bash
set -e
cd /data/${USER}
git clone https://gitlab.com/genomeinformatics/xengsort.git
cd xengsort/
module load xengsort snakemake
snakemake -j 8
</pre>
<p>Submit this job using the Slurm <a href="/docs/userguide.html">sbatch</a> command.</p>

<pre class="term">sbatch [--cpus-per-task=#] [--mem=#] xengsort.sh</pre>

<a Name="swarm"></a><div class="heading">Swarm of Jobs </div>
<div class="nudgeblock">A <a href="/apps/swarm.html">swarm of jobs</a> is an easy way to submit a set of independent commands requiring identical resources.</div>

<p>Create a swarmfile (e.g. xengsort.swarm). For example:</p>

<pre class="term">
cd /path/to/snakefile1 && snakemake -j 8
cd /path/to/snakefile2 && snakemake -j 8
cd /path/to/snakefile3 && snakemake -j 8
cd /path/to/snakefile4 && snakemake -j 8
</pre>

<p>Submit this job using the <a href='/apps/swarm.html'>swarm</a> command.</p>

<pre class="term">swarm -f xengsort.swarm [-g #] [-t #] --module xengsort snakemake</pre>
where
<table border=0>
  <tr><td width=20%><tt>-g <i>#</i> </tt><td>Number of Gigabytes of memory required for each process (1 line in the swarm command file)
  <tr><td><tt>-t <i>#</i></tt> <td>Number of threads/CPUs required for each process (1 line in the swarm command file).
  <tr><td><tt>--module xengsort</tt> <td>Loads the xengsort module for each subjob in the swarm 
</table>
















<!-- End content area - do not edit below this line -->
<script type="text/javascript" language="JavaScript" src='/js/footer.js'></script>
