<div class="term"><pre>
Path to Bowtie 2 specified as: bowtie2
Output format is BAM (default)
Alignments will be written out in BAM format. Samtools found here: '/usr/local/apps/samtools/1.3.1/bin/samtools'
Reference genome folder provided is XandY/	(absolute path is '/spin1/users/user/bismark_test/XandY/)'
FastQ format assumed (by default)

Files to be analysed:
test_data.fastq

Library is assumed to be strand-specific (directional), alignments to strands complementary to the original top or bottom strands will be ignored (i.e. not performed!)
Setting parallelization to single-threaded (default)

Current working directory is: /spin1/users/user/bismark_test

Now reading in and storing sequence information of the genome specified in: /spin1/users/user/bismark_test/XandY/

chr chrX (155270560 bp)
chr chrY (59373566 bp)

Single-core mode: setting pid to 1

Single-end alignments will be performed
=======================================

Input file is in FastQ format
Writing a C -&gt T converted version of the input file test_data.fastq to test_data.fastq_C_to_T.fastq

Created C -&gt T converted version of the FastQ file test_data.fastq (10000 sequences in total)

Input file is test_data.fastq_C_to_T.fastq (FastQ)

Now running 2 instances of Bowtie 2 against the bisulfite genome of /spin1/users/user/bismark_test/XandY/ with the specified options: -q --score-min L,0,-0.2 --ignore-quals

Now starting the Bowtie 2 aligner for CTreadCTgenome (reading in sequences from test_data.fastq_C_to_T.fastq with options -q --score-min L,0,-0.2 --ignore-quals --norc)
Using Bowtie 2 index: /spin1/users/user/bismark_test/XandY/Bisulfite_Genome/CT_conversion/BS_CT

Found first alignment:	SRR020138.15024316_SALK_2029:7:100:1672:1790_length=86	4	*	0	0	*	*	0	0	TTTTTAGAAAGTTTATATGGAAATTAAGTTTTTTGTATATGTTTGTAAAG	BCB?).4'324&1;B###################################	YT:Z:UU
Now starting the Bowtie 2 aligner for CTreadGAgenome (reading in sequences from test_data.fastq_C_to_T.fastq with options -q --score-min L,0,-0.2 --ignore-quals --nofw)
Using Bowtie 2 index: /spin1/users/user/bismark_test/XandY/Bisulfite_Genome/GA_conversion/BS_GA

Found first alignment:	SRR020138.15024316_SALK_2029:7:100:1672:1790_length=86	4	*	0	0	*	*	0	0	TTTTTAGAAAGTTTATATGGAAATTAAGTTTTTTGTATATGTTTGTAAAG	BCB?).4'324&1;B###################################	YT:Z:UU

&gt&gt&gt Writing bisulfite mapping results to test_data_bismark_bt2.bam &lt&lt&lt


Reading in the sequence file test_data.fastq
10000 reads; of these:
  10000 (100.00%) were unpaired; of these:
    9446 (94.46%) aligned 0 times
    191 (1.91%) aligned exactly 1 time
    363 (3.63%) aligned &gt1 times
5.54% overall alignment rate
10000 reads; of these:
  10000 (100.00%) were unpaired; of these:
    9451 (94.51%) aligned 0 times
    194 (1.94%) aligned exactly 1 time
    355 (3.55%) aligned &gt1 times
5.49% overall alignment rate
Processed 10000 sequences in total


Successfully deleted the temporary file test_data.fastq_C_to_T.fastq

Final Alignment report
======================
Sequences analysed in total:	10000
Number of alignments with a unique best hit from the different alignments:	378
Mapping efficiency:	3.8%

Sequences with no alignments under any condition:	9267
Sequences did not map uniquely:	355
Sequences which were discarded because genomic sequence could not be extracted:	0

Number of sequences with unique best (first) alignment came from the bowtie output:
CT/CT:	205	((converted) top strand)
CT/GA:	173	((converted) bottom strand)
GA/CT:	0	(complementary to (converted) top strand)
GA/GA:	0	(complementary to (converted) bottom strand)

Number of alignments to (merely theoretical) complementary strands being rejected in total:	0

Final Cytosine Methylation Report
=================================
Total number of C's analysed:	3258

Total methylated C's in CpG context:	88
Total methylated C's in CHG context:	4
Total methylated C's in CHH context:	13
Total methylated C's in Unknown context:	0

Total unmethylated C's in CpG context:	108
Total unmethylated C's in CHG context:	777
Total unmethylated C's in CHH context:	2268
Total unmethylated C's in Unknown context:	0

C methylated in CpG context:	44.9%
C methylated in CHG context:	0.5%
C methylated in CHH context:	0.6%
Can't determine percentage of methylated Cs in Unknown context (CN or CHN) if value was 0



====================
Bismark run complete
====================

</pre></div>
